---
title: "meiotic timecourse analysis"
author: "Lori Koch"
output: html_document
---

# Functions

```{r}

# TL functions

colMedians <- function (df, na.rm = T) {
  apply(df, 2, median, na.rm)
}

strsplit.extract <- function (x, split, index) {
  x <- as.character(x)
  x[x == ""] <- ";"
  x <- sapply(strsplit(x, split, fixed = TRUE), "[[", index)
  return(x)
}

#choosing number of clusters for k-means

wssplot <- function(data, nc=20, seed=1234, iter.max = 10){
  wss <- (nrow(data)-1)*sum(apply(data,2,var))
  for (i in 2:nc){
    set.seed(seed)
    wss[i] <- sum(kmeans(data, centers=i, iter.max = iter.max)$withinss)}
  plot(1:nc, wss, type="b", xlab="Number of Clusters",
       ylab="Within groups sum of squares")}

# LK Plotting to PDF function

print_plot<-function(plot,filename.pdf,width,height){
    print(plot)
pdf(filename.pdf, width=width, height=height) 
print(plot)
dev.off()
}


```

# data prep

```{r, message=FALSE, warning=FALSE}

#Read data tables #####
#Files have all datasets included

phosphos <- read.delim("meiotic_timecourse_Phospho (STY)Sites.txt")

protg <- read.delim("meiotic_timecourse_proteinGroups.txt")


```

# name cols
```{r, message=FALSE, warning=FALSE}

library(dplyr)
#Remove reverse & contaminant #####

phosphos <- subset(phosphos, Reverse != "+" & Potential.contaminant != "+")
protg <- subset(protg, Reverse != "+" & Potential.contaminant != "+")

#Change 'id' columns to reflect source table #####

colnames(protg)[colnames(protg) == "id"] <- "protg.id"
colnames(phosphos)[colnames(phosphos) == "id"] <- "phosphos.id"

#Set variables with cols of interest

protg.TMT.cols<-grep("Reporter.intensity.corrected.", colnames(protg), value = T) %>%
  grep("19i_N|210322|200707|201001", ., value = T)

# need to put them in the same order I had previous
# spo13 rep1
# spo13 rep2
# WT rep1
# WT rep2
protg.TMT.cols<-c(protg.TMT.cols[11:20],protg.TMT.cols[21:30],protg.TMT.cols[1:10],protg.TMT.cols[31:40])


#PE phospho TMT cols
#note only analyzing singly-phosphorylated phospho-sites AKA ___1 (not ___2 or ___3)
phos.TMT.cols<-grep("Reporter.intensity.corrected.(.+)___1", colnames(phosphos), value = T) %>%
   grep("19i_PE|210330|200710|201022", ., value = T)

# need to put them in the same order I had previous
# spo13 rep1
# spo13 rep2
# WT rep1
# WT rep2
phos.TMT.cols<-c(phos.TMT.cols[11:20],phos.TMT.cols[21:30],phos.TMT.cols[1:10],phos.TMT.cols[31:40])

#name new columns that will become the normalised columns.
#shorter names than "reporter intensity corrected[0-9].PE___1" 
reporter.cols <- c('t.0', 't.45', 't.60', 't.75', 't.90', 't.105', 't.120', 't.135', 't.150', 't.165')
reporter.cols <- paste0('raw.', reporter.cols)
reporter.cols<-rep(reporter.cols,2)
reporter.cols[1:10]<-paste0(reporter.cols[1:10],".Rep1")
reporter.cols[11:20]<-paste0(reporter.cols[11:20],".Rep2")
reporter.cols<-paste0(reporter.cols,".Spo13")
reporter.cols<-rep(reporter.cols,2)
reporter.cols[21:40]<-gsub(".Spo13",".WT",reporter.cols[21:40])

### Protg data prep
#copy raw reporter intensities to new cols named reporter.cols
protg[,reporter.cols] <- protg[,protg.TMT.cols]

#### Phosphos data prep
#same as above but for the phosphos table

phosphos[,reporter.cols] <- phosphos[,phos.TMT.cols]

### Make gene name cols ####

### #load yeast gene map table
#I downloaded this file from https://www.uniprot.org/docs/yeast
#I modified the original file so it only includes the table and not the heading part at the top

gene.map.table <-read.csv("Yeast_Uniprot2.csv")

#data prep so there isn't white space and only the first gene name is in the gene column
gene.map.table$Gene<-trimws(gene.map.table$Gene, "right")
gene.map.table$Gene.name <- strsplit.extract(gene.map.table$Gene, ";",1)

#### Create new ID columns ####
#Create a Gene.name column in phosphos table

phosphos$Gene.name <- gene.map.table$Gene.name[match(phosphos$Protein, gene.map.table$OLN)]

#### Create new ID cols in the protg table ####
#Create new Majority.protein.ID col with only FIRST (one) majority protein ID
#for cross referencing prot between tables

protg$Majority.protein.ID <- strsplit.extract(protg$Majority.protein.IDs, ";",1)

#Create Gene.name col in the protg table
protg$Gene.name <- gene.map.table$Gene.name[match(protg$Majority.protein.ID,gene.map.table$OLN)]

```

# replace Zeros with NAs
```{r,echo=F,warning=F,message=F}

# Replace zeros with NAs to make 'reporter.na.cols' so stats can be calc and plots made

library(data.table)

# new cols so reporter.cols retain zeros
reporter.na.cols<-paste0("na.",reporter.cols)

protg.DT<-as.data.table(protg)
phosphos.DT<-as.data.table(phosphos)

protg.DT[,c(reporter.na.cols) := lapply(.SD,function(x) ifelse(x==0,NA,x)), .SD=reporter.cols]
phosphos.DT[,c(reporter.na.cols) := lapply(.SD,function(x) ifelse(x==0,NA,x)), .SD=reporter.cols]

protg<-as.data.frame(protg.DT)
phosphos<-as.data.frame(phosphos.DT)

# raw data
boxplot(log2(protg[,reporter.na.cols]),main="raw protein",ylab="log2 intensity")
boxplot(log2(phosphos[,reporter.na.cols]),main="raw phosphos",ylab="log2 intensity")

```


# ColMedian Normalization

```{r, echo=F,warning=F,message=F}
# Normalization of protg to reporter medians and phospho-sites to protein levels ####

# calculate protg reporter MEDIAN intensities
tot_med <- colMedians(protg[,reporter.na.cols],na.rm=T)

# create new column names for normalised intensities
norm.tmt.cols <- gsub("raw","norm",reporter.na.cols)

## normalise protg intensities
## by dividing by median of each reporter

# # Identify the target median, which in this case is the median of the median of each experiment

target_median <- median(tot_med)

# # # Identify the scaling factors needed to scale each individual experiment to reach the target_med
scaling_factors <- as.numeric( target_median / tot_med )

# # Scale the TMT intensities of each experiment so the median becomes the target median

protg[,norm.tmt.cols] <- sweep(protg[,reporter.na.cols], 2, scaling_factors, FUN = "*")

# # check the colMedians are now the same
#colMedians(protg[,norm.tmt.cols])

# Channel/Column norm for phosphos data: ###

# calculate protg reporter MEDIAN intensities
phos_tot_med <- colMedians(phosphos[,reporter.na.cols],na.rm=T)

# # Identify the target median, which in this case is the median of the median of each experiment

phos_target_median <- median(phos_tot_med)

# # # Identify the scaling factors needed to scale each individual experiment to reach the target_med
phos_scaling_factors <- as.numeric( phos_target_median / phos_tot_med )

# # Scale the TMT intensities of each experiment so the median becomes the target median

phosphos[,norm.tmt.cols] <- sweep(phosphos[,reporter.na.cols], 2, phos_scaling_factors, FUN = "*")

# # check the colMedians
# colMedians(phosphos[,norm.tmt.cols])

# Boxplots of colNorm data
boxplot(log2(protg[,norm.tmt.cols]),main="colNorm protein",ylab="log2 intensity")
boxplot(log2(phosphos[,norm.tmt.cols]),main="colNorm phosphos",ylab="log2 intensity")
```

# Phos/Prot Norm

```{r,echo=F,message=F,warning=F}
##### New values called "stoic" for "stoichiometric" 

# extract first sequence window (some phos-sites have multiple and
# this messes up motif analysis)
phosphos$Sequence.window <- strsplit.extract(phosphos$Sequence.window, ';', 1)

#create new phos.stoic.cols 
phos.stoic.cols <- gsub("raw","stoic", reporter.cols)

# data prep
#copy protg IDs in phosphos df to new column called first.protg.id
#this is necessary for matching protg.ids between the phosphos and protg table

phosphos$first.protg.id <- strsplit.extract(phosphos$Protein.group.IDs, ';', 1)

phos.stoic.temp.df <- protg[match(phosphos$first.protg.id, protg$protg.id), norm.tmt.cols]


## ALTERNATE DF WHERE PHOSPHO-SITES ARE NOT NORM TO PROTEIN
## BUT COLNAMES ARE KEPT SAME SO ALL SUBSEQUENT ANALYSIS CAN BE RUN THE SAME WAY ON THIS DF
no_norm_phosphos<-phosphos
no_norm_phosphos[,phos.stoic.cols] <- no_norm_phosphos[,norm.tmt.cols]

#this divides the intensity of the phosphos over the protg
#normalized intensity then multiplies by 1000 (to get easier number)
#This creates "stoichiometric" phos. site changes...

phosphos[,phos.stoic.cols] <- mapply(function(x,y) {x/y},phosphos[,norm.tmt.cols],  phos.stoic.temp.df[,norm.tmt.cols]) * 1000

# check colMedians - they are close but not the same
#colMedians(phosphos[,phos.stoic.cols])

# Plot the phos.stoic.colss (normalized to protg)
boxplot(log2(phosphos[,phos.stoic.cols]),main="Phos/protg norm phosphos",ylab="log2 intensity")
# plot no norm values jut to cheeck
boxplot(log2(no_norm_phosphos[,phos.stoic.cols]),main="No protg norm phosphos",ylab="log2 intensity")

```

# add gene.name_site col
```{r, message=FALSE, warning=FALSE, echo=F}

phosphos<-phosphos %>%
  mutate(gene.name_site=paste(Gene.name,Amino.acid,Positions.within.proteins,sep="-"))

# alt df with no norm
no_norm_phosphos<-no_norm_phosphos %>%
  mutate(gene.name_site=paste(Gene.name,Amino.acid,Positions.within.proteins,sep="-"))

```

# add log2 cols
```{r,echo=F,message=F,warning=F}

# # add log2 cols
protg[,paste0("log2.",grep("norm",colnames(protg),value=T))]<-log2(protg[,grep("norm",colnames(protg),value=T)])

phosphos[,paste0("log2.",grep("stoic",colnames(phosphos),value=T))]<-log2(phosphos[,grep("stoic",colnames(phosphos),value=T)])

# alternate no norm phosphos df
no_norm_phosphos[,paste0("log2.",grep("stoic",colnames(no_norm_phosphos),value=T))]<-log2(no_norm_phosphos[,grep("stoic",colnames(no_norm_phosphos),value=T)])

```

# batch correction

```{r,echo=F,message=F,warning=F}

# Batch correction
# batch correction must be done on log2 transformed data

library(limma)

# define batches
batch<-c(rep("spo13_Rep1",10),rep("spo13_Rep2",10),rep("WT_Rep1",10),rep("WT_Rep2",10))

# remove batch effects with Limma RemoveBatchEffect function

protg[,paste0("cor.",grep("log2",colnames(protg),value=T))]<-removeBatchEffect(protg[,grep("log2",colnames(protg),value=T)],batch)

phosphos[,paste0("cor.",grep("log2",colnames(phosphos),value=T))]<-removeBatchEffect(phosphos[,grep("log2",colnames(phosphos),value=T)],batch)

# alternate no norm phos df
no_norm_phosphos[,paste0("cor.",grep("log2",colnames(no_norm_phosphos),value=T))]<-removeBatchEffect(no_norm_phosphos[,grep("log2",colnames(no_norm_phosphos),value=T)],batch)


# plot batch cor data
boxplot(protg[,grep("cor.log2.",colnames(protg),value=T)] %>% filter(if_any(grep("cor.log2.",colnames(protg),value=T),function(x) x>0)) %>% filter(if_all(grep("cor.log2.",colnames(protg),value=T),is.finite)), main="batch cor protein",ylab="log2 intensity")

boxplot(phosphos[,grep("cor.log2.",colnames(phosphos),value=T)] %>% filter(if_any(grep("cor.log2.",colnames(phosphos),value=T),function(x) x>0)) %>% filter(if_all(grep("cor.log2.",colnames(phosphos),value=T),is.finite)),main="batch cor phosphos",ylab="log2 intensity")

boxplot(no_norm_phosphos[,grep("cor.log2.",colnames(no_norm_phosphos),value=T)] %>% filter(if_any(grep("cor.log2.",colnames(no_norm_phosphos),value=T),function(x) x>0)) %>% filter(if_all(grep("cor.log2.",colnames(no_norm_phosphos),value=T),is.finite)),main="batch cor no norm phosphos",ylab="log2 intensity")

# define cor cols

cor.log2.cols<-grep("cor.log2",colnames(protg),value=T)

phos.cor.log2.cols<-grep("cor.log2",colnames(phosphos),value=T)

cor.lin.cols<-gsub("log2","lin",cor.log2.cols)

phos.cor.lin.cols<-gsub("log2","lin",phos.cor.log2.cols)

# create linear batch-corrected cols

protg[,cor.lin.cols]<-2^(protg[,cor.log2.cols])
phosphos[,phos.cor.lin.cols]<-2^(phosphos[,phos.cor.log2.cols])
# this corrects -Infs to true zeros btw
# 
# alternate no norm phos df
no_norm_phosphos[,phos.cor.lin.cols]<-2^(no_norm_phosphos[,phos.cor.log2.cols])

```

# convert NAs to Zeros
```{r,echo=F,message=F,warning=F}

# convert NAs back to zeros 

protg.DT<-as.data.table(protg)
phosphos.DT<-as.data.table(phosphos)
no_norm_phosphos.DT<-as.data.table(no_norm_phosphos)

# norm cols
protg.DT[,c(norm.tmt.cols) := lapply(.SD,function(x) ifelse(is.na(x),0,x)), .SD=norm.tmt.cols]
phosphos.DT[,c(phos.stoic.cols) := lapply(.SD,function(x) ifelse(is.na(x),0,x)), .SD=phos.stoic.cols]
no_norm_phosphos.DT[,c(phos.stoic.cols) := lapply(.SD,function(x) ifelse(is.na(x),0,x)), .SD=phos.stoic.cols]

# cor cols
protg.DT[,c(cor.log2.cols) := lapply(.SD,function(x) ifelse(is.na(x),0,x)), .SD=cor.log2.cols]
protg.DT[,c(cor.lin.cols) := lapply(.SD,function(x) ifelse(is.na(x),0,x)), .SD=cor.lin.cols]
phosphos.DT[,c(phos.cor.log2.cols) := lapply(.SD,function(x) ifelse(is.na(x),0,x)), .SD=phos.cor.log2.cols]
phosphos.DT[,c(phos.cor.lin.cols) := lapply(.SD,function(x) ifelse(is.na(x),0,x)), .SD=phos.cor.lin.cols]
no_norm_phosphos.DT[,c(phos.cor.log2.cols) := lapply(.SD,function(x) ifelse(is.na(x),0,x)), .SD=phos.cor.log2.cols]
no_norm_phosphos.DT[,c(phos.cor.lin.cols) := lapply(.SD,function(x) ifelse(is.na(x),0,x)), .SD=phos.cor.lin.cols]

protg<-as.data.frame(protg.DT)
phosphos<-as.data.frame(phosphos.DT)
no_norm_phosphos<-as.data.frame(no_norm_phosphos.DT)


```


# indicate protg MVs
```{r,echo=F,message=F,warning=F}

# count detection based on original norm cols
# not log2 since that generates Infs

# add cols to indiciate whether prot
# was detected in 2 reps of each strain or all 4
# its enough to be detected 1/10 timepoints and doesn't
# matter if its detected at diff times in each rep

protg<-protg %>%
  mutate(spo13.2reps=apply(.[,norm.tmt.cols[1:10]],1,function(x) any(x>0)) & apply(.[,norm.tmt.cols[11:20]],1,function(x) any(x>0))) %>%
  mutate(WT.2reps=apply(.[,norm.tmt.cols[21:30]],1,function(x) any(x>0)) & apply(.[,norm.tmt.cols[31:40]],1,function(x) any(x>0))) %>%
  mutate(all4reps=spo13.2reps&WT.2reps) %>%
  mutate(spo13.rep1=apply(.[,norm.tmt.cols[1:10]],1,function(x) any(x>0))) %>%
  mutate(spo13.rep2=apply(.[,norm.tmt.cols[11:20]],1,function(x) any(x>0))) %>%
  mutate(WT.rep1=apply(.[,norm.tmt.cols[21:30]],1,function(x) any(x>0))) %>%
  mutate(WT.rep2=apply(.[,norm.tmt.cols[31:40]],1,function(x) any(x>0)))


```


# indicate phosphos MVs,score
```{r,echo=F,message=F,warning=F}

# filter by whether psite was detected at least twice
# in 2 sequential timepoints
# 
# #### Remember arg number is 1 less than total number of seq timepoints you actually
#### want to filter by 
#### so 1 here actually means 2 sequential detections


# add col to indicate 2 sequential detections in both of 2 reps
# remember still have to include is.finite(x)

phosphos<-phosphos %>%
  mutate(spo13.2.detects=apply(.[,phos.stoic.cols[1:10]],1,function(x) any(rle(diff(which(x>0 & is.finite(x)))==1)$lengths>=1)) &
        apply(.[,phos.stoic.cols[11:20]],1,function(x) any(rle(diff(which(x>0 & is.finite(x)))==1)$lengths>=1))   ) %>%
  mutate(WT.2.detects=apply(.[,phos.stoic.cols[21:30]],1,function(x) any(rle(diff(which(x>0 & is.finite(x)))==1)$lengths>=1)) &
        apply(.[,phos.stoic.cols[31:40]],1,function(x) any(rle(diff(which(x>0 & is.finite(x)))==1)$lengths>=1))   ) %>%
  mutate(all4reps=spo13.2.detects & WT.2.detects) %>%
  mutate(spo13.rep1=apply(.[,phos.stoic.cols[1:10]],1,function(x) any(rle(diff(which(x>0 & is.finite(x)))==1)$lengths>=1))) %>%
  mutate(spo13.rep2=apply(.[,phos.stoic.cols[11:20]],1,function(x) any(rle(diff(which(x>0 & is.finite(x)))==1)$lengths>=1))) %>%
  mutate(WT.rep1=apply(.[,phos.stoic.cols[21:30]],1,function(x) any(rle(diff(which(x>0 & is.finite(x)))==1)$lengths>=1))) %>%
  mutate(WT.rep2=apply(.[,phos.stoic.cols[31:40]],1,function(x) any(rle(diff(which(x>0 & is.finite(x)))==1)$lengths>=1)))

# alt no norm phos df
no_norm_phosphos<-no_norm_phosphos %>%
  mutate(spo13.2.detects=apply(.[,phos.stoic.cols[1:10]],1,function(x) any(rle(diff(which(x>0 & is.finite(x)))==1)$lengths>=1)) &
        apply(.[,phos.stoic.cols[11:20]],1,function(x) any(rle(diff(which(x>0 & is.finite(x)))==1)$lengths>=1))   ) %>%
  mutate(WT.2.detects=apply(.[,phos.stoic.cols[21:30]],1,function(x) any(rle(diff(which(x>0 & is.finite(x)))==1)$lengths>=1)) &
        apply(.[,phos.stoic.cols[31:40]],1,function(x) any(rle(diff(which(x>0 & is.finite(x)))==1)$lengths>=1))   ) %>%
  mutate(all4reps=spo13.2.detects & WT.2.detects) %>%
  mutate(spo13.rep1=apply(.[,phos.stoic.cols[1:10]],1,function(x) any(rle(diff(which(x>0 & is.finite(x)))==1)$lengths>=1))) %>%
  mutate(spo13.rep2=apply(.[,phos.stoic.cols[11:20]],1,function(x) any(rle(diff(which(x>0 & is.finite(x)))==1)$lengths>=1))) %>%
  mutate(WT.rep1=apply(.[,phos.stoic.cols[21:30]],1,function(x) any(rle(diff(which(x>0 & is.finite(x)))==1)$lengths>=1))) %>%
  mutate(WT.rep2=apply(.[,phos.stoic.cols[31:40]],1,function(x) any(rle(diff(which(x>0 & is.finite(x)))==1)$lengths>=1)))

# indicate whether localization score >0.75

phosphos<-phosphos %>%
  mutate(high.localiz.score=Localization.prob>0.75)

no_norm_phosphos<-no_norm_phosphos %>%
  mutate(high.localiz.score=Localization.prob>0.75)

```


# fold changes

```{r,echo=F,message=F,warning=F}

# fold changes of interesting contrasts
# there are 45 possible contrasts per strain
# and many possible contrasts total inter+intra strain
# but I will just focus on 9 contrasts per strain (vs t0
# and 10 contrasts (each timepoint) inter-strain)

# calc median fc from indicated contrasts
# use batched corrected linear cols
# this means fc is linear
protg<-protg %>%
  mutate(spo13.fc.45_v_0=apply(.[,cor.lin.cols],1,function(x) median(c(x[2]/x[1],x[12]/x[11])))) %>%
  mutate(spo13.fc.60_v_0=apply(.[,cor.lin.cols],1,function(x) median(c(x[3]/x[1],x[13]/x[11])))) %>%
  mutate(spo13.fc.75_v_0=apply(.[,cor.lin.cols],1,function(x) median(c(x[4]/x[1],x[14]/x[11])))) %>%
  mutate(spo13.fc.90_v_0=apply(.[,cor.lin.cols],1,function(x) median(c(x[5]/x[1],x[15]/x[11])))) %>%
  mutate(spo13.fc.105_v_0=apply(.[,cor.lin.cols],1,function(x) median(c(x[6]/x[1],x[16]/x[11])))) %>%
  mutate(spo13.fc.120_v_0=apply(.[,cor.lin.cols],1,function(x) median(c(x[7]/x[1],x[17]/x[11])))) %>%
  mutate(spo13.fc.135_v_0=apply(.[,cor.lin.cols],1,function(x) median(c(x[8]/x[1],x[18]/x[11])))) %>%
  mutate(spo13.fc.150_v_0=apply(.[,cor.lin.cols],1,function(x) median(c(x[9]/x[1],x[19]/x[11])))) %>%
  mutate(spo13.fc.165_v_0=apply(.[,cor.lin.cols],1,function(x) median(c(x[10]/x[1],x[20]/x[11])))) %>%
  mutate(WT.fc.45_v_0=apply(.[,cor.lin.cols],1,function(x) median(c(x[22]/x[21],x[32]/x[31])))) %>%
  mutate(WT.fc.60_v_0=apply(.[,cor.lin.cols],1,function(x) median(c(x[23]/x[21],x[33]/x[31])))) %>%   
  mutate(WT.fc.75_v_0=apply(.[,cor.lin.cols],1,function(x) median(c(x[24]/x[21],x[34]/x[31])))) %>%
  mutate(WT.fc.90_v_0=apply(.[,cor.lin.cols],1,function(x) median(c(x[25]/x[21],x[35]/x[31])))) %>%
  mutate(WT.fc.105_v_0=apply(.[,cor.lin.cols],1,function(x) median(c(x[26]/x[21],x[36]/x[31])))) %>%
  mutate(WT.fc.120_v_0=apply(.[,cor.lin.cols],1,function(x) median(c(x[27]/x[21],x[37]/x[31])))) %>%
  mutate(WT.fc.135_v_0=apply(.[,cor.lin.cols],1,function(x) median(c(x[28]/x[21],x[38]/x[31])))) %>%
  mutate(WT.fc.150_v_0=apply(.[,cor.lin.cols],1,function(x) median(c(x[29]/x[21],x[39]/x[31])))) %>%
  mutate(WT.fc.165_v_0=apply(.[,cor.lin.cols],1,function(x) median(c(x[30]/x[21],x[40]/x[31])))) %>%
  mutate(WT.spo13.fc.0_v_0=apply(.[,cor.lin.cols],1,function(x) median(c(x[21],x[31]))/median(c(x[1],x[11])))) %>%
  mutate(WT.spo13.fc.45_v_45=apply(.[,cor.lin.cols],1,function(x) median(c(x[22],x[32]))/median(c(x[2],x[12])))) %>%
  mutate(WT.spo13.fc.60_v_60=apply(.[,cor.lin.cols],1,function(x) median(c(x[23],x[33]))/median(c(x[3],x[13])))) %>%
  mutate(WT.spo13.fc.75_v_75=apply(.[,cor.lin.cols],1,function(x) median(c(x[24],x[34]))/median(c(x[4],x[14])))) %>%
  mutate(WT.spo13.fc.90_v_90=apply(.[,cor.lin.cols],1,function(x) median(c(x[25],x[35]))/median(c(x[5],x[15])))) %>%
  mutate(WT.spo13.fc.105_v_105=apply(.[,cor.lin.cols],1,function(x) median(c(x[26],x[36]))/median(c(x[6],x[16])))) %>%
  mutate(WT.spo13.fc.120_v_120=apply(.[,cor.lin.cols],1,function(x) median(c(x[27],x[37]))/median(c(x[7],x[17])))) %>%
  mutate(WT.spo13.fc.135_v_135=apply(.[,cor.lin.cols],1,function(x) median(c(x[28],x[38]))/median(c(x[8],x[18]))))  %>%
  mutate(WT.spo13.fc.150_v_150=apply(.[,cor.lin.cols],1,function(x) median(c(x[29],x[39]))/median(c(x[9],x[19])))) %>%
  mutate(WT.spo13.fc.165_v_165=apply(.[,cor.lin.cols],1,function(x) median(c(x[30],x[40]))/median(c(x[10],x[20])))) %>%
    # comparisons spo13 to t60
  mutate(spo13.fc.75_v_60=apply(.[,cor.lin.cols],1,function(x) median(c(x[4]/x[3],x[14]/x[13])))) %>%
  mutate(spo13.fc.90_v_60=apply(.[,cor.lin.cols],1,function(x) median(c(x[5]/x[3],x[15]/x[13])))) %>%
  mutate(spo13.fc.105_v_60=apply(.[,cor.lin.cols],1,function(x) median(c(x[6]/x[3],x[16]/x[13])))) %>%
  mutate(spo13.fc.120_v_60=apply(.[,cor.lin.cols],1,function(x) median(c(x[7]/x[3],x[17]/x[13])))) %>%
  mutate(spo13.fc.135_v_60=apply(.[,cor.lin.cols],1,function(x) median(c(x[8]/x[3],x[18]/x[13])))) %>%
  mutate(spo13.fc.150_v_60=apply(.[,cor.lin.cols],1,function(x) median(c(x[9]/x[3],x[19]/x[13])))) %>%
  mutate(spo13.fc.165_v_60=apply(.[,cor.lin.cols],1,function(x) median(c(x[10]/x[3],x[20]/x[13])))) %>%
    # spo13 to t75
    mutate(spo13.fc.90_v_75=apply(.[,cor.lin.cols],1,function(x) median(c(x[5]/x[4],x[15]/x[14])))) %>%
  mutate(spo13.fc.105_v_75=apply(.[,cor.lin.cols],1,function(x) median(c(x[6]/x[4],x[16]/x[14])))) %>%
  mutate(spo13.fc.120_v_75=apply(.[,cor.lin.cols],1,function(x) median(c(x[7]/x[4],x[17]/x[14])))) %>%
  mutate(spo13.fc.135_v_75=apply(.[,cor.lin.cols],1,function(x) median(c(x[8]/x[4],x[18]/x[14])))) %>%
  mutate(spo13.fc.150_v_75=apply(.[,cor.lin.cols],1,function(x) median(c(x[9]/x[4],x[19]/x[14])))) %>%
  mutate(spo13.fc.165_v_75=apply(.[,cor.lin.cols],1,function(x) median(c(x[10]/x[4],x[20]/x[14])))) %>%
    # comparisons WT to t60
  mutate(WT.fc.75_v_60=apply(.[,cor.lin.cols],1,function(x) median(c(x[24]/x[23],x[34]/x[33])))) %>%
  mutate(WT.fc.90_v_60=apply(.[,cor.lin.cols],1,function(x) median(c(x[25]/x[23],x[35]/x[33])))) %>%
  mutate(WT.fc.105_v_60=apply(.[,cor.lin.cols],1,function(x) median(c(x[26]/x[23],x[36]/x[33])))) %>%
  mutate(WT.fc.120_v_60=apply(.[,cor.lin.cols],1,function(x) median(c(x[27]/x[23],x[37]/x[33])))) %>%
  mutate(WT.fc.135_v_60=apply(.[,cor.lin.cols],1,function(x) median(c(x[28]/x[23],x[38]/x[33])))) %>%
  mutate(WT.fc.150_v_60=apply(.[,cor.lin.cols],1,function(x) median(c(x[29]/x[23],x[39]/x[33])))) %>%
  mutate(WT.fc.165_v_60=apply(.[,cor.lin.cols],1,function(x) median(c(x[30]/x[23],x[40]/x[33])))) %>%
    # WT to t75
  mutate(WT.fc.90_v_75=apply(.[,cor.lin.cols],1,function(x) median(c(x[25]/x[24],x[35]/x[34])))) %>%
  mutate(WT.fc.105_v_75=apply(.[,cor.lin.cols],1,function(x) median(c(x[26]/x[24],x[36]/x[34])))) %>%
  mutate(WT.fc.120_v_75=apply(.[,cor.lin.cols],1,function(x) median(c(x[27]/x[24],x[37]/x[34])))) %>%
  mutate(WT.fc.135_v_75=apply(.[,cor.lin.cols],1,function(x) median(c(x[28]/x[24],x[38]/x[34])))) %>%
  mutate(WT.fc.150_v_75=apply(.[,cor.lin.cols],1,function(x) median(c(x[29]/x[24],x[39]/x[34])))) %>%
  mutate(WT.fc.165_v_75=apply(.[,cor.lin.cols],1,function(x) median(c(x[30]/x[24],x[40]/x[34])))) %>%
    # spo13 anaphase II comparisons
    # 135 to 90 -- anphase II v anaphase I
    # 135 to 105 - anaphase II v metaphase II (early)
    # 135 to 120 - anaphase II v metaphase II
mutate(spo13.fc.135_v_90=apply(.[,cor.lin.cols],1,function(x) median(c(x[8]/x[5],x[18]/x[15])))) %>%
mutate(spo13.fc.135_v_105=apply(.[,cor.lin.cols],1,function(x) median(c(x[8]/x[6],x[18]/x[16])))) %>%
mutate(spo13.fc.135_v_120=apply(.[,cor.lin.cols],1,function(x) median(c(x[8]/x[7],x[18]/x[17])))) %>%
    # WT anaphase II comparisons
    mutate(WT.fc.135_v_90=apply(.[,cor.lin.cols],1,function(x) median(c(x[28]/x[25],x[38]/x[35])))) %>%
mutate(WT.fc.135_v_105=apply(.[,cor.lin.cols],1,function(x) median(c(x[28]/x[26],x[38]/x[36])))) %>%
mutate(WT.fc.135_v_120=apply(.[,cor.lin.cols],1,function(x) median(c(x[28]/x[27],x[38]/x[37])))) %>%
 # spo13 metaphase II comparisons
    mutate(spo13.fc.150_v_120=apply(.[,cor.lin.cols],1,function(x) median(c(x[9]/x[7],x[19]/x[17])))) %>%
    mutate(spo13.fc.165_v_120=apply(.[,cor.lin.cols],1,function(x) median(c(x[10]/x[7],x[20]/x[17])))) %>%
    # metaphase II comparisons
mutate(WT.fc.150_v_120=apply(.[,cor.lin.cols],1,function(x) median(c(x[29]/x[27],x[39]/x[37])))) %>%
mutate(WT.fc.165_v_120=apply(.[,cor.lin.cols],1,function(x) median(c(x[30]/x[27],x[40]/x[37]))))
    

# cor.lin.cols[c(8,18)]    # 135 spo13
# cor.lin.cols[c(28,38)] #135 WT
# cor.lin.cols[c(5,15)] # spo13 90
# cor.lin.cols[c(6,16)] # spo13 105
# cor.lin.cols[c(7,17)] # spo13 120
# cor.lin.cols[c(25,35)] # WT 90
# cor.lin.cols[c(26,36)] # WT 105
# cor.lin.cols[c(27,37)] # WT 120

# phosphos fold changes
phosphos<-phosphos %>%
 mutate(spo13.fc.45_v_0=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[2]/x[1],x[12]/x[11])))) %>%
  mutate(spo13.fc.60_v_0=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[3]/x[1],x[13]/x[11])))) %>%
  mutate(spo13.fc.75_v_0=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[4]/x[1],x[14]/x[11])))) %>%
  mutate(spo13.fc.90_v_0=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[5]/x[1],x[15]/x[11])))) %>%
  mutate(spo13.fc.105_v_0=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[6]/x[1],x[16]/x[11])))) %>%
  mutate(spo13.fc.120_v_0=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[7]/x[1],x[17]/x[11])))) %>%
  mutate(spo13.fc.135_v_0=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[8]/x[1],x[18]/x[11])))) %>%
  mutate(spo13.fc.150_v_0=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[9]/x[1],x[19]/x[11])))) %>%
  mutate(spo13.fc.165_v_0=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[10]/x[1],x[20]/x[11])))) %>%
  mutate(WT.fc.45_v_0=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[22]/x[21],x[32]/x[31])))) %>%
  mutate(WT.fc.60_v_0=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[23]/x[21],x[33]/x[31])))) %>%   
  mutate(WT.fc.75_v_0=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[24]/x[21],x[34]/x[31])))) %>%
  mutate(WT.fc.90_v_0=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[25]/x[21],x[35]/x[31])))) %>%
  mutate(WT.fc.105_v_0=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[26]/x[21],x[36]/x[31])))) %>%
  mutate(WT.fc.120_v_0=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[27]/x[21],x[37]/x[31])))) %>%
  mutate(WT.fc.135_v_0=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[28]/x[21],x[38]/x[31])))) %>%
  mutate(WT.fc.150_v_0=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[29]/x[21],x[39]/x[31])))) %>%
  mutate(WT.fc.165_v_0=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[30]/x[21],x[40]/x[31])))) %>%
mutate(WT.spo13.fc.0_v_0=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[21],x[31]))/median(c(x[1],x[11])))) %>%
  mutate(WT.spo13.fc.45_v_45=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[22],x[32]))/median(c(x[2],x[12])))) %>%
  mutate(WT.spo13.fc.60_v_60=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[23],x[33]))/median(c(x[3],x[13])))) %>%
  mutate(WT.spo13.fc.75_v_75=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[24],x[34]))/median(c(x[4],x[14])))) %>%
  mutate(WT.spo13.fc.90_v_90=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[25],x[35]))/median(c(x[5],x[15])))) %>%
  mutate(WT.spo13.fc.105_v_105=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[26],x[36]))/median(c(x[6],x[16])))) %>%
  mutate(WT.spo13.fc.120_v_120=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[27],x[37]))/median(c(x[7],x[17])))) %>%
  mutate(WT.spo13.fc.135_v_135=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[28],x[38]))/median(c(x[8],x[18]))))  %>%
  mutate(WT.spo13.fc.150_v_150=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[29],x[39]))/median(c(x[9],x[19])))) %>%
  mutate(WT.spo13.fc.165_v_165=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[30],x[40]))/median(c(x[10],x[20])))) %>%
      # comparisons spo13 to t60
  mutate(spo13.fc.75_v_60=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[4]/x[3],x[14]/x[13])))) %>%
  mutate(spo13.fc.90_v_60=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[5]/x[3],x[15]/x[13])))) %>%
  mutate(spo13.fc.105_v_60=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[6]/x[3],x[16]/x[13])))) %>%
  mutate(spo13.fc.120_v_60=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[7]/x[3],x[17]/x[13])))) %>%
  mutate(spo13.fc.135_v_60=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[8]/x[3],x[18]/x[13])))) %>%
  mutate(spo13.fc.150_v_60=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[9]/x[3],x[19]/x[13])))) %>%
  mutate(spo13.fc.165_v_60=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[10]/x[3],x[20]/x[13])))) %>%
    # spo13 to t75
    mutate(spo13.fc.90_v_75=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[5]/x[4],x[15]/x[14])))) %>%
  mutate(spo13.fc.105_v_75=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[6]/x[4],x[16]/x[14])))) %>%
  mutate(spo13.fc.120_v_75=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[7]/x[4],x[17]/x[14])))) %>%
  mutate(spo13.fc.135_v_75=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[8]/x[4],x[18]/x[14])))) %>%
  mutate(spo13.fc.150_v_75=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[9]/x[4],x[19]/x[14])))) %>%
  mutate(spo13.fc.165_v_75=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[10]/x[4],x[20]/x[14])))) %>%
    # comparisons WT to t60
  mutate(WT.fc.75_v_60=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[24]/x[23],x[34]/x[33])))) %>%
  mutate(WT.fc.90_v_60=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[25]/x[23],x[35]/x[33])))) %>%
  mutate(WT.fc.105_v_60=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[26]/x[23],x[36]/x[33])))) %>%
  mutate(WT.fc.120_v_60=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[27]/x[23],x[37]/x[33])))) %>%
  mutate(WT.fc.135_v_60=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[28]/x[23],x[38]/x[33])))) %>%
  mutate(WT.fc.150_v_60=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[29]/x[23],x[39]/x[33])))) %>%
  mutate(WT.fc.165_v_60=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[30]/x[23],x[40]/x[33])))) %>%
    # WT to t75
  mutate(WT.fc.90_v_75=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[25]/x[24],x[35]/x[34])))) %>%
  mutate(WT.fc.105_v_75=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[26]/x[24],x[36]/x[34])))) %>%
  mutate(WT.fc.120_v_75=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[27]/x[24],x[37]/x[34])))) %>%
  mutate(WT.fc.135_v_75=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[28]/x[24],x[38]/x[34])))) %>%
  mutate(WT.fc.150_v_75=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[29]/x[24],x[39]/x[34])))) %>%
  mutate(WT.fc.165_v_75=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[30]/x[24],x[40]/x[34])))) %>%
# spo13 anaphase II comparisons
mutate(spo13.fc.135_v_90=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[8]/x[5],x[18]/x[15])))) %>%
mutate(spo13.fc.135_v_105=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[8]/x[6],x[18]/x[16])))) %>%
mutate(spo13.fc.135_v_120=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[8]/x[7],x[18]/x[17])))) %>%
    # WT anaphase II comparisons
    mutate(WT.fc.135_v_90=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[28]/x[25],x[38]/x[35])))) %>%
mutate(WT.fc.135_v_105=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[28]/x[26],x[38]/x[36])))) %>%
mutate(WT.fc.135_v_120=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[28]/x[27],x[38]/x[37])))) %>%
    # spo13 metaphase II comparisons
#mutate(spo13.fc.135_v_120=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[8]/x[7],x[18]/x[17])))) %>%
    mutate(spo13.fc.150_v_120=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[9]/x[7],x[19]/x[17])))) %>%
    mutate(spo13.fc.165_v_120=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[10]/x[7],x[20]/x[17])))) %>%
    # metaphase II comparisons
#mutate(WT.fc.135_v_120=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[28]/x[27],x[38]/x[37])))) %>%
mutate(WT.fc.150_v_120=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[29]/x[27],x[39]/x[37])))) %>%
mutate(WT.fc.165_v_120=apply(.[,phos.cor.lin.cols],1,function(x) median(c(x[30]/x[27],x[40]/x[37]))))

```

# t.tests
```{r,echo=F,message=F,warning=F}

alltimes<-c("0","45","60","75","90","105","120","135","150","165")

# # t.tests must be done in for-loop
pvals<-paste0("pval.",alltimes[2:10])

# all comparisons are to t0
# use cor.lin.cols to compare between different TMT batches
for (i in 1:length(pvals)){
    protg[,paste0("spo13.",pvals)[i]]<-as.data.frame(apply(protg[,cor.lin.cols[1:20]], 1, function(x){
        ifelse(all(is.finite(x)),t.test(x[c(1,11)],x[c(i+1,i+11)])$p.value,NA)
    }))
}
for (i in 1:length(pvals)){
    protg[,paste0("WT.",pvals)[i]]<-as.data.frame(apply(protg[,cor.lin.cols[21:40]], 1, function(x){
        ifelse(all(is.finite(x)),t.test(x[c(1,11)],x[c(i+1,i+11)])$p.value,NA)
    }))
}

# comparisons to t60 and t75
# only compare to tiempoints after t60 for the loop
#length(pvals[3:9])
#
#t60 t.tests
# testss compared with t75 (4,14) onward
for (i in 1:length(pvals[3:9])){
    protg[,paste0("spo13.t60.",pvals[3:9])[i]]<-as.data.frame(apply(protg[,cor.lin.cols[1:20]], 1, function(x){
        ifelse(all(is.finite(x)),t.test(x[c(3,13)],x[c(i+3,i+13)])$p.value,NA)
        #x[c(3,13)] # t60
    }))
}
for (i in 1:length(pvals[3:9])){
    protg[,paste0("WT.t60.",pvals[3:9])[i]]<-as.data.frame(apply(protg[,cor.lin.cols[21:40]], 1, function(x){
        ifelse(all(is.finite(x)),t.test(x[c(3,13)],x[c(i+3,i+13)])$p.value,NA)
        #x[c(3,13)] # t60
    }))
}

# t75 t.tests
# tests comparied with t90 (5,15) onward
for (i in 1:length(pvals[4:9])){
    protg[,paste0("spo13.t75.",pvals[4:9])[i]]<-as.data.frame(apply(protg[,cor.lin.cols[1:20]], 1, function(x){
        ifelse(all(is.finite(x)),t.test(x[c(4,14)],x[c(i+4,i+14)])$p.value,NA)
        #x[c(4,14)] # t75
    }))
}
for (i in 1:length(pvals[4:9])){
    protg[,paste0("WT.t75.",pvals[4:9])[i]]<-as.data.frame(apply(protg[,cor.lin.cols[21:40]], 1, function(x){
        ifelse(all(is.finite(x)),t.test(x[c(4,14)],x[c(i+4,i+14)])$p.value,NA)
        #x[c(4,14)] # t75
    }))
}

# anaphase II t.tests
# 3 tests wanted:
# 135 v 90
# 135 v 105
# 135 v 120
# pvals[4:9]
# paste0("WT.t135.",pvals[4:6])
# cor.lin.cols[c(7,17)]
# length(pvals[4:6])
# 5,15 #90
# 6,16 #105
# 7,17 #120

for (i in 1:length(pvals[4:6])){
    protg[,paste0("spo13.t135.",pvals[4:6])[i]]<-as.data.frame(apply(protg[,cor.lin.cols[1:20]], 1, function(x){
        ifelse(all(is.finite(x)),t.test(x[c(8,18)],x[c(i+4,i+14)])$p.value,NA)
        #x[c(8,18)] # t135
    }))
}

for (i in 1:length(pvals[4:6])){
    protg[,paste0("WT.t135.",pvals[4:6])[i]]<-as.data.frame(apply(protg[,cor.lin.cols[21:40]], 1, function(x){
        ifelse(all(is.finite(x)),t.test(x[c(8,18)],x[c(i+4,i+14)])$p.value,NA)
        #x[c(8,18)] # t135
    }))
}


# metapahse II t-tests
# 135 v 120 8,18
# 150 v 120 9,19
# 165 v 120 10,20

for (i in 1:length(pvals[7:9])){
    protg[,paste0("spo13.t120.",pvals[7:9])[i]]<-as.data.frame(apply(protg[,cor.lin.cols[1:20]], 1, function(x){
        ifelse(all(is.finite(x)),t.test(x[c(7,17)],x[c(i+7,i+17)])$p.value,NA)
        #x[c(7,17)] # t120
    }))
}

for (i in 1:length(pvals[7:9])){
    protg[,paste0("WT.t120.",pvals[7:9])[i]]<-as.data.frame(apply(protg[,cor.lin.cols[21:40]], 1, function(x){
        ifelse(all(is.finite(x)),t.test(x[c(7,17)],x[c(i+7,i+17)])$p.value,NA)
        #x[c(7,17)] # t120
    }))
}


# need 10 pvals for WTvspo13 to include 0v0
WT.spo13.pvals<-c("pval.0",pvals)

# WT v spo13 at each timepoint
for (i in 1:length(WT.spo13.pvals)){
    protg[,paste0("WT.spo13.",WT.spo13.pvals)[i]]<-as.data.frame(apply(protg[,cor.lin.cols], 1, function(x){
        ifelse(all(is.finite(x)),t.test(x[c(i,i+10)],x[c(i+20,i+30)])$p.value,NA)
    }))
}

# phospho- t-tests

for (i in 1:length(pvals)){
    phosphos[,paste0("spo13.",pvals)[i]]<-as.data.frame(apply(phosphos[,phos.cor.lin.cols[1:20]], 1, function(x){
        ifelse(all(is.finite(x))&sum(x>0)>=3,t.test(x[c(1,11)],x[c(i+1,i+11)])$p.value,NA)
    }))
} 

for (i in 1:length(pvals)){
    phosphos[,paste0("WT.",pvals)[i]]<-as.data.frame(apply(phosphos[,phos.cor.lin.cols[21:40]], 1, function(x){
        ifelse(all(is.finite(x))&sum(x>0)>=3,t.test(x[c(1,11)],x[c(i+1,i+11)])$p.value,NA)
    }))
} 

#t60 t.tests

for (i in 1:length(pvals[3:9])){
    phosphos[,paste0("spo13.t60.",pvals[3:9])[i]]<-as.data.frame(apply(phosphos[,phos.cor.lin.cols[1:20]], 1, function(x){
        ifelse(all(is.finite(x)),t.test(x[c(3,13)],x[c(i+3,i+13)])$p.value,NA)
        #x[c(3,13)] # t60
    }))
}
for (i in 1:length(pvals[3:9])){
    phosphos[,paste0("WT.t60.",pvals[3:9])[i]]<-as.data.frame(apply(phosphos[,phos.cor.lin.cols[21:40]], 1, function(x){
        ifelse(all(is.finite(x)),t.test(x[c(3,13)],x[c(i+3,i+13)])$p.value,NA)
        #x[c(3,13)] # t60
    }))
}

# t75 t.tests
for (i in 1:length(pvals[4:9])){
    phosphos[,paste0("spo13.t75.",pvals[4:9])[i]]<-as.data.frame(apply(phosphos[,phos.cor.lin.cols[1:20]], 1, function(x){
        ifelse(all(is.finite(x)),t.test(x[c(4,14)],x[c(i+4,i+14)])$p.value,NA)
        #x[c(4,14)] # t75
    }))
}
for (i in 1:length(pvals[4:9])){
    phosphos[,paste0("WT.t75.",pvals[4:9])[i]]<-as.data.frame(apply(phosphos[,phos.cor.lin.cols[21:40]], 1, function(x){
        ifelse(all(is.finite(x)),t.test(x[c(4,14)],x[c(i+4,i+14)])$p.value,NA)
        #x[c(4,14)] # t75
    }))
}

# anapahse II 135 tests
for (i in 1:length(pvals[4:6])){
    phosphos[,paste0("spo13.t135.",pvals[4:6])[i]]<-as.data.frame(apply(phosphos[,phos.cor.lin.cols[1:20]], 1, function(x){
        ifelse(all(is.finite(x)),t.test(x[c(8,18)],x[c(i+4,i+14)])$p.value,NA)
        #x[c(8,18)] # t135
    }))
}

for (i in 1:length(pvals[4:6])){
    phosphos[,paste0("WT.t135.",pvals[4:6])[i]]<-as.data.frame(apply(phosphos[,phos.cor.lin.cols[21:40]], 1, function(x){
        ifelse(all(is.finite(x)),t.test(x[c(8,18)],x[c(i+4,i+14)])$p.value,NA)
        #x[c(8,18)] # t135
    }))
}

# metapahse II t-tests
# 135 v 120 8,18
# 150 v 120 9,19
# 165 v 120 10,20

for (i in 1:length(pvals[7:9])){
    phosphos[,paste0("spo13.t120.",pvals[7:9])[i]]<-as.data.frame(apply(phosphos[,phos.cor.lin.cols[1:20]], 1, function(x){
        ifelse(all(is.finite(x)),t.test(x[c(7,17)],x[c(i+7,i+17)])$p.value,NA)
        #x[c(7,17)] # t120
    }))
}

for (i in 1:length(pvals[7:9])){
    phosphos[,paste0("WT.t120.",pvals[7:9])[i]]<-as.data.frame(apply(phosphos[,phos.cor.lin.cols[21:40]], 1, function(x){
        ifelse(all(is.finite(x)),t.test(x[c(7,17)],x[c(i+7,i+17)])$p.value,NA)
        #x[c(7,17)] # t120
    }))
}

# WT v spo13 at each timepoint
for (i in 1:length(WT.spo13.pvals)){
    phosphos[,paste0("WT.spo13.",WT.spo13.pvals)[i]]<-as.data.frame(apply(phosphos[,phos.cor.lin.cols], 1, function(x){
        ifelse(all(is.finite(x))&sum(x>0)>=3,t.test(x[c(i,i+10)],x[c(i+20,i+30)])$p.value,NA)
    }))
}

```


# call significance
```{r,echo=F,message=F,warning=F}

#  add col to indicate whether both fc and pval are sig

protg<-protg %>%
    #WT
   mutate(WT.sig.45_v_0=ifelse(WT.fc.45_v_0>1.5 & WT.pval.45 < 0.05,"INCR",ifelse(WT.fc.45_v_0<(1/1.5) & WT.pval.45 < 0.05,"DECR","NC"))) %>%
   mutate(WT.sig.60_v_0=ifelse(WT.fc.60_v_0>1.5 & WT.pval.60 < 0.05,"INCR",ifelse(WT.fc.60_v_0<(1/1.5) & WT.pval.60 < 0.05,"DECR","NC"))) %>%
  mutate(WT.sig.75_v_0=ifelse(WT.fc.75_v_0>1.5 & WT.pval.75 < 0.05,"INCR",ifelse(WT.fc.75_v_0<(1/1.5) & WT.pval.75 < 0.05,"DECR","NC"))) %>%
  mutate(WT.sig.90_v_0=ifelse(WT.fc.90_v_0>1.5 & WT.pval.90 < 0.05,"INCR",ifelse(WT.fc.90_v_0<(1/1.5) & WT.pval.90 < 0.05,"DECR","NC"))) %>%
  mutate(WT.sig.105_v_0=ifelse(WT.fc.105_v_0>1.5 & WT.pval.105 < 0.05,"INCR",ifelse(WT.fc.105_v_0<(1/1.5) & WT.pval.105 < 0.05,"DECR","NC"))) %>%
  mutate(WT.sig.120_v_0=ifelse(WT.fc.120_v_0>1.5 & WT.pval.120 < 0.05,"INCR",ifelse(WT.fc.120_v_0<(1/1.5) & WT.pval.120 < 0.05,"DECR","NC"))) %>%
  mutate(WT.sig.135_v_0=ifelse(WT.fc.135_v_0>1.5 & WT.pval.135 < 0.05,"INCR",ifelse(WT.fc.135_v_0<(1/1.5) & WT.pval.135 < 0.05,"DECR","NC"))) %>%
  mutate(WT.sig.150_v_0=ifelse(WT.fc.150_v_0>1.5 & WT.pval.150 < 0.05,"INCR",ifelse(WT.fc.150_v_0<(1/1.5) & WT.pval.150 < 0.05,"DECR","NC"))) %>%
  mutate(WT.sig.165_v_0=ifelse(WT.fc.165_v_0>1.5 & WT.pval.165 < 0.05,"INCR",ifelse(WT.fc.165_v_0<(1/1.5) & WT.pval.165 < 0.05,"DECR","NC"))) %>%
  # spo13
 mutate(spo13.sig.45_v_0=ifelse(spo13.fc.45_v_0>1.5 & spo13.pval.45 < 0.05,"INCR",ifelse(spo13.fc.45_v_0<(1/1.5) & spo13.pval.45 < 0.05,"DECR","NC"))) %>%
   mutate(spo13.sig.60_v_0=ifelse(spo13.fc.60_v_0>1.5 & spo13.pval.60 < 0.05,"INCR",ifelse(spo13.fc.60_v_0<(1/1.5) & spo13.pval.60 < 0.05,"DECR","NC"))) %>%
  mutate(spo13.sig.75_v_0=ifelse(spo13.fc.75_v_0>1.5 & spo13.pval.75 < 0.05,"INCR",ifelse(spo13.fc.75_v_0<(1/1.5) & spo13.pval.75 < 0.05,"DECR","NC"))) %>%
  mutate(spo13.sig.90_v_0=ifelse(spo13.fc.90_v_0>1.5 & spo13.pval.90 < 0.05,"INCR",ifelse(spo13.fc.90_v_0<(1/1.5) & spo13.pval.90 < 0.05,"DECR","NC"))) %>%
  mutate(spo13.sig.105_v_0=ifelse(spo13.fc.105_v_0>1.5 & spo13.pval.105 < 0.05,"INCR",ifelse(spo13.fc.105_v_0<(1/1.5) & spo13.pval.105 < 0.05,"DECR","NC"))) %>%
  mutate(spo13.sig.120_v_0=ifelse(spo13.fc.120_v_0>1.5 & spo13.pval.120 < 0.05,"INCR",ifelse(spo13.fc.120_v_0<(1/1.5) & spo13.pval.120 < 0.05,"DECR","NC"))) %>%
  mutate(spo13.sig.135_v_0=ifelse(spo13.fc.135_v_0>1.5 & spo13.pval.135 < 0.05,"INCR",ifelse(spo13.fc.135_v_0<(1/1.5) & spo13.pval.135 < 0.05,"DECR","NC"))) %>%
  mutate(spo13.sig.150_v_0=ifelse(spo13.fc.150_v_0>1.5 & spo13.pval.150 < 0.05,"INCR",ifelse(spo13.fc.150_v_0<(1/1.5) & spo13.pval.150 < 0.05,"DECR","NC"))) %>%
  mutate(spo13.sig.165_v_0=ifelse(spo13.fc.165_v_0>1.5 & spo13.pval.165 < 0.05,"INCR",ifelse(spo13.fc.165_v_0<(1/1.5) & spo13.pval.165 < 0.05,"DECR","NC"))) %>%
  # WT v spo13
   mutate(WT.spo13.sig.0_v_0=ifelse(WT.spo13.fc.0_v_0>1.5 & WT.spo13.pval.0 < 0.05,"INCR",ifelse(WT.spo13.fc.0_v_0<(1/1.5) & WT.spo13.pval.0 < 0.05,"DECR","NC"))) %>%
  mutate(WT.spo13.sig.45_v_45=ifelse(WT.spo13.fc.45_v_45>1.5 & WT.spo13.pval.45 < 0.05,"INCR",ifelse(WT.spo13.fc.45_v_45<(1/1.5) & WT.spo13.pval.45 < 0.05,"DECR","NC"))) %>%
   mutate(WT.spo13.sig.60_v_60=ifelse(WT.spo13.fc.60_v_60>1.5 & WT.spo13.pval.60 < 0.05,"INCR",ifelse(WT.spo13.fc.60_v_60<(1/1.5) & WT.spo13.pval.60 < 0.05,"DECR","NC"))) %>%
  mutate(WT.spo13.sig.75_v_75=ifelse(WT.spo13.fc.75_v_75>1.5 & WT.spo13.pval.75 < 0.05,"INCR",ifelse(WT.spo13.fc.75_v_75<(1/1.5) & WT.spo13.pval.75 < 0.05,"DECR","NC"))) %>%
  mutate(WT.spo13.sig.90_v_90=ifelse(WT.spo13.fc.90_v_90>1.5 & WT.spo13.pval.90 < 0.05,"INCR",ifelse(WT.spo13.fc.90_v_90<(1/1.5) & WT.spo13.pval.90 < 0.05,"DECR","NC"))) %>%
  mutate(WT.spo13.sig.105_v_105=ifelse(WT.spo13.fc.105_v_105>1.5 & WT.spo13.pval.105 < 0.05,"INCR",ifelse(WT.spo13.fc.105_v_105<(1/1.5) & WT.spo13.pval.105 < 0.05,"DECR","NC"))) %>%
  mutate(WT.spo13.sig.120_v_120=ifelse(WT.spo13.fc.120_v_120>1.5 & WT.spo13.pval.120 < 0.05,"INCR",ifelse(WT.spo13.fc.120_v_120<(1/1.5) & WT.spo13.pval.120 < 0.05,"DECR","NC"))) %>%
  mutate(WT.spo13.sig.135_v_135=ifelse(WT.spo13.fc.135_v_135>1.5 & WT.spo13.pval.135 < 0.05,"INCR",ifelse(WT.spo13.fc.135_v_135<(1/1.5) & WT.spo13.pval.135 < 0.05,"DECR","NC"))) %>%
  mutate(WT.spo13.sig.150_v_150=ifelse(WT.spo13.fc.150_v_150>1.5 & WT.spo13.pval.150 < 0.05,"INCR",ifelse(WT.spo13.fc.150_v_150<(1/1.5) & WT.spo13.pval.150 < 0.05,"DECR","NC"))) %>%
  mutate(WT.spo13.sig.165_v_165=ifelse(WT.spo13.fc.165_v_165>1.5 & WT.spo13.pval.165< 0.05,"INCR",ifelse(WT.spo13.fc.165_v_165<(1/1.5) & WT.spo13.pval.165 < 0.05,"DECR","NC"))) %>%
    # WT metaphase I comparisons
    # WT t60 comparisons
    mutate(WT.sig.75_v_60=ifelse(WT.fc.75_v_60>1.5 & WT.t60.pval.75 < 0.05,"INCR",ifelse(WT.fc.75_v_60<(1/1.5) & WT.t60.pval.75 < 0.05,"DECR","NC"))) %>%
    mutate(WT.sig.90_v_60=ifelse(WT.fc.90_v_60>1.5 & WT.t60.pval.90 < 0.05,"INCR",ifelse(WT.fc.90_v_60<(1/1.5) & WT.t60.pval.90 < 0.05,"DECR","NC"))) %>%
    mutate(WT.sig.105_v_60=ifelse(WT.fc.105_v_60>1.5 & WT.t60.pval.105 < 0.05,"INCR",ifelse(WT.fc.105_v_60<(1/1.5) & WT.t60.pval.105 < 0.05,"DECR","NC"))) %>%
    mutate(WT.sig.120_v_60=ifelse(WT.fc.120_v_60>1.5 & WT.t60.pval.120 < 0.05,"INCR",ifelse(WT.fc.120_v_60<(1/1.5) & WT.t60.pval.120 < 0.05,"DECR","NC"))) %>%
    mutate(WT.sig.135_v_60=ifelse(WT.fc.135_v_60>1.5 & WT.t60.pval.135 < 0.05,"INCR",ifelse(WT.fc.135_v_60<(1/1.5) & WT.t60.pval.135 < 0.05,"DECR","NC"))) %>%
    mutate(WT.sig.150_v_60=ifelse(WT.fc.150_v_60>1.5 & WT.t60.pval.150 < 0.05,"INCR",ifelse(WT.fc.150_v_60<(1/1.5) & WT.t60.pval.150 < 0.05,"DECR","NC"))) %>%
    mutate(WT.sig.165_v_60=ifelse(WT.fc.165_v_60>1.5 & WT.t60.pval.165 < 0.05,"INCR",ifelse(WT.fc.165_v_60<(1/1.5) & WT.t60.pval.165 < 0.05,"DECR","NC"))) %>%
    #WT t75 comparisons
    mutate(WT.sig.90_v_75=ifelse(WT.fc.90_v_75>1.5 & WT.t75.pval.90 < 0.05,"INCR",ifelse(WT.fc.90_v_75<(1/1.5) & WT.t75.pval.90 < 0.05,"DECR","NC"))) %>%
    mutate(WT.sig.105_v_75=ifelse(WT.fc.105_v_75>1.5 & WT.t75.pval.105 < 0.05,"INCR",ifelse(WT.fc.105_v_75<(1/1.5) & WT.t75.pval.105 < 0.05,"DECR","NC"))) %>%
    mutate(WT.sig.120_v_75=ifelse(WT.fc.120_v_75>1.5 & WT.t75.pval.120 < 0.05,"INCR",ifelse(WT.fc.120_v_75<(1/1.5) & WT.t75.pval.120 < 0.05,"DECR","NC"))) %>%
    mutate(WT.sig.135_v_75=ifelse(WT.fc.135_v_75>1.5 & WT.t75.pval.135 < 0.05,"INCR",ifelse(WT.fc.135_v_75<(1/1.5) & WT.t75.pval.135 < 0.05,"DECR","NC"))) %>%
    mutate(WT.sig.150_v_75=ifelse(WT.fc.150_v_75>1.5 & WT.t75.pval.150 < 0.05,"INCR",ifelse(WT.fc.150_v_75<(1/1.5) & WT.t75.pval.150 < 0.05,"DECR","NC"))) %>%
    mutate(WT.sig.165_v_75=ifelse(WT.fc.165_v_75>1.5 & WT.t75.pval.165 < 0.05,"INCR",ifelse(WT.fc.165_v_75<(1/1.5) & WT.t75.pval.165 < 0.05,"DECR","NC"))) %>%
    # spo13 metaphase I comparisons
    # spo13 t60 comparisons
    mutate(spo13.sig.75_v_60=ifelse(spo13.fc.75_v_60>1.5 & spo13.t60.pval.75 < 0.05,"INCR",ifelse(spo13.fc.75_v_60<(1/1.5) & spo13.t60.pval.75 < 0.05,"DECR","NC"))) %>%
    mutate(spo13.sig.90_v_60=ifelse(spo13.fc.90_v_60>1.5 & spo13.t60.pval.90 < 0.05,"INCR",ifelse(spo13.fc.90_v_60<(1/1.5) & spo13.t60.pval.90 < 0.05,"DECR","NC"))) %>%
    mutate(spo13.sig.105_v_60=ifelse(spo13.fc.105_v_60>1.5 & spo13.t60.pval.105 < 0.05,"INCR",ifelse(spo13.fc.105_v_60<(1/1.5) & spo13.t60.pval.105 < 0.05,"DECR","NC"))) %>%
    mutate(spo13.sig.120_v_60=ifelse(spo13.fc.120_v_60>1.5 & spo13.t60.pval.120 < 0.05,"INCR",ifelse(spo13.fc.120_v_60<(1/1.5) & spo13.t60.pval.120 < 0.05,"DECR","NC"))) %>%
    mutate(spo13.sig.135_v_60=ifelse(spo13.fc.135_v_60>1.5 & spo13.t60.pval.135 < 0.05,"INCR",ifelse(spo13.fc.135_v_60<(1/1.5) & spo13.t60.pval.135 < 0.05,"DECR","NC"))) %>%
    mutate(spo13.sig.150_v_60=ifelse(spo13.fc.150_v_60>1.5 & spo13.t60.pval.150 < 0.05,"INCR",ifelse(spo13.fc.150_v_60<(1/1.5) & spo13.t60.pval.150 < 0.05,"DECR","NC"))) %>%
    mutate(spo13.sig.165_v_60=ifelse(spo13.fc.165_v_60>1.5 & spo13.t60.pval.165 < 0.05,"INCR",ifelse(spo13.fc.165_v_60<(1/1.5) & spo13.t60.pval.165 < 0.05,"DECR","NC"))) %>%
    #spo13 t75 comparisons
    mutate(spo13.sig.90_v_75=ifelse(spo13.fc.90_v_75>1.5 & spo13.t75.pval.90 < 0.05,"INCR",ifelse(spo13.fc.90_v_75<(1/1.5) & spo13.t75.pval.90 < 0.05,"DECR","NC"))) %>%
    mutate(spo13.sig.105_v_75=ifelse(spo13.fc.105_v_75>1.5 & spo13.t75.pval.105 < 0.05,"INCR",ifelse(spo13.fc.105_v_75<(1/1.5) & spo13.t75.pval.105 < 0.05,"DECR","NC"))) %>%
    mutate(spo13.sig.120_v_75=ifelse(spo13.fc.120_v_75>1.5 & spo13.t75.pval.120 < 0.05,"INCR",ifelse(spo13.fc.120_v_75<(1/1.5) & spo13.t75.pval.120 < 0.05,"DECR","NC"))) %>%
    mutate(spo13.sig.135_v_75=ifelse(spo13.fc.135_v_75>1.5 & spo13.t75.pval.135 < 0.05,"INCR",ifelse(spo13.fc.135_v_75<(1/1.5) & spo13.t75.pval.135 < 0.05,"DECR","NC"))) %>%
    mutate(spo13.sig.150_v_75=ifelse(spo13.fc.150_v_75>1.5 & spo13.t75.pval.150 < 0.05,"INCR",ifelse(spo13.fc.150_v_75<(1/1.5) & spo13.t75.pval.150 < 0.05,"DECR","NC"))) %>%
    mutate(spo13.sig.165_v_75=ifelse(spo13.fc.165_v_75>1.5 & spo13.t75.pval.165 < 0.05,"INCR",ifelse(spo13.fc.165_v_75<(1/1.5) & spo13.t75.pval.165 < 0.05,"DECR","NC"))) %>%
    # WT t135 comparisons
     mutate(WT.sig.135_v_90=ifelse(WT.fc.135_v_90>1.5 & WT.t135.pval.90 < 0.05,"INCR",ifelse(WT.fc.135_v_90<(1/1.5) & WT.t135.pval.90 < 0.05,"DECR","NC"))) %>%
    mutate(WT.sig.135_v_105=ifelse(WT.fc.135_v_105>1.5 & WT.t135.pval.105 < 0.05,"INCR",ifelse(WT.fc.135_v_105<(1/1.5) & WT.t135.pval.105 < 0.05,"DECR","NC"))) %>%
    mutate(WT.sig.135_v_120=ifelse(WT.fc.135_v_120>1.5 & WT.t135.pval.120 < 0.05,"INCR",ifelse(WT.fc.135_v_120<(1/1.5) & WT.t135.pval.120 < 0.05,"DECR","NC"))) %>%
    # spo13 t135 comparisons
     mutate(spo13.sig.135_v_90=ifelse(spo13.fc.135_v_90>1.5 & spo13.t135.pval.90 < 0.05,"INCR",ifelse(spo13.fc.135_v_90<(1/1.5) & spo13.t135.pval.90 < 0.05,"DECR","NC"))) %>%
    mutate(spo13.sig.135_v_105=ifelse(spo13.fc.135_v_105>1.5 & spo13.t135.pval.105 < 0.05,"INCR",ifelse(spo13.fc.135_v_105<(1/1.5) & spo13.t135.pval.105 < 0.05,"DECR","NC"))) %>%
    mutate(spo13.sig.135_v_120=ifelse(spo13.fc.135_v_120>1.5 & spo13.t135.pval.120 < 0.05,"INCR",ifelse(spo13.fc.135_v_120<(1/1.5) & spo13.t135.pval.120 < 0.05,"DECR","NC"))) %>%
# WT metaphase II comparisons
    mutate(WT.sig.150_v_120=ifelse(WT.fc.150_v_120>1.5 & WT.t120.pval.150 < 0.05,"INCR",ifelse(WT.fc.150_v_120<(1/1.5) & WT.t120.pval.150 < 0.05,"DECR","NC"))) %>%
    mutate(WT.sig.165_v_120=ifelse(WT.fc.165_v_120>1.5 & WT.t120.pval.165 < 0.05,"INCR",ifelse(WT.fc.165_v_120<(1/1.5) & WT.t120.pval.165 < 0.05,"DECR","NC"))) %>%
   # spo13 metaphase II comparisons     
    mutate(spo13.sig.150_v_120=ifelse(spo13.fc.150_v_120>1.5 & spo13.t120.pval.150 < 0.05,"INCR",ifelse(spo13.fc.150_v_120<(1/1.5) & spo13.t120.pval.150 < 0.05,"DECR","NC"))) %>%
    mutate(spo13.sig.165_v_120=ifelse(spo13.fc.165_v_120>1.5 & spo13.t120.pval.165 < 0.05,"INCR",ifelse(spo13.fc.165_v_120<(1/1.5) & spo13.t120.pval.165 < 0.05,"DECR","NC")))


phosphos<-phosphos %>%
  mutate(WT.sig.45_v_0=ifelse(WT.fc.45_v_0>1.5 & WT.pval.45 < 0.05,"INCR",ifelse(WT.fc.45_v_0<(1/1.5) & WT.pval.45 < 0.05,"DECR","NC"))) %>%
   mutate(WT.sig.60_v_0=ifelse(WT.fc.60_v_0>1.5 & WT.pval.60 < 0.05,"INCR",ifelse(WT.fc.60_v_0<(1/1.5) & WT.pval.60 < 0.05,"DECR","NC"))) %>%
  mutate(WT.sig.75_v_0=ifelse(WT.fc.75_v_0>1.5 & WT.pval.75 < 0.05,"INCR",ifelse(WT.fc.75_v_0<(1/1.5) & WT.pval.75 < 0.05,"DECR","NC"))) %>%
  mutate(WT.sig.90_v_0=ifelse(WT.fc.90_v_0>1.5 & WT.pval.90 < 0.05,"INCR",ifelse(WT.fc.90_v_0<(1/1.5) & WT.pval.90 < 0.05,"DECR","NC"))) %>%
  mutate(WT.sig.105_v_0=ifelse(WT.fc.105_v_0>1.5 & WT.pval.105 < 0.05,"INCR",ifelse(WT.fc.105_v_0<(1/1.5) & WT.pval.105 < 0.05,"DECR","NC"))) %>%
  mutate(WT.sig.120_v_0=ifelse(WT.fc.120_v_0>1.5 & WT.pval.120 < 0.05,"INCR",ifelse(WT.fc.120_v_0<(1/1.5) & WT.pval.120 < 0.05,"DECR","NC"))) %>%
  mutate(WT.sig.135_v_0=ifelse(WT.fc.135_v_0>1.5 & WT.pval.135 < 0.05,"INCR",ifelse(WT.fc.135_v_0<(1/1.5) & WT.pval.135 < 0.05,"DECR","NC"))) %>%
  mutate(WT.sig.150_v_0=ifelse(WT.fc.150_v_0>1.5 & WT.pval.150 < 0.05,"INCR",ifelse(WT.fc.150_v_0<(1/1.5) & WT.pval.150 < 0.05,"DECR","NC"))) %>%
  mutate(WT.sig.165_v_0=ifelse(WT.fc.165_v_0>1.5 & WT.pval.165 < 0.05,"INCR",ifelse(WT.fc.165_v_0<(1/1.5) & WT.pval.165 < 0.05,"DECR","NC"))) %>%
  # spo13
 mutate(spo13.sig.45_v_0=ifelse(spo13.fc.45_v_0>1.5 & spo13.pval.45 < 0.05,"INCR",ifelse(spo13.fc.45_v_0<(1/1.5) & spo13.pval.45 < 0.05,"DECR","NC"))) %>%
   mutate(spo13.sig.60_v_0=ifelse(spo13.fc.60_v_0>1.5 & spo13.pval.60 < 0.05,"INCR",ifelse(spo13.fc.60_v_0<(1/1.5) & spo13.pval.60 < 0.05,"DECR","NC"))) %>%
  mutate(spo13.sig.75_v_0=ifelse(spo13.fc.75_v_0>1.5 & spo13.pval.75 < 0.05,"INCR",ifelse(spo13.fc.75_v_0<(1/1.5) & spo13.pval.75 < 0.05,"DECR","NC"))) %>%
  mutate(spo13.sig.90_v_0=ifelse(spo13.fc.90_v_0>1.5 & spo13.pval.90 < 0.05,"INCR",ifelse(spo13.fc.90_v_0<(1/1.5) & spo13.pval.90 < 0.05,"DECR","NC"))) %>%
  mutate(spo13.sig.105_v_0=ifelse(spo13.fc.105_v_0>1.5 & spo13.pval.105 < 0.05,"INCR",ifelse(spo13.fc.105_v_0<(1/1.5) & spo13.pval.105 < 0.05,"DECR","NC"))) %>%
  mutate(spo13.sig.120_v_0=ifelse(spo13.fc.120_v_0>1.5 & spo13.pval.120 < 0.05,"INCR",ifelse(spo13.fc.120_v_0<(1/1.5) & spo13.pval.120 < 0.05,"DECR","NC"))) %>%
  mutate(spo13.sig.135_v_0=ifelse(spo13.fc.135_v_0>1.5 & spo13.pval.135 < 0.05,"INCR",ifelse(spo13.fc.135_v_0<(1/1.5) & spo13.pval.135 < 0.05,"DECR","NC"))) %>%
  mutate(spo13.sig.150_v_0=ifelse(spo13.fc.150_v_0>1.5 & spo13.pval.150 < 0.05,"INCR",ifelse(spo13.fc.150_v_0<(1/1.5) & spo13.pval.150 < 0.05,"DECR","NC"))) %>%
  mutate(spo13.sig.165_v_0=ifelse(spo13.fc.165_v_0>1.5 & spo13.pval.165 < 0.05,"INCR",ifelse(spo13.fc.165_v_0<(1/1.5) & spo13.pval.165 < 0.05,"DECR","NC"))) %>%
  # WT v spo13
   mutate(WT.spo13.sig.0_v_0=ifelse(WT.spo13.fc.0_v_0>1.5 & WT.spo13.pval.0 < 0.05,"INCR",ifelse(WT.spo13.fc.0_v_0<(1/1.5) & WT.spo13.pval.0 < 0.05,"DECR","NC"))) %>%
  mutate(WT.spo13.sig.45_v_45=ifelse(WT.spo13.fc.45_v_45>1.5 & WT.spo13.pval.45 < 0.05,"INCR",ifelse(WT.spo13.fc.45_v_45<(1/1.5) & WT.spo13.pval.45 < 0.05,"DECR","NC"))) %>%
   mutate(WT.spo13.sig.60_v_60=ifelse(WT.spo13.fc.60_v_60>1.5 & WT.spo13.pval.60 < 0.05,"INCR",ifelse(WT.spo13.fc.60_v_60<(1/1.5) & WT.spo13.pval.60 < 0.05,"DECR","NC"))) %>%
  mutate(WT.spo13.sig.75_v_75=ifelse(WT.spo13.fc.75_v_75>1.5 & WT.spo13.pval.75 < 0.05,"INCR",ifelse(WT.spo13.fc.75_v_75<(1/1.5) & WT.spo13.pval.75 < 0.05,"DECR","NC"))) %>%
  mutate(WT.spo13.sig.90_v_90=ifelse(WT.spo13.fc.90_v_90>1.5 & WT.spo13.pval.90 < 0.05,"INCR",ifelse(WT.spo13.fc.90_v_90<(1/1.5) & WT.spo13.pval.90 < 0.05,"DECR","NC"))) %>%
  mutate(WT.spo13.sig.105_v_105=ifelse(WT.spo13.fc.105_v_105>1.5 & WT.spo13.pval.105 < 0.05,"INCR",ifelse(WT.spo13.fc.105_v_105<(1/1.5) & WT.spo13.pval.105 < 0.05,"DECR","NC"))) %>%
  mutate(WT.spo13.sig.120_v_120=ifelse(WT.spo13.fc.120_v_120>1.5 & WT.spo13.pval.120 < 0.05,"INCR",ifelse(WT.spo13.fc.120_v_120<(1/1.5) & WT.spo13.pval.120 < 0.05,"DECR","NC"))) %>%
  mutate(WT.spo13.sig.135_v_135=ifelse(WT.spo13.fc.135_v_135>1.5 & WT.spo13.pval.135 < 0.05,"INCR",ifelse(WT.spo13.fc.135_v_135<(1/1.5) & WT.spo13.pval.135 < 0.05,"DECR","NC"))) %>%
  mutate(WT.spo13.sig.150_v_150=ifelse(WT.spo13.fc.150_v_150>1.5 & WT.spo13.pval.150 < 0.05,"INCR",ifelse(WT.spo13.fc.150_v_150<(1/1.5) & WT.spo13.pval.150 < 0.05,"DECR","NC"))) %>%
  mutate(WT.spo13.sig.165_v_165=ifelse(WT.spo13.fc.165_v_165>1.5 & WT.spo13.pval.165< 0.05,"INCR",ifelse(WT.spo13.fc.165_v_165<(1/1.5) & WT.spo13.pval.165 < 0.05,"DECR","NC"))) %>%
    # WT metaphase I comparisons
    # WT t60 comparisons
    mutate(WT.sig.75_v_60=ifelse(WT.fc.75_v_60>1.5 & WT.t60.pval.75 < 0.05,"INCR",ifelse(WT.fc.75_v_60<(1/1.5) & WT.t60.pval.75 < 0.05,"DECR","NC"))) %>%
    mutate(WT.sig.90_v_60=ifelse(WT.fc.90_v_60>1.5 & WT.t60.pval.90 < 0.05,"INCR",ifelse(WT.fc.90_v_60<(1/1.5) & WT.t60.pval.90 < 0.05,"DECR","NC"))) %>%
    mutate(WT.sig.105_v_60=ifelse(WT.fc.105_v_60>1.5 & WT.t60.pval.105 < 0.05,"INCR",ifelse(WT.fc.105_v_60<(1/1.5) & WT.t60.pval.105 < 0.05,"DECR","NC"))) %>%
    mutate(WT.sig.120_v_60=ifelse(WT.fc.120_v_60>1.5 & WT.t60.pval.120 < 0.05,"INCR",ifelse(WT.fc.120_v_60<(1/1.5) & WT.t60.pval.120 < 0.05,"DECR","NC"))) %>%
    mutate(WT.sig.135_v_60=ifelse(WT.fc.135_v_60>1.5 & WT.t60.pval.135 < 0.05,"INCR",ifelse(WT.fc.135_v_60<(1/1.5) & WT.t60.pval.135 < 0.05,"DECR","NC"))) %>%
    mutate(WT.sig.150_v_60=ifelse(WT.fc.150_v_60>1.5 & WT.t60.pval.150 < 0.05,"INCR",ifelse(WT.fc.150_v_60<(1/1.5) & WT.t60.pval.150 < 0.05,"DECR","NC"))) %>%
    mutate(WT.sig.165_v_60=ifelse(WT.fc.165_v_60>1.5 & WT.t60.pval.165 < 0.05,"INCR",ifelse(WT.fc.165_v_60<(1/1.5) & WT.t60.pval.165 < 0.05,"DECR","NC"))) %>%
    #WT t75 comparisons
    mutate(WT.sig.90_v_75=ifelse(WT.fc.90_v_75>1.5 & WT.t75.pval.90 < 0.05,"INCR",ifelse(WT.fc.90_v_75<(1/1.5) & WT.t75.pval.90 < 0.05,"DECR","NC"))) %>%
    mutate(WT.sig.105_v_75=ifelse(WT.fc.105_v_75>1.5 & WT.t75.pval.105 < 0.05,"INCR",ifelse(WT.fc.105_v_75<(1/1.5) & WT.t75.pval.105 < 0.05,"DECR","NC"))) %>%
    mutate(WT.sig.120_v_75=ifelse(WT.fc.120_v_75>1.5 & WT.t75.pval.120 < 0.05,"INCR",ifelse(WT.fc.120_v_75<(1/1.5) & WT.t75.pval.120 < 0.05,"DECR","NC"))) %>%
    mutate(WT.sig.135_v_75=ifelse(WT.fc.135_v_75>1.5 & WT.t75.pval.135 < 0.05,"INCR",ifelse(WT.fc.135_v_75<(1/1.5) & WT.t75.pval.135 < 0.05,"DECR","NC"))) %>%
    mutate(WT.sig.150_v_75=ifelse(WT.fc.150_v_75>1.5 & WT.t75.pval.150 < 0.05,"INCR",ifelse(WT.fc.150_v_75<(1/1.5) & WT.t75.pval.150 < 0.05,"DECR","NC"))) %>%
    mutate(WT.sig.165_v_75=ifelse(WT.fc.165_v_75>1.5 & WT.t75.pval.165 < 0.05,"INCR",ifelse(WT.fc.165_v_75<(1/1.5) & WT.t75.pval.165 < 0.05,"DECR","NC"))) %>%
    # spo13 metaphase I comparisons
    # spo13 t60 comparisons
    mutate(spo13.sig.75_v_60=ifelse(spo13.fc.75_v_60>1.5 & spo13.t60.pval.75 < 0.05,"INCR",ifelse(spo13.fc.75_v_60<(1/1.5) & spo13.t60.pval.75 < 0.05,"DECR","NC"))) %>%
    mutate(spo13.sig.90_v_60=ifelse(spo13.fc.90_v_60>1.5 & spo13.t60.pval.90 < 0.05,"INCR",ifelse(spo13.fc.90_v_60<(1/1.5) & spo13.t60.pval.90 < 0.05,"DECR","NC"))) %>%
    mutate(spo13.sig.105_v_60=ifelse(spo13.fc.105_v_60>1.5 & spo13.t60.pval.105 < 0.05,"INCR",ifelse(spo13.fc.105_v_60<(1/1.5) & spo13.t60.pval.105 < 0.05,"DECR","NC"))) %>%
    mutate(spo13.sig.120_v_60=ifelse(spo13.fc.120_v_60>1.5 & spo13.t60.pval.120 < 0.05,"INCR",ifelse(spo13.fc.120_v_60<(1/1.5) & spo13.t60.pval.120 < 0.05,"DECR","NC"))) %>%
    mutate(spo13.sig.135_v_60=ifelse(spo13.fc.135_v_60>1.5 & spo13.t60.pval.135 < 0.05,"INCR",ifelse(spo13.fc.135_v_60<(1/1.5) & spo13.t60.pval.135 < 0.05,"DECR","NC"))) %>%
    mutate(spo13.sig.150_v_60=ifelse(spo13.fc.150_v_60>1.5 & spo13.t60.pval.150 < 0.05,"INCR",ifelse(spo13.fc.150_v_60<(1/1.5) & spo13.t60.pval.150 < 0.05,"DECR","NC"))) %>%
    mutate(spo13.sig.165_v_60=ifelse(spo13.fc.165_v_60>1.5 & spo13.t60.pval.165 < 0.05,"INCR",ifelse(spo13.fc.165_v_60<(1/1.5) & spo13.t60.pval.165 < 0.05,"DECR","NC"))) %>%
    #spo13 t75 comparisons
    mutate(spo13.sig.90_v_75=ifelse(spo13.fc.90_v_75>1.5 & spo13.t75.pval.90 < 0.05,"INCR",ifelse(spo13.fc.90_v_75<(1/1.5) & spo13.t75.pval.90 < 0.05,"DECR","NC"))) %>%
    mutate(spo13.sig.105_v_75=ifelse(spo13.fc.105_v_75>1.5 & spo13.t75.pval.105 < 0.05,"INCR",ifelse(spo13.fc.105_v_75<(1/1.5) & spo13.t75.pval.105 < 0.05,"DECR","NC"))) %>%
    mutate(spo13.sig.120_v_75=ifelse(spo13.fc.120_v_75>1.5 & spo13.t75.pval.120 < 0.05,"INCR",ifelse(spo13.fc.120_v_75<(1/1.5) & spo13.t75.pval.120 < 0.05,"DECR","NC"))) %>%
    mutate(spo13.sig.135_v_75=ifelse(spo13.fc.135_v_75>1.5 & spo13.t75.pval.135 < 0.05,"INCR",ifelse(spo13.fc.135_v_75<(1/1.5) & spo13.t75.pval.135 < 0.05,"DECR","NC"))) %>%
    mutate(spo13.sig.150_v_75=ifelse(spo13.fc.150_v_75>1.5 & spo13.t75.pval.150 < 0.05,"INCR",ifelse(spo13.fc.150_v_75<(1/1.5) & spo13.t75.pval.150 < 0.05,"DECR","NC"))) %>%
    mutate(spo13.sig.165_v_75=ifelse(spo13.fc.165_v_75>1.5 & spo13.t75.pval.165 < 0.05,"INCR",ifelse(spo13.fc.165_v_75<(1/1.5) & spo13.t75.pval.165 < 0.05,"DECR","NC"))) %>%
    # WT t135 comparisons
     mutate(WT.sig.135_v_90=ifelse(WT.fc.135_v_90>1.5 & WT.t135.pval.90 < 0.05,"INCR",ifelse(WT.fc.135_v_90<(1/1.5) & WT.t135.pval.90 < 0.05,"DECR","NC"))) %>%
    mutate(WT.sig.135_v_105=ifelse(WT.fc.135_v_105>1.5 & WT.t135.pval.105 < 0.05,"INCR",ifelse(WT.fc.135_v_105<(1/1.5) & WT.t135.pval.105 < 0.05,"DECR","NC"))) %>%
    mutate(WT.sig.135_v_120=ifelse(WT.fc.135_v_120>1.5 & WT.t135.pval.120 < 0.05,"INCR",ifelse(WT.fc.135_v_120<(1/1.5) & WT.t135.pval.120 < 0.05,"DECR","NC"))) %>%
    # spo13 t135 comparisons
     mutate(spo13.sig.135_v_90=ifelse(spo13.fc.135_v_90>1.5 & spo13.t135.pval.90 < 0.05,"INCR",ifelse(spo13.fc.135_v_90<(1/1.5) & spo13.t135.pval.90 < 0.05,"DECR","NC"))) %>%
    mutate(spo13.sig.135_v_105=ifelse(spo13.fc.135_v_105>1.5 & spo13.t135.pval.105 < 0.05,"INCR",ifelse(spo13.fc.135_v_105<(1/1.5) & spo13.t135.pval.105 < 0.05,"DECR","NC"))) %>%
    mutate(spo13.sig.135_v_120=ifelse(spo13.fc.135_v_120>1.5 & spo13.t135.pval.120 < 0.05,"INCR",ifelse(spo13.fc.135_v_120<(1/1.5) & spo13.t135.pval.120 < 0.05,"DECR","NC"))) %>%
    # WT metaphase II comparisons
    mutate(WT.sig.150_v_120=ifelse(WT.fc.150_v_120>1.5 & WT.t120.pval.150 < 0.05,"INCR",ifelse(WT.fc.150_v_120<(1/1.5) & WT.t120.pval.150 < 0.05,"DECR","NC"))) %>%
    mutate(WT.sig.165_v_120=ifelse(WT.fc.165_v_120>1.5 & WT.t120.pval.165 < 0.05,"INCR",ifelse(WT.fc.165_v_120<(1/1.5) & WT.t120.pval.165 < 0.05,"DECR","NC"))) %>%
   # spo13 metaphase II comparisons     
    mutate(spo13.sig.150_v_120=ifelse(spo13.fc.150_v_120>1.5 & spo13.t120.pval.150 < 0.05,"INCR",ifelse(spo13.fc.150_v_120<(1/1.5) & spo13.t120.pval.150 < 0.05,"DECR","NC"))) %>%
    mutate(spo13.sig.165_v_120=ifelse(spo13.fc.165_v_120>1.5 & spo13.t120.pval.165 < 0.05,"INCR",ifelse(spo13.fc.165_v_120<(1/1.5) & spo13.t120.pval.165 < 0.05,"DECR","NC")))
```


# Number of sig based on Inf fold change
```{r,echo=F,message=F,warning=F}

# checking how much of 'sig' proteins and sites are called sig
# based on 0 v >0 comparison, producing Inf fold change

# WT.sig.cols<-grep("sig",colnames(protg),value=T)[1:9]
# spo13.sig.cols<-grep("sig",colnames(protg),value=T)[10:18]
# WT.spo13.sig.cols<-grep("sig",colnames(protg),value=T)[19:28]
# 
# WT.t0.fc.cols<-grep("WT.fc.(.+)_v_0",colnames(protg),value=T)
# spo13.t0.fc.cols<-grep("^spo13.fc.(.+)_v_0",colnames(protg),value=T)
# 
# WT.spo13.fc.cols<-grep("WT.spo13.fc.(.+)_v_(.+)",colnames(protg),value=T)

# number of proteins called SIG from t0 at any timepoint 
# and which met all sig calls being non-NA requirement (applied later to be called sig)
# and which had a Inf fold change in 1/9 of WT.sig.cols
# because Inf WT.sig.cols are causesd by 0 at t0 and all are vs t0
# it means the whole row of 9 WT.sig.cols are fc=Inf or NA if 0/0. If 0/0 = NA not Inf so would get filtered out.
# 
# 
# WT protein sig from Inf fc
# protg %>%
#     filter(WT.2reps==T) %>%
#     filter(if_any(all_of(WT.sig.cols),function(x) {x=="DECR"|x=="INCR"})) %>%
#     filter(if_all(all_of(WT.sig.cols),function(x) {!is.na(x)})) %>%
#     filter(if_any(all_of(WT.t0.fc.cols), function(x) is.infinite(x))) %>%
#     mutate(time.sig=as.numeric(apply(.[,WT.sig.cols],1,function(x) which({x=="DECR"|x=="INCR"})[1]))) %>%
#     rowwise() %>%
# mutate(fc.at.sig = get(WT.t0.fc.cols[time.sig])) %>%
#     filter(is.infinite(fc.at.sig)) %>% 
#     nrow() #21
# 
# # WT protein total number sig
# protg %>%
#     filter(WT.2reps==T) %>%
#     filter(if_any(all_of(WT.sig.cols),function(x) {x=="DECR"|x=="INCR"})) %>%
#     filter(if_all(all_of(WT.sig.cols),function(x) {!is.na(x)})) %>% nrow() #297

#21/297 #0.07070707 # 7% of WT sig proteins are sig from Inf fold change

# WT phos sig from t0 Inf fc
# phosphos %>%
#     filter(WT.2.detects==T) %>%
#     filter(high.localiz.score==TRUE) %>% 
#     filter(if_any(all_of(WT.sig.cols),function(x) {x=="DECR"|x=="INCR"})) %>%
#     filter(if_all(all_of(WT.sig.cols),function(x) {!is.na(x)})) %>%
#     filter(if_any(all_of(WT.t0.fc.cols), function(x) is.infinite(x))) %>%
#     mutate(time.sig=as.numeric(apply(.[,WT.sig.cols],1,function(x) which({x=="DECR"|x=="INCR"})[1]))) %>%
#     rowwise() %>%
# # mutate(fc.at.sig = get(WT.t0.fc.cols[time.sig])) %>%
# #     filter(is.infinite(fc.at.sig)) %>% 
# #     nrow() #359 WT phos ssig from t0 with Inf FC
# 
# phosphos %>%
#     filter(WT.2.detects==T) %>%
#     filter(high.localiz.score==TRUE) %>% 
#     filter(if_any(all_of(WT.sig.cols),function(x) {x=="DECR"|x=="INCR"})) %>%
#     filter(if_all(all_of(WT.sig.cols),function(x) {!is.na(x)})) %>%
#     nrow() #1233 total WT phos sig
# 
# #359/1233 #0.2911598 
# 
# 
# # WT and spo13 COMBINED
# 
# # PROTEIN
# protg %>%
#     filter(WT.2reps==T & spo13.2reps==T) %>%
#     filter(if_any(all_of(c(WT.sig.cols,spo13.sig.cols)),function(x) {x=="DECR"|x=="INCR"})) %>%
#     filter(if_all(all_of(c(WT.sig.cols,spo13.sig.cols)),function(x) {!is.na(x)})) %>%
#     filter(if_any(all_of(c(WT.t0.fc.cols,spo13.t0.fc.cols)), function(x) is.infinite(x))) %>%
#     mutate(time.sig=as.numeric(apply(.[,c(WT.sig.cols,spo13.sig.cols)],1,function(x) which({x=="DECR"|x=="INCR"})[1]))) %>%
#     rowwise() %>%
#     mutate(fc.at.sig = get(c(WT.t0.fc.cols,spo13.t0.fc.cols)[time.sig])) %>%
#     filter(is.infinite(fc.at.sig)) %>% 
#     nrow() #18
# 
# protg %>%
#     filter(WT.2reps==T & spo13.2reps==T) %>%
#     filter(if_any(all_of(c(WT.sig.cols,spo13.sig.cols)),function(x) {x=="DECR"|x=="INCR"})) %>%
#     filter(if_all(all_of(c(WT.sig.cols,spo13.sig.cols)),function(x) {!is.na(x)})) %>%
#     nrow() # 414 total WT.spo13 protg from t0 sig in either or both strain
# 
# #18/414 # 0.04347826
# 
# # PHOSPHO
# # number of Wt.spo13. phos sig from t0 in either or both based on Inf FC
# phosphos %>%
#     filter(WT.2.detects==T & spo13.2.detects==T) %>%
#       filter(high.localiz.score==TRUE) %>% 
#     filter(if_any(all_of(c(WT.sig.cols,spo13.sig.cols)),function(x) {x=="DECR"|x=="INCR"})) %>%
#     filter(if_all(all_of(c(WT.sig.cols,spo13.sig.cols)),function(x) {!is.na(x)})) %>%
#     filter(if_any(all_of(c(WT.t0.fc.cols,spo13.t0.fc.cols)), function(x) is.infinite(x))) %>%
#     mutate(time.sig=as.numeric(apply(.[,c(WT.sig.cols,spo13.sig.cols)],1,function(x) which({x=="DECR"|x=="INCR"})[1]))) %>%
#     rowwise() %>%
#     mutate(fc.at.sig = get(c(WT.t0.fc.cols,spo13.t0.fc.cols)[time.sig])) %>%
#     filter(is.infinite(fc.at.sig)) %>% 
#     nrow() # 166
# 
# 
# # total number of sig WT.spo13 phos ssites from t0 in either or both
# phosphos %>%
#     filter(WT.2.detects==T & spo13.2.detects==T) %>%
#       filter(high.localiz.score==TRUE) %>% 
#     filter(if_any(all_of(c(WT.sig.cols,spo13.sig.cols)),function(x) {x=="DECR"|x=="INCR"})) %>%
#     filter(if_all(all_of(c(WT.sig.cols,spo13.sig.cols)),function(x) {!is.na(x)})) %>%
#     nrow() # 982 total WT.spo13 phos from t0 sig in either or both strain
#     
# #166/982 # 0.1690428 of phos found in 4 TMT10 tiemcourses called sig from t0 based on inf FC
# 
# 
# # WT v spo13 SIGDIFF AT MATCHED TIMEPOINTS
# 
# # PROTEIN
# 
# # proteins sigdiff between strains based on Inf FC
# protg %>%
#     filter(WT.2reps==T & spo13.2reps==T) %>%
#     filter(if_any(all_of(WT.spo13.sig.cols),function(x) {x=="DECR"|x=="INCR"})) %>%
#     filter(if_all(all_of(WT.spo13.sig.cols),function(x) {!is.na(x)})) %>% 
#     filter(if_any(all_of(WT.spo13.fc.cols), function(x) is.infinite(x))) %>%
#     mutate(time.sig=as.numeric(apply(.[,WT.spo13.sig.cols],1,function(x) which({x=="DECR"|x=="INCR"})[1]))) %>%
#     rowwise() %>%
#     mutate(fc.at.sig = get(WT.spo13.fc.cols[time.sig])) %>%
#     filter(is.infinite(fc.at.sig)) %>% 
#     nrow() # 8
# 
# # TOTAL number of SIG between strains at matched timepoints
# protg %>%
#     filter(WT.2reps==T & spo13.2reps==T) %>%
#     filter(if_any(all_of(WT.spo13.sig.cols),function(x) {x=="DECR"|x=="INCR"})) %>%
#     filter(if_all(all_of(WT.spo13.sig.cols),function(x) {!is.na(x)})) %>% nrow() #50
# 
# 8/50 # 0.16 16% of sigdiff proteins called sig based on Inf FC
# 
# # SIG DIFF SITES
# 
# # proteins sigdiff between strains based on Inf FC
# phosphos %>%
#     filter(WT.2.detects==T & spo13.2.detects==T) %>%
#       filter(high.localiz.score==TRUE) %>% 
#     filter(if_any(all_of(WT.spo13.sig.cols),function(x) {x=="DECR"|x=="INCR"})) %>%
#     filter(if_all(all_of(WT.spo13.sig.cols),function(x) {!is.na(x)})) %>% 
#     filter(if_any(all_of(WT.spo13.fc.cols), function(x) is.infinite(x))) %>%
#     mutate(time.sig=as.numeric(apply(.[,WT.spo13.sig.cols],1,function(x) which({x=="DECR"|x=="INCR"})[1]))) %>%
#     rowwise() %>%
#     mutate(fc.at.sig = get(WT.spo13.fc.cols[time.sig])) %>%
#     filter(is.infinite(fc.at.sig)) %>% 
#     nrow() # 71
# 
# # TOTAL number of SIG between strains at matched timepoints
# phosphos %>%
#     filter(WT.2.detects==T & spo13.2.detects==T) %>%
#       filter(high.localiz.score==TRUE) %>% 
#     filter(if_any(all_of(WT.spo13.sig.cols),function(x) {x=="DECR"|x=="INCR"})) %>%
#     filter(if_all(all_of(WT.spo13.sig.cols),function(x) {!is.na(x)})) %>% nrow() #305
# 
# 71/305 # 0.2327869  ~23% of sigdiff sites based on Inf FC at time called sig


```


# impute zeros in fcs for plotting

```{r,echo=F,message=F,warning=F}

# new columns to hold values with zeros imputed
imp.cor.lin.cols<-paste0("imp.",cor.lin.cols)
imp.phos.cor.lin.cols<-paste0("imp.",phos.cor.lin.cols)

# copy contents of cor.lin.cols to new columns
protg[,imp.cor.lin.cols]<-protg[,cor.lin.cols]
phosphos[,imp.phos.cor.lin.cols]<-phosphos[,phos.cor.lin.cols]

# replace zeros in cor.lin.cols with minimum value of all cor.lin.cols in protg
protg[,imp.cor.lin.cols][protg[,imp.cor.lin.cols]==0]<-min(protg[,cor.lin.cols][protg[,cor.lin.cols]>0])
# minimum protg value is 312.2495

# replace zeros in phos.cor.lin.cols with minimum value of all phos.cor.lin.cols in phosphos
phosphos[,imp.phos.cor.lin.cols][phosphos[,imp.phos.cor.lin.cols]==0]<-min(phosphos[,phos.cor.lin.cols][phosphos[,phos.cor.lin.cols]>0])
# minimum phosphos value is. 0.008987741

```

# imputed fold changes
```{r,echo=F,message=F,warning=F}

# this means fc is linear
protg<-protg %>%
  mutate(spo13.impfc.45_v_0=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[2]/x[1],x[12]/x[11])))) %>%
  mutate(spo13.impfc.60_v_0=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[3]/x[1],x[13]/x[11])))) %>%
  mutate(spo13.impfc.75_v_0=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[4]/x[1],x[14]/x[11])))) %>%
  mutate(spo13.impfc.90_v_0=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[5]/x[1],x[15]/x[11])))) %>%
  mutate(spo13.impfc.105_v_0=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[6]/x[1],x[16]/x[11])))) %>%
  mutate(spo13.impfc.120_v_0=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[7]/x[1],x[17]/x[11])))) %>%
  mutate(spo13.impfc.135_v_0=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[8]/x[1],x[18]/x[11])))) %>%
  mutate(spo13.impfc.150_v_0=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[9]/x[1],x[19]/x[11])))) %>%
  mutate(spo13.impfc.165_v_0=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[10]/x[1],x[20]/x[11])))) %>%
  mutate(WT.impfc.45_v_0=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[22]/x[21],x[32]/x[31])))) %>%
  mutate(WT.impfc.60_v_0=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[23]/x[21],x[33]/x[31])))) %>%   
  mutate(WT.impfc.75_v_0=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[24]/x[21],x[34]/x[31])))) %>%
  mutate(WT.impfc.90_v_0=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[25]/x[21],x[35]/x[31])))) %>%
  mutate(WT.impfc.105_v_0=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[26]/x[21],x[36]/x[31])))) %>%
  mutate(WT.impfc.120_v_0=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[27]/x[21],x[37]/x[31])))) %>%
  mutate(WT.impfc.135_v_0=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[28]/x[21],x[38]/x[31])))) %>%
  mutate(WT.impfc.150_v_0=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[29]/x[21],x[39]/x[31])))) %>%
  mutate(WT.impfc.165_v_0=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[30]/x[21],x[40]/x[31])))) %>%
  mutate(WT.spo13.impfc.0_v_0=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[21],x[31]))/median(c(x[1],x[11])))) %>%
  mutate(WT.spo13.impfc.45_v_45=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[22],x[32]))/median(c(x[2],x[12])))) %>%
  mutate(WT.spo13.impfc.60_v_60=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[23],x[33]))/median(c(x[3],x[13])))) %>%
  mutate(WT.spo13.impfc.75_v_75=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[24],x[34]))/median(c(x[4],x[14])))) %>%
  mutate(WT.spo13.impfc.90_v_90=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[25],x[35]))/median(c(x[5],x[15])))) %>%
  mutate(WT.spo13.impfc.105_v_105=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[26],x[36]))/median(c(x[6],x[16])))) %>%
  mutate(WT.spo13.impfc.120_v_120=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[27],x[37]))/median(c(x[7],x[17])))) %>%
  mutate(WT.spo13.impfc.135_v_135=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[28],x[38]))/median(c(x[8],x[18]))))  %>%
  mutate(WT.spo13.impfc.150_v_150=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[29],x[39]))/median(c(x[9],x[19])))) %>%
  mutate(WT.spo13.impfc.165_v_165=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[30],x[40]))/median(c(x[10],x[20])))) %>%
    # comparisons spo13 to t60
  mutate(spo13.impfc.75_v_60=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[4]/x[3],x[14]/x[13])))) %>%
  mutate(spo13.impfc.90_v_60=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[5]/x[3],x[15]/x[13])))) %>%
  mutate(spo13.impfc.105_v_60=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[6]/x[3],x[16]/x[13])))) %>%
  mutate(spo13.impfc.120_v_60=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[7]/x[3],x[17]/x[13])))) %>%
  mutate(spo13.impfc.135_v_60=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[8]/x[3],x[18]/x[13])))) %>%
  mutate(spo13.impfc.150_v_60=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[9]/x[3],x[19]/x[13])))) %>%
  mutate(spo13.impfc.165_v_60=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[10]/x[3],x[20]/x[13])))) %>%
    # spo13 to t75
    mutate(spo13.impfc.90_v_75=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[5]/x[4],x[15]/x[14])))) %>%
  mutate(spo13.impfc.105_v_75=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[6]/x[4],x[16]/x[14])))) %>%
  mutate(spo13.impfc.120_v_75=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[7]/x[4],x[17]/x[14])))) %>%
  mutate(spo13.impfc.135_v_75=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[8]/x[4],x[18]/x[14])))) %>%
  mutate(spo13.impfc.150_v_75=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[9]/x[4],x[19]/x[14])))) %>%
  mutate(spo13.impfc.165_v_75=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[10]/x[4],x[20]/x[14])))) %>%
    # comparisons WT to t60
  mutate(WT.impfc.75_v_60=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[24]/x[23],x[34]/x[33])))) %>%
  mutate(WT.impfc.90_v_60=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[25]/x[23],x[35]/x[33])))) %>%
  mutate(WT.impfc.105_v_60=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[26]/x[23],x[36]/x[33])))) %>%
  mutate(WT.impfc.120_v_60=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[27]/x[23],x[37]/x[33])))) %>%
  mutate(WT.impfc.135_v_60=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[28]/x[23],x[38]/x[33])))) %>%
  mutate(WT.impfc.150_v_60=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[29]/x[23],x[39]/x[33])))) %>%
  mutate(WT.impfc.165_v_60=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[30]/x[23],x[40]/x[33])))) %>%
    # WT to t75
  mutate(WT.impfc.90_v_75=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[25]/x[24],x[35]/x[34])))) %>%
  mutate(WT.impfc.105_v_75=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[26]/x[24],x[36]/x[34])))) %>%
  mutate(WT.impfc.120_v_75=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[27]/x[24],x[37]/x[34])))) %>%
  mutate(WT.impfc.135_v_75=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[28]/x[24],x[38]/x[34])))) %>%
  mutate(WT.impfc.150_v_75=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[29]/x[24],x[39]/x[34])))) %>%
  mutate(WT.impfc.165_v_75=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[30]/x[24],x[40]/x[34])))) %>%
    #spo13 t135 comparisons
 mutate(spo13.impfc.135_v_90=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[8]/x[5],x[18]/x[15])))) %>%
mutate(spo13.impfc.135_v_105=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[8]/x[6],x[18]/x[16])))) %>%
    mutate(spo13.impfc.135_v_120=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[8]/x[7],x[18]/x[17])))) %>%
    #WT t135 comparisons
    mutate(WT.impfc.135_v_90=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[28]/x[25],x[38]/x[35])))) %>%
mutate(WT.impfc.135_v_105=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[28]/x[26],x[38]/x[36])))) %>%
    mutate(WT.impfc.135_v_120=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[28]/x[27],x[38]/x[37])))) %>%
    # spo13 metaphse II
    mutate(spo13.impfc.150_v_120=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[9]/x[7],x[19]/x[17])))) %>%
    mutate(spo13.impfc.165_v_120=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[10]/x[7],x[20]/x[17])))) %>%
    # WT metaphase II
     mutate(WT.impfc.150_v_120=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[29]/x[27],x[39]/x[37])))) %>%
    mutate(WT.impfc.165_v_120=apply(.[,imp.cor.lin.cols],1,function(x) median(c(x[30]/x[27],x[40]/x[37]))))


# phosphos fold changes
phosphos<-phosphos %>%
 mutate(spo13.impfc.45_v_0=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[2]/x[1],x[12]/x[11])))) %>%
  mutate(spo13.impfc.60_v_0=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[3]/x[1],x[13]/x[11])))) %>%
  mutate(spo13.impfc.75_v_0=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[4]/x[1],x[14]/x[11])))) %>%
  mutate(spo13.impfc.90_v_0=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[5]/x[1],x[15]/x[11])))) %>%
  mutate(spo13.impfc.105_v_0=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[6]/x[1],x[16]/x[11])))) %>%
  mutate(spo13.impfc.120_v_0=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[7]/x[1],x[17]/x[11])))) %>%
  mutate(spo13.impfc.135_v_0=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[8]/x[1],x[18]/x[11])))) %>%
  mutate(spo13.impfc.150_v_0=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[9]/x[1],x[19]/x[11])))) %>%
  mutate(spo13.impfc.165_v_0=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[10]/x[1],x[20]/x[11])))) %>%
  mutate(WT.impfc.45_v_0=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[22]/x[21],x[32]/x[31])))) %>%
  mutate(WT.impfc.60_v_0=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[23]/x[21],x[33]/x[31])))) %>%   
  mutate(WT.impfc.75_v_0=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[24]/x[21],x[34]/x[31])))) %>%
  mutate(WT.impfc.90_v_0=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[25]/x[21],x[35]/x[31])))) %>%
  mutate(WT.impfc.105_v_0=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[26]/x[21],x[36]/x[31])))) %>%
  mutate(WT.impfc.120_v_0=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[27]/x[21],x[37]/x[31])))) %>%
  mutate(WT.impfc.135_v_0=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[28]/x[21],x[38]/x[31])))) %>%
  mutate(WT.impfc.150_v_0=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[29]/x[21],x[39]/x[31])))) %>%
  mutate(WT.impfc.165_v_0=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[30]/x[21],x[40]/x[31])))) %>%
mutate(WT.spo13.impfc.0_v_0=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[21],x[31]))/median(c(x[1],x[11])))) %>%
  mutate(WT.spo13.impfc.45_v_45=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[22],x[32]))/median(c(x[2],x[12])))) %>%
  mutate(WT.spo13.impfc.60_v_60=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[23],x[33]))/median(c(x[3],x[13])))) %>%
  mutate(WT.spo13.impfc.75_v_75=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[24],x[34]))/median(c(x[4],x[14])))) %>%
  mutate(WT.spo13.impfc.90_v_90=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[25],x[35]))/median(c(x[5],x[15])))) %>%
  mutate(WT.spo13.impfc.105_v_105=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[26],x[36]))/median(c(x[6],x[16])))) %>%
  mutate(WT.spo13.impfc.120_v_120=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[27],x[37]))/median(c(x[7],x[17])))) %>%
  mutate(WT.spo13.impfc.135_v_135=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[28],x[38]))/median(c(x[8],x[18]))))  %>%
  mutate(WT.spo13.impfc.150_v_150=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[29],x[39]))/median(c(x[9],x[19])))) %>%
  mutate(WT.spo13.impfc.165_v_165=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[30],x[40]))/median(c(x[10],x[20])))) %>%
      # comparisons spo13 to t60
  mutate(spo13.impfc.75_v_60=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[4]/x[3],x[14]/x[13])))) %>%
  mutate(spo13.impfc.90_v_60=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[5]/x[3],x[15]/x[13])))) %>%
  mutate(spo13.impfc.105_v_60=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[6]/x[3],x[16]/x[13])))) %>%
  mutate(spo13.impfc.120_v_60=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[7]/x[3],x[17]/x[13])))) %>%
  mutate(spo13.impfc.135_v_60=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[8]/x[3],x[18]/x[13])))) %>%
  mutate(spo13.impfc.150_v_60=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[9]/x[3],x[19]/x[13])))) %>%
  mutate(spo13.impfc.165_v_60=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[10]/x[3],x[20]/x[13])))) %>%
    # spo13 to t75
    mutate(spo13.impfc.90_v_75=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[5]/x[4],x[15]/x[14])))) %>%
  mutate(spo13.impfc.105_v_75=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[6]/x[4],x[16]/x[14])))) %>%
  mutate(spo13.impfc.120_v_75=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[7]/x[4],x[17]/x[14])))) %>%
  mutate(spo13.impfc.135_v_75=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[8]/x[4],x[18]/x[14])))) %>%
  mutate(spo13.impfc.150_v_75=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[9]/x[4],x[19]/x[14])))) %>%
  mutate(spo13.impfc.165_v_75=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[10]/x[4],x[20]/x[14])))) %>%
    # comparisons WT to t60
  mutate(WT.impfc.75_v_60=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[24]/x[23],x[34]/x[33])))) %>%
  mutate(WT.impfc.90_v_60=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[25]/x[23],x[35]/x[33])))) %>%
  mutate(WT.impfc.105_v_60=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[26]/x[23],x[36]/x[33])))) %>%
  mutate(WT.impfc.120_v_60=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[27]/x[23],x[37]/x[33])))) %>%
  mutate(WT.impfc.135_v_60=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[28]/x[23],x[38]/x[33])))) %>%
  mutate(WT.impfc.150_v_60=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[29]/x[23],x[39]/x[33])))) %>%
  mutate(WT.impfc.165_v_60=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[30]/x[23],x[40]/x[33])))) %>%
    # WT to t75
  mutate(WT.impfc.90_v_75=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[25]/x[24],x[35]/x[34])))) %>%
  mutate(WT.impfc.105_v_75=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[26]/x[24],x[36]/x[34])))) %>%
  mutate(WT.impfc.120_v_75=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[27]/x[24],x[37]/x[34])))) %>%
  mutate(WT.impfc.135_v_75=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[28]/x[24],x[38]/x[34])))) %>%
  mutate(WT.impfc.150_v_75=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[29]/x[24],x[39]/x[34])))) %>%
  mutate(WT.impfc.165_v_75=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[30]/x[24],x[40]/x[34])))) %>%
    #spo13 t135 comparisons
 mutate(spo13.impfc.135_v_90=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[8]/x[5],x[18]/x[15])))) %>%
mutate(spo13.impfc.135_v_105=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[8]/x[6],x[18]/x[16])))) %>%
    mutate(spo13.impfc.135_v_120=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[8]/x[7],x[18]/x[17])))) %>%
    #WT t135 comparisons
    mutate(WT.impfc.135_v_90=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[28]/x[25],x[38]/x[35])))) %>%
mutate(WT.impfc.135_v_105=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[28]/x[26],x[38]/x[36])))) %>%
    mutate(WT.impfc.135_v_120=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[28]/x[27],x[38]/x[37])))) %>%
      # spo13 metaphse II
    mutate(spo13.impfc.150_v_120=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[9]/x[7],x[19]/x[17])))) %>%
    mutate(spo13.impfc.165_v_120=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[10]/x[7],x[20]/x[17])))) %>%
    # WT metaphase II
     mutate(WT.impfc.150_v_120=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[29]/x[27],x[39]/x[37])))) %>%
    mutate(WT.impfc.165_v_120=apply(.[,imp.phos.cor.lin.cols],1,function(x) median(c(x[30]/x[27],x[40]/x[37]))))

```

# replace Inf fold changes with imputed fcs
```{r,echo=F,message=F,warning=F}

# ONLY replace Inf fcs with the imputed fcs

# # non-imputed fc cols
# fc_cols<-grep("WT.fc|spo13.fc|WT.spo13.fc",colnames(protg),value=T)
# 
# # imputed fc cols
# imp_fc_cols<-grep("impfc.",colnames(protg),value=T)

# replace Infs with imputed fcs
# couldn't think of way to make it less wordy
# 
# protg
# spo13
protg$spo13.fc.45_v_0[is.infinite(protg$spo13.fc.45_v_0)]<-protg$spo13.impfc.45_v_0[is.infinite(protg$spo13.fc.45_v_0)]
protg$spo13.fc.60_v_0[is.infinite(protg$spo13.fc.60_v_0)]<-protg$spo13.impfc.60_v_0[is.infinite(protg$spo13.fc.60_v_0)]
protg$spo13.fc.75_v_0[is.infinite(protg$spo13.fc.75_v_0)]<-protg$spo13.impfc.75_v_0[is.infinite(protg$spo13.fc.75_v_0)]
protg$spo13.fc.90_v_0[is.infinite(protg$spo13.fc.90_v_0)]<-protg$spo13.impfc.90_v_0[is.infinite(protg$spo13.fc.90_v_0)]
protg$spo13.fc.105_v_0[is.infinite(protg$spo13.fc.105_v_0)]<-protg$spo13.impfc.105_v_0[is.infinite(protg$spo13.fc.105_v_0)]
protg$spo13.fc.120_v_0[is.infinite(protg$spo13.fc.120_v_0)]<-protg$spo13.impfc.120_v_0[is.infinite(protg$spo13.fc.120_v_0)]
protg$spo13.fc.135_v_0[is.infinite(protg$spo13.fc.135_v_0)]<-protg$spo13.impfc.135_v_0[is.infinite(protg$spo13.fc.135_v_0)]
protg$spo13.fc.150_v_0[is.infinite(protg$spo13.fc.150_v_0)]<-protg$spo13.impfc.150_v_0[is.infinite(protg$spo13.fc.150_v_0)]
protg$spo13.fc.165_v_0[is.infinite(protg$spo13.fc.165_v_0)]<-protg$spo13.impfc.165_v_0[is.infinite(protg$spo13.fc.165_v_0)]
# WT
protg$WT.fc.45_v_0[is.infinite(protg$WT.fc.45_v_0)]<-protg$WT.impfc.45_v_0[is.infinite(protg$WT.fc.45_v_0)]
protg$WT.fc.60_v_0[is.infinite(protg$WT.fc.60_v_0)]<-protg$WT.impfc.60_v_0[is.infinite(protg$WT.fc.60_v_0)]
protg$WT.fc.75_v_0[is.infinite(protg$WT.fc.75_v_0)]<-protg$WT.impfc.75_v_0[is.infinite(protg$WT.fc.75_v_0)]
protg$WT.fc.90_v_0[is.infinite(protg$WT.fc.90_v_0)]<-protg$WT.impfc.90_v_0[is.infinite(protg$WT.fc.90_v_0)]
protg$WT.fc.105_v_0[is.infinite(protg$WT.fc.105_v_0)]<-protg$WT.impfc.105_v_0[is.infinite(protg$WT.fc.105_v_0)]
protg$WT.fc.120_v_0[is.infinite(protg$WT.fc.120_v_0)]<-protg$WT.impfc.120_v_0[is.infinite(protg$WT.fc.120_v_0)]
protg$WT.fc.135_v_0[is.infinite(protg$WT.fc.135_v_0)]<-protg$WT.impfc.135_v_0[is.infinite(protg$WT.fc.135_v_0)]
protg$WT.fc.150_v_0[is.infinite(protg$WT.fc.150_v_0)]<-protg$WT.impfc.150_v_0[is.infinite(protg$WT.fc.150_v_0)]
protg$WT.fc.165_v_0[is.infinite(protg$WT.fc.165_v_0)]<-protg$WT.impfc.165_v_0[is.infinite(protg$WT.fc.165_v_0)]
# WT v spo13 paired timepoints
protg$WT.spo13.fc.0_v_0[is.infinite(protg$WT.spo13.fc.0_v_0)]<-protg$WT.spo13.impfc.0_v_0[is.infinite(protg$WT.spo13.fc.0_v_0)]
protg$WT.spo13.fc.45_v_45[is.infinite(protg$WT.spo13.fc.45_v_45)]<-protg$WT.spo13.impfc.45_v_45[is.infinite(protg$WT.spo13.fc.45_v_45)]
protg$WT.spo13.fc.60_v_60[is.infinite(protg$WT.spo13.fc.60_v_60)]<-protg$WT.spo13.impfc.60_v_60[is.infinite(protg$WT.spo13.fc.60_v_60)]
protg$WT.spo13.fc.75_v_75[is.infinite(protg$WT.spo13.fc.75_v_75)]<-protg$WT.spo13.impfc.75_v_75[is.infinite(protg$WT.spo13.fc.75_v_75)]
protg$WT.spo13.fc.90_v_90[is.infinite(protg$WT.spo13.fc.90_v_90)]<-protg$WT.spo13.impfc.90_v_90[is.infinite(protg$WT.spo13.fc.90_v_90)]
protg$WT.spo13.fc.105_v_105[is.infinite(protg$WT.spo13.fc.105_v_105)]<-protg$WT.spo13.impfc.105_v_105[is.infinite(protg$WT.spo13.fc.105_v_105)]
protg$WT.spo13.fc.120_v_120[is.infinite(protg$WT.spo13.fc.120_v_120)]<-protg$WT.spo13.impfc.120_v_120[is.infinite(protg$WT.spo13.fc.120_v_120)]
protg$WT.spo13.fc.135_v_135[is.infinite(protg$WT.spo13.fc.135_v_135)]<-protg$WT.spo13.impfc.135_v_135[is.infinite(protg$WT.spo13.fc.135_v_135)]
protg$WT.spo13.fc.150_v_150[is.infinite(protg$WT.spo13.fc.150_v_150)]<-protg$WT.spo13.impfc.150_v_150[is.infinite(protg$WT.spo13.fc.150_v_150)]
protg$WT.spo13.fc.165_v_165[is.infinite(protg$WT.spo13.fc.165_v_165)]<-protg$WT.spo13.impfc.165_v_165[is.infinite(protg$WT.spo13.fc.165_v_165)]
#  spo13 t60 and t75 comparisons
#  t60
protg$spo13.fc.75_v_60[is.infinite(protg$spo13.fc.75_v_60)]<-protg$spo13.impfc.75_v_60[is.infinite(protg$spo13.fc.75_v_60)]
protg$spo13.fc.90_v_60[is.infinite(protg$spo13.fc.90_v_60)]<-protg$spo13.impfc.90_v_60[is.infinite(protg$spo13.fc.90_v_60)]
protg$spo13.fc.105_v_60[is.infinite(protg$spo13.fc.105_v_60)]<-protg$spo13.impfc.105_v_60[is.infinite(protg$spo13.fc.105_v_60)]
protg$spo13.fc.120_v_60[is.infinite(protg$spo13.fc.120_v_60)]<-protg$spo13.impfc.120_v_60[is.infinite(protg$spo13.fc.120_v_60)]
protg$spo13.fc.135_v_60[is.infinite(protg$spo13.fc.135_v_60)]<-protg$spo13.impfc.135_v_60[is.infinite(protg$spo13.fc.135_v_60)]
protg$spo13.fc.150_v_60[is.infinite(protg$spo13.fc.150_v_60)]<-protg$spo13.impfc.150_v_60[is.infinite(protg$spo13.fc.150_v_60)]
protg$spo13.fc.165_v_60[is.infinite(protg$spo13.fc.165_v_60)]<-protg$spo13.impfc.165_v_60[is.infinite(protg$spo13.fc.165_v_60)]
# t75
protg$spo13.fc.90_v_75[is.infinite(protg$spo13.fc.90_v_75)]<-protg$spo13.impfc.90_v_75[is.infinite(protg$spo13.fc.90_v_75)]
protg$spo13.fc.105_v_75[is.infinite(protg$spo13.fc.105_v_75)]<-protg$spo13.impfc.105_v_75[is.infinite(protg$spo13.fc.105_v_75)]
protg$spo13.fc.120_v_75[is.infinite(protg$spo13.fc.120_v_75)]<-protg$spo13.impfc.120_v_75[is.infinite(protg$spo13.fc.120_v_75)]
protg$spo13.fc.135_v_75[is.infinite(protg$spo13.fc.135_v_75)]<-protg$spo13.impfc.135_v_75[is.infinite(protg$spo13.fc.135_v_75)]
protg$spo13.fc.150_v_75[is.infinite(protg$spo13.fc.150_v_75)]<-protg$spo13.impfc.150_v_75[is.infinite(protg$spo13.fc.150_v_75)]
protg$spo13.fc.165_v_75[is.infinite(protg$spo13.fc.165_v_75)]<-protg$spo13.impfc.165_v_75[is.infinite(protg$spo13.fc.165_v_75)]
#  WT t60 and t75 comparisons
#  t60
protg$WT.fc.75_v_60[is.infinite(protg$WT.fc.75_v_60)]<-protg$WT.impfc.75_v_60[is.infinite(protg$WT.fc.75_v_60)]
protg$WT.fc.90_v_60[is.infinite(protg$WT.fc.90_v_60)]<-protg$WT.impfc.90_v_60[is.infinite(protg$WT.fc.90_v_60)]
protg$WT.fc.105_v_60[is.infinite(protg$WT.fc.105_v_60)]<-protg$WT.impfc.105_v_60[is.infinite(protg$WT.fc.105_v_60)]
protg$WT.fc.120_v_60[is.infinite(protg$WT.fc.120_v_60)]<-protg$WT.impfc.120_v_60[is.infinite(protg$WT.fc.120_v_60)]
protg$WT.fc.135_v_60[is.infinite(protg$WT.fc.135_v_60)]<-protg$WT.impfc.135_v_60[is.infinite(protg$WT.fc.135_v_60)]
protg$WT.fc.150_v_60[is.infinite(protg$WT.fc.150_v_60)]<-protg$WT.impfc.150_v_60[is.infinite(protg$WT.fc.150_v_60)]
protg$WT.fc.165_v_60[is.infinite(protg$WT.fc.165_v_60)]<-protg$WT.impfc.165_v_60[is.infinite(protg$WT.fc.165_v_60)]
# t75
protg$WT.fc.90_v_75[is.infinite(protg$WT.fc.90_v_75)]<-protg$WT.impfc.90_v_75[is.infinite(protg$WT.fc.90_v_75)]
protg$WT.fc.105_v_75[is.infinite(protg$WT.fc.105_v_75)]<-protg$WT.impfc.105_v_75[is.infinite(protg$WT.fc.105_v_75)]
protg$WT.fc.120_v_75[is.infinite(protg$WT.fc.120_v_75)]<-protg$WT.impfc.120_v_75[is.infinite(protg$WT.fc.120_v_75)]
protg$WT.fc.135_v_75[is.infinite(protg$WT.fc.135_v_75)]<-protg$WT.impfc.135_v_75[is.infinite(protg$WT.fc.135_v_75)]
protg$WT.fc.150_v_75[is.infinite(protg$WT.fc.150_v_75)]<-protg$WT.impfc.150_v_75[is.infinite(protg$WT.fc.150_v_75)]
protg$WT.fc.165_v_75[is.infinite(protg$WT.fc.165_v_75)]<-protg$WT.impfc.165_v_75[is.infinite(protg$WT.fc.165_v_75)]
# spo13 t135 comparisons
protg$spo13.fc.135_v_90[is.infinite(protg$spo13.fc.135_v_90)]<-protg$spo13.impfc.135_v_90[is.infinite(protg$spo13.fc.135_v_90)]
protg$spo13.fc.135_v_105[is.infinite(protg$spo13.fc.135_v_105)]<-protg$spo13.impfc.135_v_105[is.infinite(protg$spo13.fc.135_v_105)]
protg$spo13.fc.135_v_120[is.infinite(protg$spo13.fc.135_v_120)]<-protg$spo13.impfc.135_v_120[is.infinite(protg$spo13.fc.135_v_120)]

# WT t135 comparisons
protg$WT.fc.135_v_90[is.infinite(protg$WT.fc.135_v_90)]<-protg$WT.impfc.135_v_90[is.infinite(protg$WT.fc.135_v_90)]
protg$WT.fc.135_v_105[is.infinite(protg$WT.fc.135_v_105)]<-protg$WT.impfc.135_v_105[is.infinite(protg$WT.fc.135_v_105)]
protg$WT.fc.135_v_120[is.infinite(protg$WT.fc.135_v_120)]<-protg$WT.impfc.135_v_120[is.infinite(protg$WT.fc.135_v_120)]

# spo13 metaphase II comparisons
protg$spo13.fc.150_v_120[is.infinite(protg$spo13.fc.150_v_120)]<-protg$spo13.impfc.150_v_120[is.infinite(protg$spo13.fc.150_v_120)]
protg$spo13.fc.165_v_120[is.infinite(protg$spo13.fc.165_v_120)]<-protg$spo13.impfc.165_v_120[is.infinite(protg$spo13.fc.165_v_120)]

# WT metaphase II comparisons
protg$WT.fc.150_v_120[is.infinite(protg$WT.fc.150_v_120)]<-protg$WT.impfc.150_v_120[is.infinite(protg$WT.fc.150_v_120)]
protg$WT.fc.165_v_120[is.infinite(protg$WT.fc.165_v_120)]<-protg$WT.impfc.165_v_120[is.infinite(protg$WT.fc.165_v_120)]

# phosphos
# spo13
phosphos$spo13.fc.45_v_0[is.infinite(phosphos$spo13.fc.45_v_0)]<-phosphos$spo13.impfc.45_v_0[is.infinite(phosphos$spo13.fc.45_v_0)]
phosphos$spo13.fc.60_v_0[is.infinite(phosphos$spo13.fc.60_v_0)]<-phosphos$spo13.impfc.60_v_0[is.infinite(phosphos$spo13.fc.60_v_0)]
phosphos$spo13.fc.75_v_0[is.infinite(phosphos$spo13.fc.75_v_0)]<-phosphos$spo13.impfc.75_v_0[is.infinite(phosphos$spo13.fc.75_v_0)]
phosphos$spo13.fc.90_v_0[is.infinite(phosphos$spo13.fc.90_v_0)]<-phosphos$spo13.impfc.90_v_0[is.infinite(phosphos$spo13.fc.90_v_0)]
phosphos$spo13.fc.105_v_0[is.infinite(phosphos$spo13.fc.105_v_0)]<-phosphos$spo13.impfc.105_v_0[is.infinite(phosphos$spo13.fc.105_v_0)]
phosphos$spo13.fc.120_v_0[is.infinite(phosphos$spo13.fc.120_v_0)]<-phosphos$spo13.impfc.120_v_0[is.infinite(phosphos$spo13.fc.120_v_0)]
phosphos$spo13.fc.135_v_0[is.infinite(phosphos$spo13.fc.135_v_0)]<-phosphos$spo13.impfc.135_v_0[is.infinite(phosphos$spo13.fc.135_v_0)]
phosphos$spo13.fc.150_v_0[is.infinite(phosphos$spo13.fc.150_v_0)]<-phosphos$spo13.impfc.150_v_0[is.infinite(phosphos$spo13.fc.150_v_0)]
phosphos$spo13.fc.165_v_0[is.infinite(phosphos$spo13.fc.165_v_0)]<-phosphos$spo13.impfc.165_v_0[is.infinite(phosphos$spo13.fc.165_v_0)]
# WT
phosphos$WT.fc.45_v_0[is.infinite(phosphos$WT.fc.45_v_0)]<-phosphos$WT.impfc.45_v_0[is.infinite(phosphos$WT.fc.45_v_0)]
phosphos$WT.fc.60_v_0[is.infinite(phosphos$WT.fc.60_v_0)]<-phosphos$WT.impfc.60_v_0[is.infinite(phosphos$WT.fc.60_v_0)]
phosphos$WT.fc.75_v_0[is.infinite(phosphos$WT.fc.75_v_0)]<-phosphos$WT.impfc.75_v_0[is.infinite(phosphos$WT.fc.75_v_0)]
phosphos$WT.fc.90_v_0[is.infinite(phosphos$WT.fc.90_v_0)]<-phosphos$WT.impfc.90_v_0[is.infinite(phosphos$WT.fc.90_v_0)]
phosphos$WT.fc.105_v_0[is.infinite(phosphos$WT.fc.105_v_0)]<-phosphos$WT.impfc.105_v_0[is.infinite(phosphos$WT.fc.105_v_0)]
phosphos$WT.fc.120_v_0[is.infinite(phosphos$WT.fc.120_v_0)]<-phosphos$WT.impfc.120_v_0[is.infinite(phosphos$WT.fc.120_v_0)]
phosphos$WT.fc.135_v_0[is.infinite(phosphos$WT.fc.135_v_0)]<-phosphos$WT.impfc.135_v_0[is.infinite(phosphos$WT.fc.135_v_0)]
phosphos$WT.fc.150_v_0[is.infinite(phosphos$WT.fc.150_v_0)]<-phosphos$WT.impfc.150_v_0[is.infinite(phosphos$WT.fc.150_v_0)]
phosphos$WT.fc.165_v_0[is.infinite(phosphos$WT.fc.165_v_0)]<-phosphos$WT.impfc.165_v_0[is.infinite(phosphos$WT.fc.165_v_0)]
# WT v spo13 paired timepoints
phosphos$WT.spo13.fc.0_v_0[is.infinite(phosphos$WT.spo13.fc.0_v_0)]<-phosphos$WT.spo13.impfc.0_v_0[is.infinite(phosphos$WT.spo13.fc.0_v_0)]
phosphos$WT.spo13.fc.45_v_45[is.infinite(phosphos$WT.spo13.fc.45_v_45)]<-phosphos$WT.spo13.impfc.45_v_45[is.infinite(phosphos$WT.spo13.fc.45_v_45)]
phosphos$WT.spo13.fc.60_v_60[is.infinite(phosphos$WT.spo13.fc.60_v_60)]<-phosphos$WT.spo13.impfc.60_v_60[is.infinite(phosphos$WT.spo13.fc.60_v_60)]
phosphos$WT.spo13.fc.75_v_75[is.infinite(phosphos$WT.spo13.fc.75_v_75)]<-phosphos$WT.spo13.impfc.75_v_75[is.infinite(phosphos$WT.spo13.fc.75_v_75)]
phosphos$WT.spo13.fc.90_v_90[is.infinite(phosphos$WT.spo13.fc.90_v_90)]<-phosphos$WT.spo13.impfc.90_v_90[is.infinite(phosphos$WT.spo13.fc.90_v_90)]
phosphos$WT.spo13.fc.105_v_105[is.infinite(phosphos$WT.spo13.fc.105_v_105)]<-phosphos$WT.spo13.impfc.105_v_105[is.infinite(phosphos$WT.spo13.fc.105_v_105)]
phosphos$WT.spo13.fc.120_v_120[is.infinite(phosphos$WT.spo13.fc.120_v_120)]<-phosphos$WT.spo13.impfc.120_v_120[is.infinite(phosphos$WT.spo13.fc.120_v_120)]
phosphos$WT.spo13.fc.135_v_135[is.infinite(phosphos$WT.spo13.fc.135_v_135)]<-phosphos$WT.spo13.impfc.135_v_135[is.infinite(phosphos$WT.spo13.fc.135_v_135)]
phosphos$WT.spo13.fc.150_v_150[is.infinite(phosphos$WT.spo13.fc.150_v_150)]<-phosphos$WT.spo13.impfc.150_v_150[is.infinite(phosphos$WT.spo13.fc.150_v_150)]
phosphos$WT.spo13.fc.165_v_165[is.infinite(phosphos$WT.spo13.fc.165_v_165)]<-phosphos$WT.spo13.impfc.165_v_165[is.infinite(phosphos$WT.spo13.fc.165_v_165)]
#  spo13 t60 and t75 comparisons
#  t60
phosphos$spo13.fc.75_v_60[is.infinite(phosphos$spo13.fc.75_v_60)]<-phosphos$spo13.impfc.75_v_60[is.infinite(phosphos$spo13.fc.75_v_60)]
phosphos$spo13.fc.90_v_60[is.infinite(phosphos$spo13.fc.90_v_60)]<-phosphos$spo13.impfc.90_v_60[is.infinite(phosphos$spo13.fc.90_v_60)]
phosphos$spo13.fc.105_v_60[is.infinite(phosphos$spo13.fc.105_v_60)]<-phosphos$spo13.impfc.105_v_60[is.infinite(phosphos$spo13.fc.105_v_60)]
phosphos$spo13.fc.120_v_60[is.infinite(phosphos$spo13.fc.120_v_60)]<-phosphos$spo13.impfc.120_v_60[is.infinite(phosphos$spo13.fc.120_v_60)]
phosphos$spo13.fc.135_v_60[is.infinite(phosphos$spo13.fc.135_v_60)]<-phosphos$spo13.impfc.135_v_60[is.infinite(phosphos$spo13.fc.135_v_60)]
phosphos$spo13.fc.150_v_60[is.infinite(phosphos$spo13.fc.150_v_60)]<-phosphos$spo13.impfc.150_v_60[is.infinite(phosphos$spo13.fc.150_v_60)]
phosphos$spo13.fc.165_v_60[is.infinite(phosphos$spo13.fc.165_v_60)]<-phosphos$spo13.impfc.165_v_60[is.infinite(phosphos$spo13.fc.165_v_60)]
# t75
phosphos$spo13.fc.90_v_75[is.infinite(phosphos$spo13.fc.90_v_75)]<-phosphos$spo13.impfc.90_v_75[is.infinite(phosphos$spo13.fc.90_v_75)]
phosphos$spo13.fc.105_v_75[is.infinite(phosphos$spo13.fc.105_v_75)]<-phosphos$spo13.impfc.105_v_75[is.infinite(phosphos$spo13.fc.105_v_75)]
phosphos$spo13.fc.120_v_75[is.infinite(phosphos$spo13.fc.120_v_75)]<-phosphos$spo13.impfc.120_v_75[is.infinite(phosphos$spo13.fc.120_v_75)]
phosphos$spo13.fc.135_v_75[is.infinite(phosphos$spo13.fc.135_v_75)]<-phosphos$spo13.impfc.135_v_75[is.infinite(phosphos$spo13.fc.135_v_75)]
phosphos$spo13.fc.150_v_75[is.infinite(phosphos$spo13.fc.150_v_75)]<-phosphos$spo13.impfc.150_v_75[is.infinite(phosphos$spo13.fc.150_v_75)]
phosphos$spo13.fc.165_v_75[is.infinite(phosphos$spo13.fc.165_v_75)]<-phosphos$spo13.impfc.165_v_75[is.infinite(phosphos$spo13.fc.165_v_75)]
#  WT t60 and t75 comparisons
#  t60
phosphos$WT.fc.75_v_60[is.infinite(phosphos$WT.fc.75_v_60)]<-phosphos$WT.impfc.75_v_60[is.infinite(phosphos$WT.fc.75_v_60)]
phosphos$WT.fc.90_v_60[is.infinite(phosphos$WT.fc.90_v_60)]<-phosphos$WT.impfc.90_v_60[is.infinite(phosphos$WT.fc.90_v_60)]
phosphos$WT.fc.105_v_60[is.infinite(phosphos$WT.fc.105_v_60)]<-phosphos$WT.impfc.105_v_60[is.infinite(phosphos$WT.fc.105_v_60)]
phosphos$WT.fc.120_v_60[is.infinite(phosphos$WT.fc.120_v_60)]<-phosphos$WT.impfc.120_v_60[is.infinite(phosphos$WT.fc.120_v_60)]
phosphos$WT.fc.135_v_60[is.infinite(phosphos$WT.fc.135_v_60)]<-phosphos$WT.impfc.135_v_60[is.infinite(phosphos$WT.fc.135_v_60)]
phosphos$WT.fc.150_v_60[is.infinite(phosphos$WT.fc.150_v_60)]<-phosphos$WT.impfc.150_v_60[is.infinite(phosphos$WT.fc.150_v_60)]
phosphos$WT.fc.165_v_60[is.infinite(phosphos$WT.fc.165_v_60)]<-phosphos$WT.impfc.165_v_60[is.infinite(phosphos$WT.fc.165_v_60)]
# t75
phosphos$WT.fc.90_v_75[is.infinite(phosphos$WT.fc.90_v_75)]<-phosphos$WT.impfc.90_v_75[is.infinite(phosphos$WT.fc.90_v_75)]
phosphos$WT.fc.105_v_75[is.infinite(phosphos$WT.fc.105_v_75)]<-phosphos$WT.impfc.105_v_75[is.infinite(phosphos$WT.fc.105_v_75)]
phosphos$WT.fc.120_v_75[is.infinite(phosphos$WT.fc.120_v_75)]<-phosphos$WT.impfc.120_v_75[is.infinite(phosphos$WT.fc.120_v_75)]
phosphos$WT.fc.135_v_75[is.infinite(phosphos$WT.fc.135_v_75)]<-phosphos$WT.impfc.135_v_75[is.infinite(phosphos$WT.fc.135_v_75)]
phosphos$WT.fc.150_v_75[is.infinite(phosphos$WT.fc.150_v_75)]<-phosphos$WT.impfc.150_v_75[is.infinite(phosphos$WT.fc.150_v_75)]
phosphos$WT.fc.165_v_75[is.infinite(phosphos$WT.fc.165_v_75)]<-phosphos$WT.impfc.165_v_75[is.infinite(phosphos$WT.fc.165_v_75)]

# spo13 t135 comparisons
phosphos$spo13.fc.135_v_90[is.infinite(phosphos$spo13.fc.135_v_90)]<-phosphos$spo13.impfc.135_v_90[is.infinite(phosphos$spo13.fc.135_v_90)]
phosphos$spo13.fc.135_v_105[is.infinite(phosphos$spo13.fc.135_v_105)]<-phosphos$spo13.impfc.135_v_105[is.infinite(phosphos$spo13.fc.135_v_105)]
phosphos$spo13.fc.135_v_120[is.infinite(phosphos$spo13.fc.135_v_120)]<-phosphos$spo13.impfc.135_v_120[is.infinite(phosphos$spo13.fc.135_v_120)]
# WT t135 comparisons
phosphos$WT.fc.135_v_90[is.infinite(phosphos$WT.fc.135_v_90)]<-phosphos$WT.impfc.135_v_90[is.infinite(phosphos$WT.fc.135_v_90)]
phosphos$WT.fc.135_v_105[is.infinite(phosphos$WT.fc.135_v_105)]<-phosphos$WT.impfc.135_v_105[is.infinite(phosphos$WT.fc.135_v_105)]
phosphos$WT.fc.135_v_120[is.infinite(phosphos$WT.fc.135_v_120)]<-phosphos$WT.impfc.135_v_120[is.infinite(phosphos$WT.fc.135_v_120)]

# spo13 metaphase II comparisons
phosphos$spo13.fc.150_v_120[is.infinite(phosphos$spo13.fc.150_v_120)]<-phosphos$spo13.impfc.150_v_120[is.infinite(phosphos$spo13.fc.150_v_120)]
phosphos$spo13.fc.165_v_120[is.infinite(phosphos$spo13.fc.165_v_120)]<-phosphos$spo13.impfc.165_v_120[is.infinite(phosphos$spo13.fc.165_v_120)]

# WT metaphase II comparisons
phosphos$WT.fc.150_v_120[is.infinite(phosphos$WT.fc.150_v_120)]<-phosphos$WT.impfc.150_v_120[is.infinite(phosphos$WT.fc.150_v_120)]
phosphos$WT.fc.165_v_120[is.infinite(phosphos$WT.fc.165_v_120)]<-phosphos$WT.impfc.165_v_120[is.infinite(phosphos$WT.fc.165_v_120)]

```

# Mean of reps
```{r,echo=F,warning=F,message=F}

# variables to hold individual rep batch cor cols for later
spo13_protg_rep1_cols<-cor.lin.cols[1:10] # spo13 rep1
spo13_protg_rep2_cols<-cor.lin.cols[11:20] # spoo13 rep2
WT_protg_rep1_cols<-cor.lin.cols[21:30] # WT rep 1
WT_protg_rep2_cols<-cor.lin.cols[31:40] # WT rep 2

spo13_phos_rep1_cols<-phos.cor.lin.cols[1:10] # spo13 rep1
spo13_phos_rep2_cols<-phos.cor.lin.cols[11:20] # spoo13 rep2
WT_phos_rep1_cols<-phos.cor.lin.cols[21:30] # WT rep 1
WT_phos_rep2_cols<-phos.cor.lin.cols[31:40] # WT rep 2

# need the mean of 2 reps value

protg<-protg %>%
  #spo13
  mutate(spo13.mean.0=apply(.[,cor.lin.cols],1,function(x) mean(c(x[1],x[11])))) %>%
  mutate(spo13.mean.45=apply(.[,cor.lin.cols],1,function(x) mean(c(x[2],x[12])))) %>%
  mutate(spo13.mean.60=apply(.[,cor.lin.cols],1,function(x) mean(c(x[3],x[13])))) %>%
  mutate(spo13.mean.75=apply(.[,cor.lin.cols],1,function(x) mean(c(x[4],x[14])))) %>%
  mutate(spo13.mean.90=apply(.[,cor.lin.cols],1,function(x) mean(c(x[5],x[15])))) %>%
  mutate(spo13.mean.105=apply(.[,cor.lin.cols],1,function(x) mean(c(x[6],x[16])))) %>%
  mutate(spo13.mean.120=apply(.[,cor.lin.cols],1,function(x) mean(c(x[7],x[17])))) %>%
  mutate(spo13.mean.135=apply(.[,cor.lin.cols],1,function(x) mean(c(x[8],x[18])))) %>%
  mutate(spo13.mean.150=apply(.[,cor.lin.cols],1,function(x) mean(c(x[9],x[19])))) %>%
  mutate(spo13.mean.165=apply(.[,cor.lin.cols],1,function(x) mean(c(x[10],x[20])))) %>%
# WT
mutate(WT.mean.0=apply(.[,cor.lin.cols],1,function(x) mean(c(x[21],x[31])))) %>%
  mutate(WT.mean.45=apply(.[,cor.lin.cols],1,function(x) mean(c(x[22],x[32])))) %>%
  mutate(WT.mean.60=apply(.[,cor.lin.cols],1,function(x) mean(c(x[23],x[33])))) %>%
  mutate(WT.mean.75=apply(.[,cor.lin.cols],1,function(x) mean(c(x[24],x[34])))) %>%
  mutate(WT.mean.90=apply(.[,cor.lin.cols],1,function(x) mean(c(x[25],x[35])))) %>%
  mutate(WT.mean.105=apply(.[,cor.lin.cols],1,function(x) mean(c(x[26],x[36])))) %>%
  mutate(WT.mean.120=apply(.[,cor.lin.cols],1,function(x) mean(c(x[27],x[37])))) %>%
  mutate(WT.mean.135=apply(.[,cor.lin.cols],1,function(x) mean(c(x[28],x[38])))) %>%
  mutate(WT.mean.150=apply(.[,cor.lin.cols],1,function(x) mean(c(x[29],x[39])))) %>%
  mutate(WT.mean.165=apply(.[,cor.lin.cols],1,function(x) mean(c(x[30],x[40]))))


# phos
# phos.cor.lin.cols are for phos


phosphos<-phosphos %>%
  #spo13
  mutate(spo13.mean.0=apply(.[,phos.cor.lin.cols],1,function(x) mean(c(x[1],x[11])))) %>%
  mutate(spo13.mean.45=apply(.[,phos.cor.lin.cols],1,function(x) mean(c(x[2],x[12])))) %>%
  mutate(spo13.mean.60=apply(.[,phos.cor.lin.cols],1,function(x) mean(c(x[3],x[13])))) %>%
  mutate(spo13.mean.75=apply(.[,phos.cor.lin.cols],1,function(x) mean(c(x[4],x[14])))) %>%
  mutate(spo13.mean.90=apply(.[,phos.cor.lin.cols],1,function(x) mean(c(x[5],x[15])))) %>%
  mutate(spo13.mean.105=apply(.[,phos.cor.lin.cols],1,function(x) mean(c(x[6],x[16])))) %>%
  mutate(spo13.mean.120=apply(.[,phos.cor.lin.cols],1,function(x) mean(c(x[7],x[17])))) %>%
  mutate(spo13.mean.135=apply(.[,phos.cor.lin.cols],1,function(x) mean(c(x[8],x[18])))) %>%
  mutate(spo13.mean.150=apply(.[,phos.cor.lin.cols],1,function(x) mean(c(x[9],x[19])))) %>%
  mutate(spo13.mean.165=apply(.[,phos.cor.lin.cols],1,function(x) mean(c(x[10],x[20])))) %>%
# WT
mutate(WT.mean.0=apply(.[,phos.cor.lin.cols],1,function(x) mean(c(x[21],x[31])))) %>%
  mutate(WT.mean.45=apply(.[,phos.cor.lin.cols],1,function(x) mean(c(x[22],x[32])))) %>%
  mutate(WT.mean.60=apply(.[,phos.cor.lin.cols],1,function(x) mean(c(x[23],x[33])))) %>%
  mutate(WT.mean.75=apply(.[,phos.cor.lin.cols],1,function(x) mean(c(x[24],x[34])))) %>%
  mutate(WT.mean.90=apply(.[,phos.cor.lin.cols],1,function(x) mean(c(x[25],x[35])))) %>%
  mutate(WT.mean.105=apply(.[,phos.cor.lin.cols],1,function(x) mean(c(x[26],x[36])))) %>%
  mutate(WT.mean.120=apply(.[,phos.cor.lin.cols],1,function(x) mean(c(x[27],x[37])))) %>%
  mutate(WT.mean.135=apply(.[,phos.cor.lin.cols],1,function(x) mean(c(x[28],x[38])))) %>%
  mutate(WT.mean.150=apply(.[,phos.cor.lin.cols],1,function(x) mean(c(x[29],x[39])))) %>%
  mutate(WT.mean.165=apply(.[,phos.cor.lin.cols],1,function(x) mean(c(x[30],x[40]))))

```


# filter MVs and low qual
```{r, message=FALSE, warning=FALSE,echo=F}

# variables to hold colnames

spo13.mean.cols<-grep("mean",colnames(protg),value=T)[1:10]
WT.mean.cols<-grep("mean",colnames(protg),value=T)[11:20]

WT.sig.cols<-grep("sig",colnames(protg),value=T)[1:9]
spo13.sig.cols<-grep("sig",colnames(protg),value=T)[10:18]
WT.spo13.sig.cols<-grep("sig",colnames(protg),value=T)[19:28]
WT.t60.sig.cols<-grep("sig",colnames(protg),value=T)[29:35]
WT.t75.sig.cols<-grep("sig",colnames(protg),value=T)[36:41]
spo13.t60.sig.cols<-grep("sig",colnames(protg),value=T)[42:48]
spo13.t75.sig.cols<-grep("sig",colnames(protg),value=T)[49:54]

# # proteins in 2 WT reps
WT.protg<-protg %>% filter(WT.2reps==TRUE)

# proteins 2 spo13 reps
spo13.protg<-protg %>% filter(spo13.2reps==TRUE)

# proteins all 4 reps
WT.spo13.protg<-protg %>% filter(all4reps==TRUE) 

# Filter phosphos for sites found in 2 seq timepoints, 2 reps
# and high localization score

WT.phosphos<-phosphos %>% filter(WT.2.detects==TRUE & high.localiz.score==TRUE)

spo13.phosphos<-phosphos %>% filter(spo13.2.detects==TRUE & high.localiz.score==TRUE)

WT.spo13.phosphos<-phosphos %>% filter(all4reps==TRUE & high.localiz.score==TRUE)


```

# filter to finite across timecourse

```{r,echo=F,message=F,warning=F}

# filter WT.protg df to only include prots that have finite and non-NA mean values.
WT.protg.fin.df<-WT.protg[apply(WT.protg[,c(WT.mean.cols)],1,function(x) {all(is.finite(x)) & all(!is.na(x))}),] 

#spo13 protein
spo13.protg.fin.df<-spo13.protg[apply(spo13.protg[,c(spo13.mean.cols)],1,function(x) {all(is.finite(x)) & all(!is.na(x))}),] 

# WT and spo13 protg
WT.spo13.protg.fin.df<-WT.spo13.protg[apply(WT.spo13.protg[,c(WT.mean.cols,spo13.mean.cols)],1,function(x) {all(is.finite(x)) & all(!is.na(x))}),] 

# filter WT.phosphos df to only include sites that have finite and non-NA mean values.
WT.phos.fin.df<-WT.phosphos[apply(WT.phosphos[,c(WT.mean.cols)],1,function(x) {all(is.finite(x)) & all(!is.na(x))}),]

#same with spo13
spo13.phos.fin.df<-spo13.phosphos[apply(spo13.phosphos[,c(spo13.mean.cols)],1,function(x) {all(is.finite(x)) & all(!is.na(x))}),]

# filter WT.spo13.phosphos df to only include sites that have finite and non-NA mean values across the timecourse. cannot cluster NAs
WT.spo13.phos.fin.df<-WT.spo13.phosphos[apply(WT.spo13.phosphos[,c(WT.mean.cols,spo13.mean.cols)],1,function(x) {all(is.finite(x)) & all(!is.na(x))}),]


```


# add mean-scaled cols
```{r,echo=F,message=F,warning=F}

# mean-scale cols
scaled.WT.mean.cols<-paste0("scaled.",WT.mean.cols)
scaled.spo13.mean.cols<-paste0("scaled.",spo13.mean.cols)

WT.protg.fin.df[,scaled.WT.mean.cols] <- t(apply(WT.protg.fin.df[,WT.mean.cols], 1, function(x) x/mean(x, na.rm=T)))

WT.phos.fin.df[,scaled.WT.mean.cols] <- t(apply(WT.phos.fin.df[,WT.mean.cols], 1, function(x) x/mean(x, na.rm=T)))

spo13.protg.fin.df[,scaled.spo13.mean.cols] <- t(apply(spo13.protg.fin.df[,spo13.mean.cols], 1, function(x) x/mean(x, na.rm=T)))

spo13.phos.fin.df[,scaled.spo13.mean.cols] <- t(apply(spo13.phos.fin.df[,spo13.mean.cols], 1, function(x) x/mean(x, na.rm=T)))

# mean scaled cols across WT and spo13 TOGETHER
WT.spo13.scaled.mean.cols<-paste0("scaled.both.",c(WT.mean.cols,spo13.mean.cols))

WT.spo13.protg.fin.df[,WT.spo13.scaled.mean.cols]<-t(apply(WT.spo13.protg.fin.df[,c(WT.mean.cols,spo13.mean.cols)], 1, function(x) x/mean(x, na.rm=T)))

WT.spo13.phos.fin.df[,WT.spo13.scaled.mean.cols]<-t(apply(WT.spo13.phos.fin.df[,c(WT.mean.cols,spo13.mean.cols)], 1, function(x) x/mean(x, na.rm=T)))


# add metaphase scaled cols for key transitions analysis

metaI.scaled.WT.mean.cols<-paste0("metaI.scaled.",WT.mean.cols)
metaI.scaled.spo13.mean.cols<-paste0("metaI.scaled.",spo13.mean.cols)
metaII.scaled.WT.mean.cols<-paste0("metaII.scaled.",WT.mean.cols)
metaII.scaled.spo13.mean.cols<-paste0("metaII.scaled.",spo13.mean.cols)

WT.phos.fin.df[,metaI.scaled.WT.mean.cols] <- t(apply(WT.phos.fin.df[,WT.mean.cols], 1, function(x) x/x[4])) # scaled to t75
spo13.phos.fin.df[,metaI.scaled.spo13.mean.cols] <- t(apply(spo13.phos.fin.df[,spo13.mean.cols], 1, function(x) x/x[4])) #scaled to t75
WT.phos.fin.df[,metaII.scaled.WT.mean.cols] <- t(apply(WT.phos.fin.df[,WT.mean.cols], 1, function(x) x/x[7])) # scaled to t120
spo13.phos.fin.df[,metaII.scaled.spo13.mean.cols] <- t(apply(spo13.phos.fin.df[,spo13.mean.cols], 1, function(x) x/x[7])) #scaled to t120

# shared dfs # cols scaled within strain to compare kinetics between strains
WT.spo13.phos.fin.df[,metaI.scaled.WT.mean.cols] <- t(apply(WT.spo13.phos.fin.df[,WT.mean.cols], 1, function(x) x/x[4])) # scaled to t75
WT.spo13.phos.fin.df[,metaI.scaled.spo13.mean.cols] <- t(apply(WT.spo13.phos.fin.df[,spo13.mean.cols], 1, function(x) x/x[4]))
WT.spo13.phos.fin.df[,metaII.scaled.WT.mean.cols] <- t(apply(WT.spo13.phos.fin.df[,WT.mean.cols], 1, function(x) x/x[7])) # sscaled to 120
WT.spo13.phos.fin.df[,metaII.scaled.spo13.mean.cols] <- t(apply(WT.spo13.phos.fin.df[,spo13.mean.cols], 1, function(x) x/x[7]))


```


# Calling Sig Categories for Pie charts 1
```{r,echo=F,message=F,warning=F}

# replace NAs with "Zero" in sig cols for pie charts (NAs do not work for logicals)

# WT
WT.protg.fin.df[,WT.sig.cols][t(apply(WT.protg.fin.df[,WT.sig.cols],1,function(x) is.na(x)))]<-"Zero"

WT.phos.fin.df[,WT.sig.cols][t(apply(WT.phos.fin.df[,WT.sig.cols],1,function(x) is.na(x)))]<-"Zero"

# spo13
spo13.protg.fin.df[,spo13.sig.cols][t(apply(spo13.protg.fin.df[,spo13.sig.cols],1,function(x) is.na(x)))]<-"Zero"

spo13.phos.fin.df[,spo13.sig.cols][t(apply(spo13.phos.fin.df[,spo13.sig.cols],1,function(x) is.na(x)))]<-"Zero"

#WT.spo13 dfs -- both sets of wt and spo13 sig from t0 cols 
WT.spo13.protg.fin.df[,c(WT.sig.cols,spo13.sig.cols)][t(apply(WT.spo13.protg.fin.df[,c(WT.sig.cols,spo13.sig.cols)],1,function(x) is.na(x)))]<-"Zero"

WT.spo13.phos.fin.df[,c(WT.sig.cols,spo13.sig.cols)][t(apply(WT.spo13.phos.fin.df[,c(WT.sig.cols,spo13.sig.cols)],1,function(x) is.na(x)))]<-"Zero"

# #WT.spo13 dfs -- sigdiff cols
# note sigdiff sig cols are called WT.spo13.sig.cols
WT.spo13.protg.fin.df[,WT.spo13.sig.cols][t(apply(WT.spo13.protg.fin.df[,WT.spo13.sig.cols],1,function(x) is.na(x)))]<-"Zero"

WT.spo13.phos.fin.df[,WT.spo13.sig.cols][t(apply(WT.spo13.phos.fin.df[,WT.spo13.sig.cols],1,function(x) is.na(x)))]<-"Zero"

```

# Calling Sig Categories for Pie charts 2
```{r,echo=F,message=F,warning=F}

# add sig.cat and is.sig columns to facilitate plottings
# 
# sig calling conditions here need to match sig filtering step subsequently
# filter(if_any(WT.sig.cols,function(x) {x=="DECR"|x=="INCR"})) %>%
    # filter(if_all(WT.sig.cols,function(x) {!is.na(x)}))
# 
# PROTEIN ####
# 
# WT only
WT.protg.fin.df<-WT.protg.fin.df %>%
  mutate(INCR.only=apply(.[,WT.sig.cols],1,function(x) any(x=="INCR") & !any(x=="DECR") & !any(x=="Zero") )) %>%
  mutate(DECR.only=apply(.[,WT.sig.cols],1,function(x) any(x=="DECR") & !any(x=="INCR") & !any(x=="Zero"))) %>%
  mutate(variable.sig=apply(.[,WT.sig.cols],1,function(x) any(x=="DECR") & any(x=="INCR") & !any(x=="Zero"))) %>%
  mutate(sig.cat=ifelse(INCR.only==TRUE,"INCR",
                        ifelse(DECR.only==TRUE,"DECR",
                               ifelse(variable.sig==TRUE,"VAR","NC")))) %>%
  mutate(is.sig=ifelse(sig.cat=="NC",FALSE,TRUE))

# spo13 only
spo13.protg.fin.df<-spo13.protg.fin.df %>%
  mutate(INCR.only=apply(.[,spo13.sig.cols],1,function(x) any(x=="INCR") & !any(x=="DECR") & !any(x=="Zero") )) %>%
  mutate(DECR.only=apply(.[,spo13.sig.cols],1,function(x) any(x=="DECR") & !any(x=="INCR") & !any(x=="Zero"))) %>%
  mutate(variable.sig=apply(.[,spo13.sig.cols],1,function(x) any(x=="DECR") & any(x=="INCR") & !any(x=="Zero"))) %>%
  mutate(sig.cat=ifelse(INCR.only==TRUE,"INCR",
                        ifelse(DECR.only==TRUE,"DECR",
                               ifelse(variable.sig==TRUE,"VAR","NC")))) %>%
  mutate(is.sig=ifelse(sig.cat=="NC",FALSE,TRUE))

# WT v spo13 SIGDIFF protein
# 
# NOTE CHANGED for clarity in the paper
# WT_INCR = spo13_DECR
# WT_DECR = spo13_INCR
# 
WT.spo13.protg.fin.df<-WT.spo13.protg.fin.df %>%
 mutate(sigdiff_spo13_DECR=apply(.[,WT.spo13.sig.cols],1,function(x) any(x=="INCR") & !any(x=="DECR") & !any(x=="Zero") )) %>%
    mutate(sigdiff_spo13_INCR=apply(.[,WT.spo13.sig.cols],1,function(x) !any(x=="INCR") & any(x=="DECR") & !any(x=="Zero") )) %>%
    mutate(sigdiff_VAR=apply(.[,WT.spo13.sig.cols],1,function(x) any(x=="INCR") & any(x=="DECR") & !any(x=="Zero") )) %>%
    mutate(sigdiff.sig.cat=ifelse(sigdiff_spo13_DECR==T,"sigdiff_spo13_DECR",
                          ifelse(sigdiff_spo13_INCR==T,"sigdiff_spo13_INCR",
                                 ifelse(sigdiff_VAR==T,"sigdiff_VAR","NC")))) %>%
    mutate(time.first.sigdiff.DECR=apply(.[,WT.spo13.sig.cols],1,function(x) ifelse(any(x=="Zero"),NA,which(x=="INCR")[1]))) %>% # only returns time FIRST INCR
mutate(time.first.sigdiff.INCR=apply(.[,WT.spo13.sig.cols],1,function(x) ifelse(any(x=="Zero"),NA,which(x=="DECR")[1]))) %>%
mutate(time.first.sigdiff.VAR=apply(.[,WT.spo13.sig.cols],1,function(x) ifelse(any(x=="Zero"),NA,which(x=="INCR"|x=="DECR")[1]))) %>%
mutate(time.first.sigdiff.DECR=as.factor(as.numeric(alltimes[time.first.sigdiff.DECR]))) %>% # change from 1-10 to 0-165
mutate(time.first.sigdiff.INCR=as.factor(as.numeric(alltimes[time.first.sigdiff.INCR]))) %>% # change from 1-10 to 0-165
mutate(time.first.sigdiff.VAR=as.factor(as.numeric(alltimes[time.first.sigdiff.VAR]))) # change from 1-10 to 0-165

# PHOSPHO sites

# WT only
WT.phos.fin.df<-WT.phos.fin.df %>%
  mutate(INCR.only=apply(.[,WT.sig.cols],1,function(x) any(x=="INCR") & !any(x=="DECR") & !any(x=="Zero") )) %>%
  mutate(DECR.only=apply(.[,WT.sig.cols],1,function(x) any(x=="DECR") & !any(x=="INCR") & !any(x=="Zero"))) %>%
  mutate(variable.sig=apply(.[,WT.sig.cols],1,function(x) any(x=="DECR") & any(x=="INCR") & !any(x=="Zero"))) %>%
  mutate(sig.cat=ifelse(INCR.only==TRUE,"INCR",
                        ifelse(DECR.only==TRUE,"DECR",
                               ifelse(variable.sig==TRUE,"VAR","NC")))) %>%
  mutate(is.sig=ifelse(sig.cat=="NC",FALSE,TRUE))

# spo13 only
spo13.phos.fin.df<-spo13.phos.fin.df %>%
  mutate(INCR.only=apply(.[,spo13.sig.cols],1,function(x) any(x=="INCR") & !any(x=="DECR") & !any(x=="Zero") )) %>%
  mutate(DECR.only=apply(.[,spo13.sig.cols],1,function(x) any(x=="DECR") & !any(x=="INCR") & !any(x=="Zero"))) %>%
  mutate(variable.sig=apply(.[,spo13.sig.cols],1,function(x) any(x=="DECR") & any(x=="INCR") & !any(x=="Zero"))) %>%
  mutate(sig.cat=ifelse(INCR.only==TRUE,"INCR",
                        ifelse(DECR.only==TRUE,"DECR",
                               ifelse(variable.sig==TRUE,"VAR","NC")))) %>%
  mutate(is.sig=ifelse(sig.cat=="NC",FALSE,TRUE))

# WT v spo13 SIGDIFF PHOSPHO
# SIGDIFF PHOSPHOS categories within WT.spo13.phos.fin.df
WT.spo13.phos.fin.df<-WT.spo13.phos.fin.df %>%
 mutate(sigdiff_spo13_DECR=apply(.[,WT.spo13.sig.cols],1,function(x) any(x=="INCR") & !any(x=="DECR") & !any(x=="Zero") )) %>%
    mutate(sigdiff_spo13_INCR=apply(.[,WT.spo13.sig.cols],1,function(x) !any(x=="INCR") & any(x=="DECR") & !any(x=="Zero") )) %>%
    mutate(sigdiff_VAR=apply(.[,WT.spo13.sig.cols],1,function(x) any(x=="INCR") & any(x=="DECR") & !any(x=="Zero") )) %>%
    mutate(sigdiff.sig.cat=ifelse(sigdiff_spo13_DECR==T,"sigdiff_spo13_DECR",
                          ifelse(sigdiff_spo13_INCR==T,"sigdiff_spo13_INCR",
                                 ifelse(sigdiff_VAR==T,"sigdiff_VAR","NC")))) %>%
    mutate(time.first.sigdiff.DECR=apply(.[,WT.spo13.sig.cols],1,function(x) ifelse(any(x=="Zero"),NA,which(x=="INCR")[1]))) %>% # only returns time FIRST INCR
mutate(time.first.sigdiff.INCR=apply(.[,WT.spo13.sig.cols],1,function(x) ifelse(any(x=="Zero"),NA,which(x=="DECR")[1]))) %>%
mutate(time.first.sigdiff.VAR=apply(.[,WT.spo13.sig.cols],1,function(x) ifelse(any(x=="Zero"),NA,which(x=="INCR"|x=="DECR")[1]))) %>%
mutate(time.first.sigdiff.DECR=as.factor(as.numeric(alltimes[time.first.sigdiff.DECR]))) %>% # change from 1-10 to 0-165
mutate(time.first.sigdiff.INCR=as.factor(as.numeric(alltimes[time.first.sigdiff.INCR]))) %>% # change from 1-10 to 0-165
mutate(time.first.sigdiff.VAR=as.factor(as.numeric(alltimes[time.first.sigdiff.VAR]))) # change from 1-10 to 0-165


# WT. and sspo13 ssig from t0 IN EITHER --PROTEIN 
# # 10 categories total
# NOTE BECAUSE WT/SPO13 SIG CATS are called acros WT.sig.colss and sspo13.ssig.cols SEPARATELY
# need to first make "NC" category for rows with any "zeros" across BOTH WT.sig.col and spo13.sig.col
WT.spo13.protg.fin.df<-WT.spo13.protg.fin.df %>%
    mutate(NC=apply(.[,c(WT.sig.cols,spo13.sig.cols)],1,function(x) any(x=="Zero"))) %>%
    mutate(WT.INCR.only=apply(.[,WT.sig.cols],1,function(x) any(x=="INCR") & !any(x=="DECR") & !any(x=="Zero"))) %>%
  mutate(WT.DECR.only=apply(.[,WT.sig.cols],1,function(x) any(x=="DECR") & !any(x=="INCR") & !any(x=="Zero"))) %>%
  mutate(WT.variable.sig=apply(.[,WT.sig.cols],1,function(x) any(x=="DECR") & any(x=="INCR") & !any(x=="Zero"))) %>%
  mutate(spo13.INCR.only=apply(.[,spo13.sig.cols],1,function(x) any(x=="INCR") & !any(x=="DECR") & !any(x=="Zero") )) %>%
  mutate(spo13.DECR.only=apply(.[,spo13.sig.cols],1,function(x) any(x=="DECR") & !any(x=="INCR") & !any(x=="Zero"))) %>%
  mutate(spo13.variable.sig=apply(.[,spo13.sig.cols],1,function(x) any(x=="DECR") & any(x=="INCR") & !any(x=="Zero"))) %>%
  # need to check whether any Zero in OTHER strain sig cols -- there are some things that are sig and 
    mutate(WT_sig=apply(.[,c("WT.INCR.only","WT.DECR.only","WT.variable.sig")],1,function(x) any(x==T))) %>%
    mutate(spo13_sig=apply(.[,c("spo13.INCR.only","spo13.DECR.only","spo13.variable.sig")],1,function(x) any(x==T))) %>%
    mutate(WT_spo13_sig=WT_sig&spo13_sig&!NC) %>%
    mutate(WT_spo13_INCR=WT.INCR.only&spo13.INCR.only&!NC) %>%
    mutate(WT_spo13_DECR=WT.DECR.only&spo13.DECR.only&!NC) %>%
    mutate(WT_spo13_VAR=WT_spo13_sig&!WT_spo13_INCR&!WT_spo13_DECR&!NC) %>%
    mutate(WT_only_sig=WT_sig&!spo13_sig&!NC) %>%
    mutate(spo13_only_sig=spo13_sig&!WT_sig&!NC) %>%
    mutate(WT_INCR=WT_only_sig==T&WT.INCR.only==T&!NC) %>%
    mutate(WT_DECR=WT_only_sig==T&WT.DECR.only==T&!NC) %>%
    mutate(WT_VAR=WT_only_sig==T&WT.variable.sig==T&!NC) %>%
    mutate(spo13_INCR=spo13_only_sig==T&spo13.INCR.only==T&!NC) %>%
    mutate(spo13_DECR=spo13_only_sig==T&spo13.DECR.only==T&!NC) %>%
    mutate(spo13_VAR=spo13_only_sig==T&spo13.variable.sig==T&!NC) %>%
    mutate(sig.cat=ifelse(WT_spo13_INCR==T,"WT_spo13_INCR",
                    ifelse(WT_spo13_DECR==T,"WT_spo13_DECR",
                        ifelse(WT_spo13_VAR==T,"WT_spo13_VAR",
                            ifelse(WT_INCR==T,"WT_INCR",
                                 ifelse(WT_DECR==T,"WT_DECR",
                                        ifelse(WT_VAR==T,"WT_VAR",
                                        ifelse(spo13_INCR==T,"spo13_INCR",
                                 ifelse(spo13_DECR==T,"spo13_DECR",
                                        ifelse(spo13_VAR==T,"spo13_VAR","NC")))))))))) %>%
    mutate(time.first.sig_WT_INCR=apply(.[,WT.sig.cols],1,function(x) which(x=="INCR")[1])) %>% # only returns time FIRST INCR
mutate(time.first.sig_WT_DECR=apply(.[,WT.sig.cols],1,function(x) which(x=="DECR")[1])) %>%
mutate(time.first.sig_WT_VAR=apply(.[,WT.sig.cols],1,function(x) which(x=="INCR"|x=="DECR")[1])) %>%
mutate(time.first.sig_spo13_INCR=apply(.[,spo13.sig.cols],1,function(x) which(x=="INCR")[1])) %>% # only returns time FIRST INCR
mutate(time.first.sig_spo13_DECR=apply(.[,spo13.sig.cols],1,function(x) which(x=="DECR")[1])) %>%
mutate(time.first.sig_spo13_VAR=apply(.[,spo13.sig.cols],1,function(x) which(x=="INCR"|x=="DECR")[1])) %>% 
    # change all to actual time number from t0
mutate(time.first.sig_WT_INCR=as.factor(as.numeric(alltimes[2:10][time.first.sig_WT_INCR]))) %>%
mutate(time.first.sig_WT_DECR=as.factor(as.numeric(alltimes[2:10][time.first.sig_WT_DECR]))) %>%
mutate(time.first.sig_WT_VAR=as.factor(as.numeric(alltimes[2:10][time.first.sig_WT_INCR]))) %>%
mutate(time.first.sig_spo13_INCR=as.factor(as.numeric(alltimes[2:10][time.first.sig_spo13_INCR]))) %>%
mutate(time.first.sig_spo13_DECR=as.factor(as.numeric(alltimes[2:10][time.first.sig_spo13_DECR]))) %>%
mutate(time.first.sig_spo13_VAR=as.factor(as.numeric(alltimes[2:10][time.first.sig_spo13_INCR])))



# # WT and spo13 sig from t0 in either PHOSPHOS
# 10 categories total
WT.spo13.phos.fin.df<-WT.spo13.phos.fin.df %>%
    mutate(NC=apply(.[,c(WT.sig.cols,spo13.sig.cols)],1,function(x) any(x=="Zero"))) %>%
    mutate(WT.INCR.only=apply(.[,WT.sig.cols],1,function(x) any(x=="INCR") & !any(x=="DECR") & !any(x=="Zero"))) %>%
  mutate(WT.DECR.only=apply(.[,WT.sig.cols],1,function(x) any(x=="DECR") & !any(x=="INCR") & !any(x=="Zero"))) %>%
  mutate(WT.variable.sig=apply(.[,WT.sig.cols],1,function(x) any(x=="DECR") & any(x=="INCR") & !any(x=="Zero"))) %>%
  mutate(spo13.INCR.only=apply(.[,spo13.sig.cols],1,function(x) any(x=="INCR") & !any(x=="DECR") & !any(x=="Zero") )) %>%
  mutate(spo13.DECR.only=apply(.[,spo13.sig.cols],1,function(x) any(x=="DECR") & !any(x=="INCR") & !any(x=="Zero"))) %>%
  mutate(spo13.variable.sig=apply(.[,spo13.sig.cols],1,function(x) any(x=="DECR") & any(x=="INCR") & !any(x=="Zero"))) %>%
  # need to check whether any Zero in OTHER strain sig cols -- there are some things that are sig and 
    mutate(WT_sig=apply(.[,c("WT.INCR.only","WT.DECR.only","WT.variable.sig")],1,function(x) any(x==T))) %>%
    mutate(spo13_sig=apply(.[,c("spo13.INCR.only","spo13.DECR.only","spo13.variable.sig")],1,function(x) any(x==T))) %>%
    mutate(WT_spo13_sig=WT_sig&spo13_sig&!NC) %>%
    mutate(WT_spo13_INCR=WT.INCR.only&spo13.INCR.only&!NC) %>%
    mutate(WT_spo13_DECR=WT.DECR.only&spo13.DECR.only&!NC) %>%
    mutate(WT_spo13_VAR=WT_spo13_sig&!WT_spo13_INCR&!WT_spo13_DECR&!NC) %>%
    mutate(WT_only_sig=WT_sig&!spo13_sig&!NC) %>%
    mutate(spo13_only_sig=spo13_sig&!WT_sig&!NC) %>%
    mutate(WT_INCR=WT_only_sig==T&WT.INCR.only==T&!NC) %>%
    mutate(WT_DECR=WT_only_sig==T&WT.DECR.only==T&!NC) %>%
    mutate(WT_VAR=WT_only_sig==T&WT.variable.sig==T&!NC) %>%
    mutate(spo13_INCR=spo13_only_sig==T&spo13.INCR.only==T&!NC) %>%
    mutate(spo13_DECR=spo13_only_sig==T&spo13.DECR.only==T&!NC) %>%
    mutate(spo13_VAR=spo13_only_sig==T&spo13.variable.sig==T&!NC) %>%
    mutate(sig.cat=ifelse(WT_spo13_INCR==T,"WT_spo13_INCR",
                    ifelse(WT_spo13_DECR==T,"WT_spo13_DECR",
                        ifelse(WT_spo13_VAR==T,"WT_spo13_VAR",
                            ifelse(WT_INCR==T,"WT_INCR",
                                 ifelse(WT_DECR==T,"WT_DECR",
                                        ifelse(WT_VAR==T,"WT_VAR",
                                        ifelse(spo13_INCR==T,"spo13_INCR",
                                 ifelse(spo13_DECR==T,"spo13_DECR",
                                        ifelse(spo13_VAR==T,"spo13_VAR","NC")))))))))) %>%
    mutate(time.first.sig_WT_INCR=apply(.[,WT.sig.cols],1,function(x) which(x=="INCR")[1])) %>% # only returns time FIRST INCR
mutate(time.first.sig_WT_DECR=apply(.[,WT.sig.cols],1,function(x) which(x=="DECR")[1])) %>%
mutate(time.first.sig_WT_VAR=apply(.[,WT.sig.cols],1,function(x) which(x=="INCR"|x=="DECR")[1])) %>%
mutate(time.first.sig_spo13_INCR=apply(.[,spo13.sig.cols],1,function(x) which(x=="INCR")[1])) %>% # only returns time FIRST INCR
mutate(time.first.sig_spo13_DECR=apply(.[,spo13.sig.cols],1,function(x) which(x=="DECR")[1])) %>%
mutate(time.first.sig_spo13_VAR=apply(.[,spo13.sig.cols],1,function(x) which(x=="INCR"|x=="DECR")[1])) %>% 
    # change all to actual time number from t0
mutate(time.first.sig_WT_INCR=as.factor(as.numeric(alltimes[2:10][time.first.sig_WT_INCR]))) %>%
mutate(time.first.sig_WT_DECR=as.factor(as.numeric(alltimes[2:10][time.first.sig_WT_DECR]))) %>%
mutate(time.first.sig_WT_VAR=as.factor(as.numeric(alltimes[2:10][time.first.sig_WT_INCR]))) %>%
mutate(time.first.sig_spo13_INCR=as.factor(as.numeric(alltimes[2:10][time.first.sig_spo13_INCR]))) %>%
mutate(time.first.sig_spo13_DECR=as.factor(as.numeric(alltimes[2:10][time.first.sig_spo13_DECR]))) %>%
mutate(time.first.sig_spo13_VAR=as.factor(as.numeric(alltimes[2:10][time.first.sig_spo13_INCR])))

```


# Sig filtering

```{r,echo=F,message=F,warning=F}

# Filter to only include rows that have an INCR or DECR
# and have real t.test pvals at all 9 comparison points
# 
# 
# All NAs in ssig colss have been replaced with character string "Zero"

# protg filtering to sig
WT.protg.fin.df.sig<-WT.protg.fin.df %>% filter(if_any(WT.sig.cols,function(x) {x=="DECR"|x=="INCR"})) %>% filter(if_all(WT.sig.cols,function(x) {!x=="Zero"}))

spo13.protg.fin.df.sig<-spo13.protg.fin.df %>% filter(if_any(spo13.sig.cols,function(x) {x=="DECR"|x=="INCR"})) %>% filter(if_all(spo13.sig.cols,function(x) {!x=="Zero"}))


# sig in EITHER, detected in BOTH (WT.spo13.protg all4reps=T and WT.spo13.protg.fin.df meanss all WT.mean.cols finite non-NA values)
# 
WT.spo13.protg.fin.df.sig<-WT.spo13.protg.fin.df %>% filter(if_any(c(WT.sig.cols,spo13.sig.cols),function(x) {x=="DECR"|x=="INCR"})) %>% filter(if_all(c(WT.sig.cols,spo13.sig.cols),function(x) {!x=="Zero"}))


# sig DIFFERENT between WT and spo13 according to fc and t.tests
WT.spo13.protg.fin.df.sigdiff<-WT.spo13.protg.fin.df %>% filter(if_any(WT.spo13.sig.cols,function(x) {x=="DECR"|x=="INCR"})) %>% filter(if_all(WT.spo13.sig.cols,function(x) {!x=="Zero"})) 

# phospho-site dfs filtering to sig

WT.phos.fin.df.sig<-WT.phos.fin.df %>% filter(if_any(WT.sig.cols,function(x) {x=="DECR"|x=="INCR"})) %>% filter(if_all(WT.sig.cols,function(x) {!x=="Zero"}))


spo13.phos.fin.df.sig<-spo13.phos.fin.df %>% filter(if_any(spo13.sig.cols,function(x) {x=="DECR"|x=="INCR"})) %>%
filter(if_all(spo13.sig.cols,function(x) {!x=="Zero"}))


# sig in EITHER, detected in BOTH
WT.spo13.phos.fin.df.sig<-WT.spo13.phos.fin.df %>% filter(if_any(c(WT.sig.cols,spo13.sig.cols),function(x) {x=="DECR"|x=="INCR"})) %>% filter(if_all(c(WT.sig.cols,spo13.sig.cols),function(x) {!x=="Zero"}))


# sig DIFFERENT between WT and spo13 according to fc and t.tests
WT.spo13.phos.fin.df.sigdiff<-WT.spo13.phos.fin.df %>% filter(if_any(WT.spo13.sig.cols,function(x) {x=="DECR"|x=="INCR"})) %>% filter(if_all(WT.spo13.sig.cols,function(x) {!x=="Zero"}))

# #nrows all together
# WT.protg.fin.df.sig %>% nrow() #297
# spo13.protg.fin.df.sig %>% nrow() #334
# WT.spo13.protg.fin.df.sig %>% nrow() #414
# WT.spo13.protg.fin.df.sigdiff %>% nrow() #50
# WT.phos.fin.df.sig %>% nrow() #1233
# spo13.phos.fin.df.sig %>% nrow() #936
# WT.spo13.phos.fin.df.sig %>% nrow() #982
# WT.spo13.phos.fin.df.sigdiff %>% nrow() #305


```

# SIG CAT PIES
```{r,echo=F,message=F,warning=F}

library(ggplot2)
library(ggrepel)
library(patchwork)


### ACTUAL PIE CHARTS
# Pie chart of number of sig 
# thiss one includes INCR/DECR/VAR
WT_protg_sig_pie<-WT.protg.fin.df %>% 
    group_by(sig.cat) %>%
    summarize(n=n()) %>%
    ggplot(aes(x="",y=n,fill=sig.cat,label=n))+
    geom_bar(stat="identity")+
    geom_label_repel()+
    coord_polar("y", start=0)+
    theme_void()+
    scale_fill_manual(values=c("#7CAE00", "#00BFC4", "#C77CFF","blue"))+
    ggtitle("WT sig proteins")+
    theme(legend.title=element_blank())

WT_phos_sig_pie<-WT.phos.fin.df %>% 
    group_by(sig.cat) %>%
    summarize(n=n())%>%
    ggplot(aes(x="",y=n,fill=sig.cat,label=n))+
    geom_bar(stat="identity")+
    geom_label_repel()+
    coord_polar("y", start=0)+
    theme_void()+
    scale_fill_manual(values=c("#7CAE00", "#00BFC4", "#C77CFF","blue"))+
    ggtitle("WT sig sites")+
    theme(legend.title=element_blank())

print_plot(WT_protg_sig_pie+WT_phos_sig_pie,
           "WT_protg_phos_sig_pie.pdf",5,5)

# WT and spo13 SIGDIFF 
# 
# PROTEINS INCL NC
WT_spo13_protg_sigdiff_pie<-WT.spo13.protg.fin.df %>% 
    group_by(sigdiff.sig.cat) %>%
    summarize(n=n())%>%
    ggplot(aes(x="",y=n,fill=sigdiff.sig.cat,label=n))+
    geom_bar(stat="identity")+
    geom_label_repel(max.overlaps=Inf)+
    coord_polar("y", start=0)+
    theme_void()+
    ggtitle("WT v spo13 proteins sigdiff")+
    theme(legend.title=element_blank())

# SIG ONLY (no NCs)
WT_spo13_protg_sigdiff_sigonly_pie<-WT.spo13.protg.fin.df.sigdiff %>% 
    group_by(sigdiff.sig.cat) %>%
    summarize(n=n()) %>%
    ggplot(aes(x="",y=n,fill=sigdiff.sig.cat,label=n))+
    geom_bar(stat="identity")+
    geom_label_repel()+
    coord_polar("y", start=0)+
    theme_void()+
    scale_fill_manual(values=c("#7CAE00", "#00BFC4", "#C77CFF"))+
    ggtitle("WT v spo13 sig diff proteins")

WT_spo13_phos_sigdiff_pie<-WT.spo13.phos.fin.df %>% 
    group_by(sigdiff.sig.cat) %>%
    summarize(n=n())%>%
    ggplot(aes(x="",y=n,fill=sigdiff.sig.cat,label=n))+
    geom_bar(stat="identity")+
    geom_label_repel(max.overlaps=Inf)+
    coord_polar("y", start=0)+
    theme_void()+
    ggtitle("WT v spo13 sites sigdiff")+
    theme(legend.title=element_blank())

# SIGDIFF ONLY (no NCs)
WT_spo13_phos_sigdiff_sigonly_pie<-WT.spo13.phos.fin.df.sigdiff %>% 
    group_by(sigdiff.sig.cat) %>%
    summarize(n=n())%>%
    ggplot(aes(x="",y=n,fill=sigdiff.sig.cat,label=n))+
    geom_bar(stat="identity")+
    geom_label_repel()+
    coord_polar("y", start=0)+
    theme_void()+
    scale_fill_manual(values=c("#7CAE00", "#00BFC4", "#C77CFF"))+
    ggtitle("WT v spo13 sig diff sites")


print_plot((WT_spo13_protg_sigdiff_pie+WT_spo13_protg_sigdiff_sigonly_pie)/
               (WT_spo13_phos_sigdiff_pie+WT_spo13_phos_sigdiff_sigonly_pie),
           "WT_spo13_protg_phos_sigdiff_pie.pdf",10,5)

# spo13 and WT sig from t0
WT_spo13_t0_protg_sig_pie<-WT.spo13.protg.fin.df %>% 
    group_by(sig.cat) %>%
    summarize(n=n())%>%
    ggplot(aes(x="",y=n,fill=sig.cat,label=n))+
    geom_bar(stat="identity")+
    geom_label_repel(max.overlaps=Inf)+
    coord_polar("y", start=0)+
    theme_void()+
    #scale_fill_manual(values=c("#7CAE00", "#00BFC4", "#C77CFF","blue"))+
    ggtitle("WT and spo13 sig t0 protein")+
    theme(legend.title=element_blank())

WT_spo13_t0_phos_sig_pie<-WT.spo13.phos.fin.df %>% 
    group_by(sig.cat) %>%
    summarize(n=n())%>%
    ggplot(aes(x="",y=n,fill=sig.cat,label=n))+
    geom_bar(stat="identity")+
    geom_label_repel(max.overlaps=Inf)+
    coord_polar("y", start=0)+
    theme_void()+
    #scale_fill_manual(values=c("#7CAE00", "#00BFC4", "#C77CFF","blue"))+
    ggtitle("WT and spo13 sig t0 phos")+
    theme(legend.title=element_blank())

# note color categories are not the ssame between both plots
# because there were not any proteins called "VAR" from t0 in spo13 only or in WT+spo13
print_plot(WT_spo13_t0_protg_sig_pie+WT_spo13_t0_phos_sig_pie,
           "WT_spo13_t0_protg_phos_sig_pie.pdf",10,5)



```


# Sig clustering

```{r,echo=F,message=F,warning=F}

library(scales)
library(pheatmap)

#sig dfs
# WT.protg.fin.df.sig
# spo13.protg.fin.df.sig
# WT.spo13.protg.fin.df.sig
# WT.spo13.protg.fin.df.sigdiff
# 
# WT.phos.fin.df.sig
# spo13.phos.fin.df.sig
# WT.spo13.phos.fin.df.sig
# WT.spo13.phos.fin.df.sigdiff


#WT sig proteinss hclust
WT.protg.sig.hclust<-hclust(dist(t(apply(as.matrix(WT.protg.fin.df.sig[,WT.mean.cols]),1,scale))),method="ward.D2")

# how many clusters are appropriate
wssplot(as.matrix(WT.protg.fin.df.sig[,WT.mean.cols]),nc=20,iter.max=50) #6?

# clustering pheatmap function
# note the scale=row argument indicates
# that the values are centered and scaled in the row direction
# see ?pheatmap() scale argument.
# (according to a blog I read online this means that mean-scaling rowwise
# before putting into pheatmap is unnecessary)
get.hclust.ward.heatmap<-function(df,hclustdf,meancols,numclust,title){
  hclustdf.cutree<-cutree(hclustdf,k=numclust)
  hclustdf.row_annots<-data.frame(cluster = as.factor(hclustdf.cutree))
  # reassign colors of annotations (cluster key) so that they are the R default
  # this way I can make plots match the cluster cols from the heatmap more easily
  colors<-hue_pal()(numclust)
  names(colors)<-as.character(1:numclust)
  levels(colors)<-levels(hclustdf.row_annots$cluster)
  ann_colors = list(cluster = colors)
  rownames(df)<-rownames(df)
  clust.heatmap<-pheatmap(df[,meancols], main=paste0(title,", n=",nrow(df)),scale = 'row', clustering_method="ward.D2",annotation_row=hclustdf.row_annots,cutree_rows = numclust, cutree_cols = 10, show_rownames = F, show_colnames = T, labels_col=alltimes, cluster_cols = F,border_color = NA,annotation_colors=ann_colors)
  return(clust.heatmap)
}

WT.protg.sig.heatmap<-get.hclust.ward.heatmap(WT.protg.fin.df.sig,WT.protg.sig.hclust,WT.mean.cols,6,"WT sig proteins")

print_plot(WT.protg.sig.heatmap,
           "WT_protg_sig_heatmap.pdf",6,7)

#spo13 sig proteinss hclust
spo13.protg.sig.hclust<-hclust(dist(t(apply(as.matrix(spo13.protg.fin.df.sig[,spo13.mean.cols]),1,scale))),method="ward.D2")

# how many clusters are appropriate
wssplot(as.matrix(spo13.protg.fin.df.sig[,spo13.mean.cols]),nc=20,iter.max=50) #6?

spo13.protg.sig.heatmap<-get.hclust.ward.heatmap(spo13.protg.fin.df.sig,spo13.protg.sig.hclust,spo13.mean.cols,6,"spo13 sig proteins") #321

# WT v spo13 ssig from t0 protein

WT.spo13.protg.sig.hclust<-hclust(dist(t(apply(as.matrix(WT.spo13.protg.fin.df.sig[,c(WT.mean.cols, spo13.mean.cols)]),1,scale))),method="ward.D2")

# how many clusters are appropriate
wssplot(as.matrix(WT.spo13.protg.fin.df.sig[,c(WT.mean.cols,spo13.mean.cols)]),nc=20,iter.max=50) #6?

# new function for WT v spo13 20 column clustering.

get.hclust.ward.heatmap.comb<-function(df,hclustdf,meancols,numclust,title){
  hclustdf.cutree<-cutree(hclustdf,k=numclust)
  hclustdf.row_annots<-data.frame(cluster = as.factor(hclustdf.cutree))
  strain.col_annots<-data.frame(strain=as.factor(c(rep("WT",10),rep("spo13",10))))
  rownames(strain.col_annots)<-colnames(df[,meancols])
  colors<-hue_pal()(numclust)
  names(colors)<-as.character(1:numclust)
  levels(colors)<-levels(hclustdf.row_annots$cluster)
  strain_colors<-hue_pal()(2)
  names(strain_colors)<-c("WT","spo13")
  levels(strain_colors)<-levels(strain.col_annots$strain)
  ann_colors = list(cluster = colors,strain=strain_colors)
  #rownames(hclustdf.row_annots)<-rownames(df)
 rownames(df)<-rownames(df)
  clust.heatmap<-pheatmap(df[,meancols], main=paste0(title," n=",nrow(df)),scale = 'row', clustering_method = "ward.D2",annotation_row=hclustdf.row_annots,annotation_col =  strain.col_annots,cutree_rows = numclust, cutree_cols = 20, show_rownames = F, show_colnames = T, cluster_cols = F,labels_col=rep(alltimes,2),border_color = "NA",annotation_colors=ann_colors
                          )
  return(clust.heatmap)
}


WT.spo13.protg.sig.heatmap<-get.hclust.ward.heatmap.comb(WT.spo13.protg.fin.df.sig,WT.spo13.protg.sig.hclust,c(WT.mean.cols,spo13.mean.cols),10,"WT and spo13 sig protein") 

print_plot(WT.spo13.protg.sig.heatmap,
           "WT_spo13_protg_sig_heatmap.pdf",5,8)

# WT v spo13 SIG DIFF prtines n=50
# 
# new function for WT v spo13 20 column clustering. # WITH ROW LABELS

get.hclust.ward.heatmap.comb.label<-function(df,hclustdf,meancols,numclust,title){
  hclustdf.cutree<-cutree(hclustdf,k=numclust)
  hclustdf.row_annots<-data.frame(cluster = as.factor(hclustdf.cutree))
  rownames(hclustdf.row_annots)<-df$Gene.name

  strain.col_annots<-data.frame(strain=ifelse(grepl("WT",meancols),"WT","spo13")) # note this means WT must always come first in function call
colnames(df[,meancols])<-gsub("WT.mean.|spo13.mean.","",colnames(df[,meancols]))
  rownames(strain.col_annots)<-colnames(df[,meancols])

  rownames(df)<-df$Gene.name


  colors<-hue_pal()(numclust)
  names(colors)<-as.character(1:numclust)
  levels(colors)<-levels(hclustdf.row_annots$cluster)
  strain_colors<-hue_pal()(2)
  names(strain_colors)<-c("WT","spo13")
  levels(strain_colors)<-levels(strain.col_annots$strain)
  ann_colors = list(cluster = colors,strain=strain_colors)


  clust.heatmap<-pheatmap(df[,meancols], main=paste0(title," n=",nrow(df)),scale = 'row', clustering_method = "ward.D2",annotation_row=hclustdf.row_annots,annotation_col=strain.col_annots,cutree_rows = numclust, cutree_cols = 20, show_rownames = T, show_colnames = T, cluster_cols = F,border_color = "NA",annotation_colors=ann_colors,labels_col=gsub("WT.mean.|spo13.mean.","",colnames(df[,meancols]))
                          )
  return(clust.heatmap)
}

WT.spo13.protg.sigdiff.hclust<-hclust(dist(t(apply(as.matrix(WT.spo13.protg.fin.df.sigdiff[,c(WT.mean.cols, spo13.mean.cols)]),1,scale))),method="ward.D2")

# how many clusters are appropriate
wssplot(as.matrix(WT.spo13.protg.fin.df.sigdiff[,c(WT.mean.cols,spo13.mean.cols)]),nc=20,iter.max=50) #8

WT.spo13.protg.sigdiff.heatmap<-get.hclust.ward.heatmap.comb.label(WT.spo13.protg.fin.df.sigdiff,WT.spo13.protg.sigdiff.hclust,c(WT.mean.cols,spo13.mean.cols),8,"WT and spo13 sig diff protein") 

print_plot(WT.spo13.protg.sigdiff.heatmap,
           "WT_spo13_sigdiff_prot_heatmap_8_10.pdf",8,10)

## PHOSPHO ####

#WT sig SITES hclust
WT.phos.sig.hclust<-hclust(dist(t(apply(as.matrix(WT.phos.fin.df.sig[,WT.mean.cols]),1,scale))),method="ward.D2")

# how many clusters are appropriate
wssplot(as.matrix(WT.phos.fin.df.sig[,WT.mean.cols]),nc=20,iter.max=50) #11?

WT.phos.sig.heatmap<-get.hclust.ward.heatmap(WT.phos.fin.df.sig,WT.phos.sig.hclust,WT.mean.cols,11,"WT sig sites")

print_plot(WT.phos.sig.heatmap,
           "WT_phos_sig_heatmap.pdf",6,7)

#spo13 sig sites hclust
spo13.phos.sig.hclust<-hclust(dist(t(apply(as.matrix(spo13.phos.fin.df.sig[,spo13.mean.cols]),1,scale))),method="ward.D2")

# how many clusters are appropriate
wssplot(as.matrix(spo13.phos.fin.df.sig[,spo13.mean.cols]),nc=20,iter.max=50) #8?

spo13.phos.sig.heatmap<-get.hclust.ward.heatmap(spo13.phos.fin.df.sig,spo13.phos.sig.hclust,spo13.mean.cols,8,"spo13 sig sites")

# WT v spo13 sig from t0 phospho-sites

WT.spo13.phos.sig.hclust<-hclust(dist(t(apply(as.matrix(WT.spo13.phos.fin.df.sig[,c(WT.mean.cols, spo13.mean.cols)]),1,scale))),method="ward.D2")

# how many clusters are appropriate
wssplot(as.matrix(WT.spo13.phos.fin.df.sig[,c(WT.mean.cols,spo13.mean.cols)]),nc=20,iter.max=50) #9?

WT.spo13.phos.sig.heatmap<-get.hclust.ward.heatmap.comb(WT.spo13.phos.fin.df.sig,WT.spo13.phos.sig.hclust,c(WT.mean.cols,spo13.mean.cols),10,"WT and spo13 sig sites") 

print_plot(WT.spo13.phos.sig.heatmap,
           "WT_spo13_phos_sig_heatmap.pdf",5,8)

# WT v spo13 SIG DIFF sites
WT.spo13.phos.sigdiff.hclust<-hclust(dist(t(apply(as.matrix(WT.spo13.phos.fin.df.sigdiff[,c(WT.mean.cols, spo13.mean.cols)]),1,scale))),method="ward.D2")

# how many clusters are appropriate
wssplot(as.matrix(WT.spo13.phos.fin.df.sigdiff[,c(WT.mean.cols,spo13.mean.cols)]),nc=20,iter.max=50) #5?

WT.spo13.phos.sigdiff.heatmap<-get.hclust.ward.heatmap.comb(WT.spo13.phos.fin.df.sigdiff,WT.spo13.phos.sigdiff.hclust,c(WT.mean.cols,spo13.mean.cols),15,"WT and spo13 sig diff sites") 

print_plot(WT.spo13.phos.sigdiff.heatmap,
           "WT_spo13_phos_sigdiff_clust_heatmap.pdf",6,8)


```


# Add cluster col to sig dfs
```{r,echo=F,message=F,warning=F}

# add cluster col to the sig dfs
# 
# WT.protg.sig.heatmap # cluster=6
# spo13.protg.sig.heatmap # cluster=6
# WT.spo13.protg.sig.heatmap # cluster=10
# WT.spo13.protg.sigdiff.heatmap # cluster=8


#protg
WT.protg.fin.df.sig <- WT.protg.fin.df.sig %>%
  mutate(ward.cluster=cutree(WT.protg.sig.heatmap$tree_row, k= 6))
spo13.protg.fin.df.sig <- spo13.protg.fin.df.sig %>%
  mutate(ward.cluster=cutree(spo13.protg.sig.heatmap$tree_row, k= 6))
WT.spo13.protg.fin.df.sig <- WT.spo13.protg.fin.df.sig %>%
  mutate(ward.cluster=cutree(WT.spo13.protg.sig.heatmap$tree_row, k= 10))
WT.spo13.protg.fin.df.sigdiff <- WT.spo13.protg.fin.df.sigdiff %>%
  mutate(ward.cluster=cutree(WT.spo13.protg.sigdiff.heatmap$tree_row, k= 8))

# WT.phos.sig.heatmap # cluster=11
# spo13.phos.sig.heatmap # cluster=8
# WT.spo13.phos.sig.heatmap # cluster=10
# WT.spo13.phos.sigdiff.heatmap # cluster=15

#phos
WT.phos.fin.df.sig <- WT.phos.fin.df.sig %>%
  mutate(ward.cluster=cutree(WT.phos.sig.heatmap$tree_row, k= 11))
spo13.phos.fin.df.sig <- spo13.phos.fin.df.sig %>%
  mutate(ward.cluster=cutree(spo13.phos.sig.heatmap$tree_row, k= 8))
WT.spo13.phos.fin.df.sig <- WT.spo13.phos.fin.df.sig %>%
  mutate(ward.cluster=cutree(WT.spo13.phos.sig.heatmap$tree_row, k= 10))
WT.spo13.phos.fin.df.sigdiff <- WT.spo13.phos.fin.df.sigdiff %>%
  mutate(ward.cluster=cutree(WT.spo13.phos.sigdiff.heatmap$tree_row, k= 15))

```


# Med Lineplots of clusters

```{r,echo=F,message=F,warning=F}
# Median Lineplots of clustering results.

# lineplots of ward clusters median across timecourse.
# mean.scaled columns

library(tidyr)
library(gtools)
library(forcats)

# function which returns single facet_wrap plot of all clusters together
get.cluster.lineplot<-function(df,strain,var.type,title){
  prefix<-paste0("^scaled.",strain,".mean.")
  cols<-grep(prefix,colnames(df),value=T)
  var<-grep(var.type,colnames(df),value=T)
  plot<-df %>%
    select(var,all_of(cols),ward.cluster) %>%
    pivot_longer(-c(var,ward.cluster),names_prefix=prefix,names_to="time",values_to="intensity") %>% 
    mutate(cluster=as.factor(ward.cluster)) %>%
      group_by(cluster) %>%
      mutate(n=n()/10) %>%
      ungroup() %>%
      mutate(cluster=paste0(cluster,", n=",n)) %>%
      mutate(cluster_order=factor(cluster,levels=unique(mixedsort(cluster,decreasing=F)))) %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=cluster_order))+
  facet_wrap(~cluster_order,nrow=2)+
  stat_summary(aes(y=intensity,color=cluster_order),fun=median,geom="line",size=1)+
  labs(x="time",y="scaled intensity",title=paste0(title," n=", nrow(df)))+
    #ylim(c(0,2))+
    theme_classic()+
  theme(axis.text.x=element_text(angle=90), legend.position="none", axis.text=element_text(size=rel(1)),axis.title=element_text(size=rel(1)),legend.text=element_text(size=rel(1)),legend.title = element_text(size=rel(1)),strip.text=element_text(size=rel(1)))
    return(plot)
}

# function which returns a single medline of one cluster
get.sing.cluster.lineplot<-function(df,strain,var.type,sing_cluster,title){
    prefix<-paste0("^scaled.",strain,".mean.")
  cols<-grep(prefix,colnames(df),value=T)
  var<-grep(var.type,colnames(df),value=T)
  plot<-df %>%
    select(var,all_of(cols),ward.cluster) %>%
      filter(ward.cluster==sing_cluster) %>%
    pivot_longer(-c(var,ward.cluster),names_prefix=prefix,names_to="time",values_to="intensity") %>% 
    mutate(cluster=as.factor(ward.cluster)) %>%
     group_by(cluster) %>%
      mutate(n=n()/10) %>%
      ungroup() %>%
      mutate(cluster=paste0(cluster,", n=",n)) %>%
      mutate(cluster_order=factor(cluster,levels=unique(mixedsort(cluster,decreasing=F)))) %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=cluster))+
  #facet_wrap(~cluster,nrow=2)+
  stat_summary(aes(y=intensity,color=cluster),fun=median,geom="line",size=1)+
    # stat_summary(aes(y=intensity,fill=cluster),fun.min=function(z) {quantile(z,0.25)},
  #              fun.max=function(z) {quantile(z,0.75)},
  #              geom="ribbon",alpha=0.2)
  labs(x="time",y="scaled intensity",title=paste0(title," n=", nrow(df %>% filter(ward.cluster==sing_cluster))))+
   # ylim(c(NA,2))+
    theme_classic()+
  theme(axis.text.x=element_text(angle=90), legend.position="none", axis.text=element_text(size=rel(1)),axis.title=element_text(size=rel(1)),legend.text=element_text(size=rel(1)),legend.title = element_text(size=rel(1)),strip.text=element_text(size=rel(1)))+
      coord_cartesian(ylim=c(0,2))
    return(plot)
}


# WT AND spo13 phospho-sites cluster lineplot
# mean.scaled.cols
# colored by strain

# function returns a single facet_wrap plot with all clusters of shared WT.spo13 clust together
get.cluster.lineplot.both<-function(df,var.type,title){
  var<-grep(var.type,colnames(df),value=T)
  plot<-df %>%
    select(var,all_of(c(WT.spo13.scaled.mean.cols)),ward.cluster) %>%
    pivot_longer(-c(var,ward.cluster),names_prefix="scaled.both.",names_to="time",values_to="intensity") %>% 
    mutate(strain=ifelse(grepl("WT",time),"WT","spo13")) %>%
  mutate(time=gsub("WT.mean.|spo13.mean.","",time)) %>%
  mutate(cluster=as.factor(ward.cluster)) %>%
      group_by(cluster) %>%
      mutate(n=n()/20) %>%
      ungroup() %>%
      mutate(cluster=paste0(cluster,", n=",n)) %>%
    mutate(cluster_order=factor(cluster,levels=unique(mixedsort(cluster,decreasing=F)))) %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=strain))+
  facet_wrap(~cluster_order)+
  stat_summary(aes(y=intensity,color=strain),fun=median,geom="line",size=1)+
  labs(x="time",y="scaled intensity",title=paste0(title," n=", nrow(df)))+
    #ylim(c(0,2))+
    theme_classic()+
  theme(axis.text.x=element_text(angle=90), axis.text=element_text(size=rel(1)),axis.title=element_text(size=rel(1)),legend.text=element_text(size=rel(1)),legend.title = element_text(size=rel(1)),strip.text=element_text(size=rel(1)))
    return(plot)
}

WT_protg_sig_clust_line<-get.cluster.lineplot(WT.protg.fin.df.sig,"WT","Gene.name","WT sig cluster proteins median abundance")

WT_phos_sig_clust_line<-get.cluster.lineplot(WT.phos.fin.df.sig,"WT","gene.name_site","WT sig cluster phospho-sites median abundance")


# WT_protg_sig_clust_1<-get.sing.cluster.lineplot(WT.protg.fin.df.sig,"WT","Gene.name","1","WT sig cluster 1")
# WT_protg_sig_clust_2<-get.sing.cluster.lineplot(WT.protg.fin.df.sig,"WT","Gene.name","2","WT sig cluster 2")
# WT_protg_sig_clust_3<-get.sing.cluster.lineplot(WT.protg.fin.df.sig,"WT","Gene.name","3","WT sig cluster 3")
# WT_protg_sig_clust_4<-get.sing.cluster.lineplot(WT.protg.fin.df.sig,"WT","Gene.name","4","WT sig cluster 4")
# WT_protg_sig_clust_5<-get.sing.cluster.lineplot(WT.protg.fin.df.sig,"WT","Gene.name","5","WT sig cluster 5")
# WT_protg_sig_clust_6<-get.sing.cluster.lineplot(WT.protg.fin.df.sig,"WT","Gene.name","6","WT sig cluster 6")


print_plot(WT_protg_sig_clust_line,
          "WT_protg_sig_clust_line_together.pdf",8,5)
print_plot(WT_phos_sig_clust_line,
          "WT_phos_sig_clust_line_together.pdf",8,5)


# print_plot(WT_protg_sig_clust_1,
#            "WT_protg_sig_clust_1_line.pdf",4,3)
# print_plot(WT_protg_sig_clust_2,
#            "WT_protg_sig_clust_2_line.pdf",4,3)
# print_plot(WT_protg_sig_clust_3,
#            "WT_protg_sig_clust_3_line.pdf",4,3)
# print_plot(WT_protg_sig_clust_4,
#            "WT_protg_sig_clust_4_line.pdf",4,3)
# print_plot(WT_protg_sig_clust_5,
#            "WT_protg_sig_clust_5_line.pdf",4,3)
# print_plot(WT_protg_sig_clust_6,
#            "WT_protg_sig_clust_6_line.pdf",4,3)



# SIG shared proteins
WT_spo13_protg_sig_clust_medlines<-get.cluster.lineplot.both(WT.spo13.protg.fin.df.sig,var.type = "Gene.name",title = "WT and spo13 sig cluster proteins median abundance")

print_plot(WT_spo13_protg_sig_clust_medlines,
          "WT_spo13_protg_sig_clust_medlines_together.pdf",8,5)


#SIG shared sites
WT_spo13_phos_sig_clust_medlines<-get.cluster.lineplot.both(WT.spo13.phos.fin.df.sig,var.type = "gene.name_site",title = "WT and spo13 sig cluster phospho-sites median abundance")

print_plot(WT_spo13_phos_sig_clust_medlines,
          "WT_spo13_phos_sig_clust_medlines_together.pdf",8,5)

# sigdiff proteins
WT_spo13_protg_sigdiff_clust_medlines<-get.cluster.lineplot.both(WT.spo13.protg.fin.df.sigdiff,var.type = "Gene.name",title = "WT and spo13 sigdiff cluster proteins median abundance") 

print_plot(WT_spo13_protg_sigdiff_clust_medlines,
          "WT_spo13_protg_sigdiff_clust_medlines_together.pdf",8,5)

# sig diff sites
WT_spo13_phos_sigdiff_clust_medlines<-get.cluster.lineplot.both(WT.spo13.phos.fin.df.sigdiff,var.type = "gene.name_site",title = "WT and spo13 sigdiff cluster phospho-sites median abundance")

print_plot(WT_spo13_phos_sigdiff_clust_medlines,
           "WT_spo13_phos_sigdiff_clust_medlines_together.pdf",10,7)


```


# export clustering id tables
```{r,echo=F,message=F,warning=F}

library(openxlsx)

# dfs of gene name or site to export
# WT.protg.fin.df
# WT.protg.fin.df.sig
# WT.spo13.protg.fin.df
# WT.spo13.protg.fin.df.sig
# WT.spo13.protg.fin.df.sigdiff
# 
# WT.phos.fin.df
# WT.phos.fin.df.sig
# WT.spo13.phos.fin.df
# WT.spo13.phos.fin.df.sig
# WT.spo13.phos.fin.df.sigdiff


# Dynamic proteins and phospho-site lists
WT.protg.fin.df %>%
    group_by(sig.cat) %>%
    mutate(id=row_number()) %>%
    pivot_wider(names_from=sig.cat,values_from=Gene.name,id_cols=id) %>%
    select(-id) %>%
    #rename_with(~paste0("cluster_",.x,recycle0=T)) %>%
    write.xlsx(.,"WT_protg_sigcat_ids.xlsx")

WT.phos.fin.df %>%
    group_by(sig.cat) %>%
    mutate(id=row_number()) %>%
    pivot_wider(names_from=sig.cat,values_from=gene.name_site,id_cols=id) %>%
    select(-id) %>%
    #rename_with(~paste0("cluster_",.x,recycle0=T)) %>%
    write.xlsx(.,"WT_phos_sigcat_ids.xlsx")

# WT sig from t0 protein
WT.protg.fin.df.sig %>%
    group_by(ward.cluster) %>%
    mutate(id=row_number()) %>%
    pivot_wider(names_from=ward.cluster,values_from=Gene.name,id_cols=id) %>%
    select(-id) %>%
    rename_with(~paste0("cluster_",.x,recycle0=T)) %>%
    write.xlsx(.,"WT_protg_sig_cluster_ids.xlsx")


# WT and spo13 sig protein
WT.spo13.protg.fin.df.sig %>%
    group_by(ward.cluster) %>%
    mutate(id=row_number()) %>%
    pivot_wider(names_from=ward.cluster,values_from=Gene.name,id_cols=id) %>%
    select(-id) %>%
        rename_with(~paste0("cluster_",.x,recycle0=T)) %>%
    write.xlsx(.,"WT_spo13_protg_sig_cluster_ids.xlsx")

# WT and spo13 sigdiff protein
WT.spo13.protg.fin.df.sigdiff %>%
    group_by(ward.cluster) %>%
    mutate(id=row_number()) %>%
    pivot_wider(names_from=ward.cluster,values_from=Gene.name,id_cols=id) %>%
    select(-id) %>%
        rename_with(~paste0("cluster_",.x,recycle0=T)) %>%
    write.xlsx(.,"WT_spo13_protg_sigdiff_cluster_ids.xlsx")

# PHOSPHO LISTS

# WT sig phos
WT.phos.fin.df.sig %>%
    group_by(ward.cluster) %>%
    mutate(id=row_number()) %>%
    pivot_wider(names_from=ward.cluster,values_from=gene.name_site,id_cols=id) %>%
    select(-id) %>%
        rename_with(~paste0("cluster_",.x,recycle0=T)) %>%
    write.xlsx(.,"WT_phos_sig_cluster_ids.xlsx")

# WT and spo13 sig phos
WT.spo13.phos.fin.df.sig %>%
    group_by(ward.cluster) %>%
    mutate(id=row_number()) %>%
    pivot_wider(names_from=ward.cluster,values_from=gene.name_site,id_cols=id) %>%
    select(-id) %>%
        rename_with(~paste0("cluster_",.x,recycle0=T)) %>%
    write.xlsx(.,"WT_spo13_phos_sig_cluster_ids.xlsx")

# WT and spo13 sigdiff phos
WT.spo13.phos.fin.df.sigdiff %>%
    group_by(ward.cluster) %>%
    mutate(id=row_number()) %>%
    pivot_wider(names_from=ward.cluster,values_from=gene.name_site,id_cols=id) %>%
    select(-id) %>%
        rename_with(~paste0("cluster_",.x,recycle0=T)) %>%
    write.xlsx(.,"WT_spo13_phos_sigdiff_cluster_ids.xlsx")

# WT v spo13 timecourse sigdiff sites by category
WT.spo13.phos.fin.df %>%
    group_by(sigdiff.sig.cat) %>%
    mutate(id=row_number()) %>%
    pivot_wider(names_from=sigdiff.sig.cat,values_from=gene.name_site,id_cols=id) %>%
    select(-id) %>%
    #rename_with(~paste0("cluster_",.x,recycle0=T)) %>%
    write.xlsx(.,"WT_spo13_phos_sigdiff_sigcat_ids.xlsx")

# WT v spo13 timecourse sigdiff proteins by category
WT.spo13.protg.fin.df %>%
    group_by(sigdiff.sig.cat) %>%
    mutate(id=row_number()) %>%
    pivot_wider(names_from=sigdiff.sig.cat,values_from=Gene.name,id_cols=id) %>%
    select(-id) %>%
    #rename_with(~paste0("cluster_",.x,recycle0=T)) %>%
    write.xlsx(.,"WT_spo13_protg_sigdiff_sigcat_ids.xlsx")

```


# Sig Prot GO Term Barplots
```{r,echo=F,message=F,warning=F}

library(magrittr)
library(gprofiler2)

# FYI p=value=1 means genes matched that GO term in the list but do not reach significance at all but pvalue=NA occurs when no genes in the input list match that GO term

# make df with all enriched GO terms in all clusters

# WT sig protein GO terms

gost_list<-list()
terms_df<-tribble(~term_name,~source,~term_id)

# NOTE SIGNFICANCE = TRUE
for(i in 1:6){
  gost_list[[i]]<-WT.protg.fin.df.sig %>% filter(ward.cluster==i) %>% select(Gene.name) %>% unlist() %>%
  gost(.,organism='scerevisiae',evcodes=TRUE, significant=TRUE)

  terms_df<-terms_df %>% add_row(gost_list[[i]]$result[,c("term_name","source","term_id")])
}

# filter terms_df for unique rows
unique_terms_df<-terms_df %>% distinct(term_name,source,term_id)

# create dataframe with columns: term_name, source, cluster, p_val
# create separate dataframes for each cluster

# function to return pvals # EDITED FOR TERM_ID instead of term_name # NOTE SIGNFICANT=FALSE
go.term.pvals.termids<-function(genelist,termids){
  go.results<-gost(genelist,organism='scerevisiae',evcodes=TRUE, significant=FALSE)
  p_vals<-vector(length=length(termids),mode="numeric")
  for(i in 1:length(termids)){
    p_vals[[i]]<-go.results$result %>% filter(term_id==termids[i]) %>% select(p_value) %>% as.numeric()
  }
  return(p_vals)
}

# create df with term name, source, and IDs
WT.protg.sig.clust.GO_terms<-as.data.frame(unique_terms_df)

for(i in 1:6){
  WT.protg.sig.clust.GO_terms[,i+3]<-go.term.pvals.termids(WT.protg.fin.df.sig %>% filter(ward.cluster==i) %>% select(Gene.name) %>% unlist(),unique_terms_df$term_id %>% unlist() %>% unname())
}

# rename GO term pval cluster columns
colnames(WT.protg.sig.clust.GO_terms)[4:9]<-paste0("cluster_",1:6)

# export table of all enriched GO terms to table
write.xlsx(WT.protg.sig.clust.GO_terms,"WT_protg_sig_GO_terms.xlsx")

# choose selected GO terms for each cluster
protg_sig_clust1_GO_terms<-c("sporulation","meiotic cell cycle","ubiquitin protein ligase activity","anaphase-promoting complex","prospore membrane","ascospore wall formation","regulation of sister chromatid segregation")

protg_sig_clust2_GO_terms<-c("homologous recombination","meiotic sister chromatid cohesion","homologous chromosome pairing at meiosis","synaptonemal complex organization","meiosis I")

protg_sig_clust3_GO_terms<-c("Nitrogen metabolism","Glutamate biosynthesis","hydroxymethylglutaryl-CoA reductase (NADPH) activity")
    
protg_sig_clust4_GO_terms<-c("ascospore wall assembly","amino sugar metabolic process","fungal-type cell wall organization","polysaccharide metabolic process")

protg_sig_clust5_GO_terms<-c("sporulation resulting in formation of a cellular spore","meiotic cell cycle process")

protg_sig_clust6_GO_terms<-c("Monobactam biosynthesis","Sulfur amino acid biosynthesis")


# ggplot faceted by cluster so all bars between cluster groups are same width

WT_protg_GO_bar_df<-rbind(WT.protg.sig.clust.GO_terms %>%
    filter(term_name %in% protg_sig_clust1_GO_terms) %>%
    mutate(cluster_2=NA) %>%
    mutate(cluster_3=NA) %>%
    mutate(cluster_4=NA) %>%
    mutate(cluster_5=NA) %>%
    mutate(cluster_6=NA),

WT.protg.sig.clust.GO_terms %>%
    filter(term_name %in% protg_sig_clust2_GO_terms) %>%
    mutate(cluster_1=NA) %>%
    mutate(cluster_3=NA) %>%
    mutate(cluster_4=NA) %>%
    mutate(cluster_5=NA) %>%
    mutate(cluster_6=NA),

WT.protg.sig.clust.GO_terms %>%
    filter(term_name %in% protg_sig_clust3_GO_terms) %>%
    mutate(cluster_1=NA) %>%
    mutate(cluster_2=NA) %>%
    mutate(cluster_4=NA) %>%
    mutate(cluster_5=NA) %>%
    mutate(cluster_6=NA),

WT.protg.sig.clust.GO_terms %>%
    filter(term_name %in% protg_sig_clust4_GO_terms) %>%
    mutate(cluster_1=NA) %>%
    mutate(cluster_3=NA) %>%
    mutate(cluster_2=NA) %>%
    mutate(cluster_5=NA) %>%
    mutate(cluster_6=NA),

WT.protg.sig.clust.GO_terms %>%
    filter(term_name %in% protg_sig_clust5_GO_terms) %>%
    mutate(cluster_1=NA) %>%
    mutate(cluster_3=NA) %>%
    mutate(cluster_4=NA) %>%
    mutate(cluster_2=NA) %>%
    mutate(cluster_6=NA),

WT.protg.sig.clust.GO_terms %>%
    filter(term_name %in% protg_sig_clust6_GO_terms) %>%
    mutate(cluster_1=NA) %>%
    mutate(cluster_3=NA) %>%
    mutate(cluster_4=NA) %>%
    mutate(cluster_5=NA) %>%
    mutate(cluster_2=NA))

WT_protg_GO_bar_together_2<-WT_protg_GO_bar_df %>%
    select(!source) %>%
    pivot_longer(-c(term_name,term_id),names_to="cluster",values_to="p_val") %>%
    filter(!is.na(p_val)) %>%
    ggplot(aes(x=interaction(term_name,term_id),y=-log10(p_val),group=cluster))+
    geom_bar(aes(fill=cluster),stat="identity",na.rm=T)+
    geom_hline(yintercept=-log10(0.05),linetype="dashed")+
    facet_grid(cluster ~ .,scales="free_y",space="free_y")+
    coord_flip()+
labs(y="-log10(p-value)",x="",title="")+
    theme_classic()+
    scale_x_discrete(na.translate=F)+
    scale_y_continuous(expand=expansion(mult=c(0,0.2)))+ # 20% more plot area after max value
    theme(axis.text=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),title=element_text(size=rel(1.5)),legend.position="none",
          strip.text=element_text(size=rel(1.5)))

print_plot(WT_protg_GO_bar_together_2,
           "WT_protg_cluster_GO_bar_together_2.pdf",16,13)

library(ggh4x)
WT_protg_GO_bar_together_3<-WT_protg_GO_bar_df %>%
    select(!source) %>%
    pivot_longer(-c(term_name,term_id),names_to="cluster",values_to="p_val") %>%
    filter(!is.na(p_val)) %>%
    ggplot(aes(x=interaction(term_name,term_id),y=-log10(p_val),group=cluster))+
    geom_bar(aes(fill=cluster),stat="identity",na.rm=T)+
    geom_hline(yintercept=-log10(0.05),linetype="dashed")+
    #facet_grid(rows=vars(cluster),scales="free_y",space="free_y")+
    ggh4x::facet_grid2(cluster ~ .,scales = "free", independent = "x", space="free")+
    coord_flip()+
    labs(y="-log10(p-value)",x="",title="")+
    theme_classic()+
    scale_x_discrete(na.translate=F)+
    scale_y_continuous(expand=expansion(mult=c(0,0.2)))+ # 20% more plot area after max value
    theme(axis.text=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),title=element_text(size=rel(1.5)),legend.position="none",
          strip.text.y=element_text(size=rel(1.5)))

print_plot(WT_protg_GO_bar_together_3,
           "WT_protg_cluster_GO_bar_together_3.pdf",16,13)

```


# GO terms of WT phos clusters
```{r,echo=F,message=F,warning=F}

# FYI p=value=1 means genes matched that GO term in the list but do not reach significance at all but pvalue=NA occurs when no genes in the input list match that GO term

# make df with all enriched GO terms in all clusters

# WT sig phos GO terms

gost_list<-list()
terms_df<-tribble(~term_name,~source,~term_id)

# NOTE SIGNFICANCE = TRUE
for(i in 1:11){
  gost_list[[i]]<-WT.phos.fin.df.sig %>% filter(ward.cluster==i) %>% select(Gene.name) %>% unlist() %>%
  gost(.,organism='scerevisiae',evcodes=TRUE, significant=TRUE)
  
  terms_df<-terms_df %>% add_row(gost_list[[i]]$result[,c("term_name","source","term_id")])
}

# filter terms_df for unique rows
unique_terms_df<-terms_df %>% distinct(term_name,source,term_id) 

# create df with term name, source, and IDs
WT.phos.sig.clust.GO_terms<-as.data.frame(unique_terms_df)

for(i in 1:11){
  WT.phos.sig.clust.GO_terms[,i+3]<-go.term.pvals.termids(WT.phos.fin.df.sig %>% filter(ward.cluster==i) %>% select(Gene.name) %>% unlist(),unique_terms_df$term_id %>% unlist() %>% unname())
}

# rename GO term pval cluster columns
colnames(WT.phos.sig.clust.GO_terms)[4:14]<-paste0("cluster_",1:11)

# export table of all enriched GO terms to table

write.xlsx(WT.phos.sig.clust.GO_terms,"WT_phos_sig_GO_terms.xlsx")

# plot selected GO terms
# this list includes 'synaptonemal complex' which is not enriched according to default gprofiler settings

# for go.term.pvals function, need term list and GO_source list

# GO term list
GO_terms<-c("cell cycle","meiosis I","chromosome organization","chromatin remodeling","homologous recombination","synaptonemal complex","nucleolus","cell cortex","protein kinase activity","MAPK signaling pathway - yeast","autophagy","prospore membrane","ubiquitin binding","cytoplasmic stress granule","cytoskeleton","microtubule organizing center","RNA metabolic process","establishment or maintenance of cytoskeleton polarity")

GO_sources<-c("GO:BP","GO:BP","GO:BP","GO:BP","GO:BP","GO:CC","GO:CC","GO:CC","GO:MF","KEGG","GO:BP","GO:CC","GO:MF","GO:CC","GO:CC","GO:CC","GO:BP","GO:BP")


# function to return pvals # NOTE SIGNFICANCE=FALSE
go.term.pvals<-function(genelist,goterms,sourcelist){
  go.results<-gost(genelist,organism='scerevisiae',evcodes=TRUE, significant=FALSE)
  p_vals<-vector(length=length(goterms),mode="numeric")
  for(i in 1:length(goterms)){
    p_vals[[i]]<-go.results$result %>% filter(term_name==goterms[i]) %>%
        filter(source==sourcelist[i]) %>% select(p_value) %>% as.numeric()
  }
  return(p_vals)
}


# heatmap with each column is 1 cluster 1:11 and each row is a GO term
# color corresponds to -log10 pvalue

WT.phos.clust.GO_terms<-as.data.frame(GO_terms)
for(i in 1:11){
  WT.phos.clust.GO_terms[,i+1]<-go.term.pvals(WT.phos.fin.df.sig %>% filter(ward.cluster==i) %>% select(Gene.name) %>% unlist(),GO_terms,GO_sources)
}


colnames(WT.phos.clust.GO_terms)[2:12]<-paste0("cluster_",1:11)


# white and blue heatmap

WT_phos_GO_terms<-WT.phos.clust.GO_terms %>%
    filter(!(GO_terms %in% c("MAPK signaling pathway - yeast","establishment or maintenance of cytoskeleton polarity","nucleolus","microtubule organizing center"))) %>%
    arrange(desc(GO_terms)) %>%
    relocate(c(1,10,12,9,8,2,7,11,4,3,5,6)) %>% # reorder according to heatmap
pivot_longer(-GO_terms,names_to="cluster",values_to="p_vals") %>%
  mutate(cluster=gsub("cluster_(.+)","\\1",cluster)) %>%
  ggplot(aes(x=fct_rev(GO_terms),y=fct_inorder(cluster),fill=-log10(p_vals)))+
  geom_tile(aes(fill=-log10(p_vals)))+
  scale_fill_gradientn(colors=c("white","darkblue"),na.value="lightgrey")+
  coord_flip()+
  labs(y="cluster",x="")+
  theme_classic()+
  theme(axis.text.x=element_text(size=rel(1.5),angle=90,vjust=0.6),axis.text.y=element_text(size=rel(1.5)))

print_plot(WT_phos_GO_terms,
           "WT_phos_GO_terms_updated.pdf",10,10)

```


# WT sig phos clust logos
```{r}

library(ggseqlogo)
library(ggpubr)
# JUST SELECTED CLUSTERS FIRST 
# all cluster logos below

# MI cluster enriched for Cdc5
WT_sig_phos_clust3_logo_bits<-ggseqlogo(substr(WT.phos.fin.df.sig %>%
              filter(ward.cluster==3) %>% select(Sequence.window) %>% unlist(),11,21),method="bits")+
    ggtitle("cluster 3")+
    theme(legend.position="none")+
    scale_x_continuous(breaks=1:11,labels=-5:5)

WT_sig_phos_clust3_logo_prob<-ggseqlogo(substr(WT.phos.fin.df.sig %>%
              filter(ward.cluster==3) %>% select(Sequence.window) %>% unlist(),11,21),method="prob")+
    ggtitle("cluster 3")+
    theme(legend.position="none")+
    scale_x_continuous(breaks=1:11,labels=-5:5)

# MII cluster enriched for CDK
WT_sig_phos_clust2_logo_bits<-ggseqlogo(substr(WT.phos.fin.df.sig %>%
              filter(ward.cluster==2) %>% select(Sequence.window) %>% unlist(),11,21),method="bits")+
    ggtitle("cluster 2")+
    theme(legend.position="none")+
    scale_x_continuous(breaks=1:11,labels=-5:5)

WT_sig_phos_clust2_logo_prob<-ggseqlogo(substr(WT.phos.fin.df.sig %>%
              filter(ward.cluster==2) %>% select(Sequence.window) %>% unlist(),11,21),method="prob")+
    ggtitle("cluster 2")+
    theme(legend.position="none")+
    scale_x_continuous(breaks=1:11,labels=-5:5)

print_plot(WT_sig_phos_clust2_logo_bits,
           "WT_sig_phos_clust2_logo_bits.pdf",5,3)
print_plot(WT_sig_phos_clust2_logo_prob,
           "WT_sig_phos_clust2_logo_prob.pdf",5,3)
print_plot(WT_sig_phos_clust3_logo_bits,
           "WT_sig_phos_clust3_logo_bits.pdf",5,3)
print_plot(WT_sig_phos_clust3_logo_prob,
           "WT_sig_phos_clust3_logo_prob.pdf",5,3)


# custom color scheme to highlight certain amino acids in logo


cs1 <- make_col_scheme(chars=c('A','C','D','E','F','G','H','I','K','L','M','N','P','Q','R','S','T','V','W','Y'),
                      cols=c('grey','grey','darkblue','darkblue','grey','grey','grey','grey','grey','grey','grey','darkblue','black','grey','grey','darkgreen','darkgreen','grey','grey','grey'),name='cs1')

# note custom col_scheme is a variable not a quoted char
WT_sig_phos_clust3_logo_prob_custom_color<-ggseqlogo(substr(WT.phos.fin.df.sig %>%
              filter(ward.cluster==3) %>% select(Sequence.window) %>% unlist(),11,21),method="prob",col_scheme=cs1)+
    ggtitle("cluster 3")+
    theme(legend.position="none")+
    scale_x_continuous(breaks=1:11,labels=-5:5)

WT_sig_phos_clust2_logo_prob_custom_color<-ggseqlogo(substr(WT.phos.fin.df.sig %>%
              filter(ward.cluster==2) %>% select(Sequence.window) %>% unlist(),11,21),method="prob",col_scheme=cs1)+
    ggtitle("cluster 2")+
    theme(legend.position="none")+
    scale_x_continuous(breaks=1:11,labels=-5:5)


print_plot(WT_sig_phos_clust2_logo_prob_custom_color,
           "WT_sig_phos_clust2_logo_prob_custom_color.pdf",5,3)

print_plot(WT_sig_phos_clust3_logo_prob_custom_color,
           "WT_sig_phos_clust3_logo_prob_custom_color.pdf",5,3)


### ALL CLUSTER MOTIFS FOR SUPP ###

# Motif logos of clusters
# FUNCTIONS
short_motif_logo_cluster<-function(df,clusternum,meth){
  ggseqlogo(substr(df %>% filter(ward.cluster==clusternum) %>% select(Sequence.window) %>% unlist(),11,21),method=meth)+
    ggtitle(paste0("cluster ",clusternum,", n=",nrow(df%>%filter(ward.cluster==clusternum))))+
    theme(legend.position="none")+
    scale_x_continuous(breaks=1:11,labels=-5:5)
}

# bits method without central residue to show surrounding better
# prob method without central doesn't make sense since it does not change heigh of surroundings
short_motif_logo_cluster_nocentral<-function(df,clusternum,meth){
  ggseqlogo(substr(df %>% filter(ward.cluster==clusternum) %>% select(Sequence.window) %>% unlist(),11,21) %>% gsub("(.....)(\\w)(.....)","\\1 \\3",.),method=meth)+
    ggtitle(paste0("cluster ",clusternum,", n=",nrow(df%>%filter(ward.cluster==clusternum))))+
    theme(legend.position="none")+
    scale_x_continuous(breaks=1:11,labels=-5:5)
}

# SIG WT phos-sites


# bits logos
logo_list<-list()
for(i in 1:11){
  logo_list[[i]]<-short_motif_logo_cluster(WT.phos.fin.df.sig,i,"bits")
}

p<-ggarrange(logo_list[[1]],logo_list[[2]],logo_list[[3]],logo_list[[4]],logo_list[[5]],logo_list[[6]],logo_list[[7]],logo_list[[8]],logo_list[[9]],logo_list[[10]],logo_list[[11]],plot_spacer(),ncol=4,nrow=3,align="hv")

print_plot(p,
           "WT_sig_phos_cluster_logos_bits.pdf",20,10)

#prob
logo_list<-list()
for(i in 1:11){
  logo_list[[i]]<-short_motif_logo_cluster(WT.phos.fin.df.sig,i,"prob")
}

p2<-ggarrange(logo_list[[1]],logo_list[[2]],logo_list[[3]],logo_list[[4]],logo_list[[5]],logo_list[[6]],logo_list[[7]],logo_list[[8]],logo_list[[9]],logo_list[[10]],logo_list[[11]],plot_spacer(),ncol=4,nrow=3,align="hv")

print_plot(p2,
           "WT_sig_phos_cluster_logos_prob.pdf",20,10)

# no central bits
logo_list<-list()
for(i in 1:11){
  logo_list[[i]]<-short_motif_logo_cluster_nocentral(WT.phos.fin.df.sig,i,"bits")
}

p3<-ggarrange(logo_list[[1]],logo_list[[2]],logo_list[[3]],logo_list[[4]],logo_list[[5]],logo_list[[6]],logo_list[[7]],logo_list[[8]],logo_list[[9]],logo_list[[10]],logo_list[[11]],plot_spacer(),ncol=4,nrow=3,align="hv")

print_plot(p3,
           "WT_sig_phos_cluster_logos_bits_nocentral.pdf",20,10)


```

# WT v spo13 sigdiff cluster logos

```{r}

# SIGDIFF WT v spo13 phos-sites

logo_list<-list()
for(i in 1:15){
  logo_list[[i]]<-short_motif_logo_cluster(WT.spo13.phos.fin.df.sigdiff,i,"bits")
}

WT_spo13_phos_sigdiff_logos<-ggarrange(logo_list[[1]],logo_list[[2]],logo_list[[3]],logo_list[[4]],logo_list[[5]],logo_list[[6]],logo_list[[7]],logo_list[[8]],logo_list[[9]],logo_list[[10]],logo_list[[11]],logo_list[[12]],logo_list[[13]],logo_list[[14]],logo_list[[15]],ncol=5,nrow=3,align="hv")

print_plot(WT_spo13_phos_sigdiff_logos,
           "WT_spo13_phos_sigdiff_logos_bits.pdf",20,8)

logo_list<-list()
for(i in 1:15){
  logo_list[[i]]<-short_motif_logo_cluster(WT.spo13.phos.fin.df.sigdiff,i,"prob")
}

WT_spo13_phos_sigdiff_logos_prob<-ggarrange(logo_list[[1]],logo_list[[2]],logo_list[[3]],logo_list[[4]],logo_list[[5]],logo_list[[6]],logo_list[[7]],logo_list[[8]],logo_list[[9]],logo_list[[10]],logo_list[[11]],logo_list[[12]],logo_list[[13]],logo_list[[14]],logo_list[[15]],ncol=5,nrow=3,align="hv")

print_plot(WT_spo13_phos_sigdiff_logos_prob,
           "WT_spo13_phos_sigdiff_logos_prob.pdf",20,8)


```


# Export WT phos seq lists
```{r}

# just sequences
# all data in sheet 1
# number of columns = number of clusters
# one column for each cluster, listing the seqs
WT.phos.fin.df.sig %>%
    mutate(Sequence.window_5=substr(Sequence.window,11,21)) %>%
    group_by(ward.cluster) %>%
    mutate(id=row_number()) %>%
    pivot_wider(names_from=ward.cluster,values_from=Sequence.window_5,id_cols=id) %>%
    select(-id) %>%
    rename_with(~paste0("cluster_",.x,recycle0=T))%>%
    write.xlsx(.,"WT_phos_sig_cluster_seqs.xlsx")


# each cluster is separate tab and there it lists ids and seqs side by side


WT_phos_df_list<-list()
for(i in 1:11){
    WT_phos_df_list[[i]]=WT.phos.fin.df.sig %>%
    mutate(Sequence.window_5=substr(Sequence.window,11,21)) %>%
    filter(ward.cluster==i) %>%
    select(gene.name_site,Gene.name,Sequence.window_5)
}

# name each df in the list for sheet titles
names(WT_phos_df_list)<-c(paste0("cluster_",1:11))

# export as multi-sheet excel
write.xlsx(WT_phos_df_list, file = "WT_phos_sig_clust_ids_seqs_multitab.xlsx")


```

# Barplots of motif enrichment

```{r,echo=F,message=F,warning=F}

# input kinase motif df - 12 selected kinase motif
kinase.motifs<-c("...............[ST]P..............","...............[ST]P.[KR]............","............RP.[ST]...............",".............[DEN].[ST]...............",".............[RK].[ST]...............","............[RK]..[ST]...............","...............[ST].[RK].............","............[LM]..[ST].[STVILMFYW][YIFM]............","...............[ST][DEST]..............",".............[DER].[ST]...............","...............[ST]Q..............","............[ST]..[ST]...............")
kinases<-c("Cdc28_min","Cdc28_strict","Ime2","Cdc5","Ipl1/PKA","Mek1/Sgk2","Cdc15","Atg1","Cdc7/Cka1","Mps1","Mec1","Hrr25")

kinase.df<-as.data.frame(cbind(kinase.motifs,kinases))

# THIS NEEDS TO BE MODIFIED IF NEW MOTIFS ADDED # has to modified for length of the kinase df
get.fisher.cluster.numbers<-function(df,numclust,kinasedf,kinasedflength){
  kinasedf[,"fisher.pval"]<-seq(1:kinasedflength) # number or rows kinase.df
  kinasedf[,"clust.fc"]<-seq(1:kinasedflength)
  patternlist<-kinasedf[,"kinase.motifs"]
for(i in 1:nrow(kinasedf)){
    pattern<-patternlist[i]
    total.df <- df %>% nrow() 
    total.clust <- df %>% filter(ward.cluster==numclust) %>% nrow()
    match.df <- df %>% filter(grepl(pattern,Sequence.window)) %>% nrow()
    match.clust <- df  %>% filter(ward.cluster==numclust) %>% filter(grepl(pattern,Sequence.window)) %>% nrow()
    corr.mat<-matrix(c(match.df,(total.df-match.df),match.clust,(total.clust-match.clust)),
                     nrow=2,
                     dimnames= list(c("match","no match"),
                                    c("df","cluster")))
    fisher.result<-fisher.test(corr.mat)
    
    corr.mat<-corr.mat %>% as.data.frame() %>%
    rbind(colSums(.)) %>%
    cbind(rowSums(.)) %>%
    rbind(perc_match=.[1,]/.[3,])
    
    rownames(corr.mat)[3]<-"col_totals"
    colnames(corr.mat)<-c("df","cluster","row_totals")
    
    pval<-fisher.result$p.value
    
    fc<- corr.mat["perc_match",2]/corr.mat["perc_match",1]
    
    kinasedf[i,"fisher.pval"]<-pval
    kinasedf[i,"clust.fc"]<-fc
    #kinasedf[i,"scaled.fc"]<-fc*(total.clust/total.df) #this doesn't actually work to scale it
    }
  return(kinasedf)
}



WT.lineplots<-list()

for (i in 1:11){
    WT.lineplots[[i]]<-get.fisher.cluster.numbers(WT.phos.fin.df.sig,i,kinase.df,12) %>%
        mutate(stars=ifelse(fisher.pval<0.001,"***",
                      ifelse(fisher.pval<0.01,"**",
                             ifelse(fisher.pval<0.05,"*",NA))))%>%
  ggplot(aes(x=kinases,y=clust.fc,))+
  geom_bar(stat="identity",fill=hue_pal()(length(unique(WT.phos.fin.df.sig$ward.cluster)))[i])+
  geom_text(aes(label = stars), position = position_dodge(width = 0.8), vjust = 0)+
  theme_classic()+
  ylab("fold change")+
  #need to add expand to make sure bar starts at 0 y
  scale_y_continuous(expand=expansion(mult=c(0,0.2)),labels=label_number(accuracy=0.1))+
  ggtitle(paste0("cluster ",i))+
  geom_hline(yintercept=1,linetype="dashed")+
        theme(axis.text.x = element_text(angle=45,vjust=1,hjust=1),axis.title.x=element_blank(),axis.text=element_text(size=rel(1)),
        axis.title=element_text(size=rel(1)),title=element_text(size=rel(1.5)))
}

```


# Sig cluster lines and motifs
```{r,echo=F,message=F,warning=F}

# INDIVIDUAL CLUSTER LINEPLOTS

# individual sig cluster lineplots
get.single.cluster.lineplot<-function(df,strain,var.type,clusternum,title){
  prefix<-paste0("^scaled.",strain,".mean.")
  cols<-grep(prefix,colnames(df),value=T)
  var<-grep(var.type,colnames(df),value=T)
  plot<-df %>%
    filter(ward.cluster==clusternum) %>%
    select(var,all_of(cols),ward.cluster) %>%
    pivot_longer(-c(var,ward.cluster),names_prefix=prefix,names_to="time",values_to="intensity") %>% 
  ggplot(aes(x=fct_inorder(time),y=intensity,group=ward.cluster))+
  stat_summary(aes(y=intensity),fun=median,geom="line",size=0.5,color=hue_pal()(length(unique(df$ward.cluster)))[clusternum])+
  labs(x="time",y="scaled intensity",title=paste0(title," n=", nrow(df %>% filter(ward.cluster==clusternum))))+
    theme_classic()+
      scale_y_continuous(labels=label_number(accuracy=0.1))+
      theme(axis.text.x = element_text(angle=90),axis.text=element_text(size=rel(1)),
        axis.title=element_text(size=rel(1)),title=element_text(size=rel(1.5)))
    return(plot)
}


WT.lineplots[[12]]<-get.single.cluster.lineplot(WT.phos.fin.df.sig,"WT","gene.name_site",1,"cluster 1,")
WT.lineplots[[13]]<-get.single.cluster.lineplot(WT.phos.fin.df.sig,"WT","gene.name_site",2,"cluster 2,")
WT.lineplots[[14]]<-get.single.cluster.lineplot(WT.phos.fin.df.sig,"WT","gene.name_site",3,"cluster 3,")
WT.lineplots[[15]]<-get.single.cluster.lineplot(WT.phos.fin.df.sig,"WT","gene.name_site",4,"cluster 4,")
WT.lineplots[[16]]<-get.single.cluster.lineplot(WT.phos.fin.df.sig,"WT","gene.name_site",5,"cluster 5,")
WT.lineplots[[17]]<-get.single.cluster.lineplot(WT.phos.fin.df.sig,"WT","gene.name_site",6,"cluster 6,")
WT.lineplots[[18]]<-get.single.cluster.lineplot(WT.phos.fin.df.sig,"WT","gene.name_site",7,"cluster 7,")
WT.lineplots[[19]]<-get.single.cluster.lineplot(WT.phos.fin.df.sig,"WT","gene.name_site",8,"cluster 8,")
WT.lineplots[[20]]<-get.single.cluster.lineplot(WT.phos.fin.df.sig,"WT","gene.name_site",9,"cluster 9,")
WT.lineplots[[21]]<-get.single.cluster.lineplot(WT.phos.fin.df.sig,"WT","gene.name_site",10,"cluster 10,")
WT.lineplots[[22]]<-get.single.cluster.lineplot(WT.phos.fin.df.sig,"WT","gene.name_site",11,"cluster 11,")


# Cluster barplots and lineplots in one plot

# # 4 x 3, 3 rows
# # rows of 4
WT_sig_phos_clust_bar_medlines_rows_of_4<-ggarrange(WT.lineplots[[12]],WT.lineplots[[13]],WT.lineplots[[14]],WT.lineplots[[15]],WT.lineplots[[1]],WT.lineplots[[2]],WT.lineplots[[3]],WT.lineplots[[4]], WT.lineplots[[16]],WT.lineplots[[17]],WT.lineplots[[18]],WT.lineplots[[19]],WT.lineplots[[5]],WT.lineplots[[6]],WT.lineplots[[7]],WT.lineplots[[8]],WT.lineplots[[20]],WT.lineplots[[21]],WT.lineplots[[22]],plot_spacer(),WT.lineplots[[9]],WT.lineplots[[10]],WT.lineplots[[11]],plot_spacer(),nrow=6,ncol=4)
# # 20 x 15 pdf portrait
print_plot(WT_sig_phos_clust_bar_medlines_rows_of_4,
           "WT_sig_phos_clust_bar_medlines_rows_of_4.pdf",15,20)

# rows of 3
WT_sig_phos_clust_bar_medlines_rows_of_3<-ggarrange(WT.lineplots[[12]],WT.lineplots[[13]],WT.lineplots[[14]],WT.lineplots[[1]],WT.lineplots[[2]],WT.lineplots[[3]],WT.lineplots[[15]],WT.lineplots[[16]], WT.lineplots[[17]],WT.lineplots[[4]],WT.lineplots[[5]],WT.lineplots[[6]],WT.lineplots[[18]],WT.lineplots[[19]],WT.lineplots[[20]],WT.lineplots[[7]],WT.lineplots[[8]],WT.lineplots[[9]],WT.lineplots[[21]],WT.lineplots[[22]],plot_spacer(),WT.lineplots[[10]],WT.lineplots[[11]],plot_spacer(),nrow=8,ncol=3)
#12 x 25 pdf portrait
#

print_plot(WT_sig_phos_clust_bar_medlines_rows_of_3,
           "WT_sig_phos_clust_bar_medlines_rows_of_3.pdf",12,25)

# rows of 2
WT_sig_phos_clust_bar_medlines_rows_of_2<-ggarrange(WT.lineplots[[12]],WT.lineplots[[13]],WT.lineplots[[1]],WT.lineplots[[2]],WT.lineplots[[14]],WT.lineplots[[15]],WT.lineplots[[3]],WT.lineplots[[4]], WT.lineplots[[16]],WT.lineplots[[17]],WT.lineplots[[5]],WT.lineplots[[6]],WT.lineplots[[18]],WT.lineplots[[19]],WT.lineplots[[7]],WT.lineplots[[8]],WT.lineplots[[20]],WT.lineplots[[21]],WT.lineplots[[9]],WT.lineplots[[10]],WT.lineplots[[22]],plot_spacer(),WT.lineplots[[11]],plot_spacer(),nrow=12,ncol=2)
#7 x 30 pdf portrait

print_plot(WT_sig_phos_clust_bar_medlines_rows_of_2,
           "WT_sig_phos_clust_bar_medlines_rows_of_2.pdf",7,30)

```


# webVenn fisher test unique phospho-prot

```{r,echo=F,message=F,warning=F}

# WT PHOS HEATMAP
# WT.phos.sig.heatmap
# "meiosis I" clusters=3,4,9,10
# "meiosis II" clusters=5,7
# meiosis I + meiosis II = cluster 2


# export xlsx with Gene.name lists of each of 3 groups used on interactiVenn

WT.phos.fin.df.sig %>%
    mutate(mei_group=ifelse(ward.cluster %in% c(3,4,9,10), "meiosis_I",
                            ifelse(ward.cluster %in% c(5,7), "meiosis_II",
                                   ifelse(ward.cluster==2, "meiosis_I_II","other")))) %>%
    group_by(mei_group) %>%
    mutate(id=row_number()) %>%
    pivot_wider(names_from=mei_group,values_from=Gene.name,id_cols=id) %>%
    select(-c(id,other)) %>%
    apply(.,2,function(x) unique(x)) %>%
    write.xlsx(.,file="WT_meiosis_I_v_II_phos_prot_ids_interactivenn.xlsx")
    

# run fisher test for unique vs shared for each group vs other groups

# first need sub-df of WT.phos.fin.df.sig that ONLY INCLUDES these clusters (so it will be comparing enrichment in 1 cluster versus the others and not against all other clusters in WT.phos)

# only includes clusters with sites high during meiI, meiII or both (not just prophase or last 2 timepoints)
WT.meiIandIIclust.df<-WT.phos.fin.df.sig %>%
  filter(ward.cluster %in% c(2,3,4,5,7,9,10))

# calculate fisher pvalue of difference in meiosis I clusters vs others number of unique phospho-proteins


# checking these numbers match the webvenns -- they do
#number of unique phospho-proteins
# nrow(setdiff(WT.meiIandIIclust.df %>% filter(ward.cluster %in% c(3,4,9,10)) %>% select(Gene.name),WT.meiIandIIclust.df %>% filter(!ward.cluster %in% c(3,4,9,10)) %>% select(Gene.name)))
# 
# #number of shared phospho-proteins
# nrow(intersect(WT.meiIandIIclust.df %>% filter(ward.cluster %in% c(3,4,9,10)) %>% select(Gene.name),WT.meiIandIIclust.df %>% filter(!ward.cluster %in% c(3,4,9,10)) %>% select(Gene.name)))

# function to test fisher enrichment between 2 groups

meiIandII.pairwise.fish.pval<-function(group1,group2,group1name,group2name){
  # numbers for contingency table
  gp1_unique<-nrow(setdiff(WT.meiIandIIclust.df %>% filter(ward.cluster %in% group1) %>% select(Gene.name),WT.meiIandIIclust.df %>% filter(!ward.cluster %in% group1) %>% select(Gene.name)))
  gp1_shared<-nrow(intersect(WT.meiIandIIclust.df %>% filter(ward.cluster %in% group1) %>% select(Gene.name),WT.meiIandIIclust.df %>% filter(!ward.cluster %in% group1) %>% select(Gene.name)))
  gp2_unique<-nrow(setdiff(WT.meiIandIIclust.df %>% filter(ward.cluster %in% group2) %>% select(Gene.name),WT.meiIandIIclust.df %>% filter(!ward.cluster %in% group2) %>% select(Gene.name)))
  gp2_shared<-nrow(intersect(WT.meiIandIIclust.df %>% filter(ward.cluster %in% group2) %>% select(Gene.name),WT.meiIandIIclust.df %>% filter(!ward.cluster %in% group2) %>% select(Gene.name)))

  # contingency table
  corr.mat<-matrix(c(gp1_unique,gp1_shared,gp2_unique,gp2_shared),
                     nrow=2,
                     dimnames= list(c("unique","shared"),
                                    c(group1name,group2name)))
  
  #fisher test
  p_val<-fisher.test(corr.mat)$p.value
  
  corr.mat<-corr.mat %>% as.data.frame() %>%
    rbind(colSum=colSums(.)) %>%
    cbind(rowSum=rowSums(.)) %>%
    rbind(frac_unique=.[1,]/.[3,])
  
    results<-list("corrmat"=corr.mat,"pval"=p_val)
    
    return(results)
  
  #return(p_val)
}

# reminder of cluster numbers:
# "meiosis I" clusters=3,4,9,10
# "meiosis II" clusters=5,7
# meiosis I + meiosis II = cluster 2

# plot I want has 3 bars, fraction of unique (not-shared) phospho-proteins in each group
# and then comparison bars between each of 3 comparisons with stars if it is sig

# just the perc_unique of each of 3 groups compared:
meiIandII.barplot<-as.data.frame(cbind("frac_unique"=c(meiIandII.pairwise.fish.pval(c(3,4,9,10),c(5,7),"meiosis_I","meiosis_II")$corrmat["frac_unique",1:2],"meiosisI_II"=meiIandII.pairwise.fish.pval(c(3,4,9,10),2,"meiosis_I","meiosis_I_II")$corrmat["frac_unique",2]))) %>%
  mutate(group=rownames(.)) %>%
  mutate(frac_unique=as.numeric(frac_unique)) %>%
  mutate(group=ordered(as.factor(group),levels=c("meiosis_I","meiosisI_II","meiosis_II"))) %>%
  ggplot(aes(x=group,y=frac_unique))+
  geom_bar(aes(fill=group),stat="identity")+
    geom_signif(comparisons=list(c("meiosis_I","meiosis_II")),
                map_signif_level = T,
                annotations= c("**"),
                size=0.25)+
    # geom_signif(comparisons=list(c("meiosis_I","meiosis_I_II")),
    #             map_signif_level = F,
    #             annotations= "ns",
    #             size=0.25)+
    # geom_signif(comparisons=list(c("meiosis_I_II","meiosis_II")),
    #             map_signif_level = T,
    #             annotations= "**",
    #             size=0.25)+
  theme_classic()+
  scale_y_continuous(limits=c(-NA,1),expand=c(0,0))+
  labs(y="fraction unique phospho-proteins",x="")+
  theme(legend.position="none",axis.text=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)))

# add other 2 comparison bars manually because I can't control y position of each ggsignif bar

print_plot(meiIandII.barplot,
           "WT_sig_phos_mei_I_II_phos_prot_bar.pdf",5,5)

# export results of fisher test for excel showing numbers and pvals

# CONVERT TO EXCEL TABLE

meidf1<-cbind(frac_unique=meiIandII.pairwise.fish.pval(c(3,4,9,10),c(5,7),"meiosis_I","meiosis_II")$corrmat["frac_unique",1:2],pval=meiIandII.pairwise.fish.pval(c(3,4,9,10),c(5,7),"meiosis_I","meiosis_II")$pval) %>%
    mutate(stars=ifelse(pval<0.001,"***",
                      ifelse(pval<0.01,"**",
                             ifelse(pval<0.05,"*",NA))))


meidf2<-cbind(frac_unique=meiIandII.pairwise.fish.pval(c(3,4,9,10),2,"meiosis_I","meiosis_I_II")$corrmat["frac_unique",1:2],pval=meiIandII.pairwise.fish.pval(c(3,4,9,10),2,"meiosis_I","meiosis_I_II")$pval) %>%
        mutate(stars=ifelse(pval<0.001,"***",
                      ifelse(pval<0.01,"**",
                             ifelse(pval<0.05,"*",NA))))

meidf3<-cbind(frac_unique=meiIandII.pairwise.fish.pval(c(5,7),2,"meiosis_II","meiosis_I_II")$corrmat["frac_unique",1:2],pval=meiIandII.pairwise.fish.pval(c(5,7),2,"meiosis_II","meiosis_I_II")$pval) %>%
        mutate(stars=ifelse(pval<0.001,"***",
                      ifelse(pval<0.01,"**",
                             ifelse(pval<0.05,"*",NA))))

meidflist<-list(meidf1,meidf2,meidf3)

names(meidflist)<-c("meiosis_I_v_meiosis_II","meiosis_I_v_meiosis_I_II","meiosis_II_v_meiosis_I_II")

# multi-tab excel
write.xlsx(meidflist,"WT_phos_sig_meiI_v_II_phos_prot_fisher_results.xlsx")


```



# Gen Numbers plots

```{r,echo=F,warning=F,message=F}

# Number of proteins and phospho-sites detected in summarised WT protein or WT phospho-site dataset
# (which includes data from 2 TMT10 replicates)

# plot which includes sites not quantified to protein level
total_data_numbers_2<-as.data.frame(rbind(prot.rep1=protg %>%
    filter(WT.rep1==T) %>% nrow(),
    prot.rep2=protg %>%
    filter(WT.rep2==T) %>% nrow(),
    prot.2reps=protg %>%
    filter(WT.2reps==T) %>% nrow(),
    nonorm_phos_rep1=no_norm_phosphos %>%
    filter(WT.rep1==T) %>%
        nrow(),
    nonorm_phos_rep2=no_norm_phosphos %>%
    filter(WT.rep2==T) %>%
        nrow(),
    nonorm_phos_2reps=no_norm_phosphos %>%
    filter(WT.2.detects==T) %>%
        nrow(),
    norm.phos.rep1=phosphos %>%
    filter(WT.rep1==T) %>% nrow(),
    norm.phos.rep2=phosphos %>%
    filter(WT.rep2==T) %>% nrow(),
    norm.phos.2reps=phosphos %>%
    filter(WT.2.detects==T) %>% nrow(),
    nonorm_phos_loc_rep1=no_norm_phosphos %>%
    filter(WT.rep1==T) %>%
    filter(high.localiz.score==T) %>%
        nrow(),
    nonorm_phos_loc_rep2=no_norm_phosphos %>%
    filter(WT.rep2==T) %>%
    filter(high.localiz.score==T) %>%
        nrow(),
    nonorm_phos_loc_2reps=no_norm_phosphos %>%
    filter(WT.2.detects==T) %>%
    filter(high.localiz.score==T) %>%
        nrow(),
    phos.loc.rep1=phosphos %>%
    filter(WT.rep1==T) %>%
    filter(high.localiz.score==T) %>%
        nrow(),
    phos.loc.rep2=phosphos %>%
    filter(WT.rep2==T) %>%
    filter(high.localiz.score==T) %>%
        nrow(),
    phos.loc.2reps=phosphos %>%
    filter(WT.2.detects==T) %>%
    filter(high.localiz.score==T) %>%
    nrow()))

colnames(total_data_numbers_2)<-"number"

total_data_numbers_2<-cbind(dataset=c("protein rep 1","protein rep 2","protein both reps", "phospho-sites rep1","phospho-sites rep2","phospho-sites both reps",
                                    "norm. phospho-sites rep 1","norm. phospho-sites rep 2","norm. phospho-sites both reps","high local. score phospho-sites rep 1","high local. score phospho-sites rep 2","high local. score phospho-sites both reps","high local. score norm. phospho-sites rep 1","high local. score norm. phospho-sites rep 2","high local. score norm. phospho-sites both reps"),total_data_numbers_2)

total_data_numbers_2$dataset<-factor(total_data_numbers_2$dataset,levels=c("protein rep 1","protein rep 2","protein both reps", "phospho-sites rep1","phospho-sites rep2","phospho-sites both reps",
                                    "norm. phospho-sites rep 1","norm. phospho-sites rep 2","norm. phospho-sites both reps","high local. score phospho-sites rep 1","high local. score phospho-sites rep 2","high local. score phospho-sites both reps","high local. score norm. phospho-sites rep 1","high local. score norm. phospho-sites rep 2","high local. score norm. phospho-sites both reps"))

total_data_numbers_2_plot<-total_data_numbers_2 %>%
    ggplot(aes(x=dataset,y=number,fill=dataset))+
    geom_bar(aes(fill=dataset),stat="identity")+
    geom_text(aes(label=number),vjust=0.5,hjust=-0.2)+
    coord_flip()+
    theme_classic()+
    theme(legend.position="none",axis.text.x=element_blank(),axis.ticks.x=element_blank(),axis.text.y=element_text(size=rel(1.5)))+
    scale_fill_viridis_d()+
    scale_y_continuous(expand=c(0,0),limits=c(0,10000))+
    labs(x="",y="")+
    scale_x_discrete(limits = rev(levels(total_data_numbers_2$dataset)))

print_plot(total_data_numbers_2_plot,
           "WT_total_data_numbers_2.pdf",11,4)

# alternate plot that only shows phospho-sites norm to protein level
# WT
total_data_numbers<-as.data.frame(rbind(prot.rep1=protg %>%
    filter(WT.rep1==T) %>% nrow(),
    prot.rep2=protg %>%
    filter(WT.rep2==T) %>% nrow(),
    prot.2reps=protg %>%
    filter(WT.2reps==T) %>% nrow(),
    norm.phos.rep1=phosphos %>%
    filter(WT.rep1==T) %>% nrow(),
    norm.phos.rep2=phosphos %>%
    filter(WT.rep2==T) %>% nrow(),
    norm.phos.2reps=phosphos %>%
    filter(WT.2.detects==T) %>% nrow(),
    phos.loc.rep1=phosphos %>%
    filter(WT.rep1==T) %>%
    filter(high.localiz.score==T) %>%
        nrow(),
    phos.loc.rep2=phosphos %>%
    filter(WT.rep2==T) %>%
    filter(high.localiz.score==T) %>%
        nrow(),
    phos.loc.2reps=phosphos %>%
    filter(WT.2.detects==T) %>%
    filter(high.localiz.score==T) %>%
    nrow()))

colnames(total_data_numbers)<-"number"

total_data_numbers<-cbind(dataset=c("protein rep 1","protein rep 2","protein both reps",
                                    "norm. phospho-sites rep 1","norm. phospho-sites rep 2","norm. phospho-sites both reps","high local. score norm. phospho-sites rep 1","high local. score norm. phospho-sites rep 2","high local. score norm. phospho-sites both reps"),total_data_numbers)

total_data_numbers$dataset<-factor(total_data_numbers$dataset,levels=c("protein rep 1","protein rep 2","protein both reps",
                                                                       "norm. phospho-sites rep 1","norm. phospho-sites rep 2","norm. phospho-sites both reps","high local. score norm. phospho-sites rep 1","high local. score norm. phospho-sites rep 2","high local. score norm. phospho-sites both reps"))

total_data_numbers_plot<-total_data_numbers %>%
    ggplot(aes(x=dataset,y=number,fill=dataset))+
    geom_bar(aes(fill=dataset),stat="identity")+
    geom_text(aes(label=number),vjust=0.5,hjust=-0.2)+
    coord_flip()+
    theme_classic()+
    theme(legend.position="none",axis.text.x=element_blank(),axis.ticks.x=element_blank(),axis.text.y=element_text(size=rel(1.5)))+
    scale_fill_viridis_d()+
    scale_y_continuous(expand=c(0,0),limits=c(0,10000))+
    labs(x="",y="")+
    scale_x_discrete(limits = rev(levels(total_data_numbers$dataset)))

print_plot(total_data_numbers_plot,
           "WT_total_data_numbers.pdf",11,3)

# spo13 alone dataset gen numbers

# only plotting sites norm to protein level

spo13_total_data_numbers<-as.data.frame(rbind(prot.rep1=protg %>%
    filter(spo13.rep1==T) %>% nrow(),
    prot.rep2=protg %>%
    filter(spo13.rep2==T) %>% nrow(),
    prot.2reps=protg %>%
    filter(spo13.2reps==T) %>% nrow(),
    norm.phos.rep1=phosphos %>%
    filter(spo13.rep1==T) %>% nrow(),
    norm.phos.rep2=phosphos %>%
    filter(spo13.rep2==T) %>% nrow(),
    norm.phos.2reps=phosphos %>%
    filter(spo13.2.detects==T) %>% nrow(),
    phos.loc.rep1=phosphos %>%
    filter(spo13.rep1==T) %>%
    filter(high.localiz.score==T) %>%
        nrow(),
    phos.loc.rep2=phosphos %>%
    filter(spo13.rep2==T) %>%
    filter(high.localiz.score==T) %>%
        nrow(),
    phos.loc.2reps=phosphos %>%
    filter(spo13.2.detects==T) %>%
    filter(high.localiz.score==T) %>%
    nrow()))

colnames(spo13_total_data_numbers)<-"number"

spo13_total_data_numbers<-cbind(dataset=c("protein rep 1","protein rep 2","protein both reps",
                                    "norm. phospho-sites rep 1","norm. phospho-sites rep 2","norm. phospho-sites both reps","high local. score norm. phospho-sites rep 1","high local. score norm. phospho-sites rep 2","high local. score norm. phospho-sites both reps"),spo13_total_data_numbers)

spo13_total_data_numbers$dataset<-factor(spo13_total_data_numbers$dataset,levels=c("protein rep 1","protein rep 2","protein both reps",
                                                                       "norm. phospho-sites rep 1","norm. phospho-sites rep 2","norm. phospho-sites both reps","high local. score norm. phospho-sites rep 1","high local. score norm. phospho-sites rep 2","high local. score norm. phospho-sites both reps"))

spo13_total_data_numbers<-spo13_total_data_numbers %>%
    ggplot(aes(x=dataset,y=number,fill=dataset))+
    geom_bar(aes(fill=dataset),stat="identity")+
    geom_text(aes(label=number),vjust=0.5,hjust=-0.2)+
    coord_flip()+
    theme_classic()+
    theme(legend.position="none",axis.text.x=element_blank(),axis.ticks.x=element_blank(),axis.text.y=element_text(size=rel(1.5)))+
    scale_fill_viridis_d()+
    scale_y_continuous(expand=c(0,0),limits=c(0,10000))+
    labs(x="",y="")+
    scale_x_discrete(limits = rev(levels(spo13_total_data_numbers$dataset)))

print_plot(spo13_total_data_numbers,
           "spo13_total_data_numbers.pdf",11,3)


# spo13 plotting number of sites not norm to protein level
# plot which includes sites not quantified to protein level
spo13_total_data_numbers_2<-as.data.frame(rbind(prot.rep1=protg %>%
    filter(spo13.rep1==T) %>% nrow(),
    prot.rep2=protg %>%
    filter(spo13.rep2==T) %>% nrow(),
    prot.2reps=protg %>%
    filter(spo13.2reps==T) %>% nrow(),
    nonorm_phos_rep1=no_norm_phosphos %>%
    filter(spo13.rep1==T) %>%
        nrow(),
    nonorm_phos_rep2=no_norm_phosphos %>%
    filter(spo13.rep2==T) %>%
        nrow(),
    nonorm_phos_2reps=no_norm_phosphos %>%
    filter(spo13.2.detects==T) %>%
        nrow(),
    norm.phos.rep1=phosphos %>%
    filter(spo13.rep1==T) %>% nrow(),
    norm.phos.rep2=phosphos %>%
    filter(spo13.rep2==T) %>% nrow(),
    norm.phos.2reps=phosphos %>%
    filter(spo13.2.detects==T) %>% nrow(),
    nonorm_phos_loc_rep1=no_norm_phosphos %>%
    filter(spo13.rep1==T) %>%
    filter(high.localiz.score==T) %>%
        nrow(),
    nonorm_phos_loc_rep2=no_norm_phosphos %>%
    filter(spo13.rep2==T) %>%
    filter(high.localiz.score==T) %>%
        nrow(),
    nonorm_phos_loc_2reps=no_norm_phosphos %>%
    filter(spo13.2.detects==T) %>%
    filter(high.localiz.score==T) %>%
        nrow(),
    phos.loc.rep1=phosphos %>%
    filter(spo13.rep1==T) %>%
    filter(high.localiz.score==T) %>%
        nrow(),
    phos.loc.rep2=phosphos %>%
    filter(spo13.rep2==T) %>%
    filter(high.localiz.score==T) %>%
        nrow(),
    phos.loc.2reps=phosphos %>%
    filter(spo13.2.detects==T) %>%
    filter(high.localiz.score==T) %>%
    nrow()))

colnames(spo13_total_data_numbers_2)<-"number"

spo13_total_data_numbers_2<-cbind(dataset=c("protein rep 1","protein rep 2","protein both reps", "phospho-sites rep1","phospho-sites rep2","phospho-sites both reps",
                                    "norm. phospho-sites rep 1","norm. phospho-sites rep 2","norm. phospho-sites both reps","high local. score phospho-sites rep 1","high local. score phospho-sites rep 2","high local. score phospho-sites both reps","high local. score norm. phospho-sites rep 1","high local. score norm. phospho-sites rep 2","high local. score norm. phospho-sites both reps"),spo13_total_data_numbers_2)

spo13_total_data_numbers_2$dataset<-factor(spo13_total_data_numbers_2$dataset,levels=c("protein rep 1","protein rep 2","protein both reps", "phospho-sites rep1","phospho-sites rep2","phospho-sites both reps",
                                    "norm. phospho-sites rep 1","norm. phospho-sites rep 2","norm. phospho-sites both reps","high local. score phospho-sites rep 1","high local. score phospho-sites rep 2","high local. score phospho-sites both reps","high local. score norm. phospho-sites rep 1","high local. score norm. phospho-sites rep 2","high local. score norm. phospho-sites both reps"))

spo13_total_data_numbers_2<-spo13_total_data_numbers_2 %>%
    ggplot(aes(x=dataset,y=number,fill=dataset))+
    geom_bar(aes(fill=dataset),stat="identity")+
    geom_text(aes(label=number),vjust=0.5,hjust=-0.2)+
    coord_flip()+
    theme_classic()+
    theme(legend.position="none",axis.text.x=element_blank(),axis.ticks.x=element_blank(),axis.text.y=element_text(size=rel(1.5)))+
    scale_fill_viridis_d()+
    scale_y_continuous(expand=c(0,0),limits=c(0,10000))+
    labs(x="",y="")+
    scale_x_discrete(limits = rev(levels(spo13_total_data_numbers_2$dataset)))

print_plot(spo13_total_data_numbers_2,
           "spo13_total_data_numbers_2.pdf",11,4)


# WT and spo13 combined dataset

# only showing number of sites norm to protein level
total_data_numbers_WT_spo13<-as.data.frame(rbind(prot_WT=protg %>%
    filter(WT.2reps==T) %>% nrow(),
    prot_spo13=protg %>%
        filter(spo13.2reps==T) %>% nrow(),
    prot_WT_spo13=protg %>%
        filter(all4reps==T) %>% nrow(),
    phos_WT= phosphos %>%
        filter(WT.2.detects==T) %>% nrow(),
    phos_spo13= phosphos %>%
        filter(spo13.2.detects==T) %>% nrow(),
    phos_WT_spo13= phosphos %>%
        filter(all4reps==T) %>% nrow(),
    phos_loc_WT= phosphos %>%
        filter(WT.2.detects==T) %>% filter(high.localiz.score==T) %>% nrow(),
    phos_loc_spo13= phosphos %>%
        filter(spo13.2.detects==T) %>% filter(high.localiz.score==T) %>% nrow(),
    phos_loc_WT_spo13= phosphos %>%
        filter(all4reps==T) %>% filter(high.localiz.score==T) %>% nrow()))

colnames(total_data_numbers_WT_spo13)<-"number"

total_data_numbers_WT_spo13<-cbind(dataset=c("WT protein","spo13 protein","WT and spo13 protein",
                                    "WT norm phospho-sites","spo13 norm phospho-sites","WT and spo13 norm phospho-sites","high local. score norm WT phospho-sites","high local. score norm spo13 phospho-sites","high local. score norm WT and spo13 phospho-sites"),total_data_numbers_WT_spo13)

total_data_numbers_WT_spo13$dataset<-factor(total_data_numbers_WT_spo13$dataset,levels=c("WT protein","spo13 protein","WT and spo13 protein",
                                    "WT norm phospho-sites","spo13 norm phospho-sites","WT and spo13 norm phospho-sites","high local. score norm WT phospho-sites","high local. score norm spo13 phospho-sites","high local. score norm WT and spo13 phospho-sites"))

total_data_numbers_bar_WT_spo13<-total_data_numbers_WT_spo13 %>%
    ggplot(aes(x=dataset,y=number,fill=dataset))+
    geom_bar(aes(fill=dataset),stat="identity")+
    geom_text(aes(label=number),vjust=0.5,hjust=-0.2)+
    coord_flip()+
    theme_classic()+
    theme(legend.position="none",axis.text.x=element_blank(),axis.ticks.x=element_blank(),axis.text.y=element_text(size=rel(1.5)))+
    scale_fill_viridis_d()+
    scale_y_continuous(expand=c(0,0),limits=c(0,5000))+
    labs(x="",y="")+
    scale_x_discrete(limits = rev(levels(total_data_numbers_WT_spo13$dataset)))

print_plot(total_data_numbers_bar_WT_spo13,
           "total_data_numbers_bar_WT_spo13.pdf",11,3)

# WT and spo13 combined dataset including number of sites not norm to protein level

total_data_numbers_WT_spo13_2<-as.data.frame(rbind(prot_WT=protg %>%
    filter(WT.2reps==T) %>% nrow(),
    prot_spo13=protg %>%
        filter(spo13.2reps==T) %>% nrow(),
    prot_WT_spo13=protg %>%
        filter(all4reps==T) %>% nrow(),
    nonorm_WT_phos=no_norm_phosphos %>%
    filter(WT.2.detects==T) %>%
        nrow(),
    nonorm_spo13_phos=no_norm_phosphos %>%
    filter(spo13.2.detects==T) %>%
        nrow(),
    nonorm_WT_spo13_phos=no_norm_phosphos %>%
    filter(all4reps==T) %>%
        nrow(),
    phos_WT= phosphos %>%
        filter(WT.2.detects==T) %>% nrow(),
    phos_spo13= phosphos %>%
        filter(spo13.2.detects==T) %>% nrow(),
    phos_WT_spo13= phosphos %>%
        filter(all4reps==T) %>% nrow(),
    nonorm_WT_phos=no_norm_phosphos %>%
    filter(WT.2.detects==T) %>%
        filter(high.localiz.score==T) %>%
        nrow(),
    nonorm_spo13_phos=no_norm_phosphos %>%
    filter(spo13.2.detects==T) %>%
        filter(high.localiz.score==T) %>%
        nrow(),
    nonorm_WT_spo13_phos=no_norm_phosphos %>%
    filter(all4reps==T) %>%
        filter(high.localiz.score==T) %>%
        nrow(),
    phos_loc_WT= phosphos %>%
        filter(WT.2.detects==T) %>% filter(high.localiz.score==T) %>% nrow(),
    phos_loc_spo13= phosphos %>%
        filter(spo13.2.detects==T) %>% filter(high.localiz.score==T) %>% nrow(),
    phos_loc_WT_spo13= phosphos %>%
        filter(all4reps==T) %>% filter(high.localiz.score==T) %>% nrow()))

colnames(total_data_numbers_WT_spo13_2)<-"number"

total_data_numbers_WT_spo13_2<-cbind(dataset=c("WT protein","spo13 protein","WT and spo13 protein",
                                               "WT phospho-sites","spo13 phospho-sites","WT and spo13 phospho-sites","WT norm phospho-sites","spo13 norm phospho-sites","WT and spo13 norm phospho-sites", "high local. score WT phospho-sites","high local. score spo13 phospho-sites","high local. score WT and spo13 phospho-sites","high local. score norm WT phospho-sites","high local. score norm spo13 phospho-sites","high local. score norm WT and spo13 phospho-sites"),total_data_numbers_WT_spo13_2)

total_data_numbers_WT_spo13_2$dataset<-factor(total_data_numbers_WT_spo13_2$dataset,levels=c("WT protein","spo13 protein","WT and spo13 protein",
                                               "WT phospho-sites","spo13 phospho-sites","WT and spo13 phospho-sites","WT norm phospho-sites","spo13 norm phospho-sites","WT and spo13 norm phospho-sites", "high local. score WT phospho-sites","high local. score spo13 phospho-sites","high local. score WT and spo13 phospho-sites","high local. score norm WT phospho-sites","high local. score norm spo13 phospho-sites","high local. score norm WT and spo13 phospho-sites"))

total_data_numbers_bar_WT_spo13_2<-total_data_numbers_WT_spo13_2 %>%
    ggplot(aes(x=dataset,y=number,fill=dataset))+
    geom_bar(aes(fill=dataset),stat="identity")+
    geom_text(aes(label=number),vjust=0.5,hjust=-0.2)+
    coord_flip()+
    theme_classic()+
    theme(legend.position="none",axis.text.x=element_blank(),axis.ticks.x=element_blank(),axis.text.y=element_text(size=rel(1.5)))+
    scale_fill_viridis_d()+
    scale_y_continuous(expand=c(0,0),limits=c(0,5000))+
    labs(x="",y="")+
    scale_x_discrete(limits = rev(levels(total_data_numbers_WT_spo13_2$dataset)))


print_plot(total_data_numbers_bar_WT_spo13_2,
           "total_data_numbers_bar_WT_spo13_2.pdf",11,4)

# smaller plots for figures
# WT protein and phospho-site numbers for Fig 1
# split into 2 plots: one protein and one sites
# 
# protein
gen_prot_number_bar<-total_data_numbers_2 %>%
    head(3) %>%
    ggplot(aes(x=fct_rev(dataset),y=number,fill=dataset))+
    geom_bar(aes(fill=dataset),stat="identity")+
    geom_text(aes(label=number),vjust=0.5,hjust=-0.2)+
    coord_flip()+
    theme_classic()+
    theme(legend.position="none",axis.text.x=element_blank(),axis.ticks.x=element_blank(),axis.text.y=element_text(size=rel(1.5)))+
    scale_fill_viridis_d()+
    scale_y_continuous(expand=expansion(mult=c(0,0.2)))+
    #scale_y_continuous(expand=c(0,0),limits=c(0,4500))+
    labs(x="",y="")
# 3 x 6 landscape

# phospho-sites
gen_phos_number_bar<-total_data_numbers_2[7:9,] %>%
    mutate(dataset=gsub("high local. score norm. ","",dataset))%>%
    mutate(dataset=factor(dataset,levels=c("phospho-sites rep 1","phospho-sites rep 2","phospho-sites both reps")))%>%
    ggplot(aes(x=fct_rev(dataset),y=number,fill=dataset))+
    geom_bar(aes(fill=dataset),stat="identity")+
    geom_text(aes(label=number),vjust=0.5,hjust=-0.2)+
    coord_flip()+
    theme_classic()+
    theme(legend.position="none",axis.text.x=element_blank(),axis.ticks.x=element_blank(),axis.text.y=element_text(size=rel(1.5)))+
    scale_fill_viridis_d()+
    scale_y_continuous(expand=expansion(mult=c(0,0.2)))+
    labs(x="",y="")
# 3 x 6


# printing both into same pdf is safer since dimensions are then equalized
print_plot(gen_prot_number_bar/gen_phos_number_bar,
           "gen_prot_phos_number_bar.pdf",6,4)
           

# Pie chart S v T v Y
# WT
central_residue_pie<-WT.phos.fin.df %>%
mutate(central=ifelse(grepl("...............[S]...............",Sequence.window),"S",
                      ifelse(grepl("...............[T]...............",Sequence.window),"T","Y"))) %>%
group_by(central) %>%
    summarize(n=n()) %>%
ggplot(aes(x="",y=n,fill=central,label=n))+
    geom_bar(stat="identity")+
    geom_label_repel()+
    coord_polar("y", start=0)+
    theme_void()+
    scale_fill_manual(values=c("#7CAE00", "#00BFC4", "#C77CFF"))+
    ggtitle("3877 phospho-sites central residue")

print_plot(central_residue_pie,
           "central_residue_pie.pdf",4,5)

# spo13
spo13_central_residue_pie<-spo13.phos.fin.df %>%
mutate(central=ifelse(grepl("...............[S]...............",Sequence.window),"S",
                      ifelse(grepl("...............[T]...............",Sequence.window),"T","Y"))) %>%
group_by(central) %>%
    summarize(n=n()) %>%
ggplot(aes(x="",y=n,fill=central,label=n))+
    geom_bar(stat="identity")+
    geom_label_repel()+
    coord_polar("y", start=0)+
    theme_void()+
    scale_fill_manual(values=c("#7CAE00", "#00BFC4", "#C77CFF"))+
    ggtitle("3882 phospho-sites central residue")

print_plot(spo13_central_residue_pie,
           "spo13_central_residue_pie.pdf",4,5)

```

# QC numbers and intensities plots
```{r,echo=F,message=F,warning=F}

# proteins per timepoint
WT_protg_number_per_timepoint<-WT.protg.fin.df %>%
  select(WT.mean.cols) %>%
  mutate(across(WT.mean.cols,function(x) sum(x>0))) %>%
  head(1) %>%
  pivot_longer(cols=WT.mean.cols,values_to="number",names_to="time",names_prefix="WT.mean.") %>%
  ggplot(aes(x=fct_inorder(time),y=number))+
  geom_bar(aes(fill=time),stat="identity")+
  geom_text(aes(label=number),vjust=0.5,hjust=1.3)+
  theme_classic()+
  theme(legend.position="none")+
  scale_y_continuous(expand=c(0,0),limits=c(0,4100))+
  labs(y="",x="time",title="proteins per timepoint")+
  coord_flip()

print_plot(WT_protg_number_per_timepoint,
           "WT_protg_number_per_timepoint.pdf",5,4)

# protein intensity at each timepoint
# boxplot of intensity at each timepoint
WT_protg_int_box_per_timepoint<-WT.protg.fin.df %>%
  pivot_longer(cols=all_of(WT.mean.cols),names_to="time",values_to="intensity") %>%
  mutate(time=gsub("WT.mean.","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=log2(intensity),fill=time)) +
  geom_boxplot(aes(fill=time,alpha=0.2),position="dodge",outlier.size = 0.25, size = 0.25)+
  theme_classic()+
  labs(y="log2 intensity",x="time",title="Protein intensity distribution")+
  theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),legend.position="none",axis.text.y=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),plot.title=element_text(size=rel(2)))+
  coord_flip()

print_plot(WT_protg_int_box_per_timepoint,
           "WT_protg_int_box_per_timepoint.pdf",5,4)

# phospho-sites per timepoint
WT_phos_number_per_timepoint<-WT.phos.fin.df %>%
  select(WT.mean.cols) %>%
  mutate(across(WT.mean.cols,function(x) sum(x>0))) %>%
  head(1) %>%
  pivot_longer(cols=WT.mean.cols,values_to="number",names_to="time",names_prefix="WT.mean.") %>%
  ggplot(aes(x=fct_inorder(time),y=number))+
  geom_bar(aes(fill=time),stat="identity")+
  geom_text(aes(label=number),vjust=0.5,hjust=1.3)+
  theme_classic()+
  theme(legend.position="none")+
  scale_y_continuous(expand=c(0,0),limits=c(0,4100))+
  labs(y="",x="time",title="sites per timepoint")+
  coord_flip()

print_plot(WT_phos_number_per_timepoint,
           "WT_phos_number_per_timepoint.pdf",5,4)

# boxplot of phospho-site intensity at each timepoint
WT_phos_int_box_per_timepoint<-WT.phos.fin.df %>%
  pivot_longer(cols=all_of(WT.mean.cols),names_to="time",values_to="intensity") %>%
  mutate(time=gsub("WT.mean.","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=log2(intensity),fill=time)) +
  geom_boxplot(aes(fill=time,alpha=0.2),position="dodge",outlier.size = 0.25, size = 0.25)+
  theme_classic()+
  labs(y="log2 intensity",x="time",title="Site intensity distribution")+
  theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),legend.position="none",axis.text.y=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),plot.title=element_text(size=rel(2)))+
  coord_flip()

print_plot(WT_phos_int_box_per_timepoint,
           "WT_phos_int_box_per_timepoint.pdf",5,4)

# replicates separated plots ####
# note these numbers can be lower than the combined because the combiend its enough for the protein/site to be detected in 2/10 timepoints in both replicates but all valuess get carried over so for example a protein may only have >0 intenssity at t0 in 1/2 TC reps so have a lower number of t0 proteins in 1/2 tiemcourses but that 1 protein counts in thee average of both TC t0 count number.

# PROTEIN
# NUMBERS PER TIMEPOINT REP 1
WT_protg_number_per_timepoint_rep1<-WT.protg.fin.df %>%
  select(all_of(WT_protg_rep1_cols)) %>%
  mutate(across(all_of(WT_protg_rep1_cols),function(x) sum(x>0))) %>%
  head(1) %>%
  pivot_longer(cols=all_of(WT_protg_rep1_cols),values_to="number",names_to="time",names_prefix="cor.lin.na.norm.t.") %>%
    mutate(time=gsub(".Rep1.WT","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=number))+
  geom_bar(aes(fill=time),stat="identity")+
  geom_text(aes(label=number),vjust=0.5,hjust=1.3)+
  theme_classic()+
  theme(legend.position="none")+
  scale_y_continuous(expand=c(0,0),limits=c(0,4100))+
  labs(y="",x="time",title="proteins per timepoint replicate 1")+
  coord_flip()

# NUMBERS PER TIMEPOINT REP 2
WT_protg_number_per_timepoint_rep2<-WT.protg.fin.df %>%
  select(all_of(WT_protg_rep2_cols)) %>%
  mutate(across(all_of(WT_protg_rep2_cols),function(x) sum(x>0))) %>%
  head(1) %>%
  pivot_longer(cols=all_of(WT_protg_rep2_cols),values_to="number",names_to="time",names_prefix="cor.lin.na.norm.t.") %>%
    mutate(time=gsub(".Rep2.WT","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=number))+
  geom_bar(aes(fill=time),stat="identity")+
  geom_text(aes(label=number),vjust=0.5,hjust=1.3)+
  theme_classic()+
  theme(legend.position="none")+
  scale_y_continuous(expand=c(0,0),limits=c(0,4100))+
  labs(y="",x="time",title="proteins per timepoint replicate 2")+
  coord_flip()

print_plot(WT_protg_number_per_timepoint_rep1,
           "WT_protg_number_per_timepoint_rep1.pdf",5,4)
print_plot(WT_protg_number_per_timepoint_rep2,
           "WT_protg_number_per_timepoint_rep2.pdf",5,4)

# PROTEIN
# boxplot of intensity at each timepoint
WT_protg_int_box_per_timepoint_rep1<-WT.protg.fin.df %>%
  pivot_longer(cols=all_of(WT_protg_rep1_cols),names_to="time",values_to="intensity") %>%
  mutate(time=gsub("cor.lin.na.norm.t.","",time)) %>%
    mutate(time=gsub(".Rep1.WT","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=log2(intensity),fill=time)) +
  geom_boxplot(aes(fill=time,alpha=0.2),position="dodge",outlier.size = 0.25, size = 0.25)+
  theme_classic()+
  labs(y="log2 intensity",x="time",title="Protein intensity rep1")+
  theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),legend.position="none",axis.text.y=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),plot.title=element_text(size=rel(2)))+
  coord_flip()

WT_protg_int_box_per_timepoint_rep2<-WT.protg.fin.df %>%
  pivot_longer(cols=all_of(WT_protg_rep2_cols),names_to="time",values_to="intensity") %>%
  mutate(time=gsub("cor.lin.na.norm.t.","",time)) %>%
    mutate(time=gsub(".Rep2.WT","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=log2(intensity),fill=time)) +
  geom_boxplot(aes(fill=time,alpha=0.2),position="dodge",outlier.size = 0.25, size = 0.25)+
  theme_classic()+
  labs(y="log2 intensity",x="time",title="Protein intensity rep2")+
  theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),legend.position="none",axis.text.y=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),plot.title=element_text(size=rel(2)))+
  coord_flip()

print_plot(WT_protg_int_box_per_timepoint_rep1,
           "WT_protg_int_box_per_timepoint_rep1.pdf",5,4)
print_plot(WT_protg_int_box_per_timepoint_rep2,
           "WT_protg_int_box_per_timepoint_rep2.pdf",5,4)


# PHOSPHO
# NUMBERS PER TIMEPOINT REP 1
WT_phos_number_per_timepoint_rep1<-WT.phos.fin.df %>%
  select(all_of(WT_phos_rep1_cols)) %>%
  mutate(across(all_of(WT_phos_rep1_cols),function(x) sum(x>0))) %>%
  head(1) %>%
  pivot_longer(cols=all_of(WT_phos_rep1_cols),values_to="number",names_to="time",names_prefix="cor.lin.stoic.t.") %>%
    mutate(time=gsub(".Rep1.WT","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=number))+
  geom_bar(aes(fill=time),stat="identity")+
  geom_text(aes(label=number),vjust=0.5,hjust=1.3)+
  theme_classic()+
  theme(legend.position="none")+
  scale_y_continuous(expand=c(0,0),limits=c(0,4100))+
  labs(y="",x="time",title="sites per timepoint rep1")+
  coord_flip()

# NUMBERS PER TIMEPOINT REP 2
WT_phos_number_per_timepoint_rep2<-WT.phos.fin.df %>%
  select(all_of(WT_phos_rep2_cols)) %>%
  mutate(across(all_of(WT_phos_rep2_cols),function(x) sum(x>0))) %>%
  head(1) %>%
  pivot_longer(cols=all_of(WT_phos_rep2_cols),values_to="number",names_to="time",names_prefix="cor.lin.stoic.t.") %>%
    mutate(time=gsub(".Rep2.WT","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=number))+
  geom_bar(aes(fill=time),stat="identity")+
  geom_text(aes(label=number),vjust=0.5,hjust=1.3)+
  theme_classic()+
  theme(legend.position="none")+
  scale_y_continuous(expand=c(0,0),limits=c(0,4100))+
  labs(y="",x="time",title="sites per timepoint rep2")+
  coord_flip()

print_plot(WT_phos_number_per_timepoint_rep1,
           "WT_phos_number_per_timepoint_rep1.pdf",5,4)
print_plot(WT_phos_number_per_timepoint_rep2,
           "WT_phos_number_per_timepoint_rep2.pdf",5,4)

# PHOSPHO
# boxplot of intensity at each timepoint
WT_phos_int_box_per_timepoint_rep1<-WT.phos.fin.df %>%
  pivot_longer(cols=all_of(WT_phos_rep1_cols),names_to="time",values_to="intensity") %>%
  mutate(time=gsub("cor.lin.stoic.t.","",time)) %>%
    mutate(time=gsub(".Rep1.WT","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=log2(intensity),fill=time)) +
  geom_boxplot(aes(fill=time,alpha=0.2),position="dodge",outlier.size = 0.25, size = 0.25)+
  theme_classic()+
  labs(y="log2 intensity",x="time",title="Site intensity rep1")+
  theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),legend.position="none",axis.text.y=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),plot.title=element_text(size=rel(2)))+
  coord_flip()

WT_phos_int_box_per_timepoint_rep2<-WT.phos.fin.df %>%
  pivot_longer(cols=all_of(WT_phos_rep2_cols),names_to="time",values_to="intensity") %>%
  mutate(time=gsub("cor.lin.stoic.t.","",time)) %>%
    mutate(time=gsub(".Rep2.WT","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=log2(intensity),fill=time)) +
  geom_boxplot(aes(fill=time,alpha=0.2),position="dodge",outlier.size = 0.25, size = 0.25)+
  theme_classic()+
  labs(y="log2 intensity",x="time",title="Site intensity rep2")+
  theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),legend.position="none",axis.text.y=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),plot.title=element_text(size=rel(2)))+
  coord_flip()

print_plot(WT_phos_int_box_per_timepoint_rep1,
           "WT_phos_int_box_per_timepoint_rep1.pdf",5,4)
print_plot(WT_phos_int_box_per_timepoint_rep2,
           "WT_phos_int_box_per_timepoint_rep2.pdf",5,4)


# spo13 TC QC numbers plots
# proteins per timepoint
spo13_protg_number_per_timepoint<-spo13.protg.fin.df %>%
  select(spo13.mean.cols) %>%
  mutate(across(spo13.mean.cols,function(x) sum(x>0))) %>%
  head(1) %>%
  pivot_longer(cols=spo13.mean.cols,values_to="number",names_to="time",names_prefix="spo13.mean.") %>%
  ggplot(aes(x=fct_inorder(time),y=number))+
  geom_bar(aes(fill=time),stat="identity")+
  geom_text(aes(label=number),vjust=0.5,hjust=1.3)+
  theme_classic()+
  theme(legend.position="none")+
  scale_y_continuous(expand=c(0,0),limits=c(0,4100))+
  labs(y="",x="time",title="proteins per timepoint")+
  coord_flip()

print_plot(spo13_protg_number_per_timepoint,
           "spo13_protg_number_per_timepoint.pdf",5,4)

# protein intensity at each timepoint
# boxplot of intensity at each timepoint
spo13_protg_int_box_per_timepoint<-spo13.protg.fin.df %>%
  pivot_longer(cols=all_of(spo13.mean.cols),names_to="time",values_to="intensity") %>%
  mutate(time=gsub("spo13.mean.","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=log2(intensity),fill=time)) +
  geom_boxplot(aes(fill=time,alpha=0.2),position="dodge",outlier.size = 0.25, size = 0.25)+
  theme_classic()+
  labs(y="log2 intensity",x="time",title="Protein intensity distribution")+
  theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),legend.position="none",axis.text.y=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),plot.title=element_text(size=rel(2)))+
  coord_flip()

print_plot(spo13_protg_int_box_per_timepoint,
           "spo13_protg_int_box_per_timepoint.pdf",5,4)

# phospho-sites per timepoint
spo13_phos_number_per_timepoint<-spo13.phos.fin.df %>%
  select(spo13.mean.cols) %>%
  mutate(across(spo13.mean.cols,function(x) sum(x>0))) %>%
  head(1) %>%
  pivot_longer(cols=spo13.mean.cols,values_to="number",names_to="time",names_prefix="spo13.mean.") %>%
  ggplot(aes(x=fct_inorder(time),y=number))+
  geom_bar(aes(fill=time),stat="identity")+
  geom_text(aes(label=number),vjust=0.5,hjust=1.3)+
  theme_classic()+
  theme(legend.position="none")+
  scale_y_continuous(expand=c(0,0),limits=c(0,4100))+
  labs(y="",x="time",title="sites per timepoint")+
  coord_flip()

print_plot(spo13_phos_number_per_timepoint,
           "spo13_phos_number_per_timepoint.pdf",5,4)

# boxplot of phospho-site intensity at each timepoint
spo13_phos_int_box_per_timepoint<-spo13.phos.fin.df %>%
  pivot_longer(cols=all_of(spo13.mean.cols),names_to="time",values_to="intensity") %>%
  mutate(time=gsub("spo13.mean.","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=log2(intensity),fill=time)) +
  geom_boxplot(aes(fill=time,alpha=0.2),position="dodge",outlier.size = 0.25, size = 0.25)+
  theme_classic()+
  labs(y="log2 intensity",x="time",title="Site intensity distribution")+
  theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),legend.position="none",axis.text.y=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),plot.title=element_text(size=rel(2)))+
  coord_flip()

print_plot(spo13_phos_int_box_per_timepoint,
           "spo13_phos_int_box_per_timepoint.pdf",5,4)

# spo13 REPLICATES SEPARATED
# PROTEIN
# NUMBERS PER TIMEPOINT REP 1
spo13_protg_number_per_timepoint_rep1<-spo13.protg.fin.df %>%
  select(all_of(spo13_protg_rep1_cols)) %>%
  mutate(across(all_of(spo13_protg_rep1_cols),function(x) sum(x>0))) %>%
  head(1) %>%
  pivot_longer(cols=all_of(spo13_protg_rep1_cols),values_to="number",names_to="time",names_prefix="cor.lin.na.norm.t.") %>%
    mutate(time=gsub(".Rep1.Spo13","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=number))+
  geom_bar(aes(fill=time),stat="identity")+
  geom_text(aes(label=number),vjust=0.5,hjust=1.3)+
  theme_classic()+
  theme(legend.position="none")+
  scale_y_continuous(expand=c(0,0),limits=c(0,4100))+
  labs(y="",x="time",title="proteins per timepoint replicate 1")+
  coord_flip()

# NUMBERS PER TIMEPOINT REP 2
spo13_protg_number_per_timepoint_rep2<-spo13.protg.fin.df %>%
  select(all_of(spo13_protg_rep2_cols)) %>%
  mutate(across(all_of(spo13_protg_rep2_cols),function(x) sum(x>0))) %>%
  head(1) %>%
  pivot_longer(cols=all_of(spo13_protg_rep2_cols),values_to="number",names_to="time",names_prefix="cor.lin.na.norm.t.") %>%
    mutate(time=gsub(".Rep2.Spo13","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=number))+
  geom_bar(aes(fill=time),stat="identity")+
  geom_text(aes(label=number),vjust=0.5,hjust=1.3)+
  theme_classic()+
  theme(legend.position="none")+
  scale_y_continuous(expand=c(0,0),limits=c(0,4100))+
  labs(y="",x="time",title="proteins per timepoint replicate 2")+
  coord_flip()

print_plot(spo13_protg_number_per_timepoint_rep1,
           "spo13_protg_number_per_timepoint_rep1.pdf",5,4)
print_plot(spo13_protg_number_per_timepoint_rep2,
           "spo13_protg_number_per_timepoint_rep2.pdf",5,4)

# PROTEIN
# boxplot of intensity at each timepoint
spo13_protg_int_box_per_timepoint_rep1<-spo13.protg.fin.df %>%
  pivot_longer(cols=all_of(spo13_protg_rep1_cols),names_to="time",values_to="intensity") %>%
  mutate(time=gsub("cor.lin.na.norm.t.","",time)) %>%
    mutate(time=gsub(".Rep1.Spo13","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=log2(intensity),fill=time)) +
  geom_boxplot(aes(fill=time,alpha=0.2),position="dodge",outlier.size = 0.25, size = 0.25)+
  theme_classic()+
  labs(y="log2 intensity",x="time",title="Protein intensity rep1")+
  theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),legend.position="none",axis.text.y=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),plot.title=element_text(size=rel(2)))+
  coord_flip()

spo13_protg_int_box_per_timepoint_rep2<-spo13.protg.fin.df %>%
  pivot_longer(cols=all_of(spo13_protg_rep2_cols),names_to="time",values_to="intensity") %>%
  mutate(time=gsub("cor.lin.na.norm.t.","",time)) %>%
    mutate(time=gsub(".Rep2.Spo13","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=log2(intensity),fill=time)) +
  geom_boxplot(aes(fill=time,alpha=0.2),position="dodge",outlier.size = 0.25, size = 0.25)+
  theme_classic()+
  labs(y="log2 intensity",x="time",title="Protein intensity rep2")+
  theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),legend.position="none",axis.text.y=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),plot.title=element_text(size=rel(2)))+
  coord_flip()

print_plot(spo13_protg_int_box_per_timepoint_rep1,
           "spo13_protg_int_box_per_timepoint_rep1.pdf",5,4)
print_plot(spo13_protg_int_box_per_timepoint_rep2,
           "spo13_protg_int_box_per_timepoint_rep2.pdf",5,4)


# PHOSPHO
# NUMBERS PER TIMEPOINT REP 1
spo13_phos_number_per_timepoint_rep1<-spo13.phos.fin.df %>%
  select(all_of(spo13_phos_rep1_cols)) %>%
  mutate(across(all_of(spo13_phos_rep1_cols),function(x) sum(x>0))) %>%
  head(1) %>%
  pivot_longer(cols=all_of(spo13_phos_rep1_cols),values_to="number",names_to="time",names_prefix="cor.lin.stoic.t.") %>%
    mutate(time=gsub(".Rep1.Spo13","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=number))+
  geom_bar(aes(fill=time),stat="identity")+
  geom_text(aes(label=number),vjust=0.5,hjust=1.3)+
  theme_classic()+
  theme(legend.position="none")+
  scale_y_continuous(expand=c(0,0),limits=c(0,4100))+
  labs(y="",x="time",title="sites per timepoint rep1")+
  coord_flip()

# NUMBERS PER TIMEPOINT REP 2
spo13_phos_number_per_timepoint_rep2<-spo13.phos.fin.df %>%
  select(all_of(spo13_phos_rep2_cols)) %>%
  mutate(across(all_of(spo13_phos_rep2_cols),function(x) sum(x>0))) %>%
  head(1) %>%
  pivot_longer(cols=all_of(spo13_phos_rep2_cols),values_to="number",names_to="time",names_prefix="cor.lin.stoic.t.") %>%
    mutate(time=gsub(".Rep2.Spo13","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=number))+
  geom_bar(aes(fill=time),stat="identity")+
  geom_text(aes(label=number),vjust=0.5,hjust=1.3)+
  theme_classic()+
  theme(legend.position="none")+
  scale_y_continuous(expand=c(0,0),limits=c(0,4100))+
  labs(y="",x="time",title="sites per timepoint rep2")+
  coord_flip()

print_plot(spo13_phos_number_per_timepoint_rep1,
           "spo13_phos_number_per_timepoint_rep1.pdf",5,4)
print_plot(spo13_phos_number_per_timepoint_rep2,
           "spo13_phos_number_per_timepoint_rep2.pdf",5,4)

# PHOSPHO
# boxplot of intensity at each timepoint
spo13_phos_int_box_per_timepoint_rep1<-spo13.phos.fin.df %>%
  pivot_longer(cols=all_of(spo13_phos_rep1_cols),names_to="time",values_to="intensity") %>%
  mutate(time=gsub("cor.lin.stoic.t.","",time)) %>%
    mutate(time=gsub(".Rep1.Spo13","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=log2(intensity),fill=time)) +
  geom_boxplot(aes(fill=time,alpha=0.2),position="dodge",outlier.size = 0.25, size = 0.25)+
  theme_classic()+
  labs(y="log2 intensity",x="time",title="Site intensity rep1")+
  theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),legend.position="none",axis.text.y=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),plot.title=element_text(size=rel(2)))+
  coord_flip()

spo13_phos_int_box_per_timepoint_rep2<-spo13.phos.fin.df %>%
  pivot_longer(cols=all_of(spo13_phos_rep2_cols),names_to="time",values_to="intensity") %>%
  mutate(time=gsub("cor.lin.stoic.t.","",time)) %>%
    mutate(time=gsub(".Rep2.Spo13","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=log2(intensity),fill=time)) +
  geom_boxplot(aes(fill=time,alpha=0.2),position="dodge",outlier.size = 0.25, size = 0.25)+
  theme_classic()+
  labs(y="log2 intensity",x="time",title="Site intensity rep2")+
  theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),legend.position="none",axis.text.y=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),plot.title=element_text(size=rel(2)))+
  coord_flip()

print_plot(spo13_phos_int_box_per_timepoint_rep1,
           "spo13_phos_int_box_per_timepoint_rep1.pdf",5,4)
print_plot(spo13_phos_int_box_per_timepoint_rep2,
           "spo13_phos_int_box_per_timepoint_rep2.pdf",5,4)
```

# fraction proteome that is phos plot
```{r,echo=F,message=F,warning=F}

# make a plot of percent of proteome that is phosphorylated

# WT.protg.fin.df %>% nrow() #3499
# WT.phos.fin.df %>% distinct(Gene.name) %>% nrow() #1281

# Fraction of WT proteome that is phos
WT_perc_proteome_phos_pie<-as.data.frame(rbind(c("non_phospho",WT.protg.fin.df %>% nrow()-WT.phos.fin.df %>% distinct(Gene.name) %>% nrow()),
      c("phospho",WT.phos.fin.df %>% distinct(Gene.name) %>% nrow()))) %>%
    rename("type"=1,"n"=2) %>% # rename columns
   mutate(n=as.integer(n)) %>% # n NEEDS to be integer for pie slices to be correct proportions
    ggplot(aes(x="",y=n,fill=type,label=n))+
    geom_bar(stat="identity")+
    scale_y_reverse()+
    geom_label_repel()+
    coord_polar("y", start=0)+
    theme_void()+
    scale_fill_manual(values=c("#7CAE00", "#00BFC4", "#C77CFF","blue"))+
    ggtitle(paste0("WT proteome, n=",WT.protg.fin.df %>% nrow()))+
    theme(legend.title=element_blank())


# Fraction of spo13 proteome that is phos
spo13_perc_proteome_phos_pie<-as.data.frame(rbind(c("non_phospho",spo13.protg.fin.df %>% nrow()-spo13.phos.fin.df %>% distinct(Gene.name) %>% nrow()),
      c("phospho",spo13.phos.fin.df %>% distinct(Gene.name) %>% nrow()))) %>%
    rename("type"=1,"n"=2) %>% # rename columns
    mutate(n=as.integer(n)) %>% # n NEEDS to be integer for pie slices to be correct proportions
    ggplot(aes(x="",y=n,fill=type,label=n))+
    geom_bar(stat="identity")+
    scale_y_reverse()+
    geom_label_repel()+
    coord_polar("y", start=0)+
    theme_void()+
    scale_fill_manual(values=c("#7CAE00", "#00BFC4", "#C77CFF","blue"))+
    ggtitle(paste0("spo13 proteome, n=",spo13.protg.fin.df %>% nrow()))+
    theme(legend.title=element_blank())

# Fraction of WT/spo13 shhared proteome that is phos
WT_spo13_perc_proteome_phos_pie<-as.data.frame(rbind(c("non_phospho",WT.spo13.protg.fin.df %>% nrow()-WT.spo13.phos.fin.df %>% distinct(Gene.name) %>% nrow()),
      c("phospho",WT.spo13.phos.fin.df %>% distinct(Gene.name) %>% nrow()))) %>%
    rename("type"=1,"n"=2) %>% # rename columns
    mutate(n=as.integer(n)) %>% # n NEEDS to be integer for pie slices to be correct proportions
    tibble()%>%
   ggplot(aes(x="",y=n,fill=type,label=n))+
    geom_bar(stat="identity")+
    scale_y_reverse()+
    geom_label_repel()+
    coord_polar(theta="y", start=0)+
    theme_void()+
    scale_fill_manual(values=c("#7CAE00", "#00BFC4", "#C77CFF","blue"))+
    ggtitle(paste0("WT and spo13 proteome, n=",WT.spo13.protg.fin.df %>% nrow()))+
    theme(legend.title=element_blank())

print_plot(WT_perc_proteome_phos_pie+spo13_perc_proteome_phos_pie+WT_spo13_perc_proteome_phos_pie,
           "perc_proteome_phos_pies.pdf",10,4)


```


# WT sel prots and sites
```{r,echo=F,message=F,warning=F}

# representative proteins from each cluster from WT.protg.clust.heatmap
# 
# all same scale
WT_protg_clust_rep_proteins_1<-WT.protg.fin.df.sig %>%
  filter(Gene.name %in% c("APC2","REC8","HOP1","EMI2","SPO74","LSB5")) %>% 
  select(Gene.name,scaled.WT.mean.cols,ward.cluster) %>%
  pivot_longer(-c(Gene.name,ward.cluster),names_to="time",values_to="intensity",names_prefix="scaled.WT.mean.") %>%
    mutate(ward.cluster=as.character(ward.cluster)) %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=Gene.name,color=ward.cluster))+
  geom_line(aes(color=ward.cluster),size=1.5)+
  facet_grid(~Gene.name)+
  theme_classic()+
    theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),axis.text.y=element_text(size=rel(1.5)),plot.title=element_text(size=rel(1.5)),legend.title=element_text(size=rel(1.5)),legend.text=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),strip.text=element_text(size=rel(1.5)))+
  labs(y="scaled intensity",x="time",title="WT representative cluster sig proteins")+
  theme(legend.position = "none",axis.text.x = element_text(angle=90,vjust=0.6))

print_plot(WT_protg_clust_rep_proteins_1,
           "WT_protg_clust_rep1_proteins_1.pdf",20,3.5)

# free scale
WT_protg_clust_rep_proteins_2<-WT.protg.fin.df.sig %>%
  filter(Gene.name %in% c("APC2","REC8","HOP1","EMI2","SPO74","LSB5")) %>% 
  select(Gene.name,scaled.WT.mean.cols,ward.cluster) %>%
  pivot_longer(-c(Gene.name,ward.cluster),names_to="time",values_to="intensity",names_prefix="scaled.WT.mean.") %>%
    mutate(ward.cluster=as.character(ward.cluster)) %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=Gene.name,color=ward.cluster))+
  geom_line(aes(color=ward.cluster),size=1.5)+
  facet_wrap(~Gene.name,scale="free")+
    scale_y_continuous(labels=label_number(accuracy=0.1))+
  theme_classic()+
    theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),axis.text.y=element_text(size=rel(1.5)),plot.title=element_text(size=rel(1.5)),legend.title=element_text(size=rel(1.5)),legend.text=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),strip.text=element_text(size=rel(1.5)))+
  labs(y="scaled intensity",x="time",title="WT representative cluster sig proteins")+
  theme(legend.position = "none",axis.text.x = element_text(angle=90,vjust=0.6))

# 5 x10
print_plot(WT_protg_clust_rep_proteins_2,
           "WT_protg_clust_rep1_proteins_2.pdf",10,5.5)

# PDS1
pds1_line<-WT.protg.fin.df %>%
  filter(Gene.name=="PDS1") %>%
  select(Gene.name,scaled.WT.mean.cols) %>%
  pivot_longer(-Gene.name,names_to="time",values_to="intensity",names_prefix="scaled.WT.mean.") %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=Gene.name,color=Gene.name))+
  geom_line(aes(color=Gene.name),size=1)+
  theme_classic()+
scale_y_continuous(limits=c(0.1,2.0))+
  labs(y="scaled intensity",x="time",title="Pds1 protein")+
  theme(legend.position = "none",axis.text.x = element_text(angle=90,vjust=0.6),axis.text=element_text(size=rel(1)),axis.title=element_text(size=rel(1)))

print_plot(pds1_line,
           "WT_pds1_medline.pdf",6,4)

# Hrr25
hrr25_line<-WT.protg.fin.df %>%
  filter(Gene.name=="HRR25") %>%
  select(Gene.name,scaled.WT.mean.cols) %>%
  pivot_longer(-Gene.name,names_to="time",values_to="intensity",names_prefix="scaled.WT.mean.") %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=Gene.name,color=Gene.name))+
  geom_line(aes(color=Gene.name),size=1)+
  theme_classic()+
    scale_y_continuous(limits=c(0.3,1.7))+
  labs(y="scaled intensity",x="time",title="Hrr25 protein")+
  theme(legend.position = "none",axis.text.x = element_text(angle=90,vjust=0.6),axis.text=element_text(size=rel(1)),axis.title=element_text(size=rel(1)))


print_plot(hrr25_line,
           "WT_hrr25_medline.pdf",6,4)

# Rec8
rec8_line<-WT.protg.fin.df %>%
  filter(Gene.name=="REC8") %>%
  select(Gene.name,scaled.WT.mean.cols) %>%
  pivot_longer(-Gene.name,names_to="time",values_to="intensity",names_prefix="scaled.WT.mean.") %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=Gene.name,color=Gene.name))+
  geom_line(aes(color=Gene.name),size=1)+
  theme_classic()+
    scale_y_continuous(limits=c(0.1,2.0))+
  labs(y="scaled intensity",x="time",title="Rec8 protein")+
  theme(legend.position = "none",axis.text.x = element_text(angle=90,vjust=0.6),axis.text=element_text(size=rel(1)),axis.title=element_text(size=rel(1)))

print_plot(rec8_line,
           "WT_rec8_medline.pdf",6,4)

rim4_line<-WT.protg.fin.df %>%
  filter(Gene.name=="RIM4") %>%
  select(Gene.name,scaled.WT.mean.cols) %>%
  pivot_longer(-Gene.name,names_to="time",values_to="intensity",names_prefix="scaled.WT.mean.") %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=Gene.name,color=Gene.name))+
  geom_line(aes(color=Gene.name),size=1)+
  theme_classic()+
    scale_y_continuous(limits=c(0.1,2.0))+
  labs(y="scaled intensity",x="time",title="Rim4 protein")+
  theme(legend.position = "none",axis.text.x = element_text(angle=90,vjust=0.6),axis.text=element_text(size=rel(1)),axis.title=element_text(size=rel(1)))

print_plot(rim4_line,
           "WT_rim4_medline.pdf",6,4)

sgo1_line<-WT.protg.fin.df %>%
  filter(Gene.name=="SGO1") %>%
  select(Gene.name,scaled.WT.mean.cols) %>%
  pivot_longer(-Gene.name,names_to="time",values_to="intensity",names_prefix="scaled.WT.mean.") %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=Gene.name,color=Gene.name))+
  geom_line(aes(color=Gene.name),size=1)+
  theme_classic()+
  labs(y="scaled intensity",x="time",title="Sgo1 protein")+
  theme(legend.position = "none",axis.text.x = element_text(angle=90,vjust=0.6),axis.text=element_text(size=rel(1)),axis.title=element_text(size=rel(1)))

print_plot(sgo1_line,
           "WT_sgo1_medline.pdf",6,4)

ndt80_line<-WT.protg.fin.df %>%
  filter(Gene.name=="NDT80") %>%
  select(Gene.name,scaled.WT.mean.cols) %>%
  pivot_longer(-Gene.name,names_to="time",values_to="intensity",names_prefix="scaled.WT.mean.") %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=Gene.name,color=Gene.name))+
  geom_line(aes(color=Gene.name),size=1)+
  theme_classic()+
    scale_y_continuous(limits=c(0,2.0))+
  labs(y="scaled intensity",x="time",title="Ndt80 protein")+
  theme(legend.position = "none",axis.text.x = element_text(angle=90,vjust=0.6),axis.text=element_text(size=rel(1)),axis.title=element_text(size=rel(1)))

print_plot(ndt80_line,
           "WT_ndt80_medline.pdf",6,4)

spo13_line<-WT.protg.fin.df %>%
  filter(Gene.name=="SPO13") %>%
  select(Gene.name,scaled.WT.mean.cols) %>%
  pivot_longer(-Gene.name,names_to="time",values_to="intensity",names_prefix="scaled.WT.mean.") %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=Gene.name,color=Gene.name))+
  geom_line(aes(color=Gene.name),size=1)+
  theme_classic()+
  labs(y="scaled intensity",x="time",title="Spo13 protein")+
  theme(legend.position = "none",axis.text.x = element_text(angle=90,vjust=0.6),axis.text=element_text(size=rel(1)),axis.title=element_text(size=rel(1)))

print_plot(spo13_line,
           "WT_spo13_medline.pdf",6,4)

spo20_line<-WT.protg.fin.df %>%
  filter(Gene.name=="SPO20") %>%
  select(Gene.name,scaled.WT.mean.cols) %>%
  pivot_longer(-Gene.name,names_to="time",values_to="intensity",names_prefix="scaled.WT.mean.") %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=Gene.name,color=Gene.name))+
  geom_line(aes(color=Gene.name),size=1)+
  theme_classic()+
  labs(y="scaled intensity",x="time",title="SPO20 protein")+
  theme(legend.position = "none",axis.text.x = element_text(angle=90,vjust=0.6),axis.text=element_text(size=rel(1)),axis.title=element_text(size=rel(1)))

print_plot(spo20_line,
           "WT_spo20_medline.pdf",6,4)

ime2_line<-WT.protg.fin.df %>%
  filter(Gene.name=="IME2") %>%
  select(Gene.name,scaled.WT.mean.cols) %>%
  pivot_longer(-Gene.name,names_to="time",values_to="intensity",names_prefix="scaled.WT.mean.") %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=Gene.name,color=Gene.name))+
  geom_line(aes(color=Gene.name),size=1)+
  theme_classic()+
  labs(y="scaled intensity",x="time",title="Ime2 protein")+
  theme(legend.position = "none",axis.text.x = element_text(angle=90,vjust=0.6),axis.text=element_text(size=rel(1)),axis.title=element_text(size=rel(1)))

print_plot(ime2_line,
           "WT_ime2_line.pdf",6,4)

spo13_ime2_line<-spo13.protg.fin.df %>%
  filter(Gene.name=="IME2") %>%
  select(Gene.name,scaled.spo13.mean.cols) %>%
  pivot_longer(-Gene.name,names_to="time",values_to="intensity",names_prefix="scaled.spo13.mean.") %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=Gene.name,color=Gene.name))+
  geom_line(aes(color=Gene.name),size=1)+
  theme_classic()+
  labs(y="scaled intensity",x="time",title="Ime2 protein")+
  theme(legend.position = "none",axis.text.x = element_text(angle=90,vjust=0.6),axis.text=element_text(size=rel(1)),axis.title=element_text(size=rel(1)))

print_plot(spo13_ime2_line,
           "spo13_ime2_line.pdf",6,4)

WT_spo13_ime2_line<-WT.spo13.protg.fin.df %>%
  filter(Gene.name=="IME2") %>%
  select(Gene.name,WT.spo13.scaled.mean.cols) %>%
  pivot_longer(-Gene.name,names_to="time",values_to="intensity",names_prefix="scaled.both.") %>%
      mutate(strain=ifelse(grepl("WT",time),"WT","spo13")) %>%
    mutate(time=gsub("WT.mean.|spo13.mean.","",time)) %>%
  mutate(time=gsub("scaled.WT.mean.","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=strain,color=strain))+
  geom_line(aes(color=strain),size=1)+
  theme_classic()+
  labs(y="scaled intensity",x="time",title="Ime2 protein")+
  theme(axis.text.x = element_text(angle=90,vjust=0.6),axis.text=element_text(size=rel(1)),axis.title=element_text(size=rel(1)))

print_plot(WT_spo13_ime2_line,
           "WT_spo13_ime2_line.pdf",6,4)


# phospho-sites medlines by protein
#phosphoNET1
#median of all
net1_sites_medline<-WT.phos.fin.df %>%
  filter(Gene.name=="NET1") %>%
  select(Gene.name,gene.name_site,scaled.WT.mean.cols) %>%
  pivot_longer(-c(Gene.name,gene.name_site),names_to="time",values_to="intensity") %>%
  mutate(time=gsub("scaled.WT.mean.","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=Gene.name))+
  stat_summary(aes(y=intensity,group=Gene.name),fun=median,geom="line",size=1.5)+
  stat_summary(aes(y=intensity,group=Gene.name),fun.min=function(z){quantile(z,0.25)},fun.max=function(z){quantile(z,0.75)},geom="ribbon",alpha=0.1)+
    geom_vline(xintercept="90",linetype="dashed")+
    geom_vline(xintercept="135",linetype="dashed")+
  theme_classic()+
   theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),axis.text.y=element_text(size=rel(1.5)),plot.title=element_text(size=rel(1.5)),legend.title=element_text(size=rel(1.5)),legend.text=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)))+
  labs(x="time",y="scaled intensity",title=paste0("Net1 phospho-sites, n=",WT.phos.fin.df %>%
  filter(Gene.name=="NET1") %>% nrow()))

print_plot(net1_sites_medline,
           "WT_net1_sites_medline.pdf",6,5)

# IME2 phospho-sites
ime2_sites_medline<-WT.phos.fin.df %>%
  filter(Gene.name=="IME2") %>%
  select(Gene.name,gene.name_site,scaled.WT.mean.cols) %>%
  pivot_longer(-c(Gene.name,gene.name_site),names_to="time",values_to="intensity") %>%
  mutate(time=gsub("scaled.WT.mean.","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=Gene.name))+
  stat_summary(aes(y=intensity,group=Gene.name),fun=median,geom="line",size=1.5)+
  stat_summary(aes(y=intensity,group=Gene.name),fun.min=function(z){quantile(z,0.25)},fun.max=function(z){quantile(z,0.75)},geom="ribbon",alpha=0.1)+
  theme_classic()+
   theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),axis.text.y=element_text(size=rel(1.5)),plot.title=element_text(size=rel(1.5)),legend.title=element_text(size=rel(1.5)),legend.text=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)))+
  labs(x="time",y="scaled intensity",title=paste0("Ime2 phospho-sites, n=",WT.phos.fin.df %>%
  filter(Gene.name=="IME2") %>% nrow()))

print_plot(ime2_sites_medline,
           "WT_ime2_sites_medline.pdf",6,5)

# all sites lines
ime2_individual_sites_lines<-WT.phos.fin.df %>%
  filter(Gene.name=="IME2") %>%
  select(Gene.name,gene.name_site,scaled.WT.mean.cols) %>%
  pivot_longer(-c(Gene.name,gene.name_site),names_to="time",values_to="intensity") %>%
  mutate(time=gsub("scaled.WT.mean.","",time)) %>%
     mutate(gene.name_site=factor(gene.name_site,levels=unique(mixedsort(gene.name_site,decreasing=T)))) %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=gene.name_site))+
    geom_line(aes(y=intensity,color=gene.name_site),size=1.5)+
    facet_wrap(~gene.name_site)+
  theme_classic()+
   theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),axis.text.y=element_text(size=rel(1.5)),plot.title=element_text(size=rel(1.5)),legend.title=element_text(size=rel(1.5)),legend.text=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),legend.position="none")+
  labs(x="time",y="scaled intensity",title=paste0("Ime2 phospho-sites, n=",WT.phos.fin.df %>%
  filter(Gene.name=="IME2") %>% nrow()))

print_plot(ime2_individual_sites_lines,
           "WT_ime2_individual_sites_lines.pdf",8,5.5)


# WT v spo13
WT_spo13_ime2_sites_medline<-WT.spo13.phos.fin.df %>%
  filter(Gene.name=="IME2") %>%
  select(Gene.name,gene.name_site,WT.spo13.scaled.mean.cols) %>%
  pivot_longer(-c(Gene.name,gene.name_site),names_to="time",values_to="intensity",names_prefix="scaled.both.") %>%
    mutate(strain=ifelse(grepl("WT",time),"WT","spo13")) %>%
    mutate(time=gsub("WT.mean.|spo13.mean.","",time)) %>%
  mutate(time=gsub("scaled.WT.mean.","",time)) %>%
     mutate(gene.name_site=factor(gene.name_site,levels=unique(mixedsort(gene.name_site,decreasing=T)))) %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=strain))+
    geom_line(aes(y=intensity,color=strain),size=1.5)+
    facet_wrap(~gene.name_site)+
  theme_classic()+
   theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),axis.text.y=element_text(size=rel(1.5)),plot.title=element_text(size=rel(1.5)),legend.title=element_text(size=rel(1.5)),legend.text=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)))+
  labs(x="time",y="scaled intensity",title=paste0("Ime2 phospho-sites, n=",WT.spo13.phos.fin.df %>%
  filter(Gene.name=="IME2") %>% nrow()))

print_plot(WT_spo13_ime2_sites_medline,
           "WT_spo13_ime2_sites_medline.pdf",6,5)


# RIM4 phosphosites
rim4_sites_medline<-WT.phos.fin.df %>%
  filter(Gene.name=="RIM4") %>%
  select(Gene.name,gene.name_site,scaled.WT.mean.cols) %>%
  pivot_longer(-c(Gene.name,gene.name_site),names_to="time",values_to="intensity") %>%
  mutate(time=gsub("scaled.WT.mean.","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=Gene.name))+
  stat_summary(aes(y=intensity,group=Gene.name),fun=median,geom="line",size=1.5)+
  stat_summary(aes(y=intensity,group=Gene.name),fun.min=function(z){quantile(z,0.25)},fun.max=function(z){quantile(z,0.75)},geom="ribbon",alpha=0.1)+
  theme_classic()+
   theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),axis.text.y=element_text(size=rel(1.5)),plot.title=element_text(size=rel(1.5)),legend.title=element_text(size=rel(1.5)),legend.text=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)))+
  labs(x="time",y="scaled intensity",title=paste0("Rim4 phospho-sites, n=",WT.phos.fin.df %>%
  filter(Gene.name=="RIM4") %>% nrow()))+
    coord_cartesian(ylim=c(0,2))

print_plot(rim4_sites_medline,
           "WT_rim4_sites_medline.pdf",6,5)

# Sli15
sli15_sites_medline<-WT.phos.fin.df %>%
  filter(Gene.name=="SLI15") %>%
  select(Gene.name,gene.name_site,scaled.WT.mean.cols) %>%
  pivot_longer(-c(Gene.name,gene.name_site),names_to="time",values_to="intensity") %>%
  mutate(time=gsub("scaled.WT.mean.","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=Gene.name))+
  stat_summary(aes(y=intensity,group=Gene.name),fun=median,geom="line",size=1.5)+
  stat_summary(aes(y=intensity,group=Gene.name),fun.min=function(z){quantile(z,0.25)},fun.max=function(z){quantile(z,0.75)},geom="ribbon",alpha=0.1)+
  theme_classic()+
   theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),axis.text.y=element_text(size=rel(1.5)),plot.title=element_text(size=rel(1.5)),legend.title=element_text(size=rel(1.5)),legend.text=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)))+
  labs(x="time",y="scaled intensity",title=paste0("Sli15 phospho-sites, n=",WT.phos.fin.df %>%
  filter(Gene.name=="SLI15") %>% nrow()))+
    coord_cartesian(ylim=c(0,2))

print_plot(sli15_sites_medline,
           "WT_sli15_sites_medline.pdf",6,5)

spo74_sites_medline<-WT.phos.fin.df %>%
  filter(Gene.name=="SPO74") %>%
  select(Gene.name,gene.name_site,scaled.WT.mean.cols) %>%
  pivot_longer(-c(Gene.name,gene.name_site),names_to="time",values_to="intensity") %>%
  mutate(time=gsub("scaled.WT.mean.","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=Gene.name))+
  stat_summary(aes(y=intensity,group=Gene.name),fun=median,geom="line",size=1.5)+
  stat_summary(aes(y=intensity,group=Gene.name),fun.min=function(z){quantile(z,0.25)},fun.max=function(z){quantile(z,0.75)},geom="ribbon",alpha=0.1)+
  theme_classic()+
   theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),axis.text.y=element_text(size=rel(1.5)),plot.title=element_text(size=rel(1.5)),legend.title=element_text(size=rel(1.5)),legend.text=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)))+
  labs(x="time",y="scaled intensity",title=paste0("SPO74 phospho-sites, n=",WT.phos.fin.df %>%
  filter(Gene.name=="SPO74") %>% nrow()))+
    coord_cartesian(ylim=c(0,2))

print_plot(spo74_sites_medline,
           "WT_spo74_sites_medline.pdf",6,5)

# cell cycle regulated proteins
WT_cell_cycle_proteins<-WT.protg.fin.df %>%
  filter(Gene.name %in% c("SGO1","NDT80","NDC80","CLB1","CLB4","CLB3","CLB2","CLB5","CLB6","PDS1","SPO13","CDC20","AMA1","MND2","APC2","RIM4","REC8","MAM1","ZIP1")) %>%
  select(Gene.name,scaled.WT.mean.cols) %>%
  pivot_longer(-Gene.name,names_to="time",values_to="intensity",names_prefix="scaled.WT.mean.") %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=Gene.name,color=Gene.name))+
  geom_line(aes(color=Gene.name),size=1)+
  facet_wrap(~Gene.name,scales="free_y")+
    scale_y_continuous(labels=label_number(accuracy=0.1))+
  theme_classic()+
    theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),axis.text.y=element_text(size=rel(1.5)),plot.title=element_text(size=rel(1.5)),legend.title=element_text(size=rel(1.5)),legend.text=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),strip.text=element_text(size=rel(1.5)))+
  labs(y="scaled intensity",x="time",title="WT cell cycle regulated proteins")+
  theme(legend.position = "none",axis.text.x = element_text(angle=90,vjust=0.6))
# 7 x 15

print_plot(WT_cell_cycle_proteins,
           "WT_cell_cycle_proteins.pdf",15,7)

# protein kinases
# 
# kinase name list of all kinases in 12 motif df
# need this list because some motifs are overlapping
# SGK and MEK1 motifs overlap
# CDC7 and CKA motifs overlap
# TPK1-3 are PKA homologs. PKA/IPL1 motif overlap

short_kinase_name_list<-c("CDC28","IME2","CDC5","IPL1","MEK1","SGK2","CDC15","ATG1","CDC7","CKA1","MPS1","MEC1","HRR25","TPK1","TPK2","TPK3")

long_kinase_name_list<-c("AKL1","ARK1","ATG1","CAK1","CDC15","CDC28","CDC5","CDC7","CKA1","CLA4",
                         "CMK1","CMK2","FMP48","FPK1","FRK1","FUS3","GCN2","GIN4","HOG1","HRR25",
                         "HSL1","IPL1","KCC4","KDX1","KIN1","KIN2","KIN3","KIN4","KSP1","KSS1","MCK1",
                         "MEK1","MPS1","MRK1","NNK1","PHO85","PKH2","PRK1","PRR1","PSK2","PTK2","RAD53","RCK1","RCK2","RIM11","RIM15","RTK1","SAK1","SCH9","SKM1","SKS1","SKY1","SLT2","SNF1","STE20","TOS3","TPK1","TPK2","TPK3","VHS1","YAK1","YCK1","YCK2","YCK3","YGK3","YPK1","YPK2")


WT_protein_levels_kinases_short<-WT.protg.fin.df %>%
  filter(Gene.name %in% short_kinase_name_list) %>%
  select(Gene.name,scaled.WT.mean.cols) %>%
  pivot_longer(-Gene.name,names_to="time",values_to="intensity",names_prefix="scaled.WT.mean.") %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=Gene.name,color=Gene.name))+
  geom_line(aes(color=Gene.name),size=1)+
  facet_wrap(~Gene.name,scales="free_y")+
    scale_y_continuous(labels=label_number(accuracy=0.1))+
  theme_classic()+
    theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),axis.text.y=element_text(size=rel(1.5)),plot.title=element_text(size=rel(1.5)),legend.title=element_text(size=rel(1.5)),legend.text=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),strip.text=element_text(size=rel(1.5)))+
  labs(y="scaled intensity",x="time",title="WT kinases protein level")+
  theme(legend.position = "none",axis.text.x = element_text(angle=90,vjust=0.6))
# 7 x 15

print_plot(WT_protein_levels_kinases_short,
           "WT_protein_levels_kinases_short.pdf",15,7)

WT_protein_levels_kinases_long<-WT.protg.fin.df %>%
  filter(Gene.name %in% long_kinase_name_list) %>%
  select(Gene.name,scaled.WT.mean.cols) %>%
  pivot_longer(-Gene.name,names_to="time",values_to="intensity",names_prefix="scaled.WT.mean.") %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=Gene.name,color=Gene.name))+
  geom_line(aes(color=Gene.name),size=1)+
  facet_wrap(~Gene.name,scales="free_y")+
    scale_y_continuous(labels=label_number(accuracy=0.1))+
  theme_classic()+
    theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),axis.text.y=element_text(size=rel(1.5)),plot.title=element_text(size=rel(1.5)),legend.title=element_text(size=rel(1.5)),legend.text=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),strip.text=element_text(size=rel(1.5)))+
  labs(y="scaled intensity",x="time",title="WT kinases protein level")+
  theme(legend.position = "none",axis.text.x = element_text(angle=90,vjust=0.6))
# 12 x 17

print_plot(WT_protein_levels_kinases_long,
           "WT_protein_levels_kinases_long.pdf",17,12)


```

# MTA transition - PROTEIN

```{r}

# PROTEIN
# metaphase I to anaphase I

# what are relevant metaphase I comparison cols
WT_metaI_cols<-grep("^WT.sig.(.+)_v_75",colnames(WT.protg.fin.df),value=T)
spo13_metaI_cols<-grep("^spo13.sig.(.+)_v_75",colnames(spo13.protg.fin.df),value=T)

# replace NAs with "Zero" to ues logic
# what are relevant metaphase I comparison cols
WT.protg.fin.df[,WT_metaI_cols][t(apply(WT.protg.fin.df[,WT_metaI_cols],1,function(x) is.na(x)))]<-"Zero"
spo13.protg.fin.df[,spo13_metaI_cols][t(apply(spo13.protg.fin.df[,spo13_metaI_cols],1,function(x) is.na(x)))]<-"Zero"

# WT.protg.fin.df %>%
#     filter(if_any(WT_metaI_cols[1:2],function(x) x=="INCR"|x=="DECR")) %>% nrow() # 53 

# only first 2 metaI cols can be called metaphase I v anaphase I
WT_MTA1_heatmap<-WT.protg.fin.df %>%
    filter(if_any(WT_metaI_cols[1:2],function(x) x=="INCR"|x=="DECR")) %>%
    filter(if_all(WT_metaI_cols,function(x) all(x!="Zero"))) %>%
    select(Gene.name,WT.fc.90_v_75,WT.fc.105_v_75) %>%
    pivot_longer(-Gene.name,values_to="fold_change",names_to="sample") %>%
     mutate(fc_trimmed=ifelse(fold_change>2.2,2.2,
                             ifelse(fold_change<0.2,0.2,fold_change)))%>%
     mutate(Gene.name=fct_reorder2(Gene.name,sample,desc(fc_trimmed))) %>%
    # note desc(sample) not used here becauses sample name is character and 135 comes before 90
    # so to have same order ass all 120 or later metaI just do asescnding...
    ggplot(aes(x=fct_inorder(sample),y=Gene.name,fill=fc_trimmed)) +
    geom_tile()+
    scale_fill_distiller(palette="RdBu",limits=c(0.2,2.2))+
    labs(title="Metaphase I to Anaphase I",x="",y="")+
    theme_minimal()+
    theme(axis.text.x = element_text(angle=90,vjust=0.6))


# export to pdf 5x8 portrait
print_plot(WT_MTA1_heatmap,
           "WT_MTA1_prot_heatmap.pdf",5,8)


# metaphase II colss
# what are relevant metaphase II comparison cols
# note calling metaII 120 and not 105 
WT_metaII_cols<-grep("^WT.sig.(.+)_v_120",colnames(WT.protg.fin.df),value=T)
spo13_metaII_cols<-grep("^spo13.sig.(.+)_v_120",colnames(spo13.protg.fin.df),value=T)

# replace NAs with Zero to ues logic
# what are relevant metaphase I comparison cols
WT.protg.fin.df[,WT_metaII_cols][t(apply(WT.protg.fin.df[,WT_metaII_cols],1,function(x) is.na(x)))]<-"Zero"
spo13.protg.fin.df[,spo13_metaII_cols][t(apply(spo13.protg.fin.df[,spo13_metaII_cols],1,function(x) is.na(x)))]<-"Zero"

# WT.protg.fin.df %>%
#     filter(if_any(WT_metaII_cols,function(x) x=="INCR"|x=="DECR")) %>% nrow() #44

# Metaphase II v Anaphases II proteins
WT_MTA2_heatmap<-WT.protg.fin.df %>%
    filter(if_any(WT_metaII_cols,function(x) x=="INCR"|x=="DECR")) %>%
     filter(if_all(WT_metaII_cols,function(x) all(x!="Zero"))) %>%
    select(Gene.name,WT.fc.135_v_120,WT.fc.150_v_120,WT.fc.165_v_120) %>%
    pivot_longer(-Gene.name,values_to="fold_change",names_to="sample") %>%
     mutate(fc_trimmed=ifelse(fold_change>2.2,2.2,
                             ifelse(fold_change<0.2,0.2,fold_change))) %>%
    mutate(Gene.name=fct_reorder2(Gene.name,desc(sample),desc(fc_trimmed))) %>%
    ggplot(aes(x=fct_inorder(sample),y=Gene.name,fill=fc_trimmed)) +
    geom_tile()+
    scale_fill_distiller(palette="RdBu",limits=c(0.2,2.2))+
    labs(title="Metaphase II to Anaphase II",x="",y="")+
    theme_minimal()+
    theme(axis.text.x = element_text(angle=90,vjust=0.6))

# exported both to pdf 5x8 portrait
print_plot(WT_MTA2_heatmap,
           "WT_MTA2_prot_heatmap.pdf",5,8)


# GO terms of proteins at MTAs
# barplot of selected GO terms

# GO terms enriched among metaI sigdiff proteins
# 
# return terms enriched in WT metaI cols
# gost(WT.protg.fin.df %>%
#     filter(if_any(WT_metaI_cols[1:2],function(x) x=="DECR"|x=="INCR")) %>% select(Gene.name) %>% unlist(),organism='scerevisiae',evcodes=TRUE, significant=TRUE)$result %>%
#     select(term_name,p_value,source,term_id)

fav_terms_1<-c("meiotic cell cycle","septin complex","regulation of ubiquitin protein ligase activity","spore wall assembly","prospore membrane","meiosis I","cytoskeleton","anaphase promoting complex","metaphase/anaphase transition of cell cycle")

# double sided barplot with INCR on one sside and DECR terms on the other
# anaphase I v metaphase I
WT_MTA1_GO_bar<-rbind(INCR=gost(WT.protg.fin.df %>%
    filter(if_any(WT_metaI_cols[1:2],function(x) x=="INCR")) %>% select(Gene.name) %>% unlist(),organism='scerevisiae',evcodes=TRUE, significant=TRUE)$result %>%
    filter(term_name %in% fav_terms_1) %>%
    select(term_name,term_id,p_value),DECR=
gost(WT.protg.fin.df %>%
    filter(if_any(WT_metaI_cols[1:2],function(x) x=="DECR")) %>% select(Gene.name) %>% unlist(),organism='scerevisiae',evcodes=TRUE, significant=TRUE)$result %>%
    filter(term_name %in% fav_terms_1) %>%
    select(term_name,term_id,p_value)) %>%
    mutate(direction=ifelse(grepl("INCR",rownames(.)),"INCR","DECR")) %>%
    mutate(p_value=ifelse(grepl("DECR",direction),log10(p_value),-log10(p_value))) %>%
    ggplot(aes(x=term_name,y=p_value,fill=direction))+
    geom_bar(stat="identity",position="identity")+
    theme_classic()+
    theme(axis.text.x=element_text(angle=90),axis.text=element_text(size=rel(1)))+
    labs(x="",y="log10(p-value)",title="anaphase I v metaphase I proteins")+
    coord_flip()+
    scale_y_continuous(labels=label_number(accuracy=1))


print_plot(WT_MTA1_GO_bar,
           "WT_MTA1_GO_bar.pdf",8,3)

# GO termss enriched among sigdiff metaII v anaII proteinss
# gost(WT.protg.fin.df %>%
#     filter(if_any(WT_metaII_cols,function(x) x=="DECR")) %>% select(Gene.name) %>% unlist(),organism='scerevisiae',evcodes=TRUE, significant=TRUE)$result %>%
#     select(term_name,p_value,source,term_id)


fav_terms_2<-c("meiotic cell cycle","polysaccharide metabolic process","septin complex","regulation of ubiquitin protein ligase activity","spore wall assembly","glucan metabolic process","prospore membrane","anaphase promoting complex","metaphase/anaphase transition of cell cycle","condensed nuclear chromosome","meiotic chromosome segregation","meiotic nuclear division","(1->3)-beta-D-glucan metabolic process","cell wall macromolecule metabolic process","cytoskeleton","septin complex")

# gost(WT.protg.fin.df %>%
#     filter(if_any(WT_metaII_cols,function(x) x=="INCR")) %>% select(Gene.name) %>% unlist(),organism='scerevisiae',evcodes=TRUE, significant=TRUE)$result %>%
#     filter(term_name %in% fav_terms_2) %>%
#     select(term_name,term_id,p_value)


# anaphase II v metaphase II
WT_MTA2_GO_bar<-rbind(INCR=gost(WT.protg.fin.df %>%
    filter(if_any(WT_metaII_cols,function(x) x=="INCR")) %>% select(Gene.name) %>% unlist(),organism='scerevisiae',evcodes=TRUE, significant=TRUE)$result %>%
    filter(term_name %in% fav_terms_2) %>%
    select(term_name,term_id,p_value),DECR=
gost(WT.protg.fin.df %>%
    filter(if_any(WT_metaII_cols,function(x) x=="DECR")) %>% select(Gene.name) %>% unlist(),organism='scerevisiae',evcodes=TRUE, significant=TRUE)$result %>%
    filter(term_name %in% fav_terms_2) %>%
    select(term_name,term_id,p_value)) %>%
    mutate(direction=ifelse(grepl("INCR",rownames(.)),"INCR","DECR")) %>%
    mutate(p_value=ifelse(grepl("DECR",direction),log10(p_value),-log10(p_value))) %>%
    ggplot(aes(x=term_name,y=p_value,fill=direction))+
    geom_bar(stat="identity",position="identity")+
    theme_classic()+
    theme(axis.text.x=element_text(angle=90),axis.text=element_text(size=rel(1)))+
    labs(x="",y="log10(p-value)",title="anaphase II v metaphase II proteins")+
    coord_flip()+
    scale_y_continuous(labels=label_number(accuracy=1))


print_plot(WT_MTA2_GO_bar,
           "WT_MTA2_GO_bar.pdf",8,3)

```

# MTA WT v spo13 -- PROTEIN
```{r}

# of stuff found in BOTH WT and spo13 (all 4 TCs)
# what changes between WT and spo13 MTA1 and MTA2

# look at stuff that sig changes in MTA1 and MTA2 in EITHER strain 
#   --are trends the ssame even if they don't reach same level of sig?

#   MTA1
# WT.spo13.protg.fin.df %>%
#     filter(if_any(c(WT_metaI_cols[1:2],spo13_metaI_cols[1:2]),function(x) x=="INCR"|x=="DECR")) %>%
#     filter(!if_any(c(WT.fc.90_v_75,WT.fc.105_v_75,spo13.fc.90_v_75,spo13.fc.105_v_75),function(x) is.na(x))) %>%
#     select(Gene.name,WT.fc.90_v_75,WT.fc.105_v_75,spo13.fc.90_v_75,spo13.fc.105_v_75) %>% nrow() #n=66

WT_spo13_MTA1_protein_heatmap<-WT.spo13.protg.fin.df %>%
    filter(if_any(c(WT_metaI_cols[1:2],spo13_metaI_cols[1:2]),function(x) x=="INCR"|x=="DECR")) %>%
    filter(!if_any(c(WT.fc.90_v_75,WT.fc.105_v_75,spo13.fc.90_v_75,spo13.fc.105_v_75),function(x) is.na(x))) %>%
    select(Gene.name,WT.fc.90_v_75,WT.fc.105_v_75,spo13.fc.90_v_75,spo13.fc.105_v_75) %>%
    pivot_longer(-Gene.name,values_to="fold_change",names_to="sample") %>%
     mutate(fc_trimmed=ifelse(fold_change>2.2,2.2,
                             ifelse(fold_change<0.2,0.2,fold_change))) %>%
    mutate(sample=factor(sample,levels=c("WT.fc.90_v_75","spo13.fc.90_v_75","WT.fc.105_v_75","spo13.fc.105_v_75"))) %>%
     mutate(Gene.name=fct_reorder2(Gene.name,desc(sample),desc(fc_trimmed),.na_rm=F))%>%
    ggplot(aes(x=sample,y=Gene.name,fill=fc_trimmed)) +
    geom_tile()+
    scale_fill_distiller(palette="RdBu",limits=c(0.2,2.2))+
    labs(title="Metaphase I to Anaphase I",x="",y="")+
    theme_minimal()+
    theme(axis.text.x = element_text(angle=90,vjust=0.6))

# MTA2

# WT.spo13.protg.fin.df %>%
#     filter(if_any(c(WT_metaII_cols,spo13_metaII_cols),function(x) x=="INCR"|x=="DECR")) %>%
#     filter(!if_any(c(WT.fc.135_v_120,WT.fc.150_v_120,WT.fc.165_v_120,spo13.fc.135_v_120,spo13.fc.150_v_120,spo13.fc.165_v_120),function(x) is.na(x))) %>% nrow() # 64

WT_spo13_MTA2_protein_heatmap<-WT.spo13.protg.fin.df %>%
    filter(if_any(c(WT_metaII_cols,spo13_metaII_cols),function(x) x=="INCR"|x=="DECR")) %>%
    filter(!if_any(c(WT.fc.135_v_120,WT.fc.150_v_120,WT.fc.165_v_120,spo13.fc.135_v_120,spo13.fc.150_v_120,spo13.fc.165_v_120),function(x) is.na(x))) %>%
    select(Gene.name,WT.fc.135_v_120,WT.fc.150_v_120,WT.fc.165_v_120,spo13.fc.135_v_120,spo13.fc.150_v_120,spo13.fc.165_v_120) %>%
    pivot_longer(-Gene.name,values_to="fold_change",names_to="sample") %>%
     mutate(fc_trimmed=ifelse(fold_change>2.2,2.2,
                             ifelse(fold_change<0.2,0.2,fold_change))) %>%
    mutate(sample=factor(sample,levels=c("WT.fc.135_v_120","spo13.fc.135_v_120","WT.fc.150_v_120","spo13.fc.150_v_120","WT.fc.165_v_120","spo13.fc.165_v_120"))) %>%
     mutate(Gene.name=fct_reorder2(Gene.name,desc(sample),desc(fc_trimmed),.na_rm=F))%>%
    # note desc(sample) not used here becauses sample name is character and 135 comes before 90
    # so to have same order as all 120 or later metaI just do asescnding...
    ggplot(aes(x=sample,y=Gene.name,fill=fc_trimmed)) +
    geom_tile()+
    scale_fill_distiller(palette="RdBu",limits=c(0.2,2.2))+
    labs(title="Metaphase II to Anaphase II",x="",y="")+
    theme_minimal()+
    theme(axis.text.x = element_text(angle=90,vjust=0.6))


print_plot(WT_spo13_MTA1_protein_heatmap,
           "WT_spo13_MTA1_protein_heatmap.pdf",5,11)

print_plot(WT_spo13_MTA2_protein_heatmap,
           "WT_spo13_MTA2_protein_heatmap.pdf",5,11)

# MTA protein transitions Wt v spo13 quantitative trend
# MTA1
# difference between fcs 90 v 75 and 105 v 75 boxplot
WT_spo13_MTA1_boxplot<-WT.spo13.protg.fin.df %>%
    filter(if_any(c(WT_metaI_cols[1:2],spo13_metaI_cols[1:2]),function(x) x=="INCR"|x=="DECR")) %>%
    filter(!if_any(c(WT.fc.90_v_75,WT.fc.105_v_75,spo13.fc.90_v_75,spo13.fc.105_v_75),function(x) is.na(x))) %>%
    select(Gene.name,WT.fc.90_v_75,WT.fc.105_v_75,spo13.fc.90_v_75,spo13.fc.105_v_75) %>%
    pivot_longer(-Gene.name,values_to="fold_change",names_to="sample") %>%
     mutate(fc_trimmed=ifelse(fold_change>2.2,2.2,
                             ifelse(fold_change<0.2,0.2,fold_change))) %>%
    mutate(sample=factor(sample,levels=c("WT.fc.90_v_75","spo13.fc.90_v_75","WT.fc.105_v_75","spo13.fc.105_v_75"))) %>%
    ggplot(aes(x=sample,y=fc_trimmed,group=sample))+
        geom_boxplot(aes(fill=sample),position='dodge')+
    labs(title="Metaphase I to Anaphase I",x="",y="trimmed fold change")+
    geom_hline(yintercept=1.5,linetype="dashed",size=0.5)+
    geom_hline(yintercept=(1/1.5),linetype="dashed",size=0.5)+
    theme_classic()+
    theme(axis.text.x = element_text(angle=90,vjust=0.6))+
    scale_y_continuous(expand=expansion(mult=c(0.1,0.1)))


WT_spo13_MTA2_boxplot<-WT.spo13.protg.fin.df %>%
    filter(if_any(c(WT_metaII_cols,spo13_metaII_cols),function(x) x=="INCR"|x=="DECR")) %>%
    filter(!if_any(c(WT.fc.135_v_120,WT.fc.150_v_120,WT.fc.165_v_120,spo13.fc.135_v_120,spo13.fc.150_v_120,spo13.fc.165_v_120),function(x) is.na(x))) %>%
    select(Gene.name,WT.fc.135_v_120,WT.fc.150_v_120,WT.fc.165_v_120,spo13.fc.135_v_120,spo13.fc.150_v_120,spo13.fc.165_v_120) %>%
    pivot_longer(-Gene.name,values_to="fold_change",names_to="sample") %>%
     mutate(fc_trimmed=ifelse(fold_change>2.2,2.2,
                             ifelse(fold_change<0.2,0.2,fold_change))) %>%
    mutate(sample=factor(sample,levels=c("WT.fc.135_v_120","spo13.fc.135_v_120","WT.fc.150_v_120","spo13.fc.150_v_120","WT.fc.165_v_120","spo13.fc.165_v_120"))) %>%
     ggplot(aes(x=sample,y=fc_trimmed,group=sample))+
        geom_boxplot(aes(fill=sample),position='dodge',na.rm=T)+
    labs(title="Metaphase II to Anaphase II",x="",y="trimmed fold change")+
        geom_hline(yintercept=1.5,linetype="dashed",size=0.5)+
    geom_hline(yintercept=(1/1.5),linetype="dashed",size=0.5)+
    theme_classic()+
    theme(axis.text.x = element_text(angle=90,vjust=0.6))+
    scale_y_continuous(expand=expansion(mult=c(0.1,0.1)))
#5 x6

print_plot(WT_spo13_MTA1_boxplot,
           "WT_spo13_MTA1_boxplot.pdf",5,6)

print_plot(WT_spo13_MTA2_boxplot,
           "WT_spo13_MTA2_boxplot.pdf",5,6)

# boxplots split by INCR/DECR
WT_spo13_MTA1_INCR_boxplot<-WT.spo13.protg.fin.df %>%
    filter(if_any(c(WT_metaI_cols[1:2],spo13_metaI_cols[1:2]),function(x) x=="INCR")) %>%
    filter(!if_any(c(WT.fc.90_v_75,WT.fc.105_v_75,spo13.fc.90_v_75,spo13.fc.105_v_75),function(x) is.na(x))) %>%
    select(Gene.name,WT.fc.90_v_75,WT.fc.105_v_75,spo13.fc.90_v_75,spo13.fc.105_v_75) %>%
    pivot_longer(-Gene.name,values_to="fold_change",names_to="sample") %>%
     mutate(fc_trimmed=ifelse(fold_change>2.2,2.2,
                             ifelse(fold_change<0.2,0.2,fold_change))) %>%
    mutate(sample=factor(sample,levels=c("WT.fc.90_v_75","spo13.fc.90_v_75","WT.fc.105_v_75","spo13.fc.105_v_75"))) %>%
    ggplot(aes(x=sample,y=fc_trimmed,group=sample))+
        geom_boxplot(aes(fill=sample),position='dodge')+
    labs(title="Metaphase I to Anaphase I INCR",x="",y="trimmed fold change")+
    geom_hline(yintercept=1.5,linetype="dashed",size=0.5)+
    geom_hline(yintercept=(1/1.5),linetype="dashed",size=0.5)+
    theme_classic()+
    theme(axis.text.x = element_text(angle=90,vjust=0.6))+
    scale_y_continuous(expand=expansion(mult=c(0.1,0.1)))

WT_spo13_MTA1_DECR_boxplot<-WT.spo13.protg.fin.df %>%
    filter(if_any(c(WT_metaI_cols[1:2],spo13_metaI_cols[1:2]),function(x) x=="DECR")) %>%
    filter(!if_any(c(WT.fc.90_v_75,WT.fc.105_v_75,spo13.fc.90_v_75,spo13.fc.105_v_75),function(x) is.na(x))) %>%
    select(Gene.name,WT.fc.90_v_75,WT.fc.105_v_75,spo13.fc.90_v_75,spo13.fc.105_v_75) %>%
    pivot_longer(-Gene.name,values_to="fold_change",names_to="sample") %>%
     mutate(fc_trimmed=ifelse(fold_change>2.2,2.2,
                             ifelse(fold_change<0.2,0.2,fold_change))) %>%
    mutate(sample=factor(sample,levels=c("WT.fc.90_v_75","spo13.fc.90_v_75","WT.fc.105_v_75","spo13.fc.105_v_75"))) %>%
    ggplot(aes(x=sample,y=fc_trimmed,group=sample))+
        geom_boxplot(aes(fill=sample),position='dodge')+
    labs(title="Metaphase I to Anaphase I DECR",x="",y="trimmed fold change")+
    geom_hline(yintercept=1.5,linetype="dashed",size=0.5)+
    geom_hline(yintercept=(1/1.5),linetype="dashed",size=0.5)+
    theme_classic()+
    theme(axis.text.x = element_text(angle=90,vjust=0.6))+
    scale_y_continuous(expand=expansion(mult=c(0.1,0.1)))

WT_spo13_MTA2_INCR_boxplot<-WT.spo13.protg.fin.df %>%
    filter(if_any(c(WT_metaII_cols,spo13_metaII_cols),function(x) x=="INCR")) %>%
    filter(!if_any(c(WT.fc.135_v_120,WT.fc.150_v_120,WT.fc.165_v_120,spo13.fc.135_v_120,spo13.fc.150_v_120,spo13.fc.165_v_120),function(x) is.na(x))) %>%
    select(Gene.name,WT.fc.135_v_120,WT.fc.150_v_120,WT.fc.165_v_120,spo13.fc.135_v_120,spo13.fc.150_v_120,spo13.fc.165_v_120) %>%
    pivot_longer(-Gene.name,values_to="fold_change",names_to="sample") %>%
     mutate(fc_trimmed=ifelse(fold_change>2.2,2.2,
                             ifelse(fold_change<0.2,0.2,fold_change))) %>%
    mutate(sample=factor(sample,levels=c("WT.fc.135_v_120","spo13.fc.135_v_120","WT.fc.150_v_120","spo13.fc.150_v_120","WT.fc.165_v_120","spo13.fc.165_v_120"))) %>%
     ggplot(aes(x=sample,y=fc_trimmed,group=sample))+
        geom_boxplot(aes(fill=sample),position='dodge',na.rm=T)+
    labs(title="Metaphase II to Anaphase II INCR",x="",y="trimmed fold change")+
        geom_hline(yintercept=1.5,linetype="dashed",size=0.5)+
    geom_hline(yintercept=(1/1.5),linetype="dashed",size=0.5)+
    theme_classic()+
    theme(axis.text.x = element_text(angle=90,vjust=0.6))+
    scale_y_continuous(expand=expansion(mult=c(0.1,0.1)))

WT_spo13_MTA2_DECR_boxplot<-WT.spo13.protg.fin.df %>%
    filter(if_any(c(WT_metaII_cols,spo13_metaII_cols),function(x) x=="DECR")) %>%
    filter(!if_any(c(WT.fc.135_v_120,WT.fc.150_v_120,WT.fc.165_v_120,spo13.fc.135_v_120,spo13.fc.150_v_120,spo13.fc.165_v_120),function(x) is.na(x))) %>%
    select(Gene.name,WT.fc.135_v_120,WT.fc.150_v_120,WT.fc.165_v_120,spo13.fc.135_v_120,spo13.fc.150_v_120,spo13.fc.165_v_120) %>%
    pivot_longer(-Gene.name,values_to="fold_change",names_to="sample") %>%
     mutate(fc_trimmed=ifelse(fold_change>2.2,2.2,
                             ifelse(fold_change<0.2,0.2,fold_change))) %>%
    mutate(sample=factor(sample,levels=c("WT.fc.135_v_120","spo13.fc.135_v_120","WT.fc.150_v_120","spo13.fc.150_v_120","WT.fc.165_v_120","spo13.fc.165_v_120"))) %>%
     ggplot(aes(x=sample,y=fc_trimmed,group=sample))+
        geom_boxplot(aes(fill=sample),position='dodge',na.rm=T)+
    labs(title="Metaphase II to Anaphase II DECR",x="",y="trimmed fold change")+
        geom_hline(yintercept=1.5,linetype="dashed",size=0.5)+
    geom_hline(yintercept=(1/1.5),linetype="dashed",size=0.5)+
    theme_classic()+
    theme(axis.text.x = element_text(angle=90,vjust=0.6))+
    scale_y_continuous(expand=expansion(mult=c(0.1,0.1)))


print_plot(WT_spo13_MTA1_INCR_boxplot,
           "WT_spo13_MTA1_INCR_boxplot.pdf",5,6)

print_plot(WT_spo13_MTA1_DECR_boxplot,
           "WT_spo13_MTA1_DECR_boxplot.pdf",5,6)

print_plot(WT_spo13_MTA2_INCR_boxplot,
           "WT_spo13_MTA2_INCR_boxplot.pdf",5,6)

print_plot(WT_spo13_MTA2_DECR_boxplot,
           "WT_spo13_MTA2_DECR_boxplot.pdf",5,6)

```

# MTA WT site Motif trends
```{r}

# function to return ggseqlogo of non-cluster group of phospho-sites
# subset
short_logo<-function(df,meth,title){
  ggseqlogo(substr(df %>% select(Sequence.window) %>% unlist(),11,21),method=meth)+
    ggtitle(paste0(title,nrow(df)))+
    theme(legend.position="none")+
    scale_x_continuous(breaks=1:11,labels=-5:5)
}

# INCR MTA1 logo    
WT_phos_MTA1_INCR_logo<-short_logo(WT.phos.fin.df %>%
    filter(if_any(c(WT_metaI_cols[1:2]),function(x) x=="INCR")),"prob","WT MTA1 INCR sites, n=")

# DECR MTA1 logo
WT_phos_MTA1_DECR_logo<-short_logo(WT.phos.fin.df %>%
    filter(if_any(c(WT_metaI_cols[1:2]),function(x) x=="DECR")),"prob","WT MTA1 DECR sites, n=")

# INCR MTA2 logo
WT_phos_MTA2_INCR_logo<-short_logo(WT.phos.fin.df %>%
    filter(if_any(c(WT_metaII_cols),function(x) x=="INCR")),"prob","WT MTA2 INCR sites, n=")

# DECR MTA2 logo
WT_phos_MTA2_DECR_logo<-short_logo(WT.phos.fin.df %>%
    filter(if_any(c(WT_metaII_cols),function(x) x=="DECR")),"prob","WT MTA2 DECR sites, n=")


print_plot(WT_phos_MTA1_DECR_logo,
           "WT_phos_MTA1_DECR_logo.pdf",6,4)
print_plot(WT_phos_MTA1_INCR_logo,
           "WT_phos_MTA1_INCR_logo.pdf",6,4)
print_plot(WT_phos_MTA2_DECR_logo,
           "WT_phos_MTA2_DECR_logo.pdf",6,4)
print_plot(WT_phos_MTA2_INCR_logo,
           "WT_phos_MTA2_INCR_logo.pdf",6,4)


# PIE CHARTS
# number of major kinase motif matching sites that INCR/DECR at MTA1 and MTA2
# DECR MTA1
WT_phos_MTA1_DECR_pie<-WT.phos.fin.df %>%
    filter(if_any(c(WT_metaI_cols[1:2]),function(x) x=="DECR")) %>%
    mutate(CDC5=grepl(".............[DEN].[ST]...............",Sequence.window)) %>%
    mutate(CDC28_min=grepl("...............[ST]P..............",Sequence.window)) %>%
    mutate(IME2=grepl("............RP.[ST]...............",Sequence.window)) %>%
    mutate(IPL1=grepl(".............[RK].[ST]...............",Sequence.window)) %>%
    # need to tally which sites are in multiple kinase categories and call them 'mixed'
    mutate(DEN.SP=ifelse(CDC5==T&CDC28_min==T,T,F)) %>%
    mutate(IME2_SP=ifelse(CDC28_min==T&IME2==T,T,F)) %>%
    mutate(IPL1_SP=ifelse(CDC28_min==T&IPL1==T,T,F)) %>%
    mutate(kinase=ifelse(DEN.SP==T|IME2_SP==T|IPL1_SP==T,"mixed",
                                              ifelse(CDC28_min==T,"CDC28_min",
                                                            ifelse(CDC5==T,"CDC5",
                                                                   ifelse(IME2==T,"IME2",
                                                                    ifelse(IPL1==T,"IPL1","other")))))) %>% 
    mutate(kinase=factor(kinase,levels=c("mixed","CDC5","CDC28_min","IME2","IPL1","other"))) %>%
    group_by(kinase) %>% summarize(n=n()) %>%
ggplot(aes(x="",y=n,fill=kinase,label=n))+
    geom_bar(stat="identity")+
    geom_label_repel()+
    coord_polar("y", start=0)+
    theme_void()+
    scale_fill_manual(values=c("#7CAE00", "#00BFC4", "#C77CFF","blue","darkorange","darkgrey"),drop=FALSE)+
    ggtitle("DECR MTA1 sites")+
    theme(legend.title=element_blank())

# DECR MTA2
WT_phos_MTA2_DECR_pie<-WT.phos.fin.df %>%
    filter(if_any(c(WT_metaII_cols),function(x) x=="DECR")) %>%
    mutate(CDC5=grepl(".............[DEN].[ST]...............",Sequence.window)) %>%
    mutate(CDC28_min=grepl("...............[ST]P..............",Sequence.window)) %>%
    mutate(IME2=grepl("............RP.[ST]...............",Sequence.window)) %>%
    mutate(IPL1=grepl(".............[RK].[ST]...............",Sequence.window)) %>%
    # need to tally which sites are in multiple kinase categories and call them 'mixed'
    mutate(DEN.SP=ifelse(CDC5==T&CDC28_min==T,T,F)) %>%
    mutate(IME2_SP=ifelse(CDC28_min==T&IME2==T,T,F)) %>%
    mutate(IPL1_SP=ifelse(CDC28_min==T&IPL1==T,T,F)) %>%
    mutate(kinase=ifelse(DEN.SP==T|IME2_SP==T|IPL1_SP==T,"mixed",
                                              ifelse(CDC28_min==T,"CDC28_min",
                                                            ifelse(CDC5==T,"CDC5",
                                                                   ifelse(IME2==T,"IME2",
                                                                    ifelse(IPL1==T,"IPL1","other")))))) %>% 
    mutate(kinase=factor(kinase,levels=c("mixed","CDC5","CDC28_min","IME2","IPL1","other"))) %>%
    group_by(kinase) %>% summarize(n=n()) %>%
ggplot(aes(x="",y=n,fill=kinase,label=n))+
    geom_bar(stat="identity")+
    geom_label_repel()+
    coord_polar("y", start=0)+
    theme_void()+
    scale_fill_manual(values=c("#7CAE00", "#00BFC4", "#C77CFF","blue","darkorange","darkgrey"),drop=FALSE)+
    ggtitle("DECR MTA2 sites")+
    theme(legend.title=element_blank())

# INCR at MTA1 sites
WT_phos_MTA1_INCR_pie<-WT.phos.fin.df %>%
    filter(if_any(c(WT_metaI_cols[1:2]),function(x) x=="INCR")) %>%
    mutate(CDC5=grepl(".............[DEN].[ST]...............",Sequence.window)) %>%
    mutate(CDC28_min=grepl("...............[ST]P..............",Sequence.window)) %>%
    mutate(IME2=grepl("............RP.[ST]...............",Sequence.window)) %>%
    mutate(IPL1=grepl(".............[RK].[ST]...............",Sequence.window)) %>%
    # need to tally which sites are in multiple kinase categories and call them 'mixed'
    mutate(DEN.SP=ifelse(CDC5==T&CDC28_min==T,T,F)) %>%
    mutate(IME2_SP=ifelse(CDC28_min==T&IME2==T,T,F)) %>%
    mutate(IPL1_SP=ifelse(CDC28_min==T&IPL1==T,T,F)) %>%
    mutate(kinase=ifelse(DEN.SP==T|IME2_SP==T|IPL1_SP==T,"mixed",
                                              ifelse(CDC28_min==T,"CDC28_min",
                                                            ifelse(CDC5==T,"CDC5",
                                                                   ifelse(IME2==T,"IME2",
                                                                    ifelse(IPL1==T,"IPL1","other")))))) %>% 
    mutate(kinase=factor(kinase,levels=c("mixed","CDC5","CDC28_min","IME2","IPL1","other"))) %>%
    group_by(kinase) %>% summarize(n=n()) %>%
ggplot(aes(x="",y=n,fill=kinase,label=n))+
    geom_bar(stat="identity")+
    geom_label_repel()+
    coord_polar("y", start=0)+
    theme_void()+
    scale_fill_manual(values=c("#7CAE00", "#00BFC4", "#C77CFF","blue","darkorange","darkgrey"),drop=FALSE)+
    ggtitle("INCR MTA1 sites")+
    theme(legend.title=element_blank())

# INCR at MTA2
WT_phos_MTA2_INCR_pie<-WT.phos.fin.df %>%
    filter(if_any(c(WT_metaII_cols),function(x) x=="INCR")) %>%
    mutate(CDC5=grepl(".............[DEN].[ST]...............",Sequence.window)) %>%
    mutate(CDC28_min=grepl("...............[ST]P..............",Sequence.window)) %>%
    mutate(IME2=grepl("............RP.[ST]...............",Sequence.window)) %>%
    mutate(IPL1=grepl(".............[RK].[ST]...............",Sequence.window)) %>%
    # need to tally which sites are in multiple kinase categories and call them 'mixed'
    mutate(DEN.SP=ifelse(CDC5==T&CDC28_min==T,T,F)) %>%
    mutate(IME2_SP=ifelse(CDC28_min==T&IME2==T,T,F)) %>%
    mutate(IPL1_SP=ifelse(CDC28_min==T&IPL1==T,T,F)) %>%
    mutate(kinase=ifelse(DEN.SP==T|IME2_SP==T|IPL1_SP==T,"mixed",
                                              ifelse(CDC28_min==T,"CDC28_min",
                                                            ifelse(CDC5==T,"CDC5",
                                                                   ifelse(IME2==T,"IME2",
                                                                    ifelse(IPL1==T,"IPL1","other")))))) %>% 
    mutate(kinase=factor(kinase,levels=c("mixed","CDC5","CDC28_min","IME2","IPL1","other"))) %>%
    group_by(kinase) %>% summarize(n=n()) %>%
ggplot(aes(x="",y=n,fill=kinase,label=n))+
    geom_bar(stat="identity")+
    geom_label_repel()+
    coord_polar("y", start=0)+
    theme_void()+
    scale_fill_manual(values=c("#7CAE00", "#00BFC4", "#C77CFF","blue","darkorange","darkgrey"),drop=FALSE)+
    ggtitle("INCR MTA2 sites")+
    theme(legend.title=element_blank())


# all Pies together
print_plot((WT_phos_MTA1_INCR_pie+WT_phos_MTA1_DECR_pie)/
    (WT_phos_MTA2_INCR_pie+WT_phos_MTA2_DECR_pie),
    "WT_phos_MTA1_MTA2_pies.pdf",10,5)


# MOTIF LOGO OF n=95 sitess that are "other" In INCR MTA2
WT_phos_MTA2_INCR_other_logo<-WT.phos.fin.df %>%
    filter(if_any(c(WT_metaII_cols),function(x) x=="INCR")) %>%
    mutate(CDC5=grepl(".............[DEN].[ST]...............",Sequence.window)) %>%
    mutate(CDC28_min=grepl("...............[ST]P..............",Sequence.window)) %>%
    mutate(IME2=grepl("............RP.[ST]...............",Sequence.window)) %>%
    mutate(IPL1=grepl(".............[RK].[ST]...............",Sequence.window)) %>%
    # need to tally which sites are in multiple kinase categories and call them 'mixed'
    mutate(DEN.SP=ifelse(CDC5==T&CDC28_min==T,T,F)) %>%
    mutate(IME2_SP=ifelse(CDC28_min==T&IME2==T,T,F)) %>%
    mutate(IPL1_SP=ifelse(CDC28_min==T&IPL1==T,T,F)) %>%
    mutate(kinase=ifelse(DEN.SP==T|IME2_SP==T|IPL1_SP==T,"mixed",
                                              ifelse(CDC28_min==T,"CDC28_min",
                                                            ifelse(CDC5==T,"CDC5",
                                                                   ifelse(IME2==T,"IME2",
                                                                    ifelse(IPL1==T,"IPL1","other")))))) %>% 
    mutate(kinase=factor(kinase,levels=c("mixed","CDC5","CDC28_min","IME2","IPL1","other"))) %>%
    filter(kinase=="other") %>%
    select(Sequence.window) %>%
    unlist() %>%
    substr(.,11,21)%>%
    ggseqlogo(.,method="prob")+
ggtitle("WT MTA2 INCR sites, n=95")+
    theme(legend.position="none")+
    scale_x_continuous(breaks=1:11,labels=-5:5)

print_plot(WT_phos_MTA2_INCR_other_logo,
           "WT_phos_MTA2_INCR_other_logo.pdf",6,4)


# MTA1
# INCR/DECR separated plots # starting at t75
MTA1_motif_lineplot<-function(df,motif_name,motif,title){
    for(i in 1:length(motif_name)){
        df[,motif_name[i]]<-df %>%
        mutate("{motif_name[i]}" := ifelse(grepl(motif[i],Sequence.window),T,F)) %>%
            select(last_col())
    }
    df %>%
    select(gene.name_site,all_of(motif_name),metaI.scaled.WT.mean.cols,sig.cat) %>%
  pivot_longer(-c(gene.name_site,motif_name,sig.cat),names_prefix="metaI.scaled.WT.mean.",names_to="time",values_to="intensity") %>%
  pivot_longer(-c(gene.name_site,time,intensity,sig.cat),names_to="kinase",values_to="logical") %>%
  filter(logical==TRUE) %>%
    #    mutate(kinase=factor(kinase,levels=c("mixed","CDC5","CDC28_min","CDC28_strict","IME2","IPL1/PKA","other"))) %>%
  group_by(kinase) %>%
  mutate(kinase_n=n()/10) %>%
    mutate(kinase_cat_n=paste0(kinase," (",kinase_n,")")) %>%
        filter(time %in% alltimes[4:6]) %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=kinase_cat_n))+
  stat_summary(aes(y=intensity,color=kinase_cat_n),fun=median,geom="line",size=1)+
  stat_summary(aes(y=intensity,fill=kinase_cat_n),fun.min= function(z) {quantile(z,0.25)},
                 fun.max=function(z) {quantile(z,0.75)},
                 geom="ribbon",alpha=0.1)+
  theme_classic()+
    scale_y_continuous(labels=label_number(accuracy=0.1))+
  theme(axis.text.x = element_text(angle=90,vjust=0.6),axis.title=element_text(size=rel(2)),axis.text=element_text(size=rel(2)),plot.title=element_text(size=rel(2)),legend.text = element_text(size=rel(2)))+
  labs(y="scaled intensity",x="time",title=title)
}

# DEN together
WT_MTA1_DECR_motif_line<-MTA1_motif_lineplot(WT.phos.fin.df %>% filter(if_any(c(WT_metaI_cols[1:2]),function(x) x=="DECR")),c("CDC28_min","CDC28_strict","IME2","CDC5","IPL1/PKA"),c("...............[ST]P..............","...............[ST]P.[RK]............","............RP.[ST]...............",".............[DEN].[ST]...............",".............[RK].[ST]..............."),"DECR from t75") 

WT_MTA1_INCR_motif_line<-MTA1_motif_lineplot(WT.phos.fin.df %>% filter(if_any(c(WT_metaI_cols[1:2]),function(x) x=="INCR")),c("CDC28_min","CDC28_strict","IME2","CDC5","IPL1/PKA"),c("...............[ST]P..............","...............[ST]P.[RK]............","............RP.[ST]...............",".............[DEN].[ST]...............",".............[RK].[ST]..............."),"INCR from t75") 

# SSerine v Threonine
WT_MTA1_DECR_ST_line<-MTA1_motif_lineplot(WT.phos.fin.df %>% filter(if_any(c(WT_metaI_cols[1:2]),function(x) x=="DECR")),c("serine","threonine"),c("...............[S]...............","...............[T]..............."),"DECR from t75") 

WT_MTA1_INCR_ST_line<-MTA1_motif_lineplot(WT.phos.fin.df %>% filter(if_any(c(WT_metaI_cols[1:2]),function(x) x=="INCR")),c("serine","threonine"),c("...............[S]...............","...............[T]..............."),"INCR from t75") 

# SP v TP
WT_MTA1_DECR_STP_line<-MTA1_motif_lineplot(WT.phos.fin.df %>% filter(if_any(c(WT_metaI_cols[1:2]),function(x) x=="DECR")),c("SP","TP"),c("...............[S]P..............","...............[T]P.............."),"DECR from t75") 

WT_MTA1_INCR_STP_line<-MTA1_motif_lineplot(WT.phos.fin.df %>% filter(if_any(c(WT_metaI_cols[1:2]),function(x) x=="INCR")),c("SP","TP"),c("...............[S]P..............","...............[T]P.............."),"INCR from t75") 

# MTA2
# INCR/DECR separated plots # starting at t120
MTA2_motif_lineplot<-function(df,motif_name,motif,title){
    for(i in 1:length(motif_name)){
        df[,motif_name[i]]<-df %>%
        mutate("{motif_name[i]}" := ifelse(grepl(motif[i],Sequence.window),T,F)) %>%
            select(last_col())
    }
    df %>%
    select(gene.name_site,all_of(motif_name),metaII.scaled.WT.mean.cols,sig.cat) %>%
  pivot_longer(-c(gene.name_site,motif_name,sig.cat),names_prefix="metaII.scaled.WT.mean.",names_to="time",values_to="intensity") %>%
  pivot_longer(-c(gene.name_site,time,intensity,sig.cat),names_to="kinase",values_to="logical") %>%
  filter(logical==TRUE) %>%
      #  mutate(kinase=factor(kinase,levels=c("mixed","CDC5","CDC28_min","CDC28_strict","IME2","IPL1/PKA","other"))) %>%
  group_by(kinase) %>%
  mutate(kinase_n=n()/10) %>%
    mutate(kinase_cat_n=paste0(kinase," (",kinase_n,")")) %>%
        filter(time %in% alltimes[7:10]) %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=kinase_cat_n))+
  stat_summary(aes(y=intensity,color=kinase_cat_n),fun=median,geom="line",size=1)+
  stat_summary(aes(y=intensity,fill=kinase_cat_n),fun.min= function(z) {quantile(z,0.25)},
                 fun.max=function(z) {quantile(z,0.75)},
                 geom="ribbon",alpha=0.1)+
  theme_classic()+
        scale_y_continuous(labels=label_number(accuracy=0.1))+
  theme(axis.text.x = element_text(angle=90,vjust=0.6),axis.title=element_text(size=rel(2)),axis.text=element_text(size=rel(2)),plot.title=element_text(size=rel(2)),legend.text = element_text(size=rel(2)))+
  labs(y="scaled intensity",x="time",title=title)
}


WT_MTA2_DECR_motif_line<-MTA2_motif_lineplot(WT.phos.fin.df %>% filter(if_any(c(WT_metaII_cols),function(x) x=="DECR")),c("CDC28_min","CDC28_strict","IME2","CDC5","IPL1/PKA"),c("...............[ST]P..............","...............[ST]P.[RK]............","............RP.[ST]...............",".............[DEN].[ST]...............",".............[RK].[ST]..............."),"DECR from t120") 

WT_MTA2_INCR_motif_line<-MTA2_motif_lineplot(WT.phos.fin.df %>% filter(if_any(c(WT_metaII_cols),function(x) x=="INCR")),c("CDC28_min","CDC28_strict","IME2","CDC5","IPL1/PKA"),c("...............[ST]P..............","...............[ST]P.[RK]............","............RP.[ST]...............",".............[DEN].[ST]...............",".............[RK].[ST]..............."),"INCR from t120") 

# SSerine v Threonine MTA2
WT_MTA2_DECR_ST_line<-MTA2_motif_lineplot(WT.phos.fin.df %>% filter(if_any(c(WT_metaII_cols),function(x) x=="DECR")),c("serine","threonine"),c("...............[S]...............","...............[T]..............."),"DECR from t120") 

WT_MTA2_INCR_ST_line<-MTA2_motif_lineplot(WT.phos.fin.df %>% filter(if_any(c(WT_metaII_cols),function(x) x=="INCR")),c("serine","threonine"),c("...............[S]...............","...............[T]..............."),"INCR from t120") 

# SP v TP MTA2
WT_MTA2_DECR_STP_line<-MTA2_motif_lineplot(WT.phos.fin.df %>% filter(if_any(c(WT_metaII_cols),function(x) x=="DECR")),c("SP","TP"),c("...............[S]P..............","...............[T]P.............."),"DECR from t120") 

WT_MTA2_INCR_STP_line<-MTA2_motif_lineplot(WT.phos.fin.df %>% filter(if_any(c(WT_metaII_cols),function(x) x=="INCR")),c("SP","TP"),c("...............[S]P..............","...............[T]P.............."),"INCR from t120") 

# whole timecourse motif trends

get.grouped.motif.lineplot<-function(df,motif_name,motif,title){
    for(i in 1:length(motif_name)){
        df[,motif_name[i]]<-df %>%
        mutate("{motif_name[i]}" := ifelse(grepl(motif[i],Sequence.window),T,F)) %>%
            select(last_col())
    }
    df %>%
    select(gene.name_site,all_of(motif_name),scaled.WT.mean.cols) %>%
  pivot_longer(-c(gene.name_site,motif_name),names_prefix="scaled.WT.mean.",names_to="time",values_to="intensity") %>%
  pivot_longer(-c(gene.name_site,time,intensity),names_to="kinase",values_to="logical") %>%
  filter(logical==TRUE) %>%
  group_by(kinase) %>%
  mutate(kinase_n=n()/10) %>%
    mutate(kinase_cat_n=paste0(kinase," (",kinase_n,")")) %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=kinase_cat_n))+
  stat_summary(aes(y=intensity,color=kinase_cat_n),fun=median,geom="line",size=1)+
  stat_summary(aes(y=intensity,fill=kinase_cat_n),fun.min= function(z) {quantile(z,0.25)},
                 fun.max=function(z) {quantile(z,0.75)},
                 geom="ribbon",alpha=0.1)+
  theme_classic()+
        scale_y_continuous(labels=label_number(accuracy=0.1))+
 theme(axis.text.x = element_text(angle=90,vjust=0.6),axis.title=element_text(size=rel(2)),axis.text=element_text(size=rel(2)),plot.title=element_text(size=rel(2)),legend.text = element_text(size=rel(2)))+
  labs(y="scaled intensity",x="time",title=title)
   # return(df)
}

# all S v T
WT_phos_all_ST_line<-get.grouped.motif.lineplot(WT.phos.fin.df,c("serine","threonine"),c("...............[S]...............","...............[T]..............."),"S v T")

# all sig from t0 S v T
WT_phos_t0_sig_ST_line<-get.grouped.motif.lineplot(WT.phos.fin.df.sig,c("serine","threonine"),c("...............[S]...............","...............[T]..............."),"S v T sig from t0")

# all SP v TP
WT_phos_all_STP_line<-get.grouped.motif.lineplot(WT.phos.fin.df,c("SP","TP"),c("...............[S]P..............","...............[T]P.............."),"SP v TP")

# all sig from t0 SP v TP
WT_phos_t0_sig_STP_line<-get.grouped.motif.lineplot(WT.phos.fin.df.sig,c("SP","TP"),c("...............[S]P..............","...............[T]P.............."),"SP v TP sig from t0")

# all main motifs
WT_phos_all_motif_line<-get.grouped.motif.lineplot(WT.phos.fin.df,c("CDC28_min","CDC28_strict","IME2","CDC5","IPL1/PKA"),c("...............[ST]P..............","...............[ST]P.[RK]............","............RP.[ST]...............",".............[DEN].[ST]...............",".............[RK].[ST]..............."),"main motifs")

# all sig from t0 main motifs
WT_phos_t0_sig_motif_line<-get.grouped.motif.lineplot(WT.phos.fin.df.sig,c("CDC28_min","CDC28_strict","IME2","CDC5","IPL1/PKA"),c("...............[ST]P..............","...............[ST]P.[RK]............","............RP.[ST]...............",".............[DEN].[ST]...............",".............[RK].[ST]..............."),"main motifs sig from t0")

# all serine v threonine 

print_plot((WT_MTA1_INCR_ST_line+WT_MTA2_INCR_ST_line)/
    (WT_MTA1_DECR_ST_line+WT_MTA2_DECR_ST_line)/
        (WT_MTA1_INCR_STP_line+WT_MTA2_INCR_STP_line)/
        (WT_MTA1_DECR_STP_line+WT_MTA2_DECR_STP_line)/
        (WT_phos_all_ST_line+WT_phos_t0_sig_ST_line)/
        (WT_phos_all_STP_line+WT_phos_t0_sig_STP_line),
    "WT_MTA1_MTA2_ST_line.pdf",17,26)

# all main motifs

print_plot((WT_MTA1_INCR_motif_line+WT_MTA2_INCR_motif_line)/
    (WT_MTA1_DECR_motif_line+WT_MTA2_DECR_motif_line)/
        (WT_phos_all_motif_line+WT_phos_t0_sig_motif_line),
    "WT_MTA1_MTA2_motif_line.pdf",20,15)


```


# MTA WT v spo13 - PHOS

```{r}

# WT and spo13 combind dataset, MTA1 and MTA2 sig sites 
# all sig sites quantitative trend

WT_spo13_phos_MTA1_boxplot<-WT.spo13.phos.fin.df %>%
    filter(if_any(c(WT_metaI_cols[1:2],spo13_metaI_cols[1:2]),function(x) x=="INCR"|x=="DECR")) %>%
    filter(!if_any(c(WT.fc.90_v_75,WT.fc.105_v_75,spo13.fc.90_v_75,spo13.fc.105_v_75),function(x) is.na(x))) %>%
    select(Gene.name,WT.fc.90_v_75,WT.fc.105_v_75,spo13.fc.90_v_75,spo13.fc.105_v_75) %>%
    pivot_longer(-Gene.name,values_to="fold_change",names_to="sample") %>%
     mutate(fc_trimmed=ifelse(fold_change>2.2,2.2,
                             ifelse(fold_change<0.2,0.2,fold_change))) %>%
    mutate(sample=factor(sample,levels=c("WT.fc.90_v_75","spo13.fc.90_v_75","WT.fc.105_v_75","spo13.fc.105_v_75"))) %>%
    ggplot(aes(x=sample,y=fc_trimmed,group=sample))+
        geom_boxplot(aes(fill=sample),position='dodge')+
    labs(title="Metaphase I to Anaphase I",x="",y="trimmed fold change")+
    geom_hline(yintercept=1.5,linetype="dashed",size=0.5)+
    geom_hline(yintercept=(1/1.5),linetype="dashed",size=0.5)+
    theme_classic()+
    theme(axis.text.x = element_text(angle=90,vjust=0.6))+
    scale_y_continuous(expand=expansion(mult=c(0.1,0.1)),labels=label_number(accuracy=0.1))

WT_spo13_phos_MTA2_boxplot<-WT.spo13.phos.fin.df %>%
    filter(if_any(c(WT_metaII_cols,spo13_metaII_cols),function(x) x=="INCR"|x=="DECR")) %>%
    filter(!if_any(c(WT.fc.135_v_120,WT.fc.150_v_120,WT.fc.165_v_120,spo13.fc.135_v_120,spo13.fc.150_v_120,spo13.fc.165_v_120),function(x) is.na(x))) %>%
    select(Gene.name,WT.fc.135_v_120,WT.fc.150_v_120,WT.fc.165_v_120,spo13.fc.135_v_120,spo13.fc.150_v_120,spo13.fc.165_v_120) %>%
    pivot_longer(-Gene.name,values_to="fold_change",names_to="sample") %>%
     mutate(fc_trimmed=ifelse(fold_change>2.2,2.2,
                             ifelse(fold_change<0.2,0.2,fold_change))) %>%
    mutate(sample=factor(sample,levels=c("WT.fc.135_v_120","spo13.fc.135_v_120","WT.fc.150_v_120","spo13.fc.150_v_120","WT.fc.165_v_120","spo13.fc.165_v_120"))) %>%
     ggplot(aes(x=sample,y=fc_trimmed,group=sample))+
        geom_boxplot(aes(fill=sample),position='dodge',na.rm=T)+
    labs(title="Metaphase II to Anaphase II",x="",y="trimmed fold change")+
        geom_hline(yintercept=1.5,linetype="dashed",size=0.5)+
    geom_hline(yintercept=(1/1.5),linetype="dashed",size=0.5)+
    theme_classic()+
    theme(axis.text.x = element_text(angle=90,vjust=0.6))+
    scale_y_continuous(expand=expansion(mult=c(0.1,0.1)),labels=label_number(accuracy=0.1))
#5 x6

print_plot(WT_spo13_phos_MTA1_boxplot,
           "WT_spo13_phos_MTA1_boxplot.pdf",5,6)
print_plot(WT_spo13_phos_MTA2_boxplot,
           "WT_spo13_phos_MTA2_boxplot.pdf",5,6)


# boxplots split by INCR/DECR
# INCR from MTA1
WT_spo13_phos_MTA1_INCR_boxplot<-WT.spo13.phos.fin.df %>%
    filter(if_any(c(WT_metaI_cols[1:2],spo13_metaI_cols[1:2]),function(x) x=="INCR")) %>%
    filter(!if_any(c(WT.fc.90_v_75,WT.fc.105_v_75,spo13.fc.90_v_75,spo13.fc.105_v_75),function(x) is.na(x))) %>%
    select(Gene.name,WT.fc.90_v_75,WT.fc.105_v_75,spo13.fc.90_v_75,spo13.fc.105_v_75) %>%
    pivot_longer(-Gene.name,values_to="fold_change",names_to="sample") %>%
     mutate(fc_trimmed=ifelse(fold_change>2.2,2.2,
                             ifelse(fold_change<0.2,0.2,fold_change))) %>%
    mutate(sample=factor(sample,levels=c("WT.fc.90_v_75","spo13.fc.90_v_75","WT.fc.105_v_75","spo13.fc.105_v_75"))) %>%
    ggplot(aes(x=sample,y=fc_trimmed,group=sample))+
        geom_boxplot(aes(fill=sample),position='dodge')+
    labs(title="Metaphase I to Anaphase I INCR",x="",y="trimmed fold change")+
    geom_hline(yintercept=1.5,linetype="dashed",size=0.5)+
    geom_hline(yintercept=(1/1.5),linetype="dashed",size=0.5)+
    theme_classic()+
    theme(axis.text.x = element_text(angle=90,vjust=0.6))+
    scale_y_continuous(expand=expansion(mult=c(0.1,0.1)),labels=label_number(accuracy=0.1))

# DECR from MTA1
WT_spo13_phos_MTA1_DECR_boxplot<-WT.spo13.phos.fin.df %>%
    filter(if_any(c(WT_metaI_cols[1:2],spo13_metaI_cols[1:2]),function(x) x=="DECR")) %>%
    filter(!if_any(c(WT.fc.90_v_75,WT.fc.105_v_75,spo13.fc.90_v_75,spo13.fc.105_v_75),function(x) is.na(x))) %>%
    select(Gene.name,WT.fc.90_v_75,WT.fc.105_v_75,spo13.fc.90_v_75,spo13.fc.105_v_75) %>%
    pivot_longer(-Gene.name,values_to="fold_change",names_to="sample") %>%
     mutate(fc_trimmed=ifelse(fold_change>2.2,2.2,
                             ifelse(fold_change<0.2,0.2,fold_change))) %>%
    mutate(sample=factor(sample,levels=c("WT.fc.90_v_75","spo13.fc.90_v_75","WT.fc.105_v_75","spo13.fc.105_v_75"))) %>%
    ggplot(aes(x=sample,y=fc_trimmed,group=sample))+
        geom_boxplot(aes(fill=sample),position='dodge')+
    labs(title="Metaphase I to Anaphase I DECR",x="",y="trimmed fold change")+
    geom_hline(yintercept=1.5,linetype="dashed",size=0.5)+
    geom_hline(yintercept=(1/1.5),linetype="dashed",size=0.5)+
    theme_classic()+
    theme(axis.text.x = element_text(angle=90,vjust=0.6))+
    scale_y_continuous(expand=expansion(mult=c(0.1,0.1)),labels=label_number(accuracy=0.1))

# INCR from MTA2
WT_spo13_phos_MTA2_INCR_boxplot<-WT.spo13.phos.fin.df %>%
    filter(if_any(c(WT_metaII_cols,spo13_metaII_cols),function(x) x=="INCR")) %>%
    filter(!if_any(c(WT.fc.135_v_120,WT.fc.150_v_120,WT.fc.165_v_120,spo13.fc.135_v_120,spo13.fc.150_v_120,spo13.fc.165_v_120),function(x) is.na(x))) %>%
    select(Gene.name,WT.fc.135_v_120,WT.fc.150_v_120,WT.fc.165_v_120,spo13.fc.135_v_120,spo13.fc.150_v_120,spo13.fc.165_v_120) %>%
    pivot_longer(-Gene.name,values_to="fold_change",names_to="sample") %>%
     mutate(fc_trimmed=ifelse(fold_change>2.2,2.2,
                             ifelse(fold_change<0.2,0.2,fold_change))) %>%
    mutate(sample=factor(sample,levels=c("WT.fc.135_v_120","spo13.fc.135_v_120","WT.fc.150_v_120","spo13.fc.150_v_120","WT.fc.165_v_120","spo13.fc.165_v_120"))) %>%
     ggplot(aes(x=sample,y=fc_trimmed,group=sample))+
        geom_boxplot(aes(fill=sample),position='dodge',na.rm=T)+
    labs(title="Metaphase II to Anaphase II INCR",x="",y="trimmed fold change")+
        geom_hline(yintercept=1.5,linetype="dashed",size=0.5)+
    geom_hline(yintercept=(1/1.5),linetype="dashed",size=0.5)+
    theme_classic()+
    theme(axis.text.x = element_text(angle=90,vjust=0.6))+
    scale_y_continuous(expand=expansion(mult=c(0.1,0.1)),labels=label_number(accuracy=0.1))

# DECR from MTA2
WT_spo13_phos_MTA2_DECR_boxplot<-WT.spo13.phos.fin.df %>%
    filter(if_any(c(WT_metaII_cols,spo13_metaII_cols),function(x) x=="DECR")) %>%
    filter(!if_any(c(WT.fc.135_v_120,WT.fc.150_v_120,WT.fc.165_v_120,spo13.fc.135_v_120,spo13.fc.150_v_120,spo13.fc.165_v_120),function(x) is.na(x))) %>%
    select(Gene.name,WT.fc.135_v_120,WT.fc.150_v_120,WT.fc.165_v_120,spo13.fc.135_v_120,spo13.fc.150_v_120,spo13.fc.165_v_120) %>%
    pivot_longer(-Gene.name,values_to="fold_change",names_to="sample") %>%
     mutate(fc_trimmed=ifelse(fold_change>2.2,2.2,
                             ifelse(fold_change<0.2,0.2,fold_change))) %>%
    mutate(sample=factor(sample,levels=c("WT.fc.135_v_120","spo13.fc.135_v_120","WT.fc.150_v_120","spo13.fc.150_v_120","WT.fc.165_v_120","spo13.fc.165_v_120"))) %>%
     ggplot(aes(x=sample,y=fc_trimmed,group=sample))+
        geom_boxplot(aes(fill=sample),position='dodge',na.rm=T)+
    labs(title="Metaphase II to Anaphase II DECR",x="",y="trimmed fold change")+
        geom_hline(yintercept=1.5,linetype="dashed",size=0.5)+
    geom_hline(yintercept=(1/1.5),linetype="dashed",size=0.5)+
    theme_classic()+
    theme(axis.text.x = element_text(angle=90,vjust=0.6))+
    scale_y_continuous(expand=expansion(mult=c(0.1,0.1)),labels=label_number(accuracy=0.1))

print_plot(WT_spo13_phos_MTA1_INCR_boxplot,
           "WT_spo13_phos_MTA1_INCR_boxplot.pdf",5,6)
print_plot(WT_spo13_phos_MTA1_DECR_boxplot,
           "WT_spo13_phos_MTA1_DECR_boxplot.pdf",5,6)
print_plot(WT_spo13_phos_MTA2_INCR_boxplot,
           "WT_spo13_phos_MTA2_INCR_boxplot.pdf",5,6)
print_plot(WT_spo13_phos_MTA2_DECR_boxplot,
           "WT_spo13_phos_MTA2_DECR_boxplot.pdf",5,6)


# pie charts


# WT DECR and sspo13 DECR from MTA1
WT_spo13_phos_MTA1_DECR_pie<-(WT.spo13.phos.fin.df %>%
    filter(if_any(c(WT_metaI_cols[1:2]),function(x) x=="DECR")) %>%
    mutate(CDC5=grepl(".............[DEN].[ST]...............",Sequence.window)) %>%
    mutate(CDC28_min=grepl("...............[ST]P..............",Sequence.window)) %>%
    mutate(IME2=grepl("............RP.[ST]...............",Sequence.window)) %>%
    mutate(IPL1=grepl(".............[RK].[ST]...............",Sequence.window)) %>%
    # need to tally which sites are in multiple kinase categories and call them 'mixed'
    mutate(DEN.SP=ifelse(CDC5==T&CDC28_min==T,T,F)) %>%
    mutate(IME2_SP=ifelse(CDC28_min==T&IME2==T,T,F)) %>%
    mutate(IPL1_SP=ifelse(CDC28_min==T&IPL1==T,T,F)) %>%
    mutate(kinase=ifelse(DEN.SP==T|IME2_SP==T|IPL1_SP==T,"mixed",
                                              ifelse(CDC28_min==T,"CDC28_min",
                                                            ifelse(CDC5==T,"CDC5",
                                                                   ifelse(IME2==T,"IME2",
                                                                    ifelse(IPL1==T,"IPL1","other")))))) %>% 
    mutate(kinase=factor(kinase,levels=c("mixed","CDC5","CDC28_min","IME2","IPL1","other"))) %>%
    group_by(kinase) %>% summarize(n=n()) %>%
ggplot(aes(x="",y=n,fill=kinase,label=n))+
    geom_bar(stat="identity")+
    geom_label_repel()+
    coord_polar("y", start=0)+
    theme_void()+
    scale_fill_manual(values=c("#7CAE00", "#00BFC4", "#C77CFF","blue","darkorange","darkgrey"),drop=FALSE)+
    ggtitle(paste0("WT DECR MTA1 sites, n=",WT.spo13.phos.fin.df %>%
    filter(if_any(c(WT_metaI_cols[1:2]),function(x) x=="DECR")) %>% nrow()))+
    theme(legend.title=element_blank()))+
(WT.spo13.phos.fin.df %>%
    filter(if_any(c(spo13_metaI_cols[1:2]),function(x) x=="DECR")) %>%
    mutate(CDC5=grepl(".............[DEN].[ST]...............",Sequence.window)) %>%
    mutate(CDC28_min=grepl("...............[ST]P..............",Sequence.window)) %>%
    mutate(IME2=grepl("............RP.[ST]...............",Sequence.window)) %>%
    mutate(IPL1=grepl(".............[RK].[ST]...............",Sequence.window)) %>%
    # need to tally which sites are in multiple kinase categories and call them 'mixed'
    mutate(DEN.SP=ifelse(CDC5==T&CDC28_min==T,T,F)) %>%
    mutate(IME2_SP=ifelse(CDC28_min==T&IME2==T,T,F)) %>%
    mutate(IPL1_SP=ifelse(CDC28_min==T&IPL1==T,T,F)) %>%
    mutate(kinase=ifelse(DEN.SP==T|IME2_SP==T|IPL1_SP==T,"mixed",
                                              ifelse(CDC28_min==T,"CDC28_min",
                                                            ifelse(CDC5==T,"CDC5",
                                                                   ifelse(IME2==T,"IME2",
                                                                    ifelse(IPL1==T,"IPL1","other")))))) %>% 
    mutate(kinase=factor(kinase,levels=c("mixed","CDC5","CDC28_min","IME2","IPL1","other"))) %>%
    group_by(kinase) %>% summarize(n=n()) %>%
ggplot(aes(x="",y=n,fill=kinase,label=n))+
    geom_bar(stat="identity")+
    geom_label_repel()+
    coord_polar("y", start=0)+
    theme_void()+
    scale_fill_manual(values=c("#7CAE00", "#00BFC4", "#C77CFF","blue","darkorange","darkgrey"),drop=FALSE)+
    ggtitle(paste0("spo13 DECR MTA1 sites, n=",WT.spo13.phos.fin.df %>%
    filter(if_any(c(spo13_metaI_cols[1:2]),function(x) x=="DECR")) %>% nrow()))+
    theme(legend.title=element_blank()))
# 5x10



# WT and spo13 INCR from MTA1 
WT_spo13_phos_MTA1_INCR_pie<-(WT.spo13.phos.fin.df %>%
    filter(if_any(c(WT_metaI_cols[1:2]),function(x) x=="INCR")) %>%
    mutate(CDC5=grepl(".............[DEN].[ST]...............",Sequence.window)) %>%
    mutate(CDC28_min=grepl("...............[ST]P..............",Sequence.window)) %>%
    mutate(IME2=grepl("............RP.[ST]...............",Sequence.window)) %>%
    mutate(IPL1=grepl(".............[RK].[ST]...............",Sequence.window)) %>%
    # need to tally which sites are in multiple kinase categories and call them 'mixed'
    mutate(DEN.SP=ifelse(CDC5==T&CDC28_min==T,T,F)) %>%
    mutate(IME2_SP=ifelse(CDC28_min==T&IME2==T,T,F)) %>%
    mutate(IPL1_SP=ifelse(CDC28_min==T&IPL1==T,T,F)) %>%
    mutate(kinase=ifelse(DEN.SP==T|IME2_SP==T|IPL1_SP==T,"mixed",
                                              ifelse(CDC28_min==T,"CDC28_min",
                                                            ifelse(CDC5==T,"CDC5",
                                                                   ifelse(IME2==T,"IME2",
                                                                    ifelse(IPL1==T,"IPL1","other")))))) %>% 
    mutate(kinase=factor(kinase,levels=c("mixed","CDC5","CDC28_min","IME2","IPL1","other"))) %>%
    group_by(kinase) %>% summarize(n=n()) %>%
ggplot(aes(x="",y=n,fill=kinase,label=n))+
    geom_bar(stat="identity")+
    geom_label_repel()+
    coord_polar("y", start=0)+
    theme_void()+
    scale_fill_manual(values=c("#7CAE00", "#00BFC4", "#C77CFF","blue","darkorange","darkgrey"),drop=FALSE)+
    ggtitle(paste0("WT INCR MTA1 sites, n=",WT.spo13.phos.fin.df %>%
    filter(if_any(c(WT_metaI_cols[1:2]),function(x) x=="INCR")) %>% nrow()))+
    theme(legend.title=element_blank()))+
(WT.spo13.phos.fin.df %>%
    filter(if_any(c(spo13_metaI_cols[1:2]),function(x) x=="INCR")) %>%
    mutate(CDC5=grepl(".............[DEN].[ST]...............",Sequence.window)) %>%
    mutate(CDC28_min=grepl("...............[ST]P..............",Sequence.window)) %>%
    mutate(IME2=grepl("............RP.[ST]...............",Sequence.window)) %>%
    mutate(IPL1=grepl(".............[RK].[ST]...............",Sequence.window)) %>%
    # need to tally which sites are in multiple kinase categories and call them 'mixed'
    mutate(DEN.SP=ifelse(CDC5==T&CDC28_min==T,T,F)) %>%
    mutate(IME2_SP=ifelse(CDC28_min==T&IME2==T,T,F)) %>%
    mutate(IPL1_SP=ifelse(CDC28_min==T&IPL1==T,T,F)) %>%
    mutate(kinase=ifelse(DEN.SP==T|IME2_SP==T|IPL1_SP==T,"mixed",
                                              ifelse(CDC28_min==T,"CDC28_min",
                                                            ifelse(CDC5==T,"CDC5",
                                                                   ifelse(IME2==T,"IME2",
                                                                    ifelse(IPL1==T,"IPL1","other")))))) %>% 
    mutate(kinase=factor(kinase,levels=c("mixed","CDC5","CDC28_min","IME2","IPL1","other"))) %>%
    group_by(kinase) %>% summarize(n=n()) %>%
ggplot(aes(x="",y=n,fill=kinase,label=n))+
    geom_bar(stat="identity")+
    geom_label_repel()+
    coord_polar("y", start=0)+
    theme_void()+
    scale_fill_manual(values=c("#7CAE00", "#00BFC4", "#C77CFF","blue","darkorange","darkgrey"),drop=FALSE)+
    ggtitle(paste0("spo13 INCR MTA1 sites, n=",WT.spo13.phos.fin.df %>%
    filter(if_any(c(spo13_metaI_cols[1:2]),function(x) x=="INCR")) %>% nrow()))+
    theme(legend.title=element_blank()))


# MTA2
# WT.spo13.phos.fin.df %>%
#     filter(if_any(c(WT_metaII_cols),function(x) x=="DECR")) %>% nrow()
# WT.spo13.phos.fin.df %>%
#     filter(if_any(c(spo13_metaII_cols),function(x) x=="DECR")) %>% nrow()


# WT and spo13 INCR from MTA2
WT_spo13_phos_MTA2_INCR_pie<-(WT.spo13.phos.fin.df %>%
    filter(if_any(c(WT_metaII_cols),function(x) x=="INCR")) %>%
    mutate(CDC5=grepl(".............[DEN].[ST]...............",Sequence.window)) %>%
    mutate(CDC28_min=grepl("...............[ST]P..............",Sequence.window)) %>%
    mutate(IME2=grepl("............RP.[ST]...............",Sequence.window)) %>%
    mutate(IPL1=grepl(".............[RK].[ST]...............",Sequence.window)) %>%
    # need to tally which sites are in multiple kinase categories and call them 'mixed'
    mutate(DEN.SP=ifelse(CDC5==T&CDC28_min==T,T,F)) %>%
    mutate(IME2_SP=ifelse(CDC28_min==T&IME2==T,T,F)) %>%
    mutate(IPL1_SP=ifelse(CDC28_min==T&IPL1==T,T,F)) %>%
    mutate(kinase=ifelse(DEN.SP==T|IME2_SP==T|IPL1_SP==T,"mixed",
                                              ifelse(CDC28_min==T,"CDC28_min",
                                                            ifelse(CDC5==T,"CDC5",
                                                                   ifelse(IME2==T,"IME2",
                                                                    ifelse(IPL1==T,"IPL1","other")))))) %>% 
    mutate(kinase=factor(kinase,levels=c("mixed","CDC5","CDC28_min","IME2","IPL1","other"))) %>%
    group_by(kinase) %>% summarize(n=n()) %>%
ggplot(aes(x="",y=n,fill=kinase,label=n))+
    geom_bar(stat="identity")+
    geom_label_repel()+
    coord_polar("y", start=0)+
    theme_void()+
    scale_fill_manual(values=c("#7CAE00", "#00BFC4", "#C77CFF","blue","darkorange","darkgrey"),drop=FALSE)+
    ggtitle(paste0("WT INCR MTA2 sites, n=",WT.spo13.phos.fin.df %>%
    filter(if_any(c(WT_metaII_cols),function(x) x=="INCR")) %>% nrow()))+
    theme(legend.title=element_blank()))+
(WT.spo13.phos.fin.df %>%
    filter(if_any(c(spo13_metaII_cols),function(x) x=="INCR")) %>%
    mutate(CDC5=grepl(".............[DEN].[ST]...............",Sequence.window)) %>%
    mutate(CDC28_min=grepl("...............[ST]P..............",Sequence.window)) %>%
    mutate(IME2=grepl("............RP.[ST]...............",Sequence.window)) %>%
    mutate(IPL1=grepl(".............[RK].[ST]...............",Sequence.window)) %>%
    # need to tally which sites are in multiple kinase categories and call them 'mixed'
    mutate(DEN.SP=ifelse(CDC5==T&CDC28_min==T,T,F)) %>%
    mutate(IME2_SP=ifelse(CDC28_min==T&IME2==T,T,F)) %>%
    mutate(IPL1_SP=ifelse(CDC28_min==T&IPL1==T,T,F)) %>%
    mutate(kinase=ifelse(DEN.SP==T|IME2_SP==T|IPL1_SP==T,"mixed",
                                              ifelse(CDC28_min==T,"CDC28_min",
                                                            ifelse(CDC5==T,"CDC5",
                                                                   ifelse(IME2==T,"IME2",
                                                                    ifelse(IPL1==T,"IPL1","other")))))) %>% 
    mutate(kinase=factor(kinase,levels=c("mixed","CDC5","CDC28_min","IME2","IPL1","other"))) %>%
    group_by(kinase) %>% summarize(n=n()) %>%
ggplot(aes(x="",y=n,fill=kinase,label=n))+
    geom_bar(stat="identity")+
    geom_label_repel()+
    coord_polar("y", start=0)+
    theme_void()+
    scale_fill_manual(values=c("#7CAE00", "#00BFC4", "#C77CFF","blue","darkorange","darkgrey"),drop=FALSE)+
    ggtitle(paste0("spo13 INCR MTA2 sites, n=",WT.spo13.phos.fin.df %>%
    filter(if_any(c(spo13_metaII_cols),function(x) x=="INCR")) %>% nrow()))+
    theme(legend.title=element_blank()))


# WT and sspo13 DECR from MTA2 
WT_spo13_phos_MTA2_DECR_pie<-(WT.spo13.phos.fin.df %>%
    filter(if_any(c(WT_metaII_cols),function(x) x=="DECR")) %>%
    mutate(CDC5=grepl(".............[DEN].[ST]...............",Sequence.window)) %>%
    mutate(CDC28_min=grepl("...............[ST]P..............",Sequence.window)) %>%
    mutate(IME2=grepl("............RP.[ST]...............",Sequence.window)) %>%
    mutate(IPL1=grepl(".............[RK].[ST]...............",Sequence.window)) %>%
    # need to tally which sites are in multiple kinase categories and call them 'mixed'
    mutate(DEN.SP=ifelse(CDC5==T&CDC28_min==T,T,F)) %>%
    mutate(IME2_SP=ifelse(CDC28_min==T&IME2==T,T,F)) %>%
    mutate(IPL1_SP=ifelse(CDC28_min==T&IPL1==T,T,F)) %>%
    mutate(kinase=ifelse(DEN.SP==T|IME2_SP==T|IPL1_SP==T,"mixed",
                                              ifelse(CDC28_min==T,"CDC28_min",
                                                            ifelse(CDC5==T,"CDC5",
                                                                   ifelse(IME2==T,"IME2",
                                                                    ifelse(IPL1==T,"IPL1","other")))))) %>% 
    mutate(kinase=factor(kinase,levels=c("mixed","CDC5","CDC28_min","IME2","IPL1","other"))) %>%
    group_by(kinase) %>% summarize(n=n()) %>%
ggplot(aes(x="",y=n,fill=kinase,label=n))+
    geom_bar(stat="identity")+
    geom_label_repel()+
    coord_polar("y", start=0)+
    theme_void()+
    scale_fill_manual(values=c("#7CAE00", "#00BFC4", "#C77CFF","blue","darkorange","darkgrey"),drop=FALSE)+
    ggtitle(paste0("WT DECR MTA2 sites, n=",WT.spo13.phos.fin.df %>%
    filter(if_any(c(WT_metaII_cols),function(x) x=="DECR")) %>% nrow()))+
    theme(legend.title=element_blank()))+
(WT.spo13.phos.fin.df %>%
    filter(if_any(c(spo13_metaII_cols),function(x) x=="DECR")) %>%
    mutate(CDC5=grepl(".............[DEN].[ST]...............",Sequence.window)) %>%
    mutate(CDC28_min=grepl("...............[ST]P..............",Sequence.window)) %>%
    mutate(IME2=grepl("............RP.[ST]...............",Sequence.window)) %>%
    mutate(IPL1=grepl(".............[RK].[ST]...............",Sequence.window)) %>%
    # need to tally which sites are in multiple kinase categories and call them 'mixed'
    mutate(DEN.SP=ifelse(CDC5==T&CDC28_min==T,T,F)) %>%
    mutate(IME2_SP=ifelse(CDC28_min==T&IME2==T,T,F)) %>%
    mutate(IPL1_SP=ifelse(CDC28_min==T&IPL1==T,T,F)) %>%
    mutate(kinase=ifelse(DEN.SP==T|IME2_SP==T|IPL1_SP==T,"mixed",
                                              ifelse(CDC28_min==T,"CDC28_min",
                                                            ifelse(CDC5==T,"CDC5",
                                                                   ifelse(IME2==T,"IME2",
                                                                    ifelse(IPL1==T,"IPL1","other")))))) %>% 
    mutate(kinase=factor(kinase,levels=c("mixed","CDC5","CDC28_min","IME2","IPL1","other"))) %>%
    group_by(kinase) %>% summarize(n=n()) %>%
ggplot(aes(x="",y=n,fill=kinase,label=n))+
    geom_bar(stat="identity")+
    geom_label_repel()+
    coord_polar("y", start=0)+
    theme_void()+
    scale_fill_manual(values=c("#7CAE00", "#00BFC4", "#C77CFF","blue","darkorange","darkgrey"),drop=FALSE)+
    ggtitle(paste0("spo13 DECR MTA2 sites, n=",WT.spo13.phos.fin.df %>%
    filter(if_any(c(spo13_metaII_cols),function(x) x=="DECR")) %>% nrow()))+
    theme(legend.title=element_blank()))


print_plot(WT_spo13_phos_MTA1_INCR_pie/WT_spo13_phos_MTA1_DECR_pie,
           "WT_spo13_phos_MTA1_pie.pdf",10,10)
print_plot(WT_spo13_phos_MTA2_INCR_pie/WT_spo13_phos_MTA2_DECR_pie,
           "WT_spo13_phos_MTA2_pie.pdf",10,10)


# MOTIF LINEPLOTSS WT V SPO13
WT_spo13_MTA1_motif_line<-MTA1_motif_lineplot(WT.spo13.phos.fin.df %>% filter(if_any(c(WT_metaI_cols[1:2]),function(x) x=="DECR")),c("CDC28_min","CDC28_strict","IME2","CDC5","IPL1/PKA"),c("...............[ST]P..............","...............[ST]P.[RK]............","............RP.[ST]...............",".............[DEN].[ST]...............",".............[RK].[ST]..............."),"WT DECR from t75") +
    MTA1_motif_lineplot(WT.spo13.phos.fin.df %>% filter(if_any(c(spo13_metaI_cols[1:2]),function(x) x=="DECR")),c("CDC28_min","CDC28_strict","IME2","CDC5","IPL1/PKA"),c("...............[ST]P..............","...............[ST]P.[RK]............","............RP.[ST]...............",".............[DEN].[ST]...............",".............[RK].[ST]..............."),"spo13 DECR from t75")+ 
    MTA1_motif_lineplot(WT.spo13.phos.fin.df %>% filter(if_any(c(WT_metaI_cols[1:2]),function(x) x=="INCR")),c("CDC28_min","CDC28_strict","IME2","CDC5","IPL1/PKA"),c("...............[ST]P..............","...............[ST]P.[RK]............","............RP.[ST]...............",".............[DEN].[ST]...............",".............[RK].[ST]..............."),"WT INCR from t75") +
    MTA1_motif_lineplot(WT.spo13.phos.fin.df %>% filter(if_any(c(spo13_metaI_cols[1:2]),function(x) x=="INCR")),c("CDC28_min","CDC28_strict","IME2","CDC5","IPL1/PKA"),c("...............[ST]P..............","...............[ST]P.[RK]............","............RP.[ST]...............",".............[DEN].[ST]...............",".............[RK].[ST]..............."),"spo13 INCR from t75")


# MTA2
WT_spo13_MTA2_motif_line<-MTA2_motif_lineplot(WT.spo13.phos.fin.df %>% filter(if_any(c(WT_metaII_cols),function(x) x=="DECR")),c("CDC28_min","CDC28_strict","IME2","CDC5","IPL1/PKA"),c("...............[ST]P..............","...............[ST]P.[RK]............","............RP.[ST]...............",".............[DEN].[ST]...............",".............[RK].[ST]..............."),"WT DECR from t120") +
    MTA2_motif_lineplot(WT.spo13.phos.fin.df %>% filter(if_any(c(spo13_metaII_cols),function(x) x=="DECR")),c("CDC28_min","CDC28_strict","IME2","CDC5","IPL1/PKA"),c("...............[ST]P..............","...............[ST]P.[RK]............","............RP.[ST]...............",".............[DEN].[ST]...............",".............[RK].[ST]..............."),"spo13 DECR from t120")+ 
    MTA2_motif_lineplot(WT.spo13.phos.fin.df %>% filter(if_any(c(WT_metaII_cols),function(x) x=="INCR")),c("CDC28_min","CDC28_strict","IME2","CDC5","IPL1/PKA"),c("...............[ST]P..............","...............[ST]P.[RK]............","............RP.[ST]...............",".............[DEN].[ST]...............",".............[RK].[ST]..............."),"WT INCR from t120") +
    MTA2_motif_lineplot(WT.spo13.phos.fin.df %>% filter(if_any(c(spo13_metaII_cols),function(x) x=="INCR")),c("CDC28_min","CDC28_strict","IME2","CDC5","IPL1/PKA"),c("...............[ST]P..............","...............[ST]P.[RK]............","............RP.[ST]...............",".............[DEN].[ST]...............",".............[RK].[ST]..............."),"spo13 INCR from t120")

print_plot(WT_spo13_MTA1_motif_line,
           "WT_spo13_MTA1_motif_line.pdf",20,10)

print_plot(WT_spo13_MTA2_motif_line,
           "WT_spo13_MTA2_motif_line.pdf",20,10)

```


# sigdiff WT v spo13 sel protein
```{r}

# Have called n=50 proteins as sigdiff and clustered them in WT.spo13.protg.sigdiff.heatmap

get.protein.lineplot.two.strains_both_scaled_flex_row<-function(df,proteinlist,title,number){
   df %>%
    filter(Gene.name %in% proteinlist) %>%
    select(Gene.name,all_of(c(WT.spo13.scaled.mean.cols,WT.spo13.sig.cols))) %>%
    mutate(time_sig=apply(.[,WT.spo13.sig.cols],1,function(x) which(x!="NC")[1])) %>%
    mutate(time_sig=alltimes[time_sig]) %>% # convert time sig to actual min
    select(Gene.name,all_of(WT.spo13.scaled.mean.cols),time_sig) %>%
  pivot_longer(-c(Gene.name,time_sig),names_prefix="scaled.both.",names_to="time",values_to="intensity") %>%
    mutate(strain=ifelse(grepl("WT",time),"WT","spo13")) %>%
    mutate(time=gsub("WT.mean.|spo13.mean.","",time)) %>%
     #  mutate(sig_annot=ifelse(time==time_sig,"*",NA)) %>% # 4 stars are present since 4 bio reps are included and I can't think of how to only do 1
    ggplot(aes(x=fct_inorder(time),y=intensity,group=strain))+
    stat_summary(aes(y=intensity,group=strain,color=strain),fun="mean",geom="line",linewidth=1)+
    stat_summary(aes(y=intensity,group=strain,fill=strain),geom="ribbon",fun.min=min,fun.max=max,alpha=0.2)+
    #geom_text(aes(label=sig_annot, vjust=-2, hjust=0))+
    facet_wrap(~Gene.name,scales="free",nrow=number)+
        scale_y_continuous(labels = label_number(accuracy=0.1))+
    theme_classic()+ 
    theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),axis.text.y=element_text(size=rel(1.5)),plot.title=element_text(size=rel(1.5)),legend.title=element_text(size=rel(1.5)),legend.text=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),strip.text=element_text(size=rel(1.5)))+
  labs(x="time",y="scaled intensity",title=title)
}


# only 4 proteins fit in main figure to highlight and chose these because difference 
# is very clear aand they belong to different clusters
sel_sigdiff_prot_small<-get.protein.lineplot.two.strains_both_scaled_flex_row(WT.spo13.protg.fin.df,c("SGO1","SPO12","YGL036W","SPS22"),"sigdiff proteins",1)

print_plot(sel_sigdiff_prot_small,
           "sel_sigdiff_prot_small_line.pdf",17,3.5)

# cell cycle marker proteins WT v spo13

sel_cycle_marker_prot<-get.protein.lineplot.two.strains_both_scaled_flex_row(WT.spo13.protg.fin.df,c("APC1","CDC5","IME2","MAM1","NDC80","NDT80","REC8","RIM4","SCC2","ZIP1"),"cell cycle markers",2)

print_plot(sel_cycle_marker_prot,
           "sel_cycle_marker_prot_line.pdf",17,5)

clb3_co_clustered<-c("CLB3","GIP1","SPS1","YSP2","YFL012W","SPO20")

ama1_co_clustered<-c("AMA1","YDR042C","CTS2","SPR6","YGL230C","SGA1","YNL155W","YOR214C","SSP2","SPS4")

cdc5_co_clustered<-c("YDL186W","CLB4","YJL043W","YML199W","YBR250W","YEH1","YJR107W","SPO12","TEP1","SPO75","HST3","TIM18","YGL138C","SGO1","YOL047C","MUM3","YKR005C","YAL018C","PES4","HST1","SPO21","YOR024W","TID3","CDC5","STO1","CBC2")


# clb3 co clustered 
clb3_co_prot<-get.protein.lineplot.two.strains_both_scaled_flex_row(WT.spo13.protg.fin.df,clb3_co_clustered,"clb3 co-clustered",2)

ama1_co_prot<-get.protein.lineplot.two.strains_both_scaled_flex_row(WT.spo13.protg.fin.df,ama1_co_clustered,"ama1 co-clustered",2)

cdc5_co_prot<-get.protein.lineplot.two.strains_both_scaled_flex_row(WT.spo13.protg.fin.df,cdc5_co_clustered,"cdc5 co-clustered",4)

print_plot(clb3_co_prot,
           "clb3_co_prot.pdf",7,4.5)

print_plot(ama1_co_prot,
           "ama1_co_prot.pdf",7,4.5)

print_plot(cdc5_co_prot,
           "cdc5_co_prot.pdf",12,8.5)


WT_spo13_HRR25_prot_line<-get.protein.lineplot.two.strains_both_scaled_flex_row(WT.spo13.protg.fin.df,c("HRR25"),"Hrr25 protein",1)

print_plot(WT_spo13_HRR25_prot_line,
           "WT_spo13_HRR25_prot_line.pdf",6,4)



```


# WT v spo13 phos Polo clust 8
```{r}
# 
# Motif of interesting sigdiff cluster WT.spo13 sigdiff clusster 8

WT_spo13_sigdiff_phos_clust_8_logo<-short_motif_logo_cluster(WT.spo13.phos.fin.df.sigdiff,8,"prob")

print_plot(WT_spo13_sigdiff_phos_clust_8_logo,
           "WT_spo13_sigdiff_phos_clust_8_logo.pdf",6,4)

# line plot of all 15 sites in cluster 8

sigdiff_phos_cluster_8_lines<-WT.spo13.phos.fin.df.sigdiff %>%
  filter(ward.cluster==8) %>%
  select(Gene.name,gene.name_site,WT.spo13.scaled.mean.cols) %>%
  pivot_longer(-c(Gene.name,gene.name_site),names_to="time",values_to="intensity",names_prefix="scaled.both.") %>%
mutate(strain=ifelse(grepl("WT",time),"WT","spo13")) %>%
    mutate(time=gsub("WT.mean.|spo13.mean.","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=interaction(strain,gene.name_site)))+
    facet_wrap(~gene.name_site,scales="free_y")+
  geom_line(aes(color=strain))+
  theme_classic()+
    scale_y_continuous(labels=label_number(accuracy=0.1))+
   theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),axis.text.y=element_text(size=rel(1.5)),plot.title=element_text(size=rel(1.5)),legend.title=element_text(size=rel(1.5)),legend.text=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),strip.text=element_text(size=rel(1)))+
  labs(x="time",y="scaled intensity",title="cluster 8")

print_plot(sigdiff_phos_cluster_8_lines,
           "sigdiff_phos_cluster_8_lines.pdf",10,6)

sigdiff_phos_cluster_10_CDK_lines<-WT.spo13.phos.fin.df.sigdiff %>%
  filter(ward.cluster==10) %>%
    filter(grepl("...............[ST]P..............",Sequence.window)) %>%
  select(Gene.name,gene.name_site,WT.spo13.scaled.mean.cols) %>%
  pivot_longer(-c(Gene.name,gene.name_site),names_to="time",values_to="intensity",names_prefix="scaled.both.") %>%
mutate(strain=ifelse(grepl("WT",time),"WT","spo13")) %>%
    mutate(time=gsub("WT.mean.|spo13.mean.","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=interaction(strain,gene.name_site)))+
    facet_wrap(~gene.name_site,scales="free_y")+
  geom_line(aes(color=strain))+
  theme_classic()+
    scale_y_continuous(labels=label_number(accuracy=0.1))+
   theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),axis.text.y=element_text(size=rel(1.5)),plot.title=element_text(size=rel(1.5)),legend.title=element_text(size=rel(1.5)),legend.text=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),strip.text=element_text(size=rel(1)))+
  labs(x="time",y="scaled intensity",title="cluster 10 CDK sites")
  
print_plot(sigdiff_phos_cluster_10_CDK_lines,
           "sigdiff_phos_cluster_10_CDK_lines.pdf",10,5)

sigdiff_phos_cluster_3_HRR25_lines<-WT.spo13.phos.fin.df.sigdiff %>%
  filter(ward.cluster==3) %>%
    filter(grepl(".............[ST].[ST]...............",Sequence.window)) %>%
  select(Gene.name,gene.name_site,WT.spo13.scaled.mean.cols) %>%
  pivot_longer(-c(Gene.name,gene.name_site),names_to="time",values_to="intensity",names_prefix="scaled.both.") %>%
mutate(strain=ifelse(grepl("WT",time),"WT","spo13")) %>%
    mutate(time=gsub("WT.mean.|spo13.mean.","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=interaction(strain,gene.name_site)))+
    facet_wrap(~gene.name_site,scales="free_y")+
  geom_line(aes(color=strain))+
  theme_classic()+
    scale_y_continuous(labels=label_number(accuracy=0.1))+
   theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),axis.text.y=element_text(size=rel(1.5)),plot.title=element_text(size=rel(1.5)),legend.title=element_text(size=rel(1.5)),legend.text=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),strip.text=element_text(size=rel(1)))+
  labs(x="time",y="scaled intensity",title="cluster 3 HRR25 sites")

print_plot(sigdiff_phos_cluster_3_HRR25_lines,
           "sigdiff_phos_cluster_3_HRR25_lines.pdf",10,5.5)


# selected individual sites
sigdiff_phos_sel_site_lines<-WT.spo13.phos.fin.df.sigdiff %>%
  filter(gene.name_site %in% c("CDC3-S-77","ECM11-S-169","SPC110-S-36")) %>%
  select(Gene.name,gene.name_site,WT.spo13.scaled.mean.cols) %>%
  pivot_longer(-c(Gene.name,gene.name_site),names_to="time",values_to="intensity",names_prefix="scaled.both.") %>%
mutate(strain=ifelse(grepl("WT",time),"WT","spo13")) %>%
    mutate(time=gsub("WT.mean.|spo13.mean.","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=interaction(strain,gene.name_site)))+
    facet_wrap(~gene.name_site,scales="free_y")+
  geom_line(aes(color=strain))+
  theme_classic()+
    scale_y_continuous(labels=label_number(accuracy=0.1))+
   theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),axis.text.y=element_text(size=rel(1.5)),plot.title=element_text(size=rel(1.5)),legend.title=element_text(size=rel(1.5)),legend.text=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),strip.text=element_text(size=rel(1)))+
  labs(x="time",y="scaled intensity",title="")

print_plot(sigdiff_phos_sel_site_lines,
           "sigdiff_phos_sel_site_lines.pdf",12,3)

# sequences of these selected sites
# library(clipr)
# WT.spo13.phos.fin.df.sigdiff %>%
#   filter(gene.name_site %in% c("CDC3-S-77","ECM11-S-169","SPC110-S-36")) %>%
#     mutate(Sequence.window_5=substr(Sequence.window,11,21)) %>%
#     select(gene.name_site,Sequence.window_5) %>% write_clip()

```


# Export phos sigdiff SEQ tables

```{r,echo=F,message=F,warning=F}


# Export WT and spo13 sigdiff phospho site IDs and sequences ####
#WT.spo13.phos.sigdiff.heatmap # 15 clusters
#
#new sheet for each cluster

WT_spo13_phos_sigdiff_df_list<-list()
for(i in 1:15){
    WT_spo13_phos_sigdiff_df_list[[i]]=WT.spo13.phos.fin.df.sigdiff %>%
    mutate(Sequence.window_5=substr(Sequence.window,11,21)) %>%
    filter(ward.cluster==i) %>%
    select(gene.name_site,Gene.name,Sequence.window_5)
}

# name each df in the list for sheet titles
names(WT_spo13_phos_sigdiff_df_list)<-c(paste0("cluster_",1:15))

# export as multi-sheet excel
write.xlsx(WT_spo13_phos_sigdiff_df_list, file = "WT_spo13_phos_sigdiff_clust_ids_seqs.xlsx")

# Seq Lists used to make IceLogos
# single sheet with different columns for each sigdiff category

WT.spo13.phos.fin.df %>%
   mutate(Sequence.window_5=substr(Sequence.window,11,21)) %>%
    group_by(sigdiff.sig.cat) %>%
    mutate(id=row_number()) %>%
    pivot_wider(names_from=sigdiff.sig.cat,values_from=Sequence.window_5,id_cols=id) %>%
    select(-id) %>%
    write.xlsx(.,file="WT_spo13_phos_sigdiff_cat_seqs.xlsx")


```

# sigdiff fisher test barplots
```{r,echo=F,message=F,warning=F}

# fisher test function 
fisher<-function(df1,df2,motif){
    total.df1 <- df1 %>% nrow() 
    total.df2 <- df2 %>% nrow()
    match.df1 <- df1 %>% filter(grepl(motif,Sequence.window)) %>% nrow()
    match.df2 <- df2  %>% filter(grepl(motif,Sequence.window)) %>% nrow()
    corr.mat<-matrix(c(match.df1,(total.df1-match.df1),match.df2,(total.df2-match.df2)),
                     nrow=2,
                     dimnames= list(c("match","no match"),
                                    c("df1","df2")))
    fisher.result<-fisher.test(corr.mat)
    
    corr.mat<-corr.mat %>% as.data.frame() %>%
    rbind(colSums(.)) %>%
    cbind(rowSums(.)) %>%
    rbind(perc_match=(.[1,]/.[3,])*100)
    
    rownames(corr.mat)[3]<-"col_totals"
    colnames(corr.mat)<-c("df1","df2","row_totals")
    
    corr.mat<-corr.mat %>%
        cbind("pval"=fisher.result$p.value)

    
    return(corr.mat)
}

# function to make 2-bar barplot with signficance bar
fisher_barplot_TC<-function(df1,df2,motif,subset1,subset2,title){
    out_df<-fisher(df1,df2,motif) %>%
    tail(1) %>%
     mutate(stars=ifelse(pval<0.001,"***",
                      ifelse(pval<0.01,"**",
                             ifelse(pval<0.05,"*",NA)))) %>%
    select(df1,df2,pval,stars) %>%
    pivot_longer(-c(pval,stars),values_to="perc",names_to="sample") %>%
    mutate(sample=ifelse(grepl("df1",sample),subset1,subset2)) %>%
        mutate(sample=factor(sample,levels=c(subset1,subset2))) # have to make sure levels are consistent
        
    out_df %>%
    ggplot(aes(x=sample,y=perc,fill=sample))+
    geom_bar(aes(fill=sample),stat="identity")+
   geom_signif(comparisons = list(c(subset1,subset2)), 
              map_signif_level=TRUE,
              annotations = out_df$stars[1],
              size=0.25)+
    scale_y_continuous(expand=expansion(mult=c(0,0.2)))+
    theme_classic()+
    theme(axis.text.x = element_text(angle=90,vjust=0.6),legend.position="none",axis.text=element_text(size=rel(1)),plot.title = element_text(size=rel(0.5)))+
    labs(y="% matches",x="",title=title,axis.line=element_line(linewidth=0.1))
}


## Only 6 selected fisher barplot pairs:

# WT_INCR = spo13_DECR
# WT_DECR = spo13_INCR
# 
# spo13 DECR v spo13 INCR
# DEN, N
# 
# spo13 DECR v NC
# DEN, N
# 
# spo13 INCR v NC
# -3 R, -2 Q


A1<-fisher_barplot_TC(WT.spo13.phos.fin.df %>% filter(sigdiff.sig.cat=="sigdiff_spo13_DECR"),WT.spo13.phos.fin.df %>% filter(sigdiff.sig.cat=="sigdiff_spo13_INCR"),".............[DEN].[ST]...............","DECR","INCR","DECR v INCR \n [DEN].[ST]")

A2<-fisher_barplot_TC(WT.spo13.phos.fin.df %>% filter(sigdiff.sig.cat=="sigdiff_spo13_DECR"),WT.spo13.phos.fin.df %>% filter(sigdiff.sig.cat=="sigdiff_spo13_INCR"),".............[N].[ST]...............","DECR","INCR","DECR v INCR \n N.[ST]")

A3<-fisher_barplot_TC(WT.spo13.phos.fin.df %>% filter(sigdiff.sig.cat=="sigdiff_spo13_DECR"),WT.spo13.phos.fin.df %>% filter(sigdiff.sig.cat=="NC"),".............[DEN].[ST]...............","DECR","NC","DECR v NC \n [DEN].[ST]")

A4<-fisher_barplot_TC(WT.spo13.phos.fin.df %>% filter(sigdiff.sig.cat=="sigdiff_spo13_DECR"),WT.spo13.phos.fin.df %>% filter(sigdiff.sig.cat=="NC"),".............[N].[ST]...............","DECR","NC","DECR v NC \n N.[ST]")

A5<-fisher_barplot_TC(WT.spo13.phos.fin.df %>% filter(sigdiff.sig.cat=="sigdiff_spo13_INCR"),WT.spo13.phos.fin.df %>% filter(sigdiff.sig.cat=="NC"),"............[R]..[ST]...............","INCR","NC","INCR v NC \n R..[ST]")

A6<-fisher_barplot_TC(WT.spo13.phos.fin.df %>% filter(sigdiff.sig.cat=="sigdiff_spo13_INCR"),WT.spo13.phos.fin.df %>% filter(sigdiff.sig.cat=="NC"),".............[Q].[ST]...............","INCR","NC","INCR v NC \n Q.[ST]")

print_plot(A1+A2+A3+A4+A5+A6,
           "TC_WT_spo13_phos_sigdiff_fisher_bars.pdf",4,4)


```


## FINAL FIGURE

# Pies -- Percent Polo sites TC sigdiff
```{r}

# Percent of Polo sites that are sigdiff from TC experiment
TC_WT_spo13_polo_sigdiff_pie<-WT.spo13.phos.fin.df %>%
    filter(grepl(".............[DEN].[ST]...............",Sequence.window)) %>%
    group_by(sigdiff.sig.cat) %>% summarize(n=n()) %>%
ggplot(aes(x="",y=n,fill=sigdiff.sig.cat,label=n))+
    geom_bar(stat="identity")+
    geom_label_repel()+
    coord_polar("y", start=0)+
    theme_void()+
    scale_fill_manual(values=c("#7CAE00", "#00BFC4", "#C77CFF","blue"))+
    ggtitle("TC sigdiff Polo sites")+
    theme(legend.title=element_blank())

# 65/(65+16+2+569) # 10%
# (65+16+2)/(65+16+2+569) # 13% are sigdiff


# Metaphase I expt (nicer to have them in same script so plot will scale together)

Exp40E_phos<-read.csv("/Users/lkoch/Documents/Github scripts/metaphase script/Exp40E_phos.csv")

perc_polo_sigdiff_metaI_pie<-Exp40E_phos %>%
    filter(grepl(".............[DEN].[ST]...............",Sequence.window)) %>%
    group_by(spo13_WT_sig) %>% summarize(n=n()) %>%
    mutate(spo13_WT_sig=factor(spo13_WT_sig,levels=c("NC","DECR","INCR"))) %>%
ggplot(aes(x="",y=n,fill=spo13_WT_sig,label=n))+
    geom_bar(stat="identity")+
    geom_label_repel()+
    coord_polar("y", start=0)+
    theme_void()+
    scale_fill_manual(values=c("#7CAE00", "#00BFC4", "#C77CFF","blue"))+
    ggtitle("Meta I sigdiff Polo sites")+
    theme(legend.title=element_blank())

print_plot(TC_WT_spo13_polo_sigdiff_pie+perc_polo_sigdiff_metaI_pie,
           "Perc_Polo_sites_sigdiff_both_expts.pdf",10,10)


```

# GO terms of Polo sites TC + MetaI
```{r}

sigdiff_polo_dfs<-list()

sigdiff_polo_dfs[[1]]<-WT.spo13.phos.fin.df %>%
    filter(grepl(".............[DEN].[ST]...............",Sequence.window)) %>%
    filter(sigdiff.sig.cat=="sigdiff_spo13_DECR"|sigdiff.sig.cat=="sigdiff_spo13_INCR") %>%
    select(Gene.name,gene.name_site,sigdiff.sig.cat)

sigdiff_polo_dfs[[2]]<-gost(WT.spo13.phos.fin.df %>%
    filter(grepl(".............[DEN].[ST]...............",Sequence.window)) %>%
    filter(sigdiff.sig.cat=="sigdiff_spo13_DECR"|sigdiff.sig.cat=="sigdiff_spo13_INCR") %>%
    select(Gene.name) %>% unlist(),organism='scerevisiae',evcodes=T, significant=T)$result %>%
    select(term_name,term_id,p_value,intersection)
        
sigdiff_polo_dfs[[3]]<-Exp40E_phos %>%
    filter(grepl(".............[DEN].[ST]...............",Sequence.window)) %>%
    filter(spo13_WT_sig!="NC") %>%
    select(Gene.name,gene.name_site,spo13_WT_sig)

sigdiff_polo_dfs[[4]]<-gost(Exp40E_phos %>%
    filter(grepl(".............[DEN].[ST]...............",Sequence.window)) %>%
    filter(spo13_WT_sig!="NC") %>%
    select(Gene.name) %>% unlist(),organism='scerevisiae',evcodes=T, significant=T)$result %>%
    select(term_name,term_id,p_value,intersection)

names(sigdiff_polo_dfs)<-c("TC_spo13_WT_sig","TC_spo13_WT_GO_terms","meta_spo13_WT_sig","meta_spo13_WT_GO_terms")

write.xlsx(sigdiff_polo_dfs,"sigdiff_spo13_WT_polo_GO_terms.xlsx")

# GO terms barplot

#TC spo13_WT sig
#cell cycle	GO:0007049
#cytoskeleton organization	GO:0007010
#
#meta I spo13_WT sig
# cell cycle	GO:0007049 #
# chromatin remodeling	GO:0006338
# regulation of RNA metabolic process	GO:0051252
# cytoskeleton organization	GO:0007010 #
# prospore membrane	GO:0005628
# chromosome organization GO:0051276
# spore wall assembly	GO:0042244
# chromatin binding	GO:0003682
# 
# 
polo_spo13_WT_terms<-c("GO:0007049","GO:0007010","GO:0006338","GO:0051252","GO:0005628","GO:0051276","GO:0003682","GO:0042244")

meta_spo13_WT_polo_GO_bar<-rbind(meta_spo13_DECR=gost(Exp40E_phos %>%
    filter(grepl(".............[DEN].[ST]...............",Sequence.window)) %>%
    filter(spo13_WT_sig=="INCR") %>% select(Gene.name) %>% unlist(),organism='scerevisiae',evcodes=T, significant=T)$result %>%
    filter(term_id %in% polo_spo13_WT_terms) %>%
    select(term_name,term_id,p_value),
meta_spo13_INCR=gost(Exp40E_phos %>%
    filter(grepl(".............[DEN].[ST]...............",Sequence.window)) %>%
    filter(spo13_WT_sig=="DECR") %>% select(Gene.name) %>% unlist(),organism='scerevisiae',evcodes=T, significant=T)$result %>%
    filter(term_id %in% polo_spo13_WT_terms) %>%
    select(term_name,term_id,p_value)) %>%
    mutate(direction=ifelse(grepl("INCR",rownames(.)),"INCR","DECR")) %>%
    mutate(p_value=ifelse(grepl("DECR",direction),log10(p_value),-log10(p_value))) %>%
    mutate(sample=gsub("(.+)_(.+)_(.+)","\\1_\\2",rownames(.))) %>%
    ggplot(aes(x=term_name,y=p_value,fill=interaction(direction,sample)))+
    geom_bar(stat="identity",position="identity")+
    theme_classic()+
    theme(axis.text.x=element_text(angle=90),axis.text=element_text(size=rel(1)))+
    labs(x="",y="log10(p-value)",title="metaphase I spo13 v WT sig changed Polo phospho-proteins")+
    coord_flip()+
    scale_y_continuous(labels=label_number(accuracy=1))

# have to show TC in different plot because values are so different
TC_spo13_WT_polo_GO_bar<-rbind(TC_spo13_INCR=gost(WT.spo13.phos.fin.df %>%
    filter(grepl(".............[DEN].[ST]...............",Sequence.window)) %>%
    filter(sigdiff.sig.cat=="sigdiff_spo13_INCR") %>% select(Gene.name) %>% unlist(),organism='scerevisiae',evcodes=T, significant=F)$result %>%
    filter(term_id %in% polo_spo13_WT_terms) %>%
        filter(p_value!=1) %>%
    select(term_name,term_id,p_value),
    TC_spo13_DECR=gost(WT.spo13.phos.fin.df %>%
    filter(grepl(".............[DEN].[ST]...............",Sequence.window)) %>%
    filter(sigdiff.sig.cat=="sigdiff_spo13_DECR") %>% select(Gene.name) %>% unlist(),organism='scerevisiae',evcodes=T, significant=F)$result %>%
    filter(term_id %in% polo_spo13_WT_terms) %>%
        filter(p_value!=1) %>%
    select(term_name,term_id,p_value)) %>%
    mutate(direction=ifelse(grepl("INCR",rownames(.)),"INCR","DECR")) %>%
    mutate(p_value=ifelse(grepl("DECR",direction),log10(p_value),-log10(p_value))) %>%
    mutate(sample=gsub("(.+)_(.+)_(.+)","\\1_\\2",rownames(.))) %>%
    ggplot(aes(x=term_name,y=p_value,fill=interaction(direction,sample)))+
    geom_bar(stat="identity",position="identity")+
    theme_classic()+
    theme(axis.text.x=element_text(angle=90),axis.text=element_text(size=rel(1)))+
    labs(x="",y="log10(p-value)",title="timecourse spo13 v WT sig changed Polo phospho-proteins")+
    coord_flip()+
    scale_y_continuous(labels=label_number(accuracy=0.1))

print_plot(meta_spo13_WT_polo_GO_bar,
           "meta_spo13_WT_polo_GO_bar.pdf",10,3)

print_plot(TC_spo13_WT_polo_GO_bar,
           "TC_spo13_WT_polo_GO_bar.pdf",10,1.75)
    
```


## +1F motif in TCs lineplots
```{r,echo=F,message=F,warning=F}


# the Polo +1F motif in WT and WT.spo13 TCs lineplot
total_intensity_lineplot_one_strain<-function(df,strain,motif,kinasestring,col){
    cols<-grep(paste0("^scaled.",strain,".mean."),colnames(df),value=T)
    prefix<-paste0("^scaled.",strain,".mean.")
    
    df %>%
    filter(grepl(motif,Sequence.window)) %>%
    select(all_of(cols),gene.name_site) %>%
    pivot_longer(-c(gene.name_site),names_to="time",values_to="intensity",names_prefix=prefix) %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=1))+
  stat_summary(aes(y=intensity),fun=median,geom="line",size=1.5,color=col)+
  stat_summary(aes(y=intensity),fun.min= function(z) {quantile(z,0.25)},
  fun.max=function(z) {quantile(z,0.75)},
  geom="ribbon",alpha=0.1,fill=col)+
  theme_classic()+
        scale_y_continuous(labels=label_number(accuracy = 0.1))+
    #scale_y_continuous(expand=expansion(mult=c(0,0.2)))+
        theme(axis.text.x = element_text(angle=90,vjust=0.6),axis.title=element_text(size=rel(2)),axis.text=element_text(size=rel(2)),plot.title=element_text(size=rel(2)),legend.text = element_text(size=rel(2)))+
  labs(y="scaled intensity",x="time",title=paste0(kinasestring,", n=",df %>% filter(grepl(motif,Sequence.window)) %>% nrow()))
}

# WT and spo13 mean scaled across all 20 samples together  -- would preserve the abundance differences between strains
total_intensity_lineplot_two_strains<-function(df,motif,kinasestring){
    df %>%
    filter(grepl(motif,Sequence.window)) %>%
    select(all_of(WT.spo13.scaled.mean.cols),gene.name_site) %>%
    pivot_longer(-c(gene.name_site),names_to="time",values_to="intensity") %>%
    mutate(strain=ifelse(grepl("WT",time),"WT","spo13")) %>%
    mutate(time=gsub("scaled.both.WT.mean.|scaled.both.spo13.mean.","",time)) %>%
    ggplot(aes(x=fct_inorder(time),y=intensity,group=strain))+
    stat_summary(aes(y=intensity,color=strain),fun=median,geom="line",size=1.5)+
  stat_summary(aes(y=intensity,fill=strain),fun.min= function(z) {quantile(z,0.25)},
  fun.max=function(z) {quantile(z,0.75)},
  geom="ribbon",alpha=0.1)+
    theme_classic()+
        scale_y_continuous(labels=label_number(accuracy = 0.1))+
  theme(axis.text.x = element_text(angle=90,vjust=0.6),axis.title=element_text(size=rel(2)),axis.text=element_text(size=rel(2)),plot.title=element_text(size=rel(2)),legend.text = element_text(size=rel(2)))+
  labs(y="scaled intensity",x="time",title=paste0(kinasestring,", n=",df %>% filter(grepl(motif,Sequence.window)) %>% nrow()))
}

WT_all_DEN.ST_line<-total_intensity_lineplot_one_strain(WT.phos.fin.df,"WT",".............[DEN].[ST]...............","WT all [DEN].[ST]*","#00BFC4")
spo13_all_DEN.ST_line<-total_intensity_lineplot_one_strain(spo13.phos.fin.df,"spo13",".............[DEN].[ST]...............","spo13 all [DEN].[ST]*","#F8766D")
WT_spo13_all_DEN.ST_line<-total_intensity_lineplot_two_strains(WT.spo13.phos.fin.df,".............[DEN].[ST]...............","WT and spo13 [DEN].[ST]*")

WT_all_DEN.STF_line<-total_intensity_lineplot_one_strain(WT.phos.fin.df,"WT",".............[DEN].[ST]F..............","WT all [DEN].[ST]*F","#00BFC4")
spo13_all_DEN.STF_line<-total_intensity_lineplot_one_strain(spo13.phos.fin.df,"spo13",".............[DEN].[ST]F..............","spo13 all [DEN].[ST]*F","#F8766D")
WT_spo13_all_DEN.STF_line<-total_intensity_lineplot_two_strains(WT.spo13.phos.fin.df,".............[DEN].[ST]F..............","WT and spo13 [DEN].[ST]*F")

WT_all_DEN.STG_line<-total_intensity_lineplot_one_strain(WT.phos.fin.df,"WT",".............[DEN].[ST]G..............","WT all [DEN].[ST]*G","#00BFC4")
spo13_all_DEN.STG_line<-total_intensity_lineplot_one_strain(spo13.phos.fin.df,"spo13",".............[DEN].[ST]G..............","spo13 all [DEN].[ST]*G","#F8766D")
WT_spo13_all_DEN.STG_line<-total_intensity_lineplot_two_strains(WT.spo13.phos.fin.df,".............[DEN].[ST]G..............","WT and spo13 [DEN].[ST]*G")

WT_all_DEN.STFG_line<-total_intensity_lineplot_one_strain(WT.phos.fin.df,"WT",".............[DEN].[ST][FG]..............","WT all [DEN].[ST]*[FG]","#00BFC4")
spo13_all_DEN.STFG_line<-total_intensity_lineplot_one_strain(spo13.phos.fin.df,"spo13",".............[DEN].[ST][FG]..............","spo13 all [DEN].[ST]*[FG]","#F8766D")
WT_spo13_all_DEN.STFG_line<-total_intensity_lineplot_two_strains(WT.spo13.phos.fin.df,".............[DEN].[ST][FG]..............","WT and spo13 [DEN].[ST]*[FG]")

# WT_all_DEN.STY_line<-total_intensity_lineplot_one_strain(WT.phos.fin.df,"WT",".............[DEN].[ST]Y..............","WT all [DEN].[ST]Y","#00BFC4")
# spo13_all_DEN.STY_line<-total_intensity_lineplot_one_strain(spo13.phos.fin.df,"spo13",".............[DEN].[ST]Y..............","spo13 all [DEN].[ST]Y","#F8766D")
# WT_spo13_all_DEN.STY_line<-total_intensity_lineplot_two_strains(WT.spo13.phos.fin.df,".............[DEN].[ST]Y..............","WT and spo13 [DEN].[ST]Y")
# 
# WT_all_STF_line<-total_intensity_lineplot_one_strain(WT.phos.fin.df,"WT","...............[ST]F..............","WT all [ST]F","#00BFC4")
# spo13_all_STF_line<-total_intensity_lineplot_one_strain(spo13.phos.fin.df,"spo13","...............[ST]F..............","spo13 all [ST]F","#F8766D")
# WT_spo13_all_STF_line<-total_intensity_lineplot_two_strains(WT.spo13.phos.fin.df,"...............[ST]F..............","WT and spo13 [ST]F")
# 
# WT_all_STY_line<-total_intensity_lineplot_one_strain(WT.phos.fin.df,"WT","...............[ST]Y..............","WT all [ST]Y","#00BFC4")
# spo13_all_STY_line<-total_intensity_lineplot_one_strain(spo13.phos.fin.df,"spo13","...............[ST]Y..............","spo13 all [ST]Y","#F8766D")
# WT_spo13_all_STY_line<-total_intensity_lineplot_two_strains(WT.spo13.phos.fin.df,"...............[ST]Y..............","WT and spo13 [ST]Y")
# 
# WT_all_DEN.STLIMY_line<-total_intensity_lineplot_one_strain(WT.phos.fin.df,"WT",".............[DEN].[ST][LIMY]..............","WT all [DEN].[ST][LIMY]","#00BFC4")
# spo13_all_DEN.STLIMY_line<-total_intensity_lineplot_one_strain(spo13.phos.fin.df,"spo13",".............[DEN].[ST][LIMY]..............","spo13 all [DEN].[ST][LIMY]","#F8766D")
# WT_spo13_all_DEN.STLIMY_line<-total_intensity_lineplot_two_strains(WT.spo13.phos.fin.df,".............[DEN].[ST][LIMY]..............","WT and spo13 [DEN].[ST][LIMY]")

WT_all_DEN.STLIMYF_line<-total_intensity_lineplot_one_strain(WT.phos.fin.df,"WT",".............[DEN].[ST][LIMYF]..............","WT all [DEN].[ST]*[LIMYF]","#00BFC4")
spo13_all_DEN.STLIMYF_line<-total_intensity_lineplot_one_strain(spo13.phos.fin.df,"spo13",".............[DEN].[ST][LIMYF]..............","spo13 all [DEN].[ST]*[LIMYF]","#F8766D")
WT_spo13_all_DEN.STLIMYF_line<-total_intensity_lineplot_two_strains(WT.spo13.phos.fin.df,".............[DEN].[ST][LIMYF]..............","WT and spo13 [DEN].[ST]*[LIMYF]")

WT_all_DEN.STLIMYFG_line<-total_intensity_lineplot_one_strain(WT.phos.fin.df,"WT",".............[DEN].[ST][LIMYFG]..............","WT all [DEN].[ST]*[LIMYFG]","#00BFC4")
spo13_all_DEN.STLIMYFG_line<-total_intensity_lineplot_one_strain(spo13.phos.fin.df,"spo13",".............[DEN].[ST][LIMYFG]..............","spo13 all [DEN].[ST]*[LIMYFG]","#F8766D")
WT_spo13_all_DEN.STLIMYFG_line<-total_intensity_lineplot_two_strains(WT.spo13.phos.fin.df,".............[DEN].[ST][LIMYFG]..............","WT and spo13 [DEN].[ST]*[LIMYFG]")

WT_all_DEN.STLIMYF_line<-total_intensity_lineplot_one_strain(WT.phos.fin.df,"WT",".............[DEN].[ST][LIMYF]..............","WT all [DEN].[ST]*[LIMYF]","#00BFC4")
spo13_all_DEN.STLIMYF_line<-total_intensity_lineplot_one_strain(spo13.phos.fin.df,"spo13",".............[DEN].[ST][LIMYF]..............","spo13 all [DEN].[ST]*[LIMYF]","#F8766D")
WT_spo13_all_DEN.STLIMYF_line<-total_intensity_lineplot_two_strains(WT.spo13.phos.fin.df,".............[DEN].[ST][LIMYF]..............","WT and spo13 [DEN].[ST]*[LIMYF]")

# WT_all_DEN.STLIM_line<-total_intensity_lineplot_one_strain(WT.phos.fin.df,"WT",".............[DEN].[ST][LIM]..............","WT all [DEN].[ST][LIM]","#00BFC4")
# spo13_all_DEN.STLIM_line<-total_intensity_lineplot_one_strain(spo13.phos.fin.df,"spo13",".............[DEN].[ST][LIM]..............","spo13 all [DEN].[ST][LIM]","#F8766D")
# WT_spo13_all_DEN.STLIM_line<-total_intensity_lineplot_two_strains(WT.spo13.phos.fin.df,".............[DEN].[ST][LIM]..............","WT and spo13 [DEN].[ST][LIM]")

print_plot(WT_all_DEN.ST_line+spo13_all_DEN.ST_line+WT_spo13_all_DEN.ST_line,
"TC_WT_spo13_all_DEN.ST_line.pdf",20,5)
print_plot(WT_all_DEN.STF_line+spo13_all_DEN.STF_line+WT_spo13_all_DEN.STF_line,
"TC_WT_spo13_all_DEN.STF_line.pdf",20,5)
print_plot(WT_all_DEN.STG_line+spo13_all_DEN.STG_line+WT_spo13_all_DEN.STG_line,
"TC_WT_spo13_all_DEN.STG_line.pdf",20,5)
print_plot(WT_all_DEN.STFG_line+spo13_all_DEN.STFG_line+WT_spo13_all_DEN.STFG_line,
"TC_WT_spo13_all_DEN.STFG_line.pdf",20,5)
# print_plot(WT_all_DEN.STY_line+spo13_all_DEN.STY_line+WT_spo13_all_DEN.STY_line,
# "TC_WT_spo13_all_DEN.STY_line.pdf",15,4)
# print_plot(WT_all_STF_line+spo13_all_STF_line+WT_spo13_all_STF_line,
# "TC_WT_spo13_all_STF_line.pdf",15,4)
# print_plot(WT_all_STY_line+spo13_all_STY_line+WT_spo13_all_STY_line,
# "TC_WT_spo13_all_STY_line.pdf",15,4)
# print_plot(WT_all_DEN.STLIMY_line+spo13_all_DEN.STLIMY_line+WT_spo13_all_DEN.STLIMY_line,
# "TC_WT_spo13_all_DEN.STLIMY_line.pdf",15,4)
print_plot(WT_all_DEN.STLIMYF_line+spo13_all_DEN.STLIMYF_line+WT_spo13_all_DEN.STLIMYF_line,
"TC_WT_spo13_all_DEN.STLIMYF_line.pdf",20,5)
print_plot(WT_all_DEN.STLIMYFG_line+spo13_all_DEN.STLIMYFG_line+WT_spo13_all_DEN.STLIMYFG_line,
"TC_WT_spo13_all_DEN.STLIMYFG_line.pdf",20,5)
print_plot(WT_all_DEN.STLIMYF_line+spo13_all_DEN.STLIMYF_line+WT_spo13_all_DEN.STLIMYF_line,
"TC_WT_spo13_all_DEN.STLIMYF_line.pdf",20,5)
# print_plot(WT_all_DEN.STLIM_line+spo13_all_DEN.STLIM_line+WT_spo13_all_DEN.STLIM_line,
# "TC_WT_spo13_all_DEN.STLIM_line.pdf",15,4)

```

## Supp all kinase motif lineplots
```{r}

# All 12 WT motifs
WT_kinase_lineplots<-list()

for(i in 1:12){
    WT_kinase_lineplots[[i]]<-total_intensity_lineplot_one_strain(WT.phos.fin.df,"WT",kinase.df$kinase.motifs[i],paste0("WT ",kinase.df$kinases[i]),"#00BFC4")
}

# All 12 spo13 motifs
spo13_kinase_lineplots<-list()

for(i in 1:12){
    spo13_kinase_lineplots[[i]]<-total_intensity_lineplot_one_strain(spo13.phos.fin.df,"spo13",kinase.df$kinase.motifs[i],paste0("spo13 ",kinase.df$kinases[i]),"#F8766D")
}

# All 12 motifs in WT andspo13 shared df
WT_spo13_kinase_lineplots<-list()

for(i in 1:12){
    WT_spo13_kinase_lineplots[[i]]<-total_intensity_lineplot_two_strains(WT.spo13.phos.fin.df,kinase.df$kinase.motifs[i],paste0("WT and spo13 ",kinase.df$kinases[i]))
}

# print all WT together
print_plot((WT_kinase_lineplots[[1]]+WT_kinase_lineplots[[2]]+WT_kinase_lineplots[[3]])/
    (WT_kinase_lineplots[[4]]+WT_kinase_lineplots[[5]]+WT_kinase_lineplots[[6]])/
(WT_kinase_lineplots[[7]]+WT_kinase_lineplots[[8]]+WT_kinase_lineplots[[9]])/
    (WT_kinase_lineplots[[10]]+WT_kinase_lineplots[[11]]+WT_kinase_lineplots[[12]]),
"WT_all_kinase_motif_lineplots.pdf",15,18)

#15 xx 18 portrait

# print all spo13 together
print_plot((spo13_kinase_lineplots[[1]]+spo13_kinase_lineplots[[2]]+spo13_kinase_lineplots[[3]])/
    (spo13_kinase_lineplots[[4]]+spo13_kinase_lineplots[[5]]+spo13_kinase_lineplots[[6]])/
(spo13_kinase_lineplots[[7]]+spo13_kinase_lineplots[[8]]+spo13_kinase_lineplots[[9]])/
    (spo13_kinase_lineplots[[10]]+spo13_kinase_lineplots[[11]]+spo13_kinase_lineplots[[12]]),
"spo13_all_kinase_motif_lineplots.pdf",15,18)

# print all WT v spo13 together
print_plot((WT_spo13_kinase_lineplots[[1]]+WT_spo13_kinase_lineplots[[2]]+WT_spo13_kinase_lineplots[[3]])/
    (WT_spo13_kinase_lineplots[[4]]+WT_spo13_kinase_lineplots[[5]]+WT_spo13_kinase_lineplots[[6]])/
(WT_spo13_kinase_lineplots[[7]]+WT_spo13_kinase_lineplots[[8]]+WT_spo13_kinase_lineplots[[9]])/
    (WT_spo13_kinase_lineplots[[10]]+WT_spo13_kinase_lineplots[[11]]+WT_spo13_kinase_lineplots[[12]]),
"WT_spo13_all_kinase_motif_lineplots.pdf",22,20)


```

# Basophilic motif lineplots for supp figure
```{r}

# Ipl1/PKA motif as part of the 12 set RKx[ST]
WT_spo13_all_RKxST_line<-total_intensity_lineplot_two_strains(WT.spo13.phos.fin.df,"............RK.[ST]...............","WT and spo13 RK.[ST]*")

# Rxx[ST]
WT_spo13_all_RxxST_line<-total_intensity_lineplot_two_strains(WT.spo13.phos.fin.df,"............R..[ST]...............","WT and spo13 R..[ST]*")

# Rx[ST]
WT_spo13_all_RxST_line<-total_intensity_lineplot_two_strains(WT.spo13.phos.fin.df,"............R.[ST]...............","WT and spo13 R.[ST]*")

print_plot(WT_spo13_all_RKxST_line+WT_spo13_all_RxxST_line+WT_spo13_all_RxST_line,
"TC_WT_spo13_all_basophilic_line.pdf",20,5)

# [RK]xx[ST]
WT_spo13_all_RKxxST_line<-total_intensity_lineplot_two_strains(WT.spo13.phos.fin.df,"............[RK]..[ST]...............","WT and spo13 [RK]..[ST]*")

# [RK]x[ST]
WT_spo13_all_RKxST_line<-total_intensity_lineplot_two_strains(WT.spo13.phos.fin.df,"............[RK].[ST]...............","WT and spo13 [RK].[ST]*")

print_plot(WT_spo13_all_RKxxST_line+WT_spo13_all_RKxST_line+WT_spo13_all_RKxST_line,
"TC_WT_spo13_all_basophilic_line_2.pdf",20,5)



```

# Motif of WTvspo13 cluster 9
```{r}

WT_spo13_sigdiff_cluster_9_logo<-short_logo(WT.spo13.phos.fin.df.sigdiff %>% filter(ward.cluster==9),"prob","WT v spo13 cluster 9, n=")

print_plot(WT_spo13_sigdiff_cluster_9_logo+WT_spo13_sigdiff_cluster_9_logo+WT_spo13_sigdiff_cluster_9_logo,
           "sigdiff_phos_cluster_9_logo.pdf",12,3)

# choose a selected site to highlight for reviewer
# WT.spo13.phos.fin.df.sigdiff %>% filter(ward.cluster==9) %>%
#     select(gene.name_site,Sequence.window)
# FAR11-S-210 cell cycle autophagy
# KIN28-T-162 autophagy
# HRR25-S-330 cross talk between Cdc5 and Hrr25 found by Stefan


    
sigdiff_cluster_9_sel_sites<-WT.spo13.phos.fin.df.sigdiff %>%
  filter(gene.name_site %in% c("FAR11-S-210","KIN28-T-162","HRR25-S-330")) %>%
  select(Gene.name,gene.name_site,WT.spo13.scaled.mean.cols) %>%
  pivot_longer(-c(Gene.name,gene.name_site),names_to="time",values_to="intensity",names_prefix="scaled.both.") %>%
mutate(strain=ifelse(grepl("WT",time),"WT","spo13")) %>%
    mutate(time=gsub("WT.mean.|spo13.mean.","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=interaction(strain,gene.name_site)))+
    facet_wrap(~gene.name_site,scales="free_y")+
  geom_line(aes(color=strain))+
  theme_classic()+
    scale_y_continuous(labels=label_number(accuracy=0.1))+
   theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),axis.text.y=element_text(size=rel(1.5)),plot.title=element_text(size=rel(1.5)),legend.title=element_text(size=rel(1.5)),legend.text=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),strip.text=element_text(size=rel(1)))+
  labs(x="time",y="scaled intensity",title="")

print_plot(sigdiff_cluster_9_sel_sites,
           "sigdiff_cluster_9_sel_sites.pdf",12,3)


```



# WT phos cluster enrichment of DEN v DENxSTF
```{r}

fisher_barplot_TC_clust_together_2<-function(df,clusternum,motif){
    
    subset1<-list()
    subset2<-list()
    out_df<-list()
    
    for(i in 1:clusternum){
        subset1<-paste0("non-cluster ",i)
        subset2<-paste0("cluster ",i)
        
        df1<-df
        df2<-df %>% filter(ward.cluster==i)
        
       out_df[[i]]<- fisher(df1,df2,motif) %>%
    tail(1) %>%
     mutate(stars=ifelse(pval<0.001,"***",
                      ifelse(pval<0.01,"**",
                             ifelse(pval<0.05,"*",NA)))) %>%
    select(df1,df2,pval,stars) %>%
    pivot_longer(-c(pval,stars),values_to="perc",names_to="sample") %>%
    mutate(sample=ifelse(grepl("df1",sample),subset1,subset2)) %>%
        mutate(sample=factor(sample,levels=c(subset1,subset2))) %>% # have to make sure levels are consistent 
    mutate(group=subset2)
       #return(out_df)
        
    }
    
results<-do.call(rbind,out_df)
return(results)
}

WT_clusters_DEN.ST_fish_bar_new<-fisher_barplot_TC_clust_together_2(WT.phos.fin.df.sig,11,".............[DEN].[ST]...............") %>%
    mutate(within_cluster=!grepl("non-cluster",sample)) %>%
    ggplot(aes(x=fct_inorder(group),y=perc,fill=within_cluster))+
    geom_bar(position="dodge",stat="identity")+
    labs(x="cluster",y="percent matches",title="enrichment of [DEN].[ST]*")+
    theme_classic()+
    scale_y_continuous(expand=expansion(mult=c(0,0.2)),labels=label_number(accuracy=0.01))+
        theme(axis.text.x = element_text(angle=45,vjust=1,hjust=1),axis.title.x=element_blank(),axis.text=element_text(size=rel(1)),
        axis.title=element_text(size=rel(1)),title=element_text(size=rel(1.5)))

print_plot(WT_clusters_DEN.ST_fish_bar_new,"WT_clusters_DEN.ST_fish_bar_new.pdf",15,7)

WT_clusters_DEN.STF_fish_bar_new<-fisher_barplot_TC_clust_together_2(WT.phos.fin.df.sig,11,".............[DEN].[ST]F..............") %>%
    mutate(within_cluster=!grepl("non-cluster",sample)) %>%
    ggplot(aes(x=fct_inorder(group),y=perc,fill=within_cluster))+
    geom_bar(position="dodge",stat="identity")+
    labs(x="cluster",y="percent matches",title="enrichment of [DEN].[ST]*F")+
    theme_classic()+
    scale_y_continuous(expand=expansion(mult=c(0,0.2)),labels=label_number(accuracy=0.01))+
        theme(axis.text.x = element_text(angle=45,vjust=1,hjust=1),axis.title.x=element_blank(),axis.text=element_text(size=rel(1)),
        axis.title=element_text(size=rel(1)),title=element_text(size=rel(1.5)))

print_plot(WT_clusters_DEN.STF_fish_bar_new,"WT_clusters_DEN.STF_fish_bar_new.pdf",15,7)

# fisher results export
## DEN.ST
fisher_barplot_TC_clust_together_2(WT.phos.fin.df.sig,11,".............[DEN].[ST]...............") %>%
    mutate(within_cluster=!grepl("non-cluster",sample)) %>%
    write.xlsx(.,"DEN.ST_WT_phos_clust_fisher_tests_2.xlsx")
## DEN.STF
fisher_barplot_TC_clust_together_2(WT.phos.fin.df.sig,11,".............[DEN].[ST]F..............") %>%
    mutate(within_cluster=!grepl("non-cluster",sample)) %>%
    write.xlsx(.,"DEN.STF_WT_phos_clust_fisher_tests_2.xlsx")


```


# DEN.STF sites export to excel
```{r}

## DEN>STF SITE IDENTITIES

# WT DEN.STF sites # no sig cutoff #n=46
WT.phos.fin.df %>%
    filter(grepl(".............[DEN].[ST]F..............", Sequence.window)) %>% 
    mutate(Sequence.window_5=substr(Sequence.window,11,21)) %>%
    mutate(time.first.sig=apply(.[,WT.sig.cols],1,function(x) ifelse(any(x=="Zero"),NA,which(x!="NC")[1]))) %>% # remember all sites will have at least one sig change because this df only includes those sites
    mutate(time.first.sig=as.numeric(alltimes[2:10][time.first.sig])) %>% # cahnge from 1-9 to 45-165
    #filter(gene.name_site %in% Exp40E_DEN.STF_sites) %>%
    select(Gene.name,gene.name_site,sig.cat,time.first.sig,Sequence.window_5,Fasta.headers) %>%
    arrange(time.first.sig) %>%
    write.xlsx(.,"TC_WT_phos_DEN.STF_sites.xlsx")


# spo13 only DEN.STF sites # no ssig cutoff #n=51
spo13.phos.fin.df %>%
    filter(grepl(".............[DEN].[ST]F..............", Sequence.window)) %>% 
    mutate(Sequence.window_5=substr(Sequence.window,11,21)) %>%
     mutate(time.first.sig=apply(.[,spo13.sig.cols],1,function(x) ifelse(any(x=="Zero"),NA,which(x!="NC")[1]))) %>% # remember all sites will have at least one sig change because this df only includes those sites
    mutate(time.first.sig=as.numeric(alltimes[2:10][time.first.sig])) %>% # cahnge from 1-9 to 45-165
    #filter(gene.name_site %in% Exp40E_DEN.STF_sites) %>%
    select(Gene.name,gene.name_site,sig.cat,time.first.sig,Sequence.window_5,Fasta.headers) %>%
    arrange(time.first.sig) %>%
    write.xlsx(.,"TC_spo13_phos_DEN.STF_sites.xlsx")


# line plots of sites -- all DEN.STF sites found in WT + spo13 timecourses
TC_WT_spo13_DEN.STF_site_lines<-WT.spo13.phos.fin.df %>%
    filter(grepl(".............[DEN].[ST]F..............", Sequence.window)) %>% 
    mutate(Sequence.window_5=substr(Sequence.window,11,21)) %>%
    select(Gene.name,gene.name_site,WT.spo13.scaled.mean.cols) %>%
  pivot_longer(-c(Gene.name,gene.name_site),names_to="time",values_to="intensity",names_prefix="scaled.both.") %>%
mutate(strain=ifelse(grepl("WT",time),"WT","spo13")) %>%
    mutate(time=gsub("WT.mean.|spo13.mean.","",time)) %>%
  ggplot(aes(x=fct_inorder(time),y=intensity,group=interaction(strain,gene.name_site)))+
    facet_wrap(~gene.name_site,scales="free_y")+
  geom_line(aes(color=strain))+
  theme_classic()+
    scale_y_continuous(labels=label_number(accuracy=0.1))+
   theme(axis.text.x = element_text(angle=90,vjust=0.6,size=rel(1.5)),axis.text.y=element_text(size=rel(1.5)),plot.title=element_text(size=rel(1.5)),legend.title=element_text(size=rel(1.5)),legend.text=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),strip.text=element_text(size=rel(1)))+
  labs(x="time",y="scaled intensity",title="DEN.STF sites in WT + spo13 TCs")


print_plot(TC_WT_spo13_DEN.STF_site_lines,
           "TC_WT_spo13_DEN.STF_site_lines.pdf",22,12)

# Gene lists used for Venn Diagram in Figure 7
# InteractiVenn
# longest list is 74, make other dfs same length


# multi-sheet excel:
Fsite_list<-list()

Fsite_list[[1]]<-WT.phos.fin.df %>%
    filter(grepl(".............[DEN].[ST]F..............", Sequence.window)) %>% 
    distinct(Gene.name,.keep_all=T) %>% select(Gene.name,gene.name_site)

Fsite_list[[2]]<-spo13.phos.fin.df %>%
    filter(grepl(".............[DEN].[ST]F..............", Sequence.window)) %>% 
    distinct(Gene.name,.keep_all=T) %>% select(Gene.name,gene.name_site)

Fsite_list[[3]]<-Exp40E_phos %>%
    filter(grepl(".............[DEN].[ST]F..............", Sequence.window)) %>% 
    distinct(Gene.name,.keep_all=T) %>% select(Gene.name,gene.name_site)

names(Fsite_list)<-c("TC_WT","TC_spo13","metaphase")

write.xlsx(Fsite_list,"DEN.STF_sites.xlsx")

# all 94 phospho-proteins with a DEN>STF site in each of 3 datasets:
# WT TC
# spo13 TC
# metaphase I arrest (detected both WT and spo13)

DEN.STF_prot_df <- merge(merge(WT.phos.fin.df %>%
    filter(grepl(".............[DEN].[ST]F..............", Sequence.window)) %>% 
    distinct(Gene.name) %>% select(Gene.name),spo13.phos.fin.df %>%
    filter(grepl(".............[DEN].[ST]F..............", Sequence.window)) %>% 
    distinct(Gene.name) %>% select(Gene.name),by="Gene.name",all=T),
    Exp40E_phos %>%
    filter(grepl(".............[DEN].[ST]F..............", Sequence.window)) %>% 
    distinct(Gene.name) %>% select(Gene.name),by="Gene.name",all=T)


DEN.STF_prot_df %>%
    mutate(TC_WT=Gene.name %in% (WT.phos.fin.df %>%
    filter(grepl(".............[DEN].[ST]F..............", Sequence.window)) %>% 
    distinct(Gene.name) %>% select(Gene.name) %>% unlist())) %>%
    mutate(TC_spo13=Gene.name %in% (spo13.phos.fin.df %>%
    filter(grepl(".............[DEN].[ST]F..............", Sequence.window)) %>% 
    distinct(Gene.name) %>% select(Gene.name) %>% unlist())) %>%
    mutate(metaphase=Gene.name %in% (Exp40E_phos %>%
    filter(grepl(".............[DEN].[ST]F..............", Sequence.window)) %>% 
    distinct(Gene.name) %>% select(Gene.name) %>% unlist())) %>%
    arrange(desc(interaction(TC_WT,TC_spo13,metaphase))) %>%
    write.xlsx(.,"DEN.STF_phos_prot.xlsx")
    
```


# percent matches total dataset kinases barplot
```{r}

total_kin_match_list<-list()
for (i in 1:nrow(kinase.df)){
    total_kin_match_list[[i]]<-WT.phos.fin.df %>%
    filter(grepl(kinase.df$kinase.motifs[i],Sequence.window)) %>%
    nrow()
}

total_kin_perc_df<-as.data.frame(cbind(n=total_kin_match_list,total=nrow(WT.phos.fin.df),kin=kinase.df$kinases))


total_WT_timecourse_perc_matches<-total_kin_perc_df %>%
    mutate(n=as.numeric(n)) %>%
    mutate(total=as.numeric(total)) %>%
    mutate(perc=(n/total)*100) %>%
    mutate(kin=as.character(kin)) %>%
    ggplot(aes(x=fct_inorder(kin),y=perc))+
    geom_bar(stat='identity')+
  theme_classic()+
  ylab("% matches")+
    ggtitle("total WT timecourse")+
  #need to add expand to make sure bar starts at 0 y
  scale_y_continuous(expand=expansion(mult=c(0,0.1)))+
        theme(axis.text.x = element_text(angle=45,vjust=1,hjust=1),axis.title.x=element_blank(),axis.text=element_text(size=rel(1)),
        axis.title=element_text(size=rel(1)),title=element_text(size=rel(1.5)))


# WT sig TC
total_kin_match_list<-list()
for (i in 1:nrow(kinase.df)){
    total_kin_match_list[[i]]<-WT.phos.fin.df.sig %>%
    filter(grepl(kinase.df$kinase.motifs[i],Sequence.window)) %>%
    nrow()
}

total_kin_perc_df<-as.data.frame(cbind(n=total_kin_match_list,total=nrow(WT.phos.fin.df),kin=kinase.df$kinases))


sig_WT_timecourse_perc_matches<-total_kin_perc_df %>%
    mutate(n=as.numeric(n)) %>%
    mutate(total=as.numeric(total)) %>%
    mutate(perc=(n/total)*100) %>%
    mutate(kin=as.character(kin)) %>%
    ggplot(aes(x=fct_inorder(kin),y=perc))+
    geom_bar(stat='identity')+
  theme_classic()+
  ylab("% matches")+
    ggtitle("dynamic WT timecourse")+
  #need to add expand to make sure bar starts at 0 y
  scale_y_continuous(expand=expansion(mult=c(0,0.1)))+
        theme(axis.text.x = element_text(angle=45,vjust=1,hjust=1),axis.title.x=element_blank(),axis.text=element_text(size=rel(1)),
        axis.title=element_text(size=rel(1)),title=element_text(size=rel(1.5)))

print_plot(total_WT_timecourse_perc_matches,
           "total_WT_timecourse_perc_matches.pdf",6,4)

print_plot(sig_WT_timecourse_perc_matches,
           "sig_WT_timecourse_perc_matches.pdf",6,4)

```

# export datasets for shiny

```{r}
# export datasets for quicker access for plotting in sshiny app

write.xlsx(WT.protg.fin.df,file="WT_protg.xlsx")
write.xlsx(WT.phos.fin.df,file="WT_phos.xlsx")
write.xlsx(WT.spo13.protg.fin.df,file="WT_spo13_protg.xlsx")
write.xlsx(WT.spo13.phos.fin.df,file="WT_spo13_phos.xlsx")

```


# S[ST] motif analysis
```{r}

# export results table for cross analysis
write.csv(WT.spo13.phos.fin.df,"WT_spo13_timecourse_phos.csv")

# # export lists of gene names of proteins that are differentially phos
# # in WT vs spo13
# # WT v spo13 timecourse sigdiff sites by category
WT.spo13.phos.fin.df %>%
    group_by(sigdiff.sig.cat) %>%
    mutate(id=row_number()) %>%
    pivot_wider(names_from=sigdiff.sig.cat,values_from=Gene.name,id_cols=id) %>%
    select(-id) %>%
    #rename_with(~paste0("cluster_",.x,recycle0=T)) %>%
    write.xlsx(.,"WT_spo13_phos_sigdiff_sigcat_ids_gene_names.xlsx")

# new function that will print the numbers on the barplot
fisher_barplot_counts<-function(df1,df2,motif,subset1,subset2,title){
    out_df<-fisher(df1,df2,motif) %>%
    tail(1) %>%
     mutate(stars=ifelse(pval<0.001,"***",
                      ifelse(pval<0.01,"**",
                             ifelse(pval<0.05,"*",NA)))) %>%
    select(df1,df2,pval,stars) %>%
    pivot_longer(-c(pval,stars),values_to="perc",names_to="sample") %>%
    mutate(sample=ifelse(grepl("df1",sample),subset1,subset2)) %>%
        mutate(sample=factor(sample,levels=c(subset1,subset2))) # have to make sure levels are consistent
    
    ratios<-fisher(df1,df2,motif) %>%
        rbind(ratio=paste0(.[1,],"/",.[3,])) %>%
        tail(1) %>%
        select(df1,df2) %>%
        pivot_longer(cols=c(df1,df2),names_to="set",values_to="ratio") %>%
        select(ratio)
    
    out_df<-out_df %>%
        cbind(ratios)
        
    out_df %>%
    ggplot(aes(x=sample,y=perc,fill=sample))+
    geom_bar(aes(fill=sample),stat="identity")+
        geom_text(aes(label=ratio),vjust=2)+
   geom_signif(comparisons = list(c(subset1,subset2)),
              map_signif_level=TRUE,
              annotations = out_df$stars[1],
              size=0.25)+
    scale_y_continuous(expand=expansion(mult=c(0,0.2)),labels = label_number(accuracy=0.1))+
    theme_classic()+
    theme(axis.text.x = element_text(angle=90,vjust=0.6),legend.position="none",axis.text=element_text(size=rel(1)))+
    labs(y="percent matches",x="",title=title,,axis.line=element_line(linewidth=0.1))
}

# lists of proteins that have a sig DECR phos on a [DEN]x[ST] site in spo13
spo13_WT_polo_prots<-WT.spo13.phos.fin.df %>%
    filter(sigdiff.sig.cat=="sigdiff_spo13_DECR") %>%
    filter(grepl(".............[DEN].[ST]...............",Sequence.window)) %>%
    distinct(Gene.name) %>% unlist() %>% unname()

# fish bar
# of all DECR v NC sites
spo13_WT_SSTP_DECR_v_NC_fish_bar_count<-fisher_barplot_counts(WT.spo13.phos.fin.df %>% filter(sigdiff.sig.cat=="NC"),WT.spo13.phos.fin.df %>% filter(sigdiff.sig.cat=="sigdiff_spo13_DECR"),"..............S[ST]P..............","NC","DECR","S[ST]*P sites")

spo13_WT_SST_DECR_v_NC_fish_bar_count<-fisher_barplot_counts(WT.spo13.phos.fin.df %>% filter(sigdiff.sig.cat=="NC"),WT.spo13.phos.fin.df %>% filter(sigdiff.sig.cat=="sigdiff_spo13_DECR"),"..............S[ST]...............","NC","DECR","S[ST]* sites")

# of DECR v NC sites on proteins that had a sig decr polo phos in spo13
spo13_WT_polo_SSTP_DECR_v_NC_fish_bar_count<-fisher_barplot_counts(WT.spo13.phos.fin.df %>%
    filter(Gene.name %in% spo13_WT_polo_prots) %>% filter(sigdiff.sig.cat=="NC"),WT.spo13.phos.fin.df %>% filter(Gene.name %in% spo13_WT_polo_prots) %>% filter(sigdiff.sig.cat=="sigdiff_spo13_DECR"),"..............S[ST]P..............","NC","DECR","Polo prot S[ST]*P sites")

spo13_WT_polo_SST_DECR_v_NC_fish_bar_count<-fisher_barplot_counts(WT.spo13.phos.fin.df %>% filter(Gene.name %in% spo13_WT_polo_prots) %>% filter(sigdiff.sig.cat=="NC"),WT.spo13.phos.fin.df %>% filter(Gene.name %in% spo13_WT_polo_prots) %>% filter(sigdiff.sig.cat=="sigdiff_spo13_DECR"),"..............S[ST]...............","NC","DECR","Polo prot S[ST]* sites")

print_plot(spo13_WT_SSTP_DECR_v_NC_fish_bar_count+spo13_WT_SST_DECR_v_NC_fish_bar_count+spo13_WT_polo_SSTP_DECR_v_NC_fish_bar_count+spo13_WT_polo_SST_DECR_v_NC_fish_bar_count,"TC_SSTP_fisher_barplot_counts.pdf",8,13)

```

# when are baso motifs increased in spo13 timecourse
```{r}


# 86 sites in Figure 4 that are INCR in spo13 and enriched for Rxx[ST]*
# 
# all sites that are INCR in spo13
new1<-WT.spo13.phos.fin.df %>%
    filter(sigdiff.sig.cat=="sigdiff_spo13_INCR") %>%
    select(gene.name_site,sigdiff.sig.cat,WT.spo13.sig.cols) %>%
    mutate(time_first_spo13_incr=apply(.[,WT.spo13.sig.cols],1,function(x) which(x=="DECR")[1])) %>%
    mutate(new=alltimes[time_first_spo13_incr]) %>%
    #mutate(new=as.character(new)) %>%
    select(gene.name_site,new) %>%
    ggplot(aes(x=fct_inseq(new)))+
    geom_histogram(stat="count")+
    labs(x="time",y="count",title="Sites that are INCR in spo13, n=86")+
    theme_classic()+
scale_y_continuous(expand=expansion(mult=c(0,0.2)))+ # 20% more plot area after max value
    theme(axis.text=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),title=element_text(size=rel(1.5)),legend.position="none",
          strip.text=element_text(size=rel(1.5)))

# only sites that are INCR in spo13 AND matching Rxx[ST]*
new2<-WT.spo13.phos.fin.df %>%
    filter(sigdiff.sig.cat=="sigdiff_spo13_INCR") %>%
    filter(grepl("............[R]..[ST]...............",Sequence.window)) %>%
    select(gene.name_site,sigdiff.sig.cat,WT.spo13.sig.cols) %>%
    mutate(time_first_spo13_incr=apply(.[,WT.spo13.sig.cols],1,function(x) which(x=="DECR")[1])) %>%
    mutate(new=alltimes[time_first_spo13_incr]) %>%
    #mutate(new=as.character(new)) %>%
    select(gene.name_site,new) %>%
    group_by(new) %>%
    summarize(n=n()) %>%
    rbind(.,c(75,0))%>%
    ggplot(aes(x=fct_inseq(new),y=n))+
    geom_bar(stat="identity")+
    labs(x="time",y="count",title="Rxx[ST]* sites that are INCR in spo13, n=20")+
    theme_classic()+
    scale_y_continuous(expand=expansion(mult=c(0,0.2)))+ # 20% more plot area after max value
    theme(axis.text=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),title=element_text(size=rel(1.5)),legend.position="none",
          strip.text=element_text(size=rel(1.5)))

print_plot(new1,"baso_incr_spo13.pdf",8,6)
print_plot(new2,"baso_RxxST_incr_spo13.pdf",8,6)

# when are decreased sites decr 
new3<-WT.spo13.phos.fin.df %>%
    filter(sigdiff.sig.cat=="sigdiff_spo13_DECR") %>%
    select(gene.name_site,sigdiff.sig.cat,WT.spo13.sig.cols) %>%
    mutate(time_first_spo13_incr=apply(.[,WT.spo13.sig.cols],1,function(x) which(x=="INCR")[1])) %>%
    mutate(new=alltimes[time_first_spo13_incr]) %>%
    #mutate(new=as.character(new)) %>%
    select(gene.name_site,new) %>%
    ggplot(aes(x=fct_inseq(new)))+
    geom_histogram(stat="count")+
    labs(x="time",y="count",title="Sites that are DECR in spo13, n=203")+
    theme_classic()+
scale_y_continuous(expand=expansion(mult=c(0,0.2)))+ # 20% more plot area after max value
    theme(axis.text=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),title=element_text(size=rel(1.5)),legend.position="none",
          strip.text=element_text(size=rel(1.5)))

new4<-WT.spo13.phos.fin.df %>%
    filter(sigdiff.sig.cat=="sigdiff_spo13_DECR") %>%
    filter(grepl(".............[DEN].[ST]...............",Sequence.window)) %>%
    select(gene.name_site,sigdiff.sig.cat,WT.spo13.sig.cols) %>%
    mutate(time_first_spo13_incr=apply(.[,WT.spo13.sig.cols],1,function(x) which(x=="INCR")[1])) %>%
    mutate(new=alltimes[time_first_spo13_incr]) %>%
    #mutate(new=as.character(new)) %>%
    select(gene.name_site,new) %>%
    group_by(new) %>%
    summarize(n=n()) %>%
    rbind(.,c(75,0))%>%
    ggplot(aes(x=fct_inseq(new),y=n))+
    geom_bar(stat="identity")+
    labs(x="time",y="count",title="[DEN]x[ST]* sites that are DECR in spo13, n=65")+
    theme_classic()+
    scale_y_continuous(expand=expansion(mult=c(0,0.2)),breaks=pretty_breaks())+ # 20% more plot area after max value
    theme(axis.text=element_text(size=rel(1.5)),axis.title=element_text(size=rel(1.5)),title=element_text(size=rel(1.5)),legend.position="none",
          strip.text=element_text(size=rel(1.5)))

print_plot(new3,"when_decr_spo13.pdf",8,6)
print_plot(new4,"when_DENxST_decr_spo13.pdf",8,6)

```

